{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Omics Cookbook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Library Imports"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run this cell to import the necessary libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to the cptac data service package. Available datasets may be\n",
      "viewed using cptac.list_data(). In order to access a specific data\n",
      "set, import a cptac subfolder using either 'import cptac.dataset' or\n",
      "'from cptac import dataset'.\n",
      "******\n",
      "Version: 0.4.0\n",
      "******\n",
      "You have loaded the cptac endometrial dataset. To view available\n",
      "dataframes, use cptac.endometrial.list_data(). To view available\n",
      "functions for accessing and manipulating the dataframes, use\n",
      "cptac.endometrial.list_api().\n",
      "endometrial data version: 2.1\n",
      "\n",
      "Loading Dictionary...\n",
      "Loading cptac endometrial data:\n",
      "Loading proteomics data...\n",
      "Loading clinical data...\n",
      "Loading acetylproteomics data...\n",
      "Loading phosphoproteomics_site data...\n",
      "Loading somatic data...\n",
      "Loading transcriptomics_circular data...\n",
      "Loading phosphoproteomics_gene data...\n",
      "Loading transcriptomics_linear data...\n",
      "Loading miRNA data...\n",
      "Loading somatic_binary data...\n",
      "Loading CNA data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "import re\n",
    "\n",
    "import cptac.endometrial as en\n",
    "import cptac.algorithms as al"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get frequently mutated genes list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Enter the type of cancer and the cutoff for mutation frequency that you would like to use."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Gene  Percent Mutated\n",
      "0     RPL22         0.168421\n",
      "1    ARID1A         0.452632\n",
      "2    PIK3CA         0.494737\n",
      "3     FBXW7         0.189474\n",
      "4     RBM27         0.168421\n",
      "5      NSD1         0.168421\n",
      "6     SYNE1         0.168421\n",
      "7     ZFHX4         0.200000\n",
      "8      PTEN         0.789474\n",
      "9      CTCF         0.284211\n",
      "10   ZNF469         0.157895\n",
      "11     TP53         0.221053\n",
      "12    KMT2B         0.242105\n",
      "13     KRAS         0.326316\n",
      "14    LMAN1         0.157895\n",
      "15    SCAF4         0.168421\n",
      "16   PIK3R1         0.389474\n",
      "17   CTNNB1         0.305263\n",
      "18     PCLO         0.168421\n",
      "19    KMT2D         0.178947\n",
      "20    HERC2         0.178947\n",
      "21    ZFHX3         0.221053\n",
      "22     JAK1         0.168421\n",
      "23    OBSCN         0.157895\n",
      "24    AHNAK         0.157895\n",
      "25  CCDC168         0.168421\n",
      "26    DOCK3         0.200000\n",
      "27    MUC16         0.189474\n",
      "28    HUWE1         0.157895\n",
      "29   DNAH17         0.157895\n",
      "30   INPPL1         0.200000\n"
     ]
    }
   ],
   "source": [
    "cancer_type = \"Endometrial\"\n",
    "desired_cutoff = .15\n",
    "\n",
    "fm = al.get_frequently_mutated(cancer_type, cutoff=desired_cutoff)\n",
    "print(fm)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Select a gene from the list of frequently mutated genes above"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set the gene to which of the above frequently mutated genes you want to examine. For example, if you want to look at the gene PTEN, change the cell below to say \"gene = 'PTEN'\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'ARID1A'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Select desired omics comparison"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Select which type of omics you want to compare. If you want to compare proteomics data for the interacting proteins, uncomment the \"omics = en.get_proteomics()\" line, etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "#omics = en.get_proteomics()\n",
    "#omics = en.get_transcriptomics()\n",
    "omics = en.get_phosphoproteomics()\n",
    "#omics = en.get_acetylproteomics()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate interacting proteins and test omics comparisons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Simply run this cell after following the instructions above to see the results of the comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  ZFHX3\n",
      "Generating interacting protein list\n",
      "Doing t-test comparisons\n",
      "\n",
      "         Comparison   P_Value\n",
      "0  ZFHX3_proteomics  0.001988\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    '''Prevent slice degrees of freedom warning'''\n",
    "    import warnings\n",
    "    warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "    interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "    print(\"Generating interacting protein list\")\n",
    "    interacting_proteins_in_omics_df = []\n",
    "\n",
    "    '''Only do comparisons on proteins in the omics dataframe'''\n",
    "    for ip in interacting_proteins:\n",
    "        if omics.name == 'phosphoproteomics' or omics.name == 'acetylproteomics':\n",
    "            col_regex = ip + \"-.*\" # Build a regex to get all columns that match the gene\n",
    "        else:\n",
    "            col_regex = '^{}$'.format(ip)\n",
    "\n",
    "        selected = omics.filter(regex=col_regex)\n",
    "\n",
    "        if len(selected.columns) > 0:\n",
    "            interacting_proteins_in_omics_df.append(ip)\n",
    "\n",
    "    '''Ceate dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.append_mutations_to_omics(mutation_genes=[gene], omics_df=omics, omics_genes=interacting_proteins_in_omics_df)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "\n",
    "except:\n",
    "    print(\"Error in Comparison\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Look at effect on all proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This will look at the effect of the selected gene mutation on all proteins, and report any significant results."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  ARID1A\n",
      "Generating protein list\n",
      "Error in Comparison\n",
      "columns overlap but no suffix specified: Index(['ABHD14A-ACY1-K374_acetylproteomics'], dtype='object')\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    '''Prevent slice degrees of freedom warning'''\n",
    "    #import warnings\n",
    "    #warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "    proteomics = en.get_proteomics()\n",
    "    proteins = proteomics.columns\n",
    "    proteins = proteins[:300]\n",
    "\n",
    "    print(\"Generating protein list\")\n",
    "    interacting_proteins_in_omics_df = []\n",
    "\n",
    "    '''Only do comparisons on proteins in the omics dataframe'''\n",
    "    for ip in proteins:\n",
    "        if omics.name == 'phosphoproteomics' or omics.name == 'acetylproteomics':\n",
    "            col_regex = ip + \"-.*\" # Build a regex to get all columns that match the gene\n",
    "        else:\n",
    "            col_regex = '^{}$'.format(ip)\n",
    "\n",
    "        selected = omics.filter(regex=col_regex)\n",
    "\n",
    "        if len(selected.columns) > 0:\n",
    "            interacting_proteins_in_omics_df.append(ip)\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.append_mutations_to_omics(mutation_genes=[gene], omics_df=omics, omics_genes=interacting_proteins_in_omics_df)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    #print(protdf)\n",
    "\n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           A1BG_transcriptomics  A2M_transcriptomics  A2ML1_transcriptomics  \\\n",
      "Sample_ID                                                                     \n",
      "S001                       4.02                13.39                   6.79   \n",
      "S002                       4.81                13.24                   6.33   \n",
      "S003                       6.24                14.32                   9.42   \n",
      "S005                       5.31                13.77                   4.22   \n",
      "S006                       9.84                13.12                   6.83   \n",
      "S007                       5.03                14.13                   7.75   \n",
      "S008                       3.17                13.49                   5.88   \n",
      "S009                       6.03                13.67                   6.69   \n",
      "S010                       6.02                14.30                   8.47   \n",
      "S011                       5.71                14.97                   9.96   \n",
      "S012                       5.17                13.40                   8.69   \n",
      "S014                       5.38                14.13                  11.45   \n",
      "S016                       5.86                13.15                   3.65   \n",
      "S017                       3.27                12.28                   6.97   \n",
      "S018                       4.65                13.42                  10.75   \n",
      "S019                       4.33                13.06                  13.60   \n",
      "S020                       4.58                15.05                   8.90   \n",
      "S021                       4.60                12.69                   9.56   \n",
      "S022                       4.90                14.13                   5.44   \n",
      "S023                       4.03                13.07                   4.46   \n",
      "S024                       4.62                14.12                  11.77   \n",
      "S025                       5.46                13.98                   3.94   \n",
      "S026                       5.73                14.70                   8.07   \n",
      "S027                       4.13                13.13                   4.68   \n",
      "S028                       6.08                15.56                   5.92   \n",
      "S029                       5.24                14.48                   5.87   \n",
      "S030                       7.36                14.15                   8.97   \n",
      "S031                       5.73                13.36                   5.60   \n",
      "S032                       5.31                14.65                   5.02   \n",
      "S033                       4.00                14.71                  11.18   \n",
      "...                         ...                  ...                    ...   \n",
      "S073                       5.81                12.53                   5.67   \n",
      "S074                       5.63                14.82                   6.47   \n",
      "S075                       4.28                14.31                  10.19   \n",
      "S076                       5.52                13.79                   9.97   \n",
      "S077                       7.17                13.57                   6.05   \n",
      "S078                       6.11                13.75                   6.29   \n",
      "S079                       5.75                13.96                   4.20   \n",
      "S080                       5.55                14.00                   6.36   \n",
      "S081                       6.09                13.94                   5.76   \n",
      "S082                       5.82                14.34                   3.82   \n",
      "S083                       4.21                13.18                   8.37   \n",
      "S084                       4.62                14.03                   8.78   \n",
      "S085                       5.83                13.78                   6.26   \n",
      "S086                       5.11                13.74                  10.26   \n",
      "S087                       5.41                13.50                   7.49   \n",
      "S088                       6.65                13.37                   5.12   \n",
      "S090                       7.44                14.90                   5.22   \n",
      "S091                       3.79                15.16                   9.16   \n",
      "S092                       6.94                13.08                   6.25   \n",
      "S093                       4.95                14.86                   9.54   \n",
      "S094                       5.34                13.84                   7.95   \n",
      "S095                       5.22                14.73                   7.26   \n",
      "S096                       5.63                13.96                  11.41   \n",
      "S097                       7.63                13.49                   6.43   \n",
      "S098                       4.85                12.88                   8.45   \n",
      "S099                       4.84                13.37                   5.90   \n",
      "S100                       4.80                13.40                   5.12   \n",
      "S101                       5.95                14.23                  11.96   \n",
      "S102                       5.25                13.75                   2.06   \n",
      "S103                       6.79                14.96                   7.50   \n",
      "\n",
      "           A4GALT_transcriptomics  AAAS_transcriptomics  AACS_transcriptomics  \\\n",
      "Sample_ID                                                                       \n",
      "S001                        10.34                 10.26                 10.53   \n",
      "S002                        10.83                 10.44                 10.45   \n",
      "S003                        10.98                  9.24                 11.29   \n",
      "S005                         8.68                  9.67                  9.85   \n",
      "S006                        11.42                  9.87                  9.98   \n",
      "S007                         9.61                 10.17                 10.48   \n",
      "S008                         9.71                 10.23                 10.74   \n",
      "S009                         8.78                 10.21                  9.66   \n",
      "S010                         9.98                  9.65                 11.00   \n",
      "S011                        10.22                 10.30                 10.71   \n",
      "S012                        11.42                  9.58                 10.71   \n",
      "S014                        10.20                 10.11                 10.21   \n",
      "S016                         8.37                 10.33                  9.17   \n",
      "S017                        10.04                 10.11                 12.82   \n",
      "S018                        10.05                 10.51                 10.58   \n",
      "S019                        10.04                 10.16                 10.71   \n",
      "S020                        10.71                  9.74                 10.27   \n",
      "S021                         9.60                 10.48                 11.59   \n",
      "S022                         9.49                 10.37                 10.52   \n",
      "S023                         7.10                 10.69                 10.37   \n",
      "S024                        10.29                 10.27                 10.55   \n",
      "S025                        10.04                 10.34                 10.53   \n",
      "S026                         9.90                 10.52                 10.56   \n",
      "S027                        10.74                 10.06                 11.01   \n",
      "S028                        10.17                 10.24                  9.84   \n",
      "S029                        10.02                 10.00                 10.53   \n",
      "S030                        10.23                 10.33                 10.77   \n",
      "S031                        10.06                  9.63                 10.02   \n",
      "S032                        10.22                  9.87                 10.56   \n",
      "S033                        10.31                  9.84                 10.02   \n",
      "...                           ...                   ...                   ...   \n",
      "S073                        10.80                  9.90                  9.65   \n",
      "S074                         9.79                 10.07                 10.52   \n",
      "S075                        10.64                 10.06                 11.03   \n",
      "S076                        10.36                 10.52                 10.50   \n",
      "S077                        10.77                 10.55                 10.96   \n",
      "S078                         8.92                 10.21                 10.42   \n",
      "S079                        10.74                 10.04                 10.73   \n",
      "S080                         9.98                 10.35                 10.56   \n",
      "S081                         9.82                 10.40                 10.76   \n",
      "S082                         8.77                  9.93                  9.10   \n",
      "S083                        10.74                  9.82                 11.10   \n",
      "S084                        10.43                 10.23                 11.20   \n",
      "S085                        10.09                  9.88                 11.11   \n",
      "S086                         9.21                 10.39                 11.24   \n",
      "S087                        11.24                 10.08                 10.51   \n",
      "S088                        10.17                 10.18                 10.56   \n",
      "S090                         9.63                 10.73                  9.64   \n",
      "S091                         8.98                 10.67                  9.66   \n",
      "S092                        10.28                 10.90                 10.96   \n",
      "S093                        10.81                 10.15                 10.97   \n",
      "S094                         9.17                 10.10                  9.77   \n",
      "S095                         8.77                 10.41                 10.31   \n",
      "S096                         7.80                 10.44                 10.96   \n",
      "S097                         8.99                 10.48                 10.05   \n",
      "S098                         9.64                 10.20                  9.74   \n",
      "S099                         9.84                 10.76                 10.97   \n",
      "S100                        10.64                 10.35                 11.02   \n",
      "S101                         8.05                 10.03                 10.13   \n",
      "S102                         9.87                 10.08                  9.34   \n",
      "S103                         8.83                 10.29                 10.14   \n",
      "\n",
      "           AADAT_transcriptomics  AAED1_transcriptomics  \\\n",
      "Sample_ID                                                 \n",
      "S001                        9.23                   7.68   \n",
      "S002                       10.45                   7.30   \n",
      "S003                        6.66                   8.51   \n",
      "S005                       10.05                   8.42   \n",
      "S006                        5.95                   8.82   \n",
      "S007                        8.67                   8.58   \n",
      "S008                        8.65                   7.79   \n",
      "S009                        8.61                   7.34   \n",
      "S010                        8.38                   8.44   \n",
      "S011                        9.19                   8.14   \n",
      "S012                        9.06                   7.99   \n",
      "S014                        7.75                   7.29   \n",
      "S016                        6.68                   7.94   \n",
      "S017                        9.95                   7.47   \n",
      "S018                        9.02                   7.64   \n",
      "S019                        9.09                   8.08   \n",
      "S020                        8.05                   8.21   \n",
      "S021                        8.47                   7.33   \n",
      "S022                        8.79                   8.24   \n",
      "S023                        9.13                   6.43   \n",
      "S024                        8.75                   7.82   \n",
      "S025                       10.26                   7.87   \n",
      "S026                        8.98                   8.28   \n",
      "S027                        9.00                   7.33   \n",
      "S028                        8.54                   8.50   \n",
      "S029                        8.23                   8.45   \n",
      "S030                        9.53                   7.87   \n",
      "S031                        9.40                   7.75   \n",
      "S032                        8.36                   8.61   \n",
      "S033                        9.95                   7.63   \n",
      "...                          ...                    ...   \n",
      "S073                        8.88                   7.82   \n",
      "S074                        9.43                   8.46   \n",
      "S075                        8.94                   7.98   \n",
      "S076                        8.92                   8.71   \n",
      "S077                        8.94                   7.46   \n",
      "S078                        9.52                   6.35   \n",
      "S079                        8.92                   8.53   \n",
      "S080                        8.98                   8.12   \n",
      "S081                       10.38                   7.09   \n",
      "S082                        7.63                   7.21   \n",
      "S083                        8.26                   8.15   \n",
      "S084                        9.32                   7.36   \n",
      "S085                        9.25                   7.98   \n",
      "S086                       10.13                   7.91   \n",
      "S087                        8.69                   7.31   \n",
      "S088                       10.84                   8.75   \n",
      "S090                        8.91                   8.38   \n",
      "S091                        9.37                   7.59   \n",
      "S092                        8.77                   8.28   \n",
      "S093                        8.78                   8.06   \n",
      "S094                        8.58                   7.35   \n",
      "S095                        8.34                   8.43   \n",
      "S096                        7.49                   7.51   \n",
      "S097                        8.25                   7.75   \n",
      "S098                        7.44                   7.11   \n",
      "S099                        9.96                   7.11   \n",
      "S100                        8.81                   7.34   \n",
      "S101                        6.93                   7.67   \n",
      "S102                        6.04                   8.91   \n",
      "S103                        8.14                   8.15   \n",
      "\n",
      "           AAGAB_transcriptomics  AAK1_transcriptomics  ...  \\\n",
      "Sample_ID                                               ...   \n",
      "S001                       11.42                 10.91  ...   \n",
      "S002                       12.22                 10.48  ...   \n",
      "S003                       11.58                 10.71  ...   \n",
      "S005                       11.07                 11.06  ...   \n",
      "S006                       11.23                 12.08  ...   \n",
      "S007                       11.01                 12.47  ...   \n",
      "S008                       11.00                 11.68  ...   \n",
      "S009                       11.28                 12.13  ...   \n",
      "S010                       11.06                 12.37  ...   \n",
      "S011                       10.70                 11.80  ...   \n",
      "S012                       11.62                 11.42  ...   \n",
      "S014                       11.02                 11.86  ...   \n",
      "S016                       10.96                 12.31  ...   \n",
      "S017                       11.07                 11.47  ...   \n",
      "S018                       11.37                 11.98  ...   \n",
      "S019                       11.51                 12.07  ...   \n",
      "S020                       11.44                 11.82  ...   \n",
      "S021                       10.67                 11.26  ...   \n",
      "S022                       11.39                 11.73  ...   \n",
      "S023                       11.08                 11.83  ...   \n",
      "S024                       11.18                 11.25  ...   \n",
      "S025                       11.03                 12.07  ...   \n",
      "S026                       11.04                 11.41  ...   \n",
      "S027                       11.41                 11.29  ...   \n",
      "S028                       10.75                 12.20  ...   \n",
      "S029                       11.39                 11.93  ...   \n",
      "S030                       11.52                 11.83  ...   \n",
      "S031                       11.09                 12.06  ...   \n",
      "S032                       11.25                 11.28  ...   \n",
      "S033                       10.86                 12.26  ...   \n",
      "...                          ...                   ...  ...   \n",
      "S073                       11.50                 10.49  ...   \n",
      "S074                       10.93                 12.14  ...   \n",
      "S075                       11.34                 12.20  ...   \n",
      "S076                       10.68                 12.04  ...   \n",
      "S077                       11.39                 11.16  ...   \n",
      "S078                       10.90                 12.01  ...   \n",
      "S079                       11.19                 11.79  ...   \n",
      "S080                       11.42                 11.06  ...   \n",
      "S081                       10.49                 11.86  ...   \n",
      "S082                       10.74                 12.31  ...   \n",
      "S083                       10.74                 11.66  ...   \n",
      "S084                       11.02                 11.45  ...   \n",
      "S085                       10.80                 11.81  ...   \n",
      "S086                       10.51                 11.08  ...   \n",
      "S087                       10.83                 11.51  ...   \n",
      "S088                       11.29                 11.22  ...   \n",
      "S090                       10.80                 11.21  ...   \n",
      "S091                       10.89                 11.40  ...   \n",
      "S092                       10.97                 11.03  ...   \n",
      "S093                       10.91                 11.66  ...   \n",
      "S094                       10.95                 11.77  ...   \n",
      "S095                       10.55                 12.17  ...   \n",
      "S096                       11.21                 12.30  ...   \n",
      "S097                       10.80                 11.75  ...   \n",
      "S098                       10.95                 11.67  ...   \n",
      "S099                       11.42                 11.46  ...   \n",
      "S100                       10.85                 11.24  ...   \n",
      "S101                       10.98                 12.26  ...   \n",
      "S102                       11.50                 12.49  ...   \n",
      "S103                       10.83                 12.74  ...   \n",
      "\n",
      "           AKAP10_transcriptomics  AKAP11_transcriptomics  \\\n",
      "Sample_ID                                                   \n",
      "S001                        10.64                   12.51   \n",
      "S002                        10.77                   10.66   \n",
      "S003                        10.83                   12.16   \n",
      "S005                        11.23                   12.24   \n",
      "S006                        10.84                   11.43   \n",
      "S007                        10.66                   12.94   \n",
      "S008                        10.83                   12.42   \n",
      "S009                        11.04                   12.57   \n",
      "S010                        11.01                   13.01   \n",
      "S011                        10.63                   12.29   \n",
      "S012                        11.17                   12.28   \n",
      "S014                        10.66                   12.46   \n",
      "S016                        11.52                   11.59   \n",
      "S017                        10.91                   12.37   \n",
      "S018                        10.67                   12.66   \n",
      "S019                        10.72                   12.14   \n",
      "S020                        10.83                   12.09   \n",
      "S021                        10.74                   12.51   \n",
      "S022                        10.75                   12.31   \n",
      "S023                        10.86                   12.13   \n",
      "S024                        10.73                   11.83   \n",
      "S025                        10.67                   12.49   \n",
      "S026                        10.82                   12.16   \n",
      "S027                        11.12                   12.89   \n",
      "S028                        10.73                   12.46   \n",
      "S029                        10.76                   12.09   \n",
      "S030                        10.74                   12.09   \n",
      "S031                        10.42                   11.10   \n",
      "S032                        10.72                   11.88   \n",
      "S033                        10.87                   12.93   \n",
      "...                           ...                     ...   \n",
      "S073                        11.07                   12.70   \n",
      "S074                        10.68                   12.46   \n",
      "S075                        10.84                   13.12   \n",
      "S076                        10.90                   12.63   \n",
      "S077                        10.74                   11.85   \n",
      "S078                        10.79                   12.69   \n",
      "S079                        10.83                   12.29   \n",
      "S080                        10.91                   12.29   \n",
      "S081                        10.83                   12.48   \n",
      "S082                        11.26                   11.99   \n",
      "S083                        10.55                   12.25   \n",
      "S084                        10.56                   12.31   \n",
      "S085                        10.79                   12.22   \n",
      "S086                        10.57                   11.85   \n",
      "S087                        10.89                   12.19   \n",
      "S088                        10.69                   11.81   \n",
      "S090                        10.99                   12.34   \n",
      "S091                        10.82                   12.30   \n",
      "S092                        10.65                   11.58   \n",
      "S093                        10.81                   12.54   \n",
      "S094                        11.18                   12.71   \n",
      "S095                        10.50                   12.29   \n",
      "S096                        10.35                   12.40   \n",
      "S097                        10.97                   12.10   \n",
      "S098                        11.16                   12.41   \n",
      "S099                        10.57                   11.92   \n",
      "S100                        10.34                   11.01   \n",
      "S101                        10.85                   12.73   \n",
      "S102                        10.23                   11.72   \n",
      "S103                        11.22                   12.32   \n",
      "\n",
      "           AKAP12_transcriptomics  AKAP13_transcriptomics  \\\n",
      "Sample_ID                                                   \n",
      "S001                        11.06                   13.26   \n",
      "S002                        10.05                   13.23   \n",
      "S003                        10.65                   13.39   \n",
      "S005                        10.18                   13.24   \n",
      "S006                         8.21                   13.06   \n",
      "S007                        12.28                   13.41   \n",
      "S008                        10.71                   13.68   \n",
      "S009                        12.59                   13.26   \n",
      "S010                        10.07                   14.13   \n",
      "S011                        12.11                   13.72   \n",
      "S012                         9.81                   13.62   \n",
      "S014                        13.01                   14.55   \n",
      "S016                        13.23                   12.94   \n",
      "S017                         9.19                   12.92   \n",
      "S018                        10.75                   14.05   \n",
      "S019                        12.27                   13.97   \n",
      "S020                        12.09                   14.08   \n",
      "S021                         8.83                   13.54   \n",
      "S022                        11.30                   14.21   \n",
      "S023                        14.64                   13.08   \n",
      "S024                        10.63                   13.70   \n",
      "S025                        10.78                   13.66   \n",
      "S026                        11.97                   13.79   \n",
      "S027                        11.60                   13.43   \n",
      "S028                        13.25                   14.62   \n",
      "S029                        10.61                   14.01   \n",
      "S030                        11.95                   13.81   \n",
      "S031                        12.48                   13.69   \n",
      "S032                        10.85                   13.65   \n",
      "S033                        12.17                   14.09   \n",
      "...                           ...                     ...   \n",
      "S073                        11.14                   13.86   \n",
      "S074                        11.51                   13.58   \n",
      "S075                        10.91                   13.77   \n",
      "S076                        12.62                   15.20   \n",
      "S077                        13.69                   13.36   \n",
      "S078                        11.39                   14.24   \n",
      "S079                        11.64                   13.49   \n",
      "S080                        11.17                   13.33   \n",
      "S081                        11.93                   14.24   \n",
      "S082                        13.21                   13.61   \n",
      "S083                        11.92                   13.53   \n",
      "S084                        11.30                   13.72   \n",
      "S085                        10.69                   13.32   \n",
      "S086                        10.69                   13.50   \n",
      "S087                         9.72                   13.64   \n",
      "S088                        10.28                   13.18   \n",
      "S090                        11.43                   13.69   \n",
      "S091                        10.67                   13.67   \n",
      "S092                        12.36                   13.79   \n",
      "S093                        11.90                   13.64   \n",
      "S094                        10.55                   13.63   \n",
      "S095                        12.19                   13.68   \n",
      "S096                        11.15                   13.91   \n",
      "S097                        13.22                   13.75   \n",
      "S098                        13.28                   14.05   \n",
      "S099                        12.20                   13.78   \n",
      "S100                        10.85                   12.76   \n",
      "S101                        11.21                   13.70   \n",
      "S102                        15.11                   14.49   \n",
      "S103                        12.59                   13.81   \n",
      "\n",
      "           AKAP17A_transcriptomics  AKAP5_transcriptomics  \\\n",
      "Sample_ID                                                   \n",
      "S001                         10.33                   8.89   \n",
      "S002                         10.88                   8.62   \n",
      "S003                         10.58                   8.28   \n",
      "S005                         10.00                   8.41   \n",
      "S006                         11.29                   9.15   \n",
      "S007                         10.46                   9.61   \n",
      "S008                         11.13                   8.90   \n",
      "S009                         10.19                   9.53   \n",
      "S010                         10.62                  10.29   \n",
      "S011                         10.92                   8.95   \n",
      "S012                         10.90                   8.73   \n",
      "S014                         10.89                   8.70   \n",
      "S016                         10.40                   9.05   \n",
      "S017                         11.19                   9.09   \n",
      "S018                         10.83                   8.67   \n",
      "S019                         10.35                   8.84   \n",
      "S020                         10.54                   8.55   \n",
      "S021                         11.48                   8.85   \n",
      "S022                         11.07                   9.26   \n",
      "S023                         10.81                   8.95   \n",
      "S024                         10.88                   8.61   \n",
      "S025                         11.21                   8.97   \n",
      "S026                         10.74                   8.78   \n",
      "S027                         11.11                   8.71   \n",
      "S028                         10.73                   9.17   \n",
      "S029                         11.26                   9.06   \n",
      "S030                         10.52                   9.14   \n",
      "S031                         10.91                   8.33   \n",
      "S032                         10.79                   8.55   \n",
      "S033                         10.36                   9.67   \n",
      "...                            ...                    ...   \n",
      "S073                         10.34                   8.58   \n",
      "S074                         10.60                   9.14   \n",
      "S075                         10.82                   8.61   \n",
      "S076                         11.13                   8.74   \n",
      "S077                         11.13                   8.62   \n",
      "S078                         11.18                   9.73   \n",
      "S079                         10.72                   8.47   \n",
      "S080                         10.70                   8.41   \n",
      "S081                         10.85                   9.17   \n",
      "S082                         10.34                  10.13   \n",
      "S083                         11.53                   8.85   \n",
      "S084                         11.01                   8.74   \n",
      "S085                         11.10                   9.32   \n",
      "S086                         11.12                   8.76   \n",
      "S087                         11.25                   9.04   \n",
      "S088                         10.64                   9.03   \n",
      "S090                         10.85                   8.77   \n",
      "S091                         10.81                   9.14   \n",
      "S092                         10.86                   9.37   \n",
      "S093                         10.85                   8.82   \n",
      "S094                         11.06                   9.41   \n",
      "S095                         11.36                   8.71   \n",
      "S096                         11.02                   9.56   \n",
      "S097                         10.77                   8.94   \n",
      "S098                         11.50                   9.01   \n",
      "S099                         11.26                   8.85   \n",
      "S100                         10.86                   8.57   \n",
      "S101                         10.53                   9.47   \n",
      "S102                         11.06                   9.80   \n",
      "S103                         10.72                   9.30   \n",
      "\n",
      "           AKAP7_transcriptomics  AKAP8_transcriptomics  \\\n",
      "Sample_ID                                                 \n",
      "S001                        9.59                   9.94   \n",
      "S002                       11.04                  10.67   \n",
      "S003                        8.56                  10.16   \n",
      "S005                        9.46                  10.56   \n",
      "S006                        9.42                  11.31   \n",
      "S007                        7.58                  10.19   \n",
      "S008                        9.26                  10.40   \n",
      "S009                        9.50                  11.49   \n",
      "S010                        8.71                  10.55   \n",
      "S011                        8.67                  10.30   \n",
      "S012                       10.52                  10.36   \n",
      "S014                        9.26                  10.38   \n",
      "S016                        8.44                  10.79   \n",
      "S017                        9.98                  10.82   \n",
      "S018                        7.71                  10.26   \n",
      "S019                        7.46                  10.71   \n",
      "S020                        8.02                  10.33   \n",
      "S021                        8.88                  10.74   \n",
      "S022                        9.09                  10.25   \n",
      "S023                        8.08                  10.10   \n",
      "S024                        8.91                  10.56   \n",
      "S025                        8.76                  10.59   \n",
      "S026                        9.21                  10.24   \n",
      "S027                        9.66                  10.10   \n",
      "S028                        8.14                  10.57   \n",
      "S029                        8.54                  10.41   \n",
      "S030                       10.19                  10.40   \n",
      "S031                        9.50                  10.66   \n",
      "S032                        8.73                  10.35   \n",
      "S033                        9.82                  10.56   \n",
      "...                          ...                    ...   \n",
      "S073                        9.70                  10.25   \n",
      "S074                        8.87                  10.42   \n",
      "S075                       10.12                  10.21   \n",
      "S076                        8.24                  10.31   \n",
      "S077                        9.89                  10.59   \n",
      "S078                       10.43                  10.55   \n",
      "S079                       10.01                  10.59   \n",
      "S080                        9.45                  10.67   \n",
      "S081                        8.11                  10.72   \n",
      "S082                        8.29                  10.91   \n",
      "S083                        9.80                  10.58   \n",
      "S084                        9.04                  10.53   \n",
      "S085                        9.85                  10.47   \n",
      "S086                       10.09                  10.44   \n",
      "S087                        9.86                  10.34   \n",
      "S088                        9.37                  10.10   \n",
      "S090                        8.86                  10.60   \n",
      "S091                        8.77                  10.74   \n",
      "S092                        9.43                  10.72   \n",
      "S093                        9.93                  10.41   \n",
      "S094                        9.71                  10.42   \n",
      "S095                        8.57                  10.38   \n",
      "S096                        8.30                  10.37   \n",
      "S097                        8.14                   9.68   \n",
      "S098                        8.54                  10.16   \n",
      "S099                        9.65                  10.43   \n",
      "S100                        8.52                  10.33   \n",
      "S101                        9.05                  10.74   \n",
      "S102                        7.61                  10.62   \n",
      "S103                        8.71                  10.36   \n",
      "\n",
      "           AKAP8L_transcriptomics     Label  \n",
      "Sample_ID                                    \n",
      "S001                        10.72   Mutated  \n",
      "S002                        11.04   Mutated  \n",
      "S003                        10.45  Wildtype  \n",
      "S005                        10.49  Wildtype  \n",
      "S006                        11.52  Wildtype  \n",
      "S007                        10.73  Wildtype  \n",
      "S008                        11.14   Mutated  \n",
      "S009                        12.17  Wildtype  \n",
      "S010                        10.95  Wildtype  \n",
      "S011                        10.83  Wildtype  \n",
      "S012                        11.50   Mutated  \n",
      "S014                        10.60   Mutated  \n",
      "S016                        11.42  Wildtype  \n",
      "S017                        11.09  Wildtype  \n",
      "S018                        10.27   Mutated  \n",
      "S019                        10.88   Mutated  \n",
      "S020                        10.70  Wildtype  \n",
      "S021                        11.37  Wildtype  \n",
      "S022                        11.28   Mutated  \n",
      "S023                        11.48   Mutated  \n",
      "S024                        10.92   Mutated  \n",
      "S025                        11.32  Wildtype  \n",
      "S026                        10.89   Mutated  \n",
      "S027                        10.81   Mutated  \n",
      "S028                        10.90  Wildtype  \n",
      "S029                        10.91  Wildtype  \n",
      "S030                        10.95   Mutated  \n",
      "S031                        11.63  Wildtype  \n",
      "S032                        10.95  Wildtype  \n",
      "S033                        10.99  Wildtype  \n",
      "...                           ...       ...  \n",
      "S073                        10.62   Mutated  \n",
      "S074                        10.78  Wildtype  \n",
      "S075                        11.09  Wildtype  \n",
      "S076                        10.82  Wildtype  \n",
      "S077                        10.92   Mutated  \n",
      "S078                        10.99   Mutated  \n",
      "S079                        10.99   Mutated  \n",
      "S080                        11.12   Mutated  \n",
      "S081                        11.28  Wildtype  \n",
      "S082                        11.46  Wildtype  \n",
      "S083                        11.52   Mutated  \n",
      "S084                        10.98   Mutated  \n",
      "S085                        11.22   Mutated  \n",
      "S086                        11.82  Wildtype  \n",
      "S087                        11.43   Mutated  \n",
      "S088                        11.13  Wildtype  \n",
      "S090                        10.79   Mutated  \n",
      "S091                        10.84   Mutated  \n",
      "S092                        10.48   Mutated  \n",
      "S093                        10.83   Mutated  \n",
      "S094                        11.39   Mutated  \n",
      "S095                        10.76  Wildtype  \n",
      "S096                        11.32  Wildtype  \n",
      "S097                        10.66  Wildtype  \n",
      "S098                        11.01   Mutated  \n",
      "S099                        10.97   Mutated  \n",
      "S100                        11.12   Mutated  \n",
      "S101                        11.38  Wildtype  \n",
      "S102                        10.73  Wildtype  \n",
      "S103                        10.92  Wildtype  \n",
      "\n",
      "[95 rows x 301 columns]\n"
     ]
    }
   ],
   "source": [
    "print(protdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['A1BG', 'A2M', 'A2ML1', 'A4GALT', 'AAAS', 'AACS', 'AADAT', 'AAED1',\n",
      "       'AAGAB', 'AAK1',\n",
      "       ...\n",
      "       'AKAP1', 'AKAP10', 'AKAP11', 'AKAP12', 'AKAP13', 'AKAP17A', 'AKAP5',\n",
      "       'AKAP7', 'AKAP8', 'AKAP8L'],\n",
      "      dtype='object', name='idx', length=300)\n",
      "300\n",
      "['A2M', 'AACS', 'AAGAB', 'AAK1', 'AARS', 'AASS', 'AATF', 'ABAT', 'ABCB7', 'ABCE1', 'ABCF1', 'ABHD10', 'ABHD11', 'ABHD14A', 'ABHD14A-ACY1', 'ABHD16A', 'ABI3BP', 'ABLIM1', 'ABLIM3', 'ABRACL', 'ABRAXAS2', 'ACAA1', 'ACAA2', 'ACAD11', 'ACAD8', 'ACAD9', 'ACADM', 'ACADS', 'ACADSB', 'ACADVL', 'ACAP2', 'ACAT1', 'ACAT2', 'ACIN1', 'ACLY', 'ACO1', 'ACO2', 'ACOT11', 'ACOT13', 'ACOT4', 'ACOT7', 'ACOT8', 'ACOT9', 'ACOX1', 'ACOX3', 'ACP1', 'ACSF2', 'ACSL3', 'ACSL4', 'ACSL5', 'ACSM3', 'ACSS1', 'ACSS2', 'ACSS3', 'ACTA1', 'ACTB', 'ACTL6A', 'ACTN1', 'ACTN2', 'ACTN4', 'ACTR1A', 'ACTR1B', 'ACTR2', 'ACTR3', 'ACTR6', 'ACTR8', 'ACYP1', 'ACYP2', 'ADAM10', 'ADAR', 'ADD1', 'ADD3', 'ADH1B', 'ADH1C', 'ADH5', 'ADH7', 'ADI1', 'ADIRF', 'ADK', 'ADM', 'ADNP', 'ADPRHL2', 'ADRM1', 'ADSL', 'ADSS', 'ADSSL1', 'AFDN', 'AFF1', 'AFF4', 'AFG3L2', 'AFM', 'AGFG1', 'AGFG2', 'AGK', 'AGL', 'AGO2', 'AGPS', 'AHCTF1', 'AHCY', 'AHCYL1', 'AHCYL2', 'AHDC1', 'AHNAK', 'AHNAK2', 'AHSA1', 'AHSG', 'AIDA', 'AIF1', 'AIFM1', 'AIMP1', 'AIMP2', 'AIP', 'AK1', 'AK2', 'AK3', 'AK4', 'AKAP12']\n",
      "117\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "columns overlap but no suffix specified: Index(['ABHD14A-ACY1-K374_acetylproteomics'], dtype='object')",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-62-4c756f341649>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     23\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minteracting_proteins_in_omics_df\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 25\u001b[0;31m \u001b[0mprotdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0men\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend_mutations_to_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmutation_genes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mgene\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0momics\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_genes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0minteracting_proteins_in_omics_df\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     26\u001b[0m \u001b[0mprotdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/cptac/endometrial/__init__.py\u001b[0m in \u001b[0;36mappend_mutations_to_omics\u001b[0;34m(omics_df, mutation_genes, omics_genes, show_location)\u001b[0m\n\u001b[1;32m    219\u001b[0m     \u001b[0msomatic_mutation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_mutations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    220\u001b[0m     \u001b[0msample_status_map\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_sample_status_map\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 221\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mUtilities\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend_mutations_to_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmutation_genes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_genes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mshow_location\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msample_status_map\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    222\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    223\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mdefine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mterm\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/cptac/endometrial/utilities.py\u001b[0m in \u001b[0;36mappend_mutations_to_omics\u001b[0;34m(self, somatic_mutation, omics_df, mutation_genes, omics_genes, show_location, sample_status_map)\u001b[0m\n\u001b[1;32m    296\u001b[0m         \u001b[0mpandas\u001b[0m \u001b[0mDataFrame\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mThe\u001b[0m \u001b[0mmutations\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mspecified\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mappended\u001b[0m \u001b[0mto\u001b[0m \u001b[0mall\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mpart\u001b[0m \u001b[0mof\u001b[0m \u001b[0mthe\u001b[0m \u001b[0momics\u001b[0m \u001b[0mdataframe\u001b[0m\u001b[0;34m.\u001b[0m \u001b[0mEach\u001b[0m \u001b[0mlocation\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mmutation\u001b[0m \u001b[0mcell\u001b[0m \u001b[0mcontains\u001b[0m \u001b[0ma\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwhich\u001b[0m \u001b[0mcontains\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mone\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mmore\u001b[0m \u001b[0mlocation\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mmutation\u001b[0m \u001b[0mvalues\u001b[0m \u001b[0mcorresponding\u001b[0m \u001b[0mto\u001b[0m \u001b[0mthat\u001b[0m \u001b[0msample\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mthat\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0ma\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0mindicating\u001b[0m \u001b[0mthat\u001b[0m \u001b[0mthe\u001b[0m \u001b[0msample\u001b[0m \u001b[0mdidn\u001b[0m\u001b[0;31m'\u001b[0m\u001b[0mt\u001b[0m \u001b[0mhave\u001b[0m \u001b[0ma\u001b[0m \u001b[0mmutation\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mthat\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    297\u001b[0m         \"\"\"\n\u001b[0;32m--> 298\u001b[0;31m         \u001b[0momics\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_omics_from_str_or_list\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_genes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    299\u001b[0m         \u001b[0mmutations\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_mutations_from_str_or_list\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmutation_genes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    300\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/cptac/endometrial/utilities.py\u001b[0m in \u001b[0;36mget_omics_from_str_or_list\u001b[0;34m(self, omics_df, genes)\u001b[0m\n\u001b[1;32m     75\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_col_from_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     76\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseries\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindexes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbase\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mIndex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# If it's a str, or list or array-like of str of genes, feed it to the proper function\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 77\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_cols_from_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     78\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mgenes\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# If it's the default of None, rename columns and return the entire dataframe\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     79\u001b[0m             \u001b[0mreturn_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m'{}_{}'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Append dataframe name to end of each column header, to preserve info when we merge dataframes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/cptac/endometrial/utilities.py\u001b[0m in \u001b[0;36mget_cols_from_omics\u001b[0;34m(self, omics_df, genes)\u001b[0m\n\u001b[1;32m     58\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mgene\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m             \u001b[0mselected\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_col_from_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Get the columns for that gene from the dataframe\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 60\u001b[0;31m             \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mselected\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Otherwise, append the columns to our dataframe we'll return.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     61\u001b[0m         \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"{} for {} genes\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Name the dataframe!\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mjoin\u001b[0;34m(self, other, on, how, lsuffix, rsuffix, sort)\u001b[0m\n\u001b[1;32m   6813\u001b[0m         \u001b[0;31m# For SparseDataFrame's benefit\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6814\u001b[0m         return self._join_compat(other, on=on, how=how, lsuffix=lsuffix,\n\u001b[0;32m-> 6815\u001b[0;31m                                  rsuffix=rsuffix, sort=sort)\n\u001b[0m\u001b[1;32m   6816\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6817\u001b[0m     def _join_compat(self, other, on=None, how='left', lsuffix='', rsuffix='',\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_join_compat\u001b[0;34m(self, other, on, how, lsuffix, rsuffix, sort)\u001b[0m\n\u001b[1;32m   6828\u001b[0m             return merge(self, other, left_on=on, how=how,\n\u001b[1;32m   6829\u001b[0m                          \u001b[0mleft_index\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mon\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mright_index\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 6830\u001b[0;31m                          suffixes=(lsuffix, rsuffix), sort=sort)\n\u001b[0m\u001b[1;32m   6831\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6832\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mon\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mmerge\u001b[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[1;32m     46\u001b[0m                          \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindicator\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mindicator\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     47\u001b[0m                          validate=validate)\n\u001b[0;32m---> 48\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     49\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     50\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mget_result\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    550\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    551\u001b[0m         llabels, rlabels = items_overlap_with_suffix(ldata.items, lsuf,\n\u001b[0;32m--> 552\u001b[0;31m                                                      rdata.items, rsuf)\n\u001b[0m\u001b[1;32m    553\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    554\u001b[0m         \u001b[0mlindexers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mleft_indexer\u001b[0m\u001b[0;34m}\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mleft_indexer\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36mitems_overlap_with_suffix\u001b[0;34m(left, lsuffix, right, rsuffix)\u001b[0m\n\u001b[1;32m   1970\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mlsuffix\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mrsuffix\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1971\u001b[0m             raise ValueError('columns overlap but no suffix specified: '\n\u001b[0;32m-> 1972\u001b[0;31m                              '{rename}'.format(rename=to_rename))\n\u001b[0m\u001b[1;32m   1973\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1974\u001b[0m         \u001b[0;32mdef\u001b[0m \u001b[0mlrenamer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: columns overlap but no suffix specified: Index(['ABHD14A-ACY1-K374_acetylproteomics'], dtype='object')"
     ]
    }
   ],
   "source": [
    "proteomics = en.get_proteomics()\n",
    "proteins = proteomics.columns\n",
    "proteins = proteins[0:300]\n",
    "\n",
    "print(proteins)\n",
    "print(len(proteins))\n",
    "\n",
    "interacting_proteins_in_omics_df = []\n",
    "\n",
    "'''Only do comparisons on proteins in the omics dataframe'''\n",
    "for ip in proteins:\n",
    "    if omics.name == 'phosphoproteomics' or omics.name == 'acetylproteomics':\n",
    "        col_regex = ip + \"-.*\" # Build a regex to get all columns that match the gene\n",
    "    else:\n",
    "        col_regex = '^{}$'.format(ip)\n",
    "\n",
    "    selected = omics.filter(regex=col_regex)\n",
    "\n",
    "    if len(selected.columns) > 0:\n",
    "        interacting_proteins_in_omics_df.append(ip)\n",
    "\n",
    "print(interacting_proteins_in_omics_df)\n",
    "print(len(interacting_proteins_in_omics_df))\n",
    "\n",
    "protdf = en.append_mutations_to_omics(mutation_genes=[gene], omics_df=omics, omics_genes=interacting_proteins_in_omics_df)\n",
    "protdf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A1BG did not match any columns in phosphoproteomics dataframe. A1BG_phosphoproteomics column inserted, but filled with NaN.\n",
      "A2M did not match any columns in phosphoproteomics dataframe. A2M_phosphoproteomics column inserted, but filled with NaN.\n",
      "A2ML1 did not match any columns in phosphoproteomics dataframe. A2ML1_phosphoproteomics column inserted, but filled with NaN.\n",
      "A4GALT did not match any columns in phosphoproteomics dataframe. A4GALT_phosphoproteomics column inserted, but filled with NaN.\n",
      "AADAT did not match any columns in phosphoproteomics dataframe. AADAT_phosphoproteomics column inserted, but filled with NaN.\n",
      "AAMP did not match any columns in phosphoproteomics dataframe. AAMP_phosphoproteomics column inserted, but filled with NaN.\n",
      "AAR2 did not match any columns in phosphoproteomics dataframe. AAR2_phosphoproteomics column inserted, but filled with NaN.\n",
      "AARS2 did not match any columns in phosphoproteomics dataframe. AARS2_phosphoproteomics column inserted, but filled with NaN.\n",
      "AARSD1 did not match any columns in phosphoproteomics dataframe. AARSD1_phosphoproteomics column inserted, but filled with NaN.\n",
      "AASDHPPT did not match any columns in phosphoproteomics dataframe. AASDHPPT_phosphoproteomics column inserted, but filled with NaN.\n",
      "AASS did not match any columns in phosphoproteomics dataframe. AASS_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABAT did not match any columns in phosphoproteomics dataframe. ABAT_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABCA8 did not match any columns in phosphoproteomics dataframe. ABCA8_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABCB10 did not match any columns in phosphoproteomics dataframe. ABCB10_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABCB7 did not match any columns in phosphoproteomics dataframe. ABCB7_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABCB8 did not match any columns in phosphoproteomics dataframe. ABCB8_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABCC10 did not match any columns in phosphoproteomics dataframe. ABCC10_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABCD1 did not match any columns in phosphoproteomics dataframe. ABCD1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABCD4 did not match any columns in phosphoproteomics dataframe. ABCD4_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABCG1 did not match any columns in phosphoproteomics dataframe. ABCG1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD10 did not match any columns in phosphoproteomics dataframe. ABHD10_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD11 did not match any columns in phosphoproteomics dataframe. ABHD11_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD13 did not match any columns in phosphoproteomics dataframe. ABHD13_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD14A did not match any columns in phosphoproteomics dataframe. ABHD14A_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD14A-ACY1 did not match any columns in phosphoproteomics dataframe. ABHD14A-ACY1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD14B did not match any columns in phosphoproteomics dataframe. ABHD14B_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD17A did not match any columns in phosphoproteomics dataframe. ABHD17A_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD17B did not match any columns in phosphoproteomics dataframe. ABHD17B_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD17C did not match any columns in phosphoproteomics dataframe. ABHD17C_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD2 did not match any columns in phosphoproteomics dataframe. ABHD2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD3 did not match any columns in phosphoproteomics dataframe. ABHD3_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD4 did not match any columns in phosphoproteomics dataframe. ABHD4_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD5 did not match any columns in phosphoproteomics dataframe. ABHD5_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABHD6 did not match any columns in phosphoproteomics dataframe. ABHD6_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABO did not match any columns in phosphoproteomics dataframe. ABO_phosphoproteomics column inserted, but filled with NaN.\n",
      "ABRACL did not match any columns in phosphoproteomics dataframe. ABRACL_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACAA1 did not match any columns in phosphoproteomics dataframe. ACAA1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACAA2 did not match any columns in phosphoproteomics dataframe. ACAA2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACAD10 did not match any columns in phosphoproteomics dataframe. ACAD10_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACAD11 did not match any columns in phosphoproteomics dataframe. ACAD11_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACAD8 did not match any columns in phosphoproteomics dataframe. ACAD8_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACAD9 did not match any columns in phosphoproteomics dataframe. ACAD9_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACADL did not match any columns in phosphoproteomics dataframe. ACADL_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACADS did not match any columns in phosphoproteomics dataframe. ACADS_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACADSB did not match any columns in phosphoproteomics dataframe. ACADSB_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACAP1 did not match any columns in phosphoproteomics dataframe. ACAP1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACAT1 did not match any columns in phosphoproteomics dataframe. ACAT1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACAT2 did not match any columns in phosphoproteomics dataframe. ACAT2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACBD7 did not match any columns in phosphoproteomics dataframe. ACBD7_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACE did not match any columns in phosphoproteomics dataframe. ACE_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACO2 did not match any columns in phosphoproteomics dataframe. ACO2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACOT1 did not match any columns in phosphoproteomics dataframe. ACOT1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACOT11 did not match any columns in phosphoproteomics dataframe. ACOT11_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACOT13 did not match any columns in phosphoproteomics dataframe. ACOT13_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACOT2 did not match any columns in phosphoproteomics dataframe. ACOT2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACOT4 did not match any columns in phosphoproteomics dataframe. ACOT4_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACOT7 did not match any columns in phosphoproteomics dataframe. ACOT7_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACOT8 did not match any columns in phosphoproteomics dataframe. ACOT8_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACOX2 did not match any columns in phosphoproteomics dataframe. ACOX2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACP2 did not match any columns in phosphoproteomics dataframe. ACP2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACP5 did not match any columns in phosphoproteomics dataframe. ACP5_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACP6 did not match any columns in phosphoproteomics dataframe. ACP6_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACPP did not match any columns in phosphoproteomics dataframe. ACPP_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACSF2 did not match any columns in phosphoproteomics dataframe. ACSF2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACSF3 did not match any columns in phosphoproteomics dataframe. ACSF3_phosphoproteomics column inserted, but filled with NaN.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACSL1 did not match any columns in phosphoproteomics dataframe. ACSL1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACSM3 did not match any columns in phosphoproteomics dataframe. ACSM3_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACSS1 did not match any columns in phosphoproteomics dataframe. ACSS1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTA1 did not match any columns in phosphoproteomics dataframe. ACTA1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTA2 did not match any columns in phosphoproteomics dataframe. ACTA2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTB did not match any columns in phosphoproteomics dataframe. ACTB_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTG1 did not match any columns in phosphoproteomics dataframe. ACTG1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTG2 did not match any columns in phosphoproteomics dataframe. ACTG2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTL6B did not match any columns in phosphoproteomics dataframe. ACTL6B_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTL8 did not match any columns in phosphoproteomics dataframe. ACTL8_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTR1A did not match any columns in phosphoproteomics dataframe. ACTR1A_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTR1B did not match any columns in phosphoproteomics dataframe. ACTR1B_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTR3B did not match any columns in phosphoproteomics dataframe. ACTR3B_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTR5 did not match any columns in phosphoproteomics dataframe. ACTR5_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACTRT2 did not match any columns in phosphoproteomics dataframe. ACTRT2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACYP1 did not match any columns in phosphoproteomics dataframe. ACYP1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ACYP2 did not match any columns in phosphoproteomics dataframe. ACYP2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADA2 did not match any columns in phosphoproteomics dataframe. ADA2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAL did not match any columns in phosphoproteomics dataframe. ADAL_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAM15 did not match any columns in phosphoproteomics dataframe. ADAM15_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAM28 did not match any columns in phosphoproteomics dataframe. ADAM28_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAMDEC1 did not match any columns in phosphoproteomics dataframe. ADAMDEC1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAMTS1 did not match any columns in phosphoproteomics dataframe. ADAMTS1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAMTS15 did not match any columns in phosphoproteomics dataframe. ADAMTS15_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAMTS2 did not match any columns in phosphoproteomics dataframe. ADAMTS2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAMTS5 did not match any columns in phosphoproteomics dataframe. ADAMTS5_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAMTS8 did not match any columns in phosphoproteomics dataframe. ADAMTS8_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAMTSL3 did not match any columns in phosphoproteomics dataframe. ADAMTSL3_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAMTSL5 did not match any columns in phosphoproteomics dataframe. ADAMTSL5_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAP1 did not match any columns in phosphoproteomics dataframe. ADAP1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADAT2 did not match any columns in phosphoproteomics dataframe. ADAT2_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADCK1 did not match any columns in phosphoproteomics dataframe. ADCK1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADGRG3 did not match any columns in phosphoproteomics dataframe. ADGRG3_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADGRL4 did not match any columns in phosphoproteomics dataframe. ADGRL4_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADH1A did not match any columns in phosphoproteomics dataframe. ADH1A_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADH1C did not match any columns in phosphoproteomics dataframe. ADH1C_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADH4 did not match any columns in phosphoproteomics dataframe. ADH4_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADH7 did not match any columns in phosphoproteomics dataframe. ADH7_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADHFE1 did not match any columns in phosphoproteomics dataframe. ADHFE1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADI1 did not match any columns in phosphoproteomics dataframe. ADI1_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADIPOQ did not match any columns in phosphoproteomics dataframe. ADIPOQ_phosphoproteomics column inserted, but filled with NaN.\n",
      "ADIPOR1 did not match any columns in phosphoproteomics dataframe. ADIPOR1_phosphoproteomics column inserted, but filled with NaN.\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "columns overlap but no suffix specified: Index(['ACADM-S240_phosphoproteomics'], dtype='object')",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-8-c3e04a82d703>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0mphos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0men\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_phosphoproteomics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m \u001b[0mprotdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0men\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend_mutations_to_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmutation_genes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mgene\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mphos\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_genes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mproteins\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprotdf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/cptac/endometrial/__init__.py\u001b[0m in \u001b[0;36mappend_mutations_to_omics\u001b[0;34m(omics_df, mutation_genes, omics_genes, show_location)\u001b[0m\n\u001b[1;32m    219\u001b[0m     \u001b[0msomatic_mutation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_mutations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    220\u001b[0m     \u001b[0msample_status_map\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_sample_status_map\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 221\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mUtilities\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend_mutations_to_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmutation_genes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_genes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mshow_location\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msample_status_map\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    222\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    223\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mdefine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mterm\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/cptac/endometrial/utilities.py\u001b[0m in \u001b[0;36mappend_mutations_to_omics\u001b[0;34m(self, somatic_mutation, omics_df, mutation_genes, omics_genes, show_location, sample_status_map)\u001b[0m\n\u001b[1;32m    296\u001b[0m         \u001b[0mpandas\u001b[0m \u001b[0mDataFrame\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mThe\u001b[0m \u001b[0mmutations\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mspecified\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mappended\u001b[0m \u001b[0mto\u001b[0m \u001b[0mall\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mpart\u001b[0m \u001b[0mof\u001b[0m \u001b[0mthe\u001b[0m \u001b[0momics\u001b[0m \u001b[0mdataframe\u001b[0m\u001b[0;34m.\u001b[0m \u001b[0mEach\u001b[0m \u001b[0mlocation\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mmutation\u001b[0m \u001b[0mcell\u001b[0m \u001b[0mcontains\u001b[0m \u001b[0ma\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwhich\u001b[0m \u001b[0mcontains\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mone\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mmore\u001b[0m \u001b[0mlocation\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mmutation\u001b[0m \u001b[0mvalues\u001b[0m \u001b[0mcorresponding\u001b[0m \u001b[0mto\u001b[0m \u001b[0mthat\u001b[0m \u001b[0msample\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mthat\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0ma\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0mindicating\u001b[0m \u001b[0mthat\u001b[0m \u001b[0mthe\u001b[0m \u001b[0msample\u001b[0m \u001b[0mdidn\u001b[0m\u001b[0;31m'\u001b[0m\u001b[0mt\u001b[0m \u001b[0mhave\u001b[0m \u001b[0ma\u001b[0m \u001b[0mmutation\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mthat\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    297\u001b[0m         \"\"\"\n\u001b[0;32m--> 298\u001b[0;31m         \u001b[0momics\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_omics_from_str_or_list\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_genes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    299\u001b[0m         \u001b[0mmutations\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_mutations_from_str_or_list\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmutation_genes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    300\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/cptac/endometrial/utilities.py\u001b[0m in \u001b[0;36mget_omics_from_str_or_list\u001b[0;34m(self, omics_df, genes)\u001b[0m\n\u001b[1;32m     75\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_col_from_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     76\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseries\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindexes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbase\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mIndex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# If it's a str, or list or array-like of str of genes, feed it to the proper function\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 77\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_cols_from_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     78\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mgenes\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# If it's the default of None, rename columns and return the entire dataframe\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     79\u001b[0m             \u001b[0mreturn_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m'{}_{}'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Append dataframe name to end of each column header, to preserve info when we merge dataframes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/cptac/endometrial/utilities.py\u001b[0m in \u001b[0;36mget_cols_from_omics\u001b[0;34m(self, omics_df, genes)\u001b[0m\n\u001b[1;32m     58\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mgene\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m             \u001b[0mselected\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_col_from_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Get the columns for that gene from the dataframe\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 60\u001b[0;31m             \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mselected\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Otherwise, append the columns to our dataframe we'll return.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     61\u001b[0m         \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"{} for {} genes\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Name the dataframe!\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mjoin\u001b[0;34m(self, other, on, how, lsuffix, rsuffix, sort)\u001b[0m\n\u001b[1;32m   6813\u001b[0m         \u001b[0;31m# For SparseDataFrame's benefit\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6814\u001b[0m         return self._join_compat(other, on=on, how=how, lsuffix=lsuffix,\n\u001b[0;32m-> 6815\u001b[0;31m                                  rsuffix=rsuffix, sort=sort)\n\u001b[0m\u001b[1;32m   6816\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6817\u001b[0m     def _join_compat(self, other, on=None, how='left', lsuffix='', rsuffix='',\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_join_compat\u001b[0;34m(self, other, on, how, lsuffix, rsuffix, sort)\u001b[0m\n\u001b[1;32m   6828\u001b[0m             return merge(self, other, left_on=on, how=how,\n\u001b[1;32m   6829\u001b[0m                          \u001b[0mleft_index\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mon\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mright_index\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 6830\u001b[0;31m                          suffixes=(lsuffix, rsuffix), sort=sort)\n\u001b[0m\u001b[1;32m   6831\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6832\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mon\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mmerge\u001b[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[1;32m     46\u001b[0m                          \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindicator\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mindicator\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     47\u001b[0m                          validate=validate)\n\u001b[0;32m---> 48\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     49\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     50\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mget_result\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    550\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    551\u001b[0m         llabels, rlabels = items_overlap_with_suffix(ldata.items, lsuf,\n\u001b[0;32m--> 552\u001b[0;31m                                                      rdata.items, rsuf)\n\u001b[0m\u001b[1;32m    553\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    554\u001b[0m         \u001b[0mlindexers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mleft_indexer\u001b[0m\u001b[0;34m}\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mleft_indexer\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36mitems_overlap_with_suffix\u001b[0;34m(left, lsuffix, right, rsuffix)\u001b[0m\n\u001b[1;32m   1970\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mlsuffix\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mrsuffix\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1971\u001b[0m             raise ValueError('columns overlap but no suffix specified: '\n\u001b[0;32m-> 1972\u001b[0;31m                              '{rename}'.format(rename=to_rename))\n\u001b[0m\u001b[1;32m   1973\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1974\u001b[0m         \u001b[0;32mdef\u001b[0m \u001b[0mlrenamer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: columns overlap but no suffix specified: Index(['ACADM-S240_phosphoproteomics'], dtype='object')"
     ]
    }
   ],
   "source": [
    "import cptac.endometrial as en\n",
    "\n",
    "proteomics = en.get_proteomics()\n",
    "proteins = proteomics.columns\n",
    "proteins = proteins[0:220]\n",
    "gene = \"ARID1A\"\n",
    "phos = en.get_phosphoproteomics()\n",
    "\n",
    "protdf = en.append_mutations_to_omics(mutation_genes=[gene], omics_df=phos, omics_genes=proteins)\n",
    "print(protdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
