{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# WDFY4 Mutation Trans Effect on Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook analyzes the trans effect of WDFY4 mutation on interacting and other proteins Transcriptomics, in Endometrial, Colon, and Ovarian cancer."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Library Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "import cptac\n",
    "import cptac.algorithms as al\n",
    "\n",
    "en = cptac.Endometrial()\n",
    "co = cptac.Colon()\n",
    "ov = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = \"WDFY4\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Investigate Proteomics, Phosphoproteomics, Acetylproteomics, or Transcriptomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#omics = \"proteomics\"\n",
    "omics = \"transcriptomics\"\n",
    "#omics = \"phosphoproteomics\"\n",
    "#omics = \"acetylproteomics\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Track all significant comparisons in Dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_significant_comparisons = pd.DataFrame(columns=['Cancer_Type', 'Gene', 'Comparison','Interacting_Protein','P_Value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_to_all_significant_comparisons(df, cancer, interacting, all_sig_comp):\n",
    "    expanded = df\n",
    "    expanded['Gene'] = gene\n",
    "    expanded['Cancer_Type'] = cancer\n",
    "    expanded['Interacting_Protein'] = interacting\n",
    "    \n",
    "    updated_all_comparisons = pd.concat([all_sig_comp, expanded], sort=False)\n",
    "    \n",
    "    return updated_all_comparisons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Interacting Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate interacting protein list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a call to the cptac.algorithms get interacting proteins method, which interacts with the uniprot and string databases to generate a list of known interacting partners with the given gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Interacting Proteins:\n",
      "PLEK\n",
      "SPI1\n",
      "GABARAPL1\n",
      "GABARAPL2\n",
      "MAP1LC3B\n",
      "IBTK\n",
      "GABARAP\n",
      "WDFY4\n",
      "HERC2\n",
      "DCP1B\n",
      "LRRC18\n",
      "IKZF1\n",
      "WDR25\n",
      "MAP1LC3C\n",
      "PTPN22\n",
      "MPEG1\n",
      "MAP1LC3A\n",
      "RCCD1\n",
      "HERC4\n",
      "ITGB2\n",
      "CSF2RB\n",
      "HERC3\n",
      "HERC1\n",
      "MAP1LC3B2\n",
      "DCP1A\n",
      "FMN1\n",
      "CUL3\n"
     ]
    }
   ],
   "source": [
    "'''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "'''Show interacting protein list'''\n",
    "print(\"Interacting Proteins:\")\n",
    "for interacting_protein in interacting_proteins:\n",
    "    print(interacting_protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", True, all_significant_comparisons)\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WDFY4 did not match any columns in transcriptomics dataframe. WDFY4_transcriptomics column inserted, but filled with NaN.\n",
      "LRRC18 did not match any columns in transcriptomics dataframe. LRRC18_transcriptomics column inserted, but filled with NaN.\n",
      "MAP1LC3C did not match any columns in transcriptomics dataframe. MAP1LC3C_transcriptomics column inserted, but filled with NaN.\n",
      "MAP1LC3B2 did not match any columns in transcriptomics dataframe. MAP1LC3B2_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "                 Comparison   P_Value\n",
      "0  MAP1LC3A_transcriptomics  0.000196\n",
      "1      IBTK_transcriptomics  0.000646\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MPEG1 did not match any columns in transcriptomics dataframe. MPEG1_transcriptomics column inserted, but filled with NaN.\n",
      "RCCD1 did not match any columns in transcriptomics dataframe. RCCD1_transcriptomics column inserted, but filled with NaN.\n",
      "HERC3 did not match any columns in transcriptomics dataframe. HERC3_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Expand our search for significant comparisons to all proteins in our dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  WDFY4\n",
      "Doing t-test comparisons\n",
      "\n",
      "               Comparison       P_Value\n",
      "0  MBD3L3_transcriptomics  2.244869e-07\n",
      "1   GHRHR_transcriptomics  9.109165e-07\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  WDFY4\n",
      "Doing t-test comparisons\n",
      "\n",
      "                      Comparison       P_Value\n",
      "0           VIL1_transcriptomics  8.701471e-08\n",
      "1         DUSP16_transcriptomics  1.053527e-07\n",
      "2        SLC39A5_transcriptomics  2.464849e-07\n",
      "3          NR1I2_transcriptomics  3.336888e-07\n",
      "4          CDK20_transcriptomics  3.676987e-07\n",
      "5          PRSS8_transcriptomics  4.335563e-07\n",
      "6           STYX_transcriptomics  4.768290e-07\n",
      "7          DPEP1_transcriptomics  5.349111e-07\n",
      "8           CHML_transcriptomics  6.460697e-07\n",
      "9         ANTXR2_transcriptomics  8.853763e-07\n",
      "10          ZXDA_transcriptomics  1.074115e-06\n",
      "11        MOGAT3_transcriptomics  1.142925e-06\n",
      "12      PPP1R14D_transcriptomics  1.279212e-06\n",
      "13        HS3ST1_transcriptomics  1.349387e-06\n",
      "14     LINC01006_transcriptomics  1.554528e-06\n",
      "15         CDHR1_transcriptomics  1.575942e-06\n",
      "16           IHH_transcriptomics  1.743245e-06\n",
      "17      FLJ20021_transcriptomics  1.889396e-06\n",
      "18         GLOD5_transcriptomics  2.381289e-06\n",
      "19         MEP1A_transcriptomics  2.429942e-06\n",
      "20         AXIN2_transcriptomics  3.066553e-06\n",
      "21         AMACR_transcriptomics  3.435852e-06\n",
      "22  LOC102723354_transcriptomics  3.567232e-06\n",
      "23        SLC3A1_transcriptomics  3.578921e-06\n",
      "24        PMFBP1_transcriptomics  3.671909e-06\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  WDFY4\n",
      "Doing t-test comparisons\n",
      "\n",
      "                     Comparison       P_Value\n",
      "0       SNORA55_transcriptomics  7.257617e-09\n",
      "1  LOC101927406_transcriptomics  1.102317e-06\n",
      "2         TRPC3_transcriptomics  2.092095e-06\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(\"\\nGene: \", gene)\n",
    "\n",
    "'''Use all proteins'''\n",
    "\n",
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "protdf = protdf.loc[:,~protdf.columns.duplicated()]\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the datafram correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", False, all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Print All Signififant Comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Interacting_Protein</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>MAP1LC3A_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>1.962419e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>IBTK_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>6.463247e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>MBD3L3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.244869e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>GHRHR_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.109165e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>VIL1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.701471e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>DUSP16_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.053527e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>SLC39A5_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.464849e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>NR1I2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.336888e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>CDK20_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.676987e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>PRSS8_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.335563e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>STYX_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.768290e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>DPEP1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.349111e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>CHML_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.460697e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>ANTXR2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.853763e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>ZXDA_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.074115e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>MOGAT3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.142925e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>PPP1R14D_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.279212e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>HS3ST1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.349387e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>LINC01006_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.554528e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>CDHR1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.575942e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>IHH_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.743245e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>FLJ20021_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.889396e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>GLOD5_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.381289e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>MEP1A_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.429942e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>AXIN2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.066553e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>AMACR_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.435852e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>LOC102723354_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.567232e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>SLC3A1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.578921e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Colon</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>PMFBP1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.671909e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>SNORA55_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.257617e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>LOC101927406_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.102317e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>WDFY4</td>\n",
       "      <td>TRPC3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.092095e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Cancer_Type   Gene                    Comparison Interacting_Protein  \\\n",
       "0         Colon  WDFY4      MAP1LC3A_transcriptomics                True   \n",
       "1         Colon  WDFY4          IBTK_transcriptomics                True   \n",
       "0   Endometrial  WDFY4        MBD3L3_transcriptomics               False   \n",
       "1   Endometrial  WDFY4         GHRHR_transcriptomics               False   \n",
       "0         Colon  WDFY4          VIL1_transcriptomics               False   \n",
       "1         Colon  WDFY4        DUSP16_transcriptomics               False   \n",
       "2         Colon  WDFY4       SLC39A5_transcriptomics               False   \n",
       "3         Colon  WDFY4         NR1I2_transcriptomics               False   \n",
       "4         Colon  WDFY4         CDK20_transcriptomics               False   \n",
       "5         Colon  WDFY4         PRSS8_transcriptomics               False   \n",
       "6         Colon  WDFY4          STYX_transcriptomics               False   \n",
       "7         Colon  WDFY4         DPEP1_transcriptomics               False   \n",
       "8         Colon  WDFY4          CHML_transcriptomics               False   \n",
       "9         Colon  WDFY4        ANTXR2_transcriptomics               False   \n",
       "10        Colon  WDFY4          ZXDA_transcriptomics               False   \n",
       "11        Colon  WDFY4        MOGAT3_transcriptomics               False   \n",
       "12        Colon  WDFY4      PPP1R14D_transcriptomics               False   \n",
       "13        Colon  WDFY4        HS3ST1_transcriptomics               False   \n",
       "14        Colon  WDFY4     LINC01006_transcriptomics               False   \n",
       "15        Colon  WDFY4         CDHR1_transcriptomics               False   \n",
       "16        Colon  WDFY4           IHH_transcriptomics               False   \n",
       "17        Colon  WDFY4      FLJ20021_transcriptomics               False   \n",
       "18        Colon  WDFY4         GLOD5_transcriptomics               False   \n",
       "19        Colon  WDFY4         MEP1A_transcriptomics               False   \n",
       "20        Colon  WDFY4         AXIN2_transcriptomics               False   \n",
       "21        Colon  WDFY4         AMACR_transcriptomics               False   \n",
       "22        Colon  WDFY4  LOC102723354_transcriptomics               False   \n",
       "23        Colon  WDFY4        SLC3A1_transcriptomics               False   \n",
       "24        Colon  WDFY4        PMFBP1_transcriptomics               False   \n",
       "0       Ovarian  WDFY4       SNORA55_transcriptomics               False   \n",
       "1       Ovarian  WDFY4  LOC101927406_transcriptomics               False   \n",
       "2       Ovarian  WDFY4         TRPC3_transcriptomics               False   \n",
       "\n",
       "         P_Value  \n",
       "0   1.962419e-04  \n",
       "1   6.463247e-04  \n",
       "0   2.244869e-07  \n",
       "1   9.109165e-07  \n",
       "0   8.701471e-08  \n",
       "1   1.053527e-07  \n",
       "2   2.464849e-07  \n",
       "3   3.336888e-07  \n",
       "4   3.676987e-07  \n",
       "5   4.335563e-07  \n",
       "6   4.768290e-07  \n",
       "7   5.349111e-07  \n",
       "8   6.460697e-07  \n",
       "9   8.853763e-07  \n",
       "10  1.074115e-06  \n",
       "11  1.142925e-06  \n",
       "12  1.279212e-06  \n",
       "13  1.349387e-06  \n",
       "14  1.554528e-06  \n",
       "15  1.575942e-06  \n",
       "16  1.743245e-06  \n",
       "17  1.889396e-06  \n",
       "18  2.381289e-06  \n",
       "19  2.429942e-06  \n",
       "20  3.066553e-06  \n",
       "21  3.435852e-06  \n",
       "22  3.567232e-06  \n",
       "23  3.578921e-06  \n",
       "24  3.671909e-06  \n",
       "0   7.257617e-09  \n",
       "1   1.102317e-06  \n",
       "2   2.092095e-06  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if len(all_significant_comparisons) == 0:\n",
    "    print('No Significant Comparisons!')\n",
    "    \n",
    "else:\n",
    "    display(all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Write Significant Comparisons (if any) to Shared CSV file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "existing_results = pd.read_csv(gene+'_Trans_Results.csv')\n",
    "\n",
    "updated_results = pd.concat([existing_results, all_significant_comparisons], sort=False)\n",
    "\n",
    "updated_results.to_csv(path_or_buf = gene + '_Trans_Results.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
