{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "    \n",
    "#   Copyright 2018 Samuel Payne sam_payne@byu.edu\n",
    "#   Licensed under the Apache License, Version 2.0 (the \"License\");\n",
    "#   you may not use this file except in compliance with the License.\n",
    "#   You may obtain a copy of the License at\n",
    "#       http://www.apache.org/licenses/LICENSE-2.0\n",
    "#   Unless required by applicable law or agreed to in writing, software\n",
    "#   distributed under the License is distributed on an \"AS IS\" BASIS,\n",
    "#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
    "#   See the License for the specific language governing permissions and\n",
    "#   limitations under the License.\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import cptac.endometrial as en\n",
    "from utilities import Utilities\n",
    "\n",
    "def print_test_result(PASS):\n",
    "    \"\"\"Prints the result of a test, based on a bool.\n",
    "    Parameters:\n",
    "    PASS (bool): Whether or not the test passed.\n",
    "    \"\"\"\n",
    "    if PASS:\n",
    "        print('\\tPASS')\n",
    "    else:\n",
    "        print('\\tFAIL\\n')\n",
    "\n",
    "def check_returned_is_df(returned):\n",
    "    \"\"\"Checks that an object is a dataframe. Prints a specific message if it's actually None, or a general message if it's something else.\n",
    "    Parameters:\n",
    "    returned: The object to test\n",
    "    Returns:\n",
    "    bool: Indicates whether the object was a dataframe.\n",
    "    \"\"\"\n",
    "    if returned is None:\n",
    "        print(\"Function under test returned None.\")\n",
    "        return False\n",
    "    \n",
    "    if not isinstance(returned, pd.core.frame.DataFrame):\n",
    "        print(\"Returned object was not a dataframe. Type of object: {}\".format(type(returned)))\n",
    "        return False\n",
    "    return True\n",
    "\n",
    "def check_df_name(df, expected_name):\n",
    "    \"\"\"Checks that a dataframe has a \"name\" attribute, and that it has the proper value.\n",
    "    Parameters:\n",
    "    df (pandas.core.frame.DataFrame): The dataframe to test.\n",
    "    expected_name (str): The expected name of the dataframe.\n",
    "    Returns:\n",
    "    bool: Whether the dataframe had a name, and it was the correct name.\n",
    "    \"\"\"\n",
    "    # Check that the dataframe has a name\n",
    "    if not hasattr(df, 'name'):\n",
    "        print('Dataframe did not have a \"name\" attribute.')\n",
    "        return False\n",
    "\n",
    "    # Check that the dataframe has the correct name\n",
    "    if df.name != expected_name:\n",
    "        print(\"Dataframe had incorrect name.\\n\\tExpected: {}\\n\\tActual: {}\".format(expected_name, df.name))\n",
    "        return False\n",
    "    return True\n",
    "\n",
    "def check_df_shape(df, exp_shape):\n",
    "    \"\"\"Checks that a dataframe has the proper shape.\n",
    "    Parameters:\n",
    "    df (pandas.core.frame.DataFrame): The dataframe to test.\n",
    "    exp_shape (tuple): A tuple with two elements. First element is expected number of rows, second is expected number of columns.\n",
    "    Returns:\n",
    "    bool: Indicates whether the dataframe had the proper shape.\n",
    "    \"\"\"\n",
    "    act_shape = df.shape\n",
    "    if exp_shape != act_shape:\n",
    "        print(\"Dataframe dimensions did not match expected values.\\n\\tExpected: {}\\n\\tActual: {}\\n\".format(exp_shape, act_shape))\n",
    "        return False\n",
    "    return True\n",
    "\n",
    "def check_getter(df, exp_name, exp_dim, exp_headers, coordinates, values): \n",
    "    \"\"\"Test a dataframe's name, dimensions and headers, and three test values, then print whether it passed the test.\n",
    "    Parameters\n",
    "    df: the dataframe gotten by the getter we are testing\n",
    "    exp_name: string containing the expected name of the dataframe gotten by the getter we're testing\n",
    "    exp_dim: a tuple containing the expected dimensions of the dataframe, in the format (rows, columns)\n",
    "    exp_headers: if the dataframe has up to 20 columns, all of the headers for the dataframe, in order. If it has more than 20 columns, then a list containing the first ten and last ten headers, in order.\n",
    "    coordinates: a tuple with three elements, each element being a tuple with two elements, the first element being the int index of the row of a test value, and the second element being the int index of the column of a test value\n",
    "    values: a tuple with three elements, each element being the expected value of the test value corresponding to the coordinates at the same index in the coordinates parameter \n",
    "    Returns\n",
    "    bool indicating if the dataframe had the correct data.\n",
    "    \"\"\"\n",
    "    PASS = True\n",
    "\n",
    "    # Check that df is a dataframe, not None or something else.\n",
    "    if not check_returned_is_df(df):\n",
    "        return False # End test, because other tests will be useless.\n",
    "\n",
    "    # Check name\n",
    "    if not check_df_name(df, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Check dimensions\n",
    "    if not check_df_shape(df, exp_dim):\n",
    "        PASS = False\n",
    "\n",
    "    # Check headers\n",
    "    act_headers_all = list(df.columns.values)\n",
    "    if len(df.columns.values) <= 20:\n",
    "        act_headers = act_headers_all\n",
    "    else:\n",
    "        act_headers = act_headers_all[:10] + act_headers_all[-10:]\n",
    "\n",
    "    if len(exp_headers) != len(act_headers):\n",
    "        print(\"Unexpected number of test headers in dataframe. Expected number of headers: {}. You passed {} headers.\\n\".format(len(act_headers), len(exp_headers)))\n",
    "        PASS = False\n",
    "    else:\n",
    "        for i, header in enumerate(exp_headers):\n",
    "            if header != act_headers[i]:\n",
    "                print(\"Dataframe header did not match expected value.\\n\\tExpected: {}\\n\\tActual: {}\\n\".format(header, act_headers[i]))\n",
    "                PASS = False\n",
    "\n",
    "    # Check test values\n",
    "    act_values = [\n",
    "        df.iloc[coordinates[0][0], coordinates[0][1]],\n",
    "        df.iloc[coordinates[1][0], coordinates[1][1]],\n",
    "        df.iloc[coordinates[2][0], coordinates[2][1]]]\n",
    "\n",
    "    for i, value in enumerate(values):\n",
    "        if act_values[i] != value:\n",
    "            print(\"Dataframe value did not match expected value.\\n\\tColumn: {}\\n\\tIndex: {}\\n\\tExpected: {}\\n\\tActual: {}\\n\".format(df.columns.values[coordinates[i][1]], df.index.values[coordinates[i][0]], value, act_values[i]))\n",
    "            PASS = False\n",
    "\n",
    "    # Return whether the dataframe passed the test\n",
    "    return PASS\n",
    "\n",
    "def build_omics_regex(genes, suffix=\"\"):\n",
    "    \"\"\"Builds a regex from a list of genes to grab all columns corresponding to those genes from an omics dataframe. Optional suffix allows you to search acetylproteomics and phosphoproteomics dataframes.\n",
    "    Parameters:\n",
    "    genes (str or list): one gene as a string, or a list of the genes, as strings\n",
    "    suffix (str, optional): a suffix to put on the end of the regex, if needed. Default is empty string.\n",
    "    Returns:\n",
    "    str: regex to get the columns for those genes.\n",
    "    \"\"\"\n",
    "    regex = '^('\n",
    "    if isinstance(genes, str):\n",
    "        genes = [genes]\n",
    "    for gene in genes:\n",
    "        regex = regex + gene + '|'\n",
    "    regex = regex[:-1] + ')' + suffix + '$'\n",
    "    return regex\n",
    "\n",
    "def check_appended_columns(source_df, dest_df, source_headers, dest_headers=None):\n",
    "    \"\"\"Checks whether a column or list of columns appended to a dataframe have the same values for each index in that dataframe as they had in the dataframe they were taken from.\n",
    "    Parameters:\n",
    "    source_df (pandas.core.frame.DataFrame): The dataframe the columns were taken from.\n",
    "    dest_df (pandas.core.frame.DataFrame): The dataframe the columns were appended to (with them appended to it).\n",
    "    source_headers (str, or list or array-like of str): The header(s) of the columns to test in source_df. str if one, list or array-like of str if multiple. \n",
    "    dest_headers (str, or list or array-like of str, optional): The header(s) of the columns to test in dest_df. str if one, list or array-like of str if multiple. If provided, must be in the same order as their corresponding headers in source_headers. If not provided, header(s) in dest_df will be constructed by appending an underscore and source_df.name to each of the source_headers.\n",
    "    \n",
    "    Returns:\n",
    "    bool: Indicates whether the specified column(s) in dest_df had the same values for each index as they did in source_df.\n",
    "    \"\"\"\n",
    "    PASS = True\n",
    "    if isinstance(source_headers, str):\n",
    "        source_headers = [source_headers]\n",
    "\n",
    "    original_dest_headers = dest_headers # Copy for later, so we can continue running conditionals based on whether dest_headers was provided\n",
    "    if dest_headers is None:\n",
    "        dest_headers = [header + '_' + source_df.name for header in source_headers] # Construct what the source_headers will be in the merged dataframe\n",
    "    elif isinstance(dest_headers, str):\n",
    "        dest_headers = [dest_headers]\n",
    "\n",
    "    source_selection = source_df[source_headers] # Select all of the columns to test from the source dataframe\n",
    "    dest_selection = dest_df[dest_headers]\n",
    "\n",
    "    if original_dest_headers is None:\n",
    "        dest_selection = dest_selection.rename(columns=lambda x: x.split('_')[0])\n",
    "    else:\n",
    "        dest_rename_dict = dict(zip(dest_headers, source_headers)) # Create a dict mapping a dest header to its corresponding source header. This is why if dest_headers is provided, the headers must be in the same order as their corresponding headers in source_headers.\n",
    "        dest_selection = dest_selection.rename(columns=dest_rename_dict)\n",
    "\n",
    "    if not source_selection.equals(dest_selection):\n",
    "        PASS = False\n",
    "        diff_bools = source_selection.eq(dest_selection)\n",
    "        diff_cols = diff_bools.columns\n",
    "        for col in diff_cols:\n",
    "            if not diff_bools[col].all():\n",
    "                source_col = source_selection[col]\n",
    "                dest_col = dest_selection[col]\n",
    "                col_bools = source_col.eq(dest_col)\n",
    "                source_col_diffs = source_col[~col_bools]\n",
    "                dest_col_diffs = dest_col[~col_bools]\n",
    "                diffs_df = pd.DataFrame({\"Source\":source_col_diffs, \"Merged\":dest_col_diffs})\n",
    "                print(\"Merged dataframe had incorrect values. Column: {}\\nValues:\\n{}\\n\\n\".format(col, diffs_df))\n",
    "\n",
    "    return PASS\n",
    "\n",
    "def check_invalid_columns(invalid_cols, exp_num_invalid):\n",
    "    \"\"\"Check that we have the expected number of columns returned for an invalid key from an omics dataframe, and that they are all NaN.\n",
    "    \n",
    "    Parameters:\n",
    "    invalid_cols(pandas DataFrame): The invalid columns found in the merged dataframe.\n",
    "    exp_num_invalid (int): The expected number of invalid columns.\n",
    "    Returns:\n",
    "    bool: Indicates whether the test passed.\n",
    "    \"\"\"\n",
    "    PASS = True\n",
    "\n",
    "    # Check that we got the expected number of invalid columns\n",
    "    if len(invalid_cols.columns) != exp_num_invalid:\n",
    "        print(\"Unexpected number of columns from invalid key. Expected number: {} Actual number: {}\".format(exp_num_invalid, len(invalid_cols.columns)))\n",
    "        PASS = False\n",
    "    \n",
    "    # Check that they were all NaN\n",
    "    null_bools = invalid_cols.isnull()\n",
    "    if not null_bools.all().all():\n",
    "        print(\"Unexpected values in dataframe of columns for invalid keys. Expected all NaN. Actual:\\n{}\".format(invalid_cols))\n",
    "        PASS = False\n",
    "\n",
    "    if PASS:\n",
    "        print(\"(NOTE: The invalid key message above was expected.)\")\n",
    "\n",
    "    return PASS\n",
    "\n",
    "def check_mutation_columns(mutations, merged_df, genes, show_location=True):\n",
    "    \"\"\"\n",
    "    Parameters\n",
    "    mutations (pandas.core.frame.DataFrame): The somatic_mutation dataframe.\n",
    "    merged_df (pandas.core.frame.DataFrame): The merged dataframe.\n",
    "    gene (str or list): The gene(s) the mutation data was collected for. str if one, list of str if multiple.\n",
    "    show_location (bool, optional): Whether the location column was included in merged_df. Default is True.\n",
    "    Returns\n",
    "    bool: Indicates whether the mutation data for that gene and each sample in the merged dataframe matched the data in the somatic_mutation dataframe.\n",
    "    \"\"\"\n",
    "    PASS = True\n",
    "\n",
    "    # Set our column names for use later\n",
    "    gene_col = 'Gene'\n",
    "    sample_col = 'Sample_ID'\n",
    "    location_col = 'Location'\n",
    "    mutation_col = 'Mutation'\n",
    "    sample_status_col = 'Sample_Status'\n",
    "    mutation_status_col = \"Mutation_Status\"\n",
    "\n",
    "    # Get a map of sample IDs to sample statuses, so we can check the Sample_Status column\n",
    "    sample_status_map = en.get_sample_status_map()\n",
    "\n",
    "    if isinstance(genes, str):\n",
    "        genes = [genes]\n",
    "\n",
    "    # Loop through all the genes\n",
    "    for gene in genes:\n",
    "        # Set two more column names that vary by gene\n",
    "        merged_location_col = gene + '_' + location_col\n",
    "        merged_mutation_col = gene + '_' + mutation_col\n",
    "        merged_mutation_status_col = gene + '_' + mutation_status_col\n",
    "\n",
    "        # Load all the mutations for the gene\n",
    "        gene_df = mutations.loc[mutations[gene_col] == gene] \n",
    "\n",
    "        for sample in merged_df.index.values:\n",
    "            # Get the rows for just this sample from our two dataframes\n",
    "            sample_df = gene_df.loc[gene_df.index == sample]\n",
    "            merged_sample_df = merged_df.loc[merged_df.index == sample]\n",
    "\n",
    "            if len(sample_df.index) == 0: # There were no mutations for that gene in this sample\n",
    "                original_location = [['No_mutation']]\n",
    "                if sample_status_map[sample] == \"Tumor\":\n",
    "                    original_mutation = [['Wildtype_Tumor']]\n",
    "                    original_mutation_status = \"Wildtype_Tumor\"\n",
    "                else:\n",
    "                    original_mutation = [['Wildtype_Normal']]\n",
    "                    original_mutation_status = \"Wildtype_Normal\"\n",
    "                sample_dict = { # Create a prep dictionary with what the values for the different columns should be\n",
    "                    sample_col:[sample], # This will be our index\n",
    "                    mutation_col:original_mutation, \n",
    "                    location_col:original_location}\n",
    "                sample_df = pd.DataFrame(data=sample_dict, index=sample_dict[sample_col]) # Make that dict a dataframe, and set it as our sample_df\n",
    "\n",
    "            elif len(sample_df.index) == 1: # There was one mutation for that gene in this sample\n",
    "                original_mutation = sample_df.loc[sample, mutation_col] # Get the values\n",
    "                original_location = sample_df.loc[sample, location_col]\n",
    "\n",
    "                sample_df.at[sample, mutation_col] = [original_mutation] # Put them inside of lists, to match the merged dataframe.\n",
    "                sample_df.at[sample, location_col] = [original_location] \n",
    "\n",
    "                original_mutation_status = \"Single_mutation\"\n",
    "\n",
    "            else: # There were multiple mutations for that gene in this sample\n",
    "                original_mutations = sample_df[mutation_col].tolist() \n",
    "                original_locations = sample_df[location_col].tolist()\n",
    "\n",
    "                original_mutations = [original_mutations] # In the merged dataframe, all of the multiple mutations/location values will be in one cell, in a list\n",
    "                original_locations = [original_locations] # So, we nest each mutation/location list inside another list, so that the inner list isn't unpacked when we create our df.\n",
    "\n",
    "                sample_dict = { # Create a prep dictionary with the column values\n",
    "                    sample_col:[sample], # This will be our index\n",
    "                    mutation_col:original_mutations,\n",
    "                    location_col:original_locations}\n",
    "                sample_df = pd.DataFrame(data=sample_dict, index=sample_dict[sample_col]) # Make it a dataframe, and set it as our sample_df\n",
    "\n",
    "                original_mutation_status = \"Multiple_mutation\"\n",
    "\n",
    "            # Add a Sample_Status column to our original data df\n",
    "            if sample_status_map[sample] == \"Tumor\": # Figure out what our sample status should be\n",
    "                original_sample_status = 'Tumor'\n",
    "            else:\n",
    "                original_sample_status = 'Normal'\n",
    "            sample_df = sample_df.assign(**{sample_status_col:original_sample_status}) # Append a Sample_Status column to our original values dataframe, with what the value should be, for comparison to the merged dataframe.\n",
    "\n",
    "            # Append a Sample_Status column to our original values dataframe, with what the value should be, for comparison to the merged dataframe.\n",
    "            sample_df = sample_df.assign(**{mutation_status_col:original_mutation_status}) \n",
    "\n",
    "            # TODO: Add mutation status column, check, change dest to merge\n",
    "\n",
    "            # Make a list of our columns to test\n",
    "            source_cols_to_test = [mutation_col, sample_status_col, mutation_status_col]\n",
    "            dest_cols_to_test = [merged_mutation_col, sample_status_col, merged_mutation_status_col]\n",
    "            if show_location: # Only test the location column if it was included\n",
    "                source_cols_to_test.append(location_col)\n",
    "                dest_cols_to_test.append(merged_location_col)\n",
    "\n",
    "            # Test the columns\n",
    "            if not check_appended_columns(sample_df, merged_sample_df, source_cols_to_test, dest_cols_to_test):\n",
    "                PASS = False\n",
    "\n",
    "    return PASS\n",
    "\n",
    "# Test functions that get dataframes\n",
    "def test_get_clinical():\n",
    "    \"\"\"Test get_clinical.\"\"\"\n",
    "\n",
    "    print('Running test_get_clinical...')\n",
    "\n",
    "    df = en.get_clinical()\n",
    "    name = \"clinical\"\n",
    "    dimensions = (144, 26)\n",
    "    headers = ['Patient_ID', 'Proteomics_Tumor_Normal', 'Country', 'Histologic_Grade_FIGO', 'Myometrial_invasion_Specify', 'Histologic_type', 'Treatment_naive', 'Tumor_purity', 'Path_Stage_Primary_Tumor-pT', 'Path_Stage_Reg_Lymph_Nodes-pN', 'Age', 'Diabetes', 'Race', 'Ethnicity', 'Gender', 'Tumor_Site', 'Tumor_Site_Other', 'Tumor_Focality', 'Tumor_Size_cm', 'Num_full_term_pregnancies']\n",
    "    test_coord = ((79, 16), (15, 25), (88, 2))\n",
    "    test_vals = (77.0, '3', 'Poland')\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_derived_molecular():\n",
    "    \"\"\"Test get_derived_molecular.\"\"\"\n",
    "\n",
    "    print('Running test_get_derived_molecular...')\n",
    "\n",
    "    df = en.get_derived_molecular()\n",
    "    name = 'derived_molecular'\n",
    "    dimensions = (144, 125) \n",
    "    headers = ['Estrogen_Receptor', 'Estrogen_Receptor_%', 'Progesterone_Receptor', 'Progesterone_Receptor_%', 'MLH1', 'MLH2', 'MSH6', 'PMS2', 'p53', 'Other_IHC_specify', 'Log2_variant_total', 'Log2_SNP_total', 'Log2_INDEL_total', 'Genomics_subtype', 'Mutation_signature_C>A', 'Mutation_signature_C>G', 'Mutation_signature_C>T', 'Mutation_signature_T>C', 'Mutation_signature_T>A', 'Mutation_signature_T>G']\n",
    "    test_coord = ((3, 4), (30, 117), (80, 52))\n",
    "    test_vals = ('Intact nuclear expression', 5.459431619, -0.34)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_experimental_setup():\n",
    "    \"\"\"Test get_experimental_setup.\"\"\"\n",
    "\n",
    "    print('Running test_get_experimental_setup...')\n",
    "\n",
    "    df = en.get_experimental_setup()\n",
    "    name = 'experimental_setup'\n",
    "    dimensions = (144, 26) \n",
    "    headers = ['Proteomics_TMT_batch', 'Proteomics_TMT_plex', 'Proteomics_TMT_channel', 'Proteomics_Parent_Sample_IDs', 'Proteomics_Aliquot_ID', 'Proteomics_OCT', 'WXS_normal_sample_type', 'WXS_normal_filename', 'WXS_normal_UUID', 'WXS_tumor_sample_type', 'RNAseq_R1_sample_type', 'RNAseq_R1_filename', 'RNAseq_R1_UUID', 'RNAseq_R2_sample_type', 'RNAseq_R2_filename', 'RNAseq_R2_UUID', 'miRNAseq_sample_type', 'miRNAseq_UUID', 'Methylation_available', 'Methylation_quality']\n",
    "    test_coord = ((2, 13), (143, 2), (67, 25))\n",
    "    test_vals = ('a16b07d8-46c1-4fd9-8204-4f866aacfbec', '130N', 'PASS')\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_acetylproteomics():\n",
    "    \"\"\"Test get_acetylproteomics.\"\"\"\n",
    "\n",
    "    print('Running test_get_acetylproteomics...')\n",
    "\n",
    "    df = en.get_acetylproteomics()\n",
    "    name = 'acetylproteomics'\n",
    "    dimensions = (144, 10862)\n",
    "    headers = ['A2M-K1168', 'A2M-K1176', 'A2M-K135', 'A2M-K145', 'A2M-K516', 'A2M-K664', 'A2M-K682', 'AACS-K391', 'AAGAB-K290', 'AAK1-K201', 'ZSCAN31-K215', 'ZSCAN32-K659', 'ZW10-K634', 'ZYX-K24', 'ZYX-K25', 'ZYX-K265', 'ZYX-K272', 'ZYX-K279', 'ZYX-K533', 'ZZZ3-K117']\n",
    "    test_coord = ((1, 1), (12, 10861), (90, 5849))\n",
    "    test_vals = (0.47700000000000004, 0.16, 0.4098)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_proteomics():\n",
    "    \"\"\"Test get_proteomics.\"\"\"\n",
    "\n",
    "    print('Running test_get_proteomics...')\n",
    "\n",
    "    df = en.get_proteomics()\n",
    "    name = \"proteomics\"\n",
    "    dimensions = (144, 10999)\n",
    "    headers = ['A1BG', 'A2M', 'A2ML1', 'A4GALT', 'AAAS', 'AACS', 'AADAT', 'AAED1', 'AAGAB', 'AAK1', 'ZSWIM8', 'ZSWIM9', 'ZW10', 'ZWILCH', 'ZWINT', 'ZXDC', 'ZYG11B', 'ZYX', 'ZZEF1', 'ZZZ3']\n",
    "    test_coord = ((34, 6003), (99, 9544), (143, 32))\n",
    "    test_vals = (0.0461, 1.68, 0.904)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_transcriptomics():\n",
    "    \"\"\"Test get_transcriptomics.\"\"\"\n",
    "\n",
    "    print('Running test_get_transcriptomics...')\n",
    "\n",
    "    df = en.get_transcriptomics()\n",
    "    name = \"transcriptomics\"\n",
    "    dimensions = (109, 28057)\n",
    "    headers = ['A1BG', 'A1BG-AS1', 'A1CF', 'A2M', 'A2M-AS1', 'A2ML1', 'A2MP1', 'A3GALT2', 'A4GALT', 'A4GNT', 'ZWILCH', 'ZWINT', 'ZXDA', 'ZXDB', 'ZXDC', 'ZYG11A', 'ZYG11B', 'ZYX', 'ZZEF1', 'ZZZ3']\n",
    "    test_coord = ((22, 25483), (108, 23), (101, 17748))\n",
    "    test_vals = (0.82, 12.0, 6.19)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_circular_RNA():\n",
    "    \"\"\"Test get_circular_RNA.\"\"\"\n",
    "\n",
    "    print('Running test_get_circular_RNA...')\n",
    "\n",
    "    df = en.get_circular_RNA()\n",
    "    name = \"circular_RNA\"\n",
    "    dimensions = (109, 4945)\n",
    "    headers = ['circ_chr10_100260218_100262063_CWF19L1', 'circ_chr10_100923975_100926019_SLF2', 'circ_chr10_100923978_100926019_SLF2', 'circ_chr10_100937402_100944128_SLF2', 'circ_chr10_100937402_100950753_SLF2', 'circ_chr10_101584602_101586156_POLL', 'circ_chr10_101667886_101676436_FBXW4', 'circ_chr10_101672915_101676436_FBXW4', 'circ_chr10_101792839_101807901_OGA', 'circ_chr10_101792839_101810314_OGA', 'circ_chrX_80288906_80310233_CHMP1B2P', 'circ_chrX_80289664_80310233_CHMP1B2P', 'circ_chrX_80707427_80719656_BRWD3', 'circ_chrX_80791854_80793772_BRWD3', 'circ_chrX_84096194_84164387_RPS6KA6', 'circ_chrX_84134782_84164387_RPS6KA6', 'circ_chrX_85067127_85074391_APOOL', 'circ_chrX_85978767_85981809_CHM', 'circ_chrX_91414904_91418871_PABPC5-AS1', 'circ_chrX_9691579_9693419_TBL1X']\n",
    "    test_coord = ((108, 1), (30, 4935), (73, 2003))\n",
    "    test_vals = (9.08, 6.56, 0.0)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_miRNA():\n",
    "    \"\"\"Test get_miRNA.\"\"\"\n",
    "\n",
    "    print('Running test_get_miRNA...')\n",
    "\n",
    "    df = en.get_miRNA()\n",
    "    name = \"miRNA\"\n",
    "    dimensions = (99, 2337)\n",
    "    headers = ['hsa-let-7a-2-3p', 'hsa-let-7a-3p', 'hsa-let-7a-5p', 'hsa-let-7b-3p', 'hsa-let-7b-5p', 'hsa-let-7c-3p', 'hsa-let-7c-5p', 'hsa-let-7d-3p', 'hsa-let-7d-5p', 'hsa-let-7e-3p', 'hsa-miR-9901', 'hsa-miR-9902', 'hsa-miR-9903', 'hsa-miR-9983-3p', 'hsa-miR-9985', 'hsa-miR-9986', 'hsa-miR-99a-3p', 'hsa-miR-99a-5p', 'hsa-miR-99b-3p', 'hsa-miR-99b-5p']\n",
    "    test_coord = ((5, 0), (98, 1597), (54, 2231))\n",
    "    test_vals = (1.79, 1.36, 0.26)\n",
    "    \n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_CNA():\n",
    "    \"\"\"Test get_CNA.\"\"\"\n",
    "\n",
    "    print('Running test_get_CNA...')\n",
    "\n",
    "    df = en.get_CNA()\n",
    "    name = \"CNA\"\n",
    "    dimensions = (95, 28057)\n",
    "    headers = ['MFSD14A', 'SASS6', 'TRMT13', 'LRRC39', 'DBT', 'RTCA-AS1', 'RTCA', 'MIR553', 'UBE4B', 'CDC14A', 'TSPY8', 'FAM197Y2', 'FAM197Y4', 'FAM197Y5', 'FAM197Y7', 'FAM197Y8', 'FAM197Y6', 'FAM197Y3', 'RBMY3AP', 'TTTY22']\n",
    "    test_coord = ((12, 27865), (60, 8), (94, 15439))\n",
    "    test_vals = (-0.07, 0.01, 0.03)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_phosphoproteomics():\n",
    "    \"\"\"Test get_phosphoproteomics.\"\"\"\n",
    "\n",
    "    print('Running test_get_phosphoproteomics...')\n",
    "\n",
    "    df =  en.get_phosphoproteomics()\n",
    "    name = \"phosphoproteomics\"\n",
    "    dimensions = (144, 73212)\n",
    "    headers = ['AAAS-S495', 'AAAS-S541', 'AAAS-Y485', 'AACS-S618', 'AAED1-S12', 'AAGAB-S310', 'AAGAB-S311', 'AAK1-S14', 'AAK1-S18', 'AAK1-S20', 'ZZZ3-S397', 'ZZZ3-S411', 'ZZZ3-S420', 'ZZZ3-S424', 'ZZZ3-S426', 'ZZZ3-S468', 'ZZZ3-S89', 'ZZZ3-T415', 'ZZZ3-T418', 'ZZZ3-Y399']\n",
    "    test_coord = ((36, 46), (12, 72436), (96, 45361))\n",
    "    test_vals = (0.579, 0.669, 0.156)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_phosphoproteomics_gene():\n",
    "    \"\"\"Test get_phosphoproteomics_gene.\"\"\"\n",
    "\n",
    "    print('Running test_get_phosphoproteomics_gene...')\n",
    "\n",
    "    df = en.get_phosphoproteomics_gene()\n",
    "    name = \"phosphoproteomics_gene\"\n",
    "    dimensions = (144, 8466)\n",
    "    headers = ['AAAS', 'AACS', 'AAED1', 'AAGAB', 'AAK1', 'AAMDC', 'AARS', 'AASDH', 'AATF', 'ABCA1', 'ZSCAN5C', 'ZSWIM3', 'ZSWIM8', 'ZUP1', 'ZW10', 'ZXDA', 'ZXDC', 'ZYX', 'ZZEF1', 'ZZZ3']\n",
    "    test_coord =  ((2, 7999), (143, 1045), (71, 6543))\n",
    "    test_vals = (-0.0879, 0.929, 0.153)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_phosphosites():\n",
    "    \"\"\"Test get_phosphosites.\"\"\"\n",
    "\n",
    "    print('Running test_get_phosphosites...')\n",
    "\n",
    "    gene = 'AAK1'\n",
    "    df = en.get_phosphosites(gene)\n",
    "    name = 'phosphoproteomics for ' + gene\n",
    "    dimensions = (144, 37)\n",
    "    headers = ['AAK1-S14_phosphoproteomics', 'AAK1-S18_phosphoproteomics', 'AAK1-S20_phosphoproteomics', 'AAK1-S21_phosphoproteomics', 'AAK1-S26_phosphoproteomics', 'AAK1-S618_phosphoproteomics', 'AAK1-S623_phosphoproteomics', 'AAK1-S624_phosphoproteomics', 'AAK1-S637_phosphoproteomics', 'AAK1-S642_phosphoproteomics', 'AAK1-T448_phosphoproteomics', 'AAK1-T606_phosphoproteomics', 'AAK1-T620_phosphoproteomics', 'AAK1-T640_phosphoproteomics', 'AAK1-T653_phosphoproteomics', 'AAK1-T674_phosphoproteomics', 'AAK1-T681_phosphoproteomics', 'AAK1-T694_phosphoproteomics', 'AAK1-T848_phosphoproteomics', 'AAK1-Y687_phosphoproteomics']\n",
    "    test_coord = ((5, 33), (64, 14), (128, 0))\n",
    "    test_vals = (0.547, -0.5379999999999999, 0.1395)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_mutations():\n",
    "    \"\"\"Test get_mutations.\"\"\"\n",
    "\n",
    "    print('Running test_get_mutations...')\n",
    "\n",
    "    df = en.get_mutations()\n",
    "    name = \"somatic_mutation\"\n",
    "    dimensions = (52560, 3)\n",
    "    headers = ['Gene', 'Mutation', 'Location']\n",
    "    test_coord = ((52000, 2), (12, 0), (34567, 1))\n",
    "    test_vals = ('p.V167L', 'ARID1A', 'Missense_Mutation')\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_get_mutations_binary():\n",
    "    \"\"\"Test get_mutations_binary.\"\"\"\n",
    "\n",
    "    print('Running test_get_mutations_binary...')\n",
    "\n",
    "    df = en.get_mutations_binary()\n",
    "    name = \"somatic_mutation_binary\"\n",
    "    dimensions = (95, 51559)\n",
    "    headers = ['A1BG_p.E298K', 'A1BG_p.S181N', 'A1CF_p.F487L', 'A1CF_p.S236Y', 'A2ML1_p.A8V', 'A2ML1_p.G1306D', 'A2ML1_p.L1347F', 'A2ML1_p.L82I', 'A2ML1_p.P712S', 'A2ML1_p.R443Q', 'ZYG11A_p.Q442H', 'ZYG11B_p.H315R', 'ZYG11B_p.R495M', 'ZYG11B_p.R728C', 'ZYX_p.C447Y', 'ZZEF1_p.A2723V', 'ZZEF1_p.D845Y', 'ZZEF1_p.K1251E', 'ZZEF1_p.K2387Sfs*40', 'ZZZ3_p.Y891C']\n",
    "    test_coord = ((94, 51558), (0, 0), (45, 25436))\n",
    "    test_vals = (0, 0, 0)\n",
    "\n",
    "    PASS = check_getter(df, name, dimensions, headers, test_coord, test_vals)\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# Test merging and appending functions\n",
    "def test_compare_omics_source_preservation():\n",
    "    \"\"\"Test that compare_omics does not alter the dataframes it pulls data from.\"\"\"\n",
    "    print(\"Running test_compare_omics_source_preservation...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    prot = en.get_proteomics()\n",
    "    acet = en.get_acetylproteomics() # Acetylproteomics and phosphoproteomics have multiple columns for one gene. We use acetylproteomics to make sure compare_omics can grab all those values.\n",
    "\n",
    "    # Copy the source dataframes so we can make sure later on that compare_omics doesn't alter them.\n",
    "    prot_copy = prot.copy()\n",
    "    acet_copy = acet.copy()\n",
    "\n",
    "    # Call compare_omics on the dataframes, and make sure it doesn't return None.\n",
    "    compared = en.compare_omics(prot, acet)\n",
    "    if compared is None:\n",
    "        print('compare_omics returned None.')\n",
    "        PASS = False\n",
    "\n",
    "    # Use the copies we made at the beginning to make sure that compare_omics didn't alter the source dataframes\n",
    "    if not prot.equals(prot_copy):\n",
    "        print(\"Proteomics dataframe was altered by compare_omics.\")\n",
    "        PASS = False\n",
    "\n",
    "    if not acet.equals(acet_copy):\n",
    "        print(\"Acetylproteomics dataframe was altered by compare_omics.\")\n",
    "        PASS = False\n",
    "\n",
    "    # Indicate whether the test passed.\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_default_parameters():\n",
    "    \"\"\"Test compare_omics with default parameters cols1=None and cols2=None.\"\"\"\n",
    "    print(\"Running test_compare_omics_default_parameters...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    prot = en.get_proteomics()\n",
    "    acet = en.get_acetylproteomics() # Acetylproteomics and phosphoproteomics have multiple columns for one gene. We use acetylproteomics to make sure compare_omics can grab all those values.\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    compared = en.compare_omics(prot, acet) \n",
    "    if not check_returned_is_df(compared):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = prot.name + ', with ' + acet.name\n",
    "    if not check_df_name(compared, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(prot.index.intersection(acet.index))\n",
    "    exp_num_cols = len(prot.columns) + len(acet.columns)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(compared, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Test that all columns from proteomics were appended, and that values were preserved\n",
    "    if not check_appended_columns(prot, compared, prot.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Test that all columns from acetylproteomics were appended, and that values were preserved\n",
    "    if not check_appended_columns(acet, compared, acet.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_one_gene():\n",
    "    \"\"\"Test compare_omics with one genes for cols1 and cols2.\"\"\"\n",
    "    print(\"Running test_compare_omics_one_gene...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    prot = en.get_proteomics()\n",
    "    acet = en.get_acetylproteomics() # Acetylproteomics and phosphoproteomics have multiple columns for one gene. We use acetylproteomics to make sure compare_omics can grab all those values.\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    prot_gene = 'TP53'\n",
    "    acet_gene = 'A2M'\n",
    "    compared = en.compare_omics(prot, acet, prot_gene, acet_gene) \n",
    "    if not check_returned_is_df(compared):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} for {}, with {} for {}\".format(prot.name, prot_gene, acet.name, acet_gene)\n",
    "    if not check_df_name(compared, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which columns from proteomics correspond to prot_gene (should be just one)\n",
    "    prot_regex = build_omics_regex(prot_gene)\n",
    "    prot_cols = prot.filter(regex=prot_regex)\n",
    "    if len(prot_cols.columns) != 1:\n",
    "        print(\"Unexpected number of matching proteomics columns in test.\\n\\tExpected: 1\\n\\tActual: {}\".format(len(prot_cols)))\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which columns from acetylproteomics correspond to acet_gene\n",
    "    acet_suffix = \"-.*\"\n",
    "    acet_regex = build_omics_regex(acet_gene, suffix=acet_suffix)\n",
    "    acet_cols = acet.filter(regex=acet_regex)\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(prot.index.intersection(acet.index))\n",
    "    exp_num_cols = len(prot_cols.columns) + len(acet_cols.columns)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(compared, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(prot, compared, prot_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(acet, compared, acet_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_multiple_genes():\n",
    "    \"\"\"Test compare_omics with lists of genes for cols1 and cols2.\"\"\"\n",
    "    print(\"Running test_compare_omics_multiple_genes...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    prot = en.get_proteomics()\n",
    "    acet = en.get_acetylproteomics() # Acetylproteomics and phosphoproteomics have multiple columns for one gene. We use acetylproteomics to make sure compare_omics can grab all those values.\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    prot_genes = ['A4GALT', 'TP53', 'ZSCAN30']\n",
    "    acet_genes = ['AAGAB', 'AACS', 'ZW10', 'ZYX']\n",
    "    compared = en.compare_omics(prot, acet, prot_genes, acet_genes) \n",
    "    if not check_returned_is_df(compared):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} for {} genes, with {} for {} genes\".format(prot.name, len(prot_genes), acet.name, len(acet_genes))\n",
    "    if not check_df_name(compared, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which columns from proteomics correspond to prot_gene (should be the same number as number of genes in prot_genes)\n",
    "    prot_regex = build_omics_regex(prot_genes)\n",
    "    prot_cols = prot.filter(regex=prot_regex) # Use the regex to get all matching columns\n",
    "    if len(prot_cols.columns) != len(prot_genes):\n",
    "        print(\"Unexpected number of matching proteomics columns in test.\\n\\tExpected: {}\\n\\tActual: {}\".format(len(prot_genes), len(prot_cols.columns)))\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which columns from acetylproteomics correspond to acet_gene\n",
    "    acet_suffix = '-.*'\n",
    "    acet_regex = build_omics_regex(acet_genes, suffix=acet_suffix)\n",
    "    acet_cols = acet.filter(regex=acet_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(prot.index.intersection(acet.index))\n",
    "    exp_num_cols = len(prot_cols.columns) + len(acet_cols.columns)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(compared, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(prot, compared, prot_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(acet, compared, acet_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_all_dfs():\n",
    "    \"\"\"Test that compare_omics works will all dataframes that are valid for the function.\"\"\"\n",
    "    print(\"Running test_compare_omics_all_dfs...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load our dataframes to test, and set our genes. We call individual parameters, to make sure the columns are formatted properly.\n",
    "    acet = en.get_acetylproteomics()\n",
    "    CNA = en.get_CNA()\n",
    "    phosg = en.get_phosphoproteomics_gene()\n",
    "    phoss = en.get_phosphoproteomics()\n",
    "    prot = en.get_proteomics()\n",
    "    tran = en.get_transcriptomics()\n",
    "    gene1 = 'TP53'\n",
    "    gene2 = 'AAGAB'\n",
    "\n",
    "    # Call compare_omics on the dataframes\n",
    "    acet_CNA = en.compare_omics(acet, CNA, gene1, gene2)\n",
    "    phosg_phoss = en.compare_omics(phosg, phoss, gene1, gene2)\n",
    "    prot_tran = en.compare_omics(prot, tran, gene1, gene2)\n",
    "\n",
    "    # Check the return values\n",
    "    if not check_returned_is_df(acet_CNA):\n",
    "        print(\"Dataframes compared: acetylproetomics and CNA.\")\n",
    "        PASS = False\n",
    "    if not check_returned_is_df(phosg_phoss):\n",
    "        print(\"Dataframes compared: phosphoproteomics_gene and phosphoproteomics.\")\n",
    "        PASS = False\n",
    "    if not check_returned_is_df(prot_tran):\n",
    "        print(\"Dataframes compared: proteomics and transcriptomics.\")\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_invalid_dfs():\n",
    "    \"\"\"Test that compare_omics will not accept non-omics dataframes, and not accept omics dataframes of the wrong format.\"\"\"\n",
    "    print(\"Running test_compare_omics_invalid_dfs...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load our dataframes to test with\n",
    "    prot = en.get_proteomics() # We want to try a mix of valid and invalid dataframes, so we need to load this valid dataframe\n",
    "    clin = en.get_clinical()\n",
    "    tran_cir = en.get_circular_RNA() # Although circular_RNA is an omics dataframe, it's of the wrong format to work with compare_omics\n",
    "\n",
    "    # Test with one valid dataframe and one invalid one\n",
    "    comp = en.compare_omics(prot, tran_cir)\n",
    "    if comp is not None:\n",
    "        print(\"compare_omics should have returned None when passed the {} dataframe, but instead returned a {}\".format(tran_cir.name, type(comp)))\n",
    "        PASS = False\n",
    "    else:\n",
    "        print(\"(NOTE: The invalid dataframe message above was expected.)\")\n",
    "\n",
    "    # Test with two invalid dataframes\n",
    "    comp = en.compare_omics(clin, tran_cir)\n",
    "    if comp is not None:\n",
    "        print(\"compare_omics should have returned None when passed the {} and {} dataframes, but instead returned a {}\".format(clin.name, tran_cir.name, type(comp)))\n",
    "        PASS = False\n",
    "    else:\n",
    "        print(\"(NOTE: The invalid dataframe message above was expected.)\")\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_one_invalid_key():\n",
    "    \"\"\"Test that compare_omics will gracefully handle one invalid key.\"\"\"\n",
    "    print(\"Running test_compare_omics_one_invalid_key...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    prot = en.get_proteomics()\n",
    "    acet = en.get_acetylproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    invalid = 'gobbledegook'\n",
    "    acet_valid = 'AACS' \n",
    "    compared = en.compare_omics(prot, acet, invalid, acet_valid)\n",
    "    if not check_returned_is_df(compared):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} for {}, with {} for {}\".format(prot.name, invalid, acet.name, acet_valid)\n",
    "    if not check_df_name(compared, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get our columns corresponding to the invalid key\n",
    "    invalid_suffix = \"_.*\"\n",
    "    invalid_regex = build_omics_regex(invalid, suffix=invalid_suffix)\n",
    "    invalid_cols = compared.filter(regex=invalid_regex)\n",
    "    exp_num_invalid = 1\n",
    "\n",
    "    # Figure out which columns from acetylproteomics should have been grabbed for acet_valid\n",
    "    acet_suffix = '-.*'\n",
    "    acet_regex = build_omics_regex(acet_valid, suffix=acet_suffix)\n",
    "    acet_cols = acet.filter(regex=acet_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(prot.index.intersection(acet.index))\n",
    "    exp_num_cols = exp_num_invalid + len(acet_cols.columns)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(compared, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check invalid columns\n",
    "    if not check_invalid_columns(invalid_cols, exp_num_invalid):\n",
    "        PASS = False\n",
    "\n",
    "    # Check columns for acet_gene\n",
    "    if not check_appended_columns(acet, compared, acet_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_both_invalid_keys():\n",
    "    \"\"\"Test that compare_omics will gracefully handle two invalid keys.\"\"\"\n",
    "    print(\"Running test_compare_omics_both_invalid_keys...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load dataframes to test with, and set keys to use\n",
    "    prot = en.get_proteomics()\n",
    "    acet = en.get_acetylproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    invalid = 'gobbledegook'\n",
    "    compared = en.compare_omics(prot, acet, invalid, invalid)\n",
    "    if not check_returned_is_df(compared):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} for {}, with {} for {}\".format(prot.name, invalid, acet.name, invalid)\n",
    "    if not check_df_name(compared, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get our columns corresponding to the invalid key\n",
    "    invalid_suffix = \"_.*\"\n",
    "    invalid_regex = build_omics_regex(invalid, suffix=invalid_suffix)\n",
    "    invalid_cols = compared.filter(regex=invalid_regex)\n",
    "    exp_num_invalid = 2\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(prot.index.intersection(acet.index))\n",
    "    exp_num_cols = exp_num_invalid\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(compared, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check invalid columns\n",
    "    if not check_invalid_columns(invalid_cols, exp_num_invalid):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_one_list_with_invalid_key():\n",
    "    \"\"\"Test that compare_omics will gracefully handle a list of valid keys containing one invalid key.\"\"\"\n",
    "    print(\"Running test_compare_omics_one_list_with_invalid_key...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load dataframes to test with, and set keys to use\n",
    "    prot = en.get_proteomics()\n",
    "    acet = en.get_acetylproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    invalid = \"gobbledegook\"\n",
    "\n",
    "    prot_valid_list = ['A4GALT', 'TP53', 'ZSCAN30']\n",
    "    prot_invalid_list = prot_valid_list.copy()\n",
    "    prot_invalid_list.append(invalid)\n",
    "\n",
    "    acet_valid_list = ['AAGAB', 'AACS', 'ZW10', 'ZYX']\n",
    "    \n",
    "    compared = en.compare_omics(prot, acet, prot_invalid_list, acet_valid_list)\n",
    "    if not check_returned_is_df(compared):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} for {} genes, with {} for {} genes\".format(prot.name, len(prot_invalid_list), acet.name, len(acet_valid_list))\n",
    "    if not check_df_name(compared, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get our columns corresponding to the invalid key\n",
    "    invalid_suffix = \"_.*\"\n",
    "    invalid_regex = build_omics_regex(invalid, suffix=invalid_suffix)\n",
    "    invalid_cols = compared.filter(regex=invalid_regex)\n",
    "    exp_num_invalid = 1\n",
    "\n",
    "    # Figure out which columns from proteomics should have been grabbed for the valid genes from prot_invalid_list (should be the same number as number of valid genes\n",
    "    prot_regex = build_omics_regex(prot_valid_list)\n",
    "    prot_cols = prot.filter(regex=prot_regex) # Use the regex to get all matching columns\n",
    "    if len(prot_cols.columns) != len(prot_valid_list):\n",
    "        print(\"Unexpected number of matching proteomics columns in test.\\n\\tExpected: {}\\n\\tActual: {}\".format(len(prot_valid_list), len(prot_cols.columns)))\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which columns from acetylproteomics should have been grabbed for acet_valid_list\n",
    "    acet_suffix = '-.*'\n",
    "    acet_regex = build_omics_regex(acet_valid_list, suffix=acet_suffix)\n",
    "    acet_cols = acet.filter(regex=acet_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(prot.index.intersection(acet.index))\n",
    "    exp_num_cols = len(prot_cols.columns) + len(acet_cols.columns) + exp_num_invalid \n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(compared, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check invalid columns\n",
    "    if not check_invalid_columns(invalid_cols, exp_num_invalid):\n",
    "        PASS = False\n",
    "\n",
    "    # Check columns for valid genes in prot_invalid_list\n",
    "    if not check_appended_columns(prot, compared, prot_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Check columns for acet_valid_list\n",
    "    if not check_appended_columns(acet, compared, acet_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_both_list_with_invalid_key():\n",
    "    \"\"\"Test that compare_omics will gracefully handle two lists of valid keys, each containing one invalid key.\"\"\"\n",
    "    print(\"Running test_compare_omics_both_list_with_invalid_key...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load dataframes to test with, and set keys to use\n",
    "    prot = en.get_proteomics()\n",
    "    acet = en.get_acetylproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    invalid = \"gobbledegook\"\n",
    "\n",
    "    prot_valid_list = ['A4GALT', 'TP53', 'ZSCAN30']\n",
    "    prot_invalid_list = prot_valid_list.copy()\n",
    "    prot_invalid_list.append(invalid)\n",
    "\n",
    "    acet_valid_list = ['AAGAB', 'AACS', 'ZW10', 'ZYX']\n",
    "    acet_invalid_list = acet_valid_list.copy()\n",
    "    acet_invalid_list.append(invalid)\n",
    "\n",
    "    compared = en.compare_omics(prot, acet, prot_invalid_list, acet_invalid_list)\n",
    "    if not check_returned_is_df(compared):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} for {} genes, with {} for {} genes\".format(prot.name, len(prot_invalid_list), acet.name, len(acet_invalid_list))\n",
    "    if not check_df_name(compared, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get our columns corresponding to the invalid key\n",
    "    invalid_suffix = \"_.*\"\n",
    "    invalid_regex = build_omics_regex(invalid, suffix=invalid_suffix)\n",
    "    invalid_cols = compared.filter(regex=invalid_regex)\n",
    "    exp_num_invalid = 2\n",
    "\n",
    "    # Figure out which columns from proteomics should have been grabbed for the valid genes from prot_invalid_list (should be the same number as number of valid genes\n",
    "    prot_regex = build_omics_regex(prot_valid_list)\n",
    "    prot_cols = prot.filter(regex=prot_regex) # Use the regex to get all matching columns\n",
    "    if len(prot_cols.columns) != len(prot_valid_list):\n",
    "        print(\"Unexpected number of matching proteomics columns in test.\\n\\tExpected: {}\\n\\tActual: {}\".format(len(prot_valid_list), len(prot_cols.columns)))\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which columns from acetylproteomics should have been grabbed for acet_valid_list\n",
    "    acet_suffix = '-.*'\n",
    "    acet_regex = build_omics_regex(acet_valid_list, suffix=acet_suffix)\n",
    "    acet_cols = acet.filter(regex=acet_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(prot.index.intersection(acet.index))\n",
    "    exp_num_cols = len(prot_cols.columns) + len(acet_cols.columns) + exp_num_invalid \n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(compared, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check invalid columns\n",
    "    if not check_invalid_columns(invalid_cols, exp_num_invalid):\n",
    "        PASS = False\n",
    "\n",
    "    # Check columns for valid genes in prot_invalid_list\n",
    "    if not check_appended_columns(prot, compared, prot_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Check columns for valid genes in acet_invalid_list\n",
    "    if not check_appended_columns(acet, compared, acet_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_compare_omics_invalid_key_types():\n",
    "    \"\"\"Test that compare_omics will gracefully handle a key of an invalid type.\"\"\"\n",
    "    print(\"Running test_compare_omics_invalid_key_types...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load our dataframes\n",
    "    prot = en.get_proteomics()\n",
    "    acet = en.get_acetylproteomics()\n",
    "\n",
    "    # Set our keys to use\n",
    "    prot_valid = 'TP53'\n",
    "    acet_valid = 'AACS'\n",
    "    int_key = 100\n",
    "    prot_valid_list = ['TP53', 'AURKA', 'PIK3R1']\n",
    "\n",
    "    # Test a key of type int\n",
    "    comp = en.compare_omics(prot, acet, prot_valid, int_key)\n",
    "    if comp is not None:\n",
    "        print(\"compare_omics should have returned None when passed a key of type int, but instead returned a {}\".format(type(comp)))\n",
    "        PASS = False\n",
    "    else:\n",
    "        print(\"(NOTE: The invalid key message above was expected.)\")\n",
    "\n",
    "    # Test two keys of type int\n",
    "    comp = en.compare_omics(prot, acet, int_key, int_key)\n",
    "    if comp is not None:\n",
    "        print(\"compare_omics should have returned None when passed two keys of type int, but instead returned a {}\".format(type(comp)))\n",
    "        PASS = False\n",
    "    else:\n",
    "        print(\"(NOTE: The invalid key messages above were expected.)\")\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_source_preservation():\n",
    "    \"\"\"Test that append_mutations_to_omics does not alter the dataframes it pulls data from.\"\"\"\n",
    "    print(\"Running test_append_mutations_source_preservation...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes, set our variables\n",
    "    mut = en.get_mutations()\n",
    "    acet = en.get_acetylproteomics()\n",
    "    mut_gene = 'TP53'\n",
    "    mut_genes = ['TP53', 'PIK3R1']\n",
    "    acet_gene = 'AAGAB'\n",
    "    acet_genes = ['AACS', 'ZW10']\n",
    "\n",
    "    # Copy the source dataframes, to compare at the end\n",
    "    mut_copy = mut.copy()\n",
    "    acet_copy = acet.copy()\n",
    "\n",
    "    # Call append_mutations_to_omics a bunch of times\n",
    "    en.append_mutations_to_omics(acet, mut_gene)\n",
    "    en.append_mutations_to_omics(acet, mut_genes)\n",
    "    en.append_mutations_to_omics(acet, mut_gene, acet_gene)\n",
    "    en.append_mutations_to_omics(acet, mut_gene, acet_genes)\n",
    "    en.append_mutations_to_omics(acet, mut_genes, acet_gene)\n",
    "    en.append_mutations_to_omics(acet, mut_genes, acet_genes)\n",
    "    en.append_mutations_to_omics(acet, mut_genes, acet_genes, show_location=False)\n",
    "\n",
    "    # Check that the source dataframes weren't changed\n",
    "    if not mut.equals(mut_copy):\n",
    "        print(\"Mutations dataframe was altered by append_mutations_to_omics.\")\n",
    "        PASS = False\n",
    "\n",
    "    if not acet.equals(acet_copy):\n",
    "        print(\"Acetylproteomics dataframe was altered by append_mutations_to_omics.\")\n",
    "        PASS = False\n",
    "\n",
    "    # Indicate whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_one_mut_one_omics():\n",
    "    \"\"\"Test append_mutations_to_omics with one mutation gene and one omics gene.\"\"\"\n",
    "    print(\"Running test_append_mutations_one_mut_one_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    phos_gene = 'AAGAB'\n",
    "    mut_gene = 'PIK3R1'\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_gene, omics_genes=phos_gene)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{} for {}, with somatic mutation data for {} gene'.format(phos.name, phos_gene, mut_gene)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which phosphoproteomics columns should have been grabbed\n",
    "    phos_suffix = '-.*'\n",
    "    phos_regex = build_omics_regex(phos_gene, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos_cols.columns) + 4\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_gene):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_one_mut_three_omics():\n",
    "    \"\"\"Test append_mutations_to_omics with one mutation gene and three omics genes.\"\"\"\n",
    "    print(\"Running test_append_mutations_one_mut_three_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    phos_genes = ['AAGAB', 'AACS', 'ZZZ3']\n",
    "    mut_gene = 'PIK3R1'\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_gene, omics_genes=phos_genes)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{} for {} genes, with somatic mutation data for {} gene'.format(phos.name, len(phos_genes), mut_gene)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which phosphoproteomics columns should have been grabbed\n",
    "    phos_suffix = '-.*'\n",
    "    phos_regex = build_omics_regex(phos_genes, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos_cols.columns) + 4\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_gene):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_one_mut_all_omics():\n",
    "    \"\"\"Test append_mutations_to_omics with one mutation gene, and the default parameter of None for the omics gene, which should give the entire omics dataframe.\"\"\"\n",
    "    print(\"Running test_append_mutations_one_mut_all_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    mut_gene = 'PIK3R1'\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_gene)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{}, with somatic mutation data for {} gene'.format(phos.name, mut_gene)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos.columns) + 4\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_gene):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_three_mut_one_omics():\n",
    "    \"\"\"Test append_mutations_to_omics with three mutation genes and one omics gene.\"\"\"\n",
    "    print(\"Running test_append_mutations_three_mut_one_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    phos_gene = 'AAGAB'\n",
    "    mut_genes = ['PIK3R1', 'TP53', 'AURKA']\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_genes, omics_genes=phos_gene)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{} for {}, with somatic mutation data for {} genes'.format(phos.name, phos_gene, len(mut_genes))\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which phosphoproteomics columns should have been grabbed\n",
    "    phos_suffix = '-.*'\n",
    "    phos_regex = build_omics_regex(phos_gene, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos_cols.columns) + 10\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_genes):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_three_mut_three_omics():\n",
    "    \"\"\"Test append_mutations_to_omics with three mutation genes and three omics genes.\"\"\"\n",
    "    print(\"Running test_append_mutations_three_mut_three_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    phos_genes = ['AAGAB', 'AACS', 'ZZZ3']\n",
    "    mut_genes = ['PIK3R1', 'TP53', 'AURKA']\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_genes, omics_genes=phos_genes)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{} for {} genes, with somatic mutation data for {} genes'.format(phos.name, len(phos_genes), len(mut_genes))\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which phosphoproteomics columns should have been grabbed\n",
    "    phos_suffix = '-.*'\n",
    "    phos_regex = build_omics_regex(phos_genes, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos_cols.columns) + 10\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_genes):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_three_mut_all_omics():\n",
    "    \"\"\"Test append_mutations_to_omics with three mutation genes, and the default parameter of None for the omics gene, which should give the entire omics dataframe.\"\"\"\n",
    "    print(\"Running test_append_mutations_three_mut_all_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    mut_genes = ['PIK3R1', 'TP53', 'AURKA']\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_genes)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{}, with somatic mutation data for {} genes'.format(phos.name, len(mut_genes))\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos.columns) + 10\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_genes):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_one_mut_one_omics_no_location():\n",
    "    \"\"\"Test append_mutations_to_omics with one mutation gene and one omics gene, and no location column.\"\"\"\n",
    "    print(\"Running test_append_mutations_one_mut_one_omics_no_location...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    phos_gene = 'AAGAB'\n",
    "    mut_gene = 'PIK3R1'\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_gene, omics_genes=phos_gene, show_location=False)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{} for {}, with somatic mutation data for {} gene'.format(phos.name, phos_gene, mut_gene)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which phosphoproteomics columns should have been grabbed\n",
    "    phos_suffix = '-.*'\n",
    "    phos_regex = build_omics_regex(phos_gene, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos_cols.columns) + 3\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_gene, show_location=False):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_one_mut_three_omics_no_location():\n",
    "    \"\"\"Test append_mutations_to_omics with one mutation gene and three omics genes, and no location column.\"\"\"\n",
    "    print(\"Running test_append_mutations_one_mut_three_omics_no_location...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    phos_genes = ['AAGAB', 'AACS', 'ZZZ3']\n",
    "    mut_gene = 'PIK3R1'\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_gene, omics_genes=phos_genes, show_location=False)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{} for {} genes, with somatic mutation data for {} gene'.format(phos.name, len(phos_genes), mut_gene)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which phosphoproteomics columns should have been grabbed\n",
    "    phos_suffix = '-.*'\n",
    "    phos_regex = build_omics_regex(phos_genes, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos_cols.columns) + 3\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_gene, show_location=False):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_one_mut_all_omics_no_location():\n",
    "    \"\"\"Test append_mutations_to_omics with one mutation gene, default of None for omics gene (to select all omics), and no location column.\"\"\"\n",
    "    print(\"Running test_append_mutations_one_mut_all_omics_no_location...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    mut_gene = 'PIK3R1'\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_gene, show_location=False)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{}, with somatic mutation data for {} gene'.format(phos.name, mut_gene)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos.columns) + 3\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_gene, show_location=False):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_three_mut_one_omics_no_location():\n",
    "    \"\"\"Test append_mutations_to_omics with three mutation genes and one omics gene, and no location column.\"\"\"\n",
    "    print(\"Running test_append_mutations_three_mut_one_omics_no_location...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    phos_gene = 'AAGAB'\n",
    "    mut_genes = ['PIK3R1', 'TP53', 'AURKA']\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_genes, omics_genes=phos_gene, show_location=False)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{} for {}, with somatic mutation data for {} genes'.format(phos.name, phos_gene, len(mut_genes))\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which phosphoproteomics columns should have been grabbed\n",
    "    phos_suffix = '-.*'\n",
    "    phos_regex = build_omics_regex(phos_gene, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos_cols.columns) + 7\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_genes, show_location=False):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_three_mut_three_omics_no_location():\n",
    "    \"\"\"Test append_mutations_to_omics with three mutation genes and three omics genes, and no location column.\"\"\"\n",
    "    print(\"Running test_append_mutations_three_mut_three_omics_no_location...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    phos_genes = ['AAGAB', 'AACS', 'ZZZ3']\n",
    "    mut_genes = ['PIK3R1', 'TP53', 'AURKA']\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_genes, omics_genes=phos_genes, show_location=False)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{} for {} genes, with somatic mutation data for {} genes'.format(phos.name, len(phos_genes), len(mut_genes))\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Figure out which phosphoproteomics columns should have been grabbed\n",
    "    phos_suffix = '-.*'\n",
    "    phos_regex = build_omics_regex(phos_genes, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex) # Use the regex to get all matching columns\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos_cols.columns) + 7\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_genes, show_location=False):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_three_mut_all_omics_no_location():\n",
    "    \"\"\"Test append_mutations_to_omics with three mutation genes, default of None for the omics gene (to select all omics), and no location column.\"\"\"\n",
    "    print(\"Running test_append_mutations_three_mut_all_omics_no_location...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframe and set our keys\n",
    "    phos = en.get_phosphoproteomics()\n",
    "    mut_genes = ['PIK3R1', 'TP53', 'AURKA']\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_mutations_to_omics(phos, mut_genes, show_location=False)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip remaining steps, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = '{}, with somatic mutation data for {} genes'.format(phos.name, len(mut_genes))\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(phos.index)\n",
    "    exp_num_cols = len(phos.columns) + 7\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check values in columns\n",
    "    if not check_appended_columns(phos, appended, phos.columns):\n",
    "        PASS = False\n",
    "\n",
    "    mutations = en.get_mutations() # Load the somatic_mutation dataframe, which the mutation data was drawn from\n",
    "    if not check_mutation_columns(mutations, appended, mut_genes, show_location=False):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_invalid_key():\n",
    "    \"\"\"Test that append_mutations_to_omics gracefully handles invalid mutation gene keys.\"\"\"\n",
    "    print(\"Running test_append_mutations_invalid_key...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load our dataframe to test with, and set our keys to use\n",
    "    acet = en.get_acetylproteomics()\n",
    "    invalid = 'lorem ipsum'\n",
    "    invalid_list = ['PIK3R1', 'TAF1', 'GP6', 'lorem ipsum']\n",
    "\n",
    "    # Test one invalid key\n",
    "    appended = en.append_mutations_to_omics(acet, invalid)\n",
    "    if appended is not None:\n",
    "        print(\"append_mutations_to_omics should have returned None when passed an invalid key, but instead returned a {}\".format(type(appended)))\n",
    "        PASS = False\n",
    "    else:\n",
    "        print(\"(NOTE: The invalid key message above was expected.)\")\n",
    "\n",
    "    # Test one invalid key in a list of valid keys\n",
    "    appended = en.append_mutations_to_omics(acet, invalid_list)\n",
    "    if appended is not None:\n",
    "        print(\"append_mutations_to_omics should have returned None when passed a list of valid keys containing one invalid key, but instead returned a {}\".format(type(appended)))\n",
    "        PASS = False\n",
    "    else:\n",
    "        print(\"(NOTE: The invalid key message above was expected.)\")\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_mutations_invalid_key_types():\n",
    "    \"\"\"Test that append_mutations_to_omics gracefully handles invalid mutation gene key types.\"\"\"\n",
    "    print(\"Running test_append_mutations_invalid_key_types...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load our dataframe to test with, and set our keys to use\n",
    "    prot = en.get_proteomics()\n",
    "    int_key = 44\n",
    "    dict_key = {0:\"TP53\", 1:\"PIK3R1\", 2:\"AURKA\"}\n",
    "\n",
    "    # Test a key of type int\n",
    "    appended = en.append_mutations_to_omics(acet, int_key)\n",
    "    if appended is not None:\n",
    "        print(\"append_mutations_to_omics should have returned None when passed a key of invalid type int, but instead returned a {}\".format(type(appended)))\n",
    "        PASS = False\n",
    "    else:\n",
    "        print(\"(NOTE: The invalid key message above was expected.)\")\n",
    "\n",
    "    # Test a key of type dict\n",
    "    appended = en.append_mutations_to_omics(acet, dict_key)\n",
    "    if appended is not None:\n",
    "        print(\"append_mutations_to_omics should have returned None when passed a key of invalid type dict, but instead returned a {}\".format(type(appended)))\n",
    "        PASS = False\n",
    "    else:\n",
    "        print(\"(NOTE: The invalid key message above was expected.)\")\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "def test_append_metadata_source_preservation():\n",
    "    \"\"\"Test that append_metadata_to_omics does not alter the dataframes it pulls data from.\"\"\"\n",
    "    print(\"Running test_append_metadata_source_preservation...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes, and set our variables\n",
    "    clin = en.get_clinical()\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    exp_setup = en.get_experimental_setup()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    clin_col = \"Country\"\n",
    "    clin_cols = [\"Country\", \"tumor_Stage-Pathological\", \"LVSI\"]\n",
    "    derived_mol_col = \"CIBERSORT_Eosinophils\" \n",
    "    derived_mol_cols = [\"CIBERSORT_Eosinophils\", \"Pathway_activity_JAK.STAT\", \"Progesterone_Receptor_%\"]\n",
    "    exp_setup_col = \"RNAseq_R1_UUID\"\n",
    "    exp_setup_cols = [\"RNAseq_R1_UUID\", \"Methylation_available\", \"WXS_tumor_UUID\"]\n",
    "    phos_col = \"ZZZ3\"\n",
    "    phos_cols = [\"AAAS\", \"AAED1\"]\n",
    "\n",
    "    # Copy the source dataframes, to compare at the end\n",
    "    clin_copy = clin.copy()\n",
    "    derived_mol_copy = derived_mol.copy()\n",
    "    exp_setup_copy = exp_setup.copy()\n",
    "    phos_copy = phos.copy()\n",
    "\n",
    "    # Call append_metadata_to_omics a bunch of times\n",
    "    en.append_metadata_to_omics(clin, phos)\n",
    "    en.append_metadata_to_omics(clin, phos, metadata_cols=clin_col)\n",
    "    en.append_metadata_to_omics(clin, phos, omics_genes=phos_col)\n",
    "    en.append_metadata_to_omics(clin, phos, metadata_cols=clin_col, omics_genes=phos_col)\n",
    "    en.append_metadata_to_omics(clin, phos, metadata_cols=clin_cols)\n",
    "    en.append_metadata_to_omics(clin, phos, omics_genes=phos_cols)\n",
    "    en.append_metadata_to_omics(clin, phos, metadata_cols=clin_cols, omics_genes=phos_col)\n",
    "    en.append_metadata_to_omics(clin, phos, metadata_cols=clin_col, omics_genes=phos_cols)\n",
    "    en.append_metadata_to_omics(clin, phos, metadata_cols=clin_cols, omics_genes=phos_cols)\n",
    "\n",
    "    en.append_metadata_to_omics(derived_mol, phos)\n",
    "    en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_col)\n",
    "    en.append_metadata_to_omics(derived_mol, phos, omics_genes=phos_col)\n",
    "    en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_col, omics_genes=phos_col)\n",
    "    en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_cols)\n",
    "    en.append_metadata_to_omics(derived_mol, phos, omics_genes=phos_cols)\n",
    "    en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_cols, omics_genes=phos_col)\n",
    "    en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_col, omics_genes=phos_cols)\n",
    "    en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_cols, omics_genes=phos_cols)\n",
    "\n",
    "    en.append_metadata_to_omics(exp_setup, phos)\n",
    "    en.append_metadata_to_omics(exp_setup, phos, metadata_cols=exp_setup_col)\n",
    "    en.append_metadata_to_omics(exp_setup, phos, omics_genes=phos_col)\n",
    "    en.append_metadata_to_omics(exp_setup, phos, metadata_cols=exp_setup_col, omics_genes=phos_col)\n",
    "    en.append_metadata_to_omics(exp_setup, phos, metadata_cols=exp_setup_cols)\n",
    "    en.append_metadata_to_omics(exp_setup, phos, omics_genes=phos_cols)\n",
    "    en.append_metadata_to_omics(exp_setup, phos, metadata_cols=exp_setup_cols, omics_genes=phos_col)\n",
    "    en.append_metadata_to_omics(exp_setup, phos, metadata_cols=exp_setup_col, omics_genes=phos_cols)\n",
    "    en.append_metadata_to_omics(exp_setup, phos, metadata_cols=exp_setup_cols, omics_genes=phos_cols)\n",
    "\n",
    "    # Check that the source dataframes weren't changed\n",
    "    if not clin.equals(clin_copy):\n",
    "        print(\"clinical dataframe was altered by append_metadata_to_omics.\")\n",
    "        PASS = False\n",
    "\n",
    "    if not derived_mol.equals(derived_mol_copy):\n",
    "        print(\"derived_molecular dataframe was altered by append_metadata_to_omics.\")\n",
    "        PASS = False\n",
    "\n",
    "    if not exp_setup.equals(exp_setup_copy):\n",
    "        print(\"experimental_setup dataframe was altered by append_metadata_to_omics.\")\n",
    "        PASS = False\n",
    "\n",
    "    if not phos.equals(phos_copy):\n",
    "        print(\"phosphoproteomics dataframe was altered by append_metadata_to_omics.\")\n",
    "        PASS = False\n",
    "\n",
    "    # Indicate whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# One meta one omics\n",
    "def test_append_metadata_one_meta_one_omics():\n",
    "    \"\"\"Test append_metadata_to_omics with one metadata column and one omics gene.\"\"\"\n",
    "    print(\"Running test_append_metadata_one_meta_one_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    derived_mol_col = \"Purity_Stroma\"\n",
    "    phos_gene = \"USP36\"\n",
    "    appended = en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_col, omics_genes=phos_gene)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} from {}, with {} for {}\".format(derived_mol_col, derived_mol.name, phos.name, phos_gene)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get the columns that should've been selected from phosphoproteomics\n",
    "    phos_suffix = \"-.*\"\n",
    "    phos_regex = build_omics_regex(phos_gene, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex)\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(derived_mol.index.intersection(phos.index))\n",
    "    exp_num_cols = len(phos_cols.columns) + 1\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(derived_mol, appended, derived_mol_col, derived_mol_col):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# One meta three omics\n",
    "def test_append_metadata_one_meta_three_omics():\n",
    "    \"\"\"Test append_metadata_to_omics with one metadata column and three omics genes.\"\"\"\n",
    "    print(\"Running test_append_metadata_one_meta_three_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    derived_mol_col = \"Purity_Stroma\"\n",
    "    phos_genes = [\"USP36\", \"TMEM209\", \"STXBP5\"]\n",
    "    appended = en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_col, omics_genes=phos_genes)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} from {}, with {} for {} genes\".format(derived_mol_col, derived_mol.name, phos.name, len(phos_genes))\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get the columns that should've been selected from phosphoproteomics\n",
    "    phos_suffix = \"-.*\"\n",
    "    phos_regex = build_omics_regex(phos_genes, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex)\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(derived_mol.index.intersection(phos.index))\n",
    "    exp_num_cols = len(phos_cols.columns) + 1\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(derived_mol, appended, derived_mol_col, derived_mol_col):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# One meta all omics\n",
    "def test_append_metadata_one_meta_all_omics():\n",
    "    \"\"\"Test append_metadata_to_omics with one metadata column, and the default parameter of None for the omics gene, which should cause the entire omics dataframe to be selected.\"\"\"\n",
    "    print(\"Running test_append_metadata_one_meta_all_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    derived_mol_col = \"Purity_Stroma\"\n",
    "    appended = en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_col)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} from {}, with {}\".format(derived_mol_col, derived_mol.name, phos.name)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(derived_mol.index.intersection(phos.index))\n",
    "    exp_num_cols = len(phos.columns) + 1\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(derived_mol, appended, derived_mol_col, derived_mol_col):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(phos, appended, phos.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# Three meta one omics\n",
    "def test_append_metadata_three_meta_one_omics():\n",
    "    \"\"\"Test append_metadata_to_omics with three metadata columns and one omics gene.\"\"\"\n",
    "    print(\"Running test_append_metadata_three_meta_one_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    derived_mol_cols = [\"Purity_Stroma\", \"POLE_subtype\", \"CIBERSORT_T _cells _CD4 _memory _resting\"]\n",
    "    phos_gene = \"USP36\"\n",
    "    appended = en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_cols, omics_genes=phos_gene)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} columns from {}, with {} for {}\".format(len(derived_mol_cols), derived_mol.name, phos.name, phos_gene)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get the columns that should've been selected from phosphoproteomics\n",
    "    phos_suffix = \"-.*\"\n",
    "    phos_regex = build_omics_regex(phos_gene, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex)\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(derived_mol.index.intersection(phos.index))\n",
    "    exp_num_cols = len(phos_cols.columns) + len(derived_mol_cols)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(derived_mol, appended, derived_mol_cols, derived_mol_cols):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# Three meta three omics\n",
    "def test_append_metadata_three_meta_three_omics():\n",
    "    \"\"\"Test append_metadata_to_omics with three metadata columns and three omics genes.\"\"\"\n",
    "    print(\"Running test_append_metadata_three_meta_three_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    derived_mol_cols = [\"Purity_Stroma\", \"POLE_subtype\", \"CIBERSORT_T _cells _CD4 _memory _resting\"]\n",
    "    phos_genes = [\"USP36\", \"TMEM209\", \"STXBP5\"]\n",
    "    appended = en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_cols, omics_genes=phos_genes)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} columns from {}, with {} for {} genes\".format(len(derived_mol_cols), derived_mol.name, phos.name, len(phos_genes))\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get the columns that should've been selected from phosphoproteomics\n",
    "    phos_suffix = \"-.*\"\n",
    "    phos_regex = build_omics_regex(phos_genes, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex)\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(derived_mol.index.intersection(phos.index))\n",
    "    exp_num_cols = len(phos_cols.columns) + len(derived_mol_cols)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(derived_mol, appended, derived_mol_cols, derived_mol_cols):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# Three meta all omics\n",
    "def test_append_metadata_three_meta_all_omics():\n",
    "    \"\"\"Test append_metadata_to_omics with three metadata columns, and the default parameter of None for the omics gene, which should cause the entire omics dataframe to be selected.\"\"\"\n",
    "    print(\"Running test_append_metadata_three_meta_all_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    derived_mol_cols = [\"Purity_Stroma\", \"POLE_subtype\", \"CIBERSORT_T _cells _CD4 _memory _resting\"]\n",
    "    appended = en.append_metadata_to_omics(derived_mol, phos, metadata_cols=derived_mol_cols)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{} columns from {}, with {}\".format(len(derived_mol_cols), derived_mol.name, phos.name)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(derived_mol.index.intersection(phos.index))\n",
    "    exp_num_cols = len(phos.columns) + len(derived_mol_cols)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(derived_mol, appended, derived_mol_cols, derived_mol_cols):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(phos, appended, phos.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# All meta one omics\n",
    "def test_append_metadata_all_meta_one_omics():\n",
    "    \"\"\"Test append_metadata_to_omics with the default of None for metadata_cols, which should select the entire dataframe, and one omics gene.\"\"\"\n",
    "    print(\"Running test_append_metadata_all_meta_one_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    phos_gene = \"USP36\"\n",
    "    appended = en.append_metadata_to_omics(derived_mol, phos, omics_genes=phos_gene)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{}, with {} for {}\".format(derived_mol.name, phos.name, phos_gene)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get the columns that should've been selected from phosphoproteomics\n",
    "    phos_suffix = \"-.*\"\n",
    "    phos_regex = build_omics_regex(phos_gene, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex)\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(derived_mol.index.intersection(phos.index))\n",
    "    exp_num_cols = len(phos_cols.columns) + len(derived_mol.columns)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(derived_mol, appended, derived_mol.columns, derived_mol.columns):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# All meta three omics\n",
    "def test_append_metadata_all_meta_three_omics():\n",
    "    \"\"\"Test append_metadata_to_omics with the default of None for metadata_cols, which should select the entire dataframe, and three omics genes.\"\"\"\n",
    "    print(\"Running test_append_metadata_all_meta_three_omics...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    phos_genes = [\"USP36\", \"TMEM209\", \"STXBP5\"]\n",
    "    appended = en.append_metadata_to_omics(derived_mol, phos, omics_genes=phos_genes)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{}, with {} for {} genes\".format(derived_mol.name, phos.name, len(phos_genes))\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Get the columns that should've been selected from phosphoproteomics\n",
    "    phos_suffix = \"-.*\"\n",
    "    phos_regex = build_omics_regex(phos_genes, suffix=phos_suffix)\n",
    "    phos_cols = phos.filter(regex=phos_regex)\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(derived_mol.index.intersection(phos.index))\n",
    "    exp_num_cols = len(phos_cols.columns) + len(derived_mol.columns)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(derived_mol, appended, derived_mol.columns, derived_mol.columns):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(phos, appended, phos_cols.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "# All meta all omics (default parameters)\n",
    "def test_append_metadata_default_parameters():\n",
    "    \"\"\"Test append_metadata_to_omics with the default parameters of None for metadata_cols and omics_genes, which should cause it to select the entire metadata and omics dataframes.\"\"\"\n",
    "    print(\"Running test_append_metadata_default_parameters...\")\n",
    "    PASS = True\n",
    "\n",
    "    # Load the source dataframes\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    phos = en.get_phosphoproteomics()\n",
    "\n",
    "    # Run the function, make sure it returned properly\n",
    "    appended = en.append_metadata_to_omics(derived_mol, phos)\n",
    "    if not check_returned_is_df(appended):\n",
    "        PASS = False\n",
    "        print_test_result(PASS)\n",
    "        return # Skip other tests, since they won't work if it's not a dataframe.\n",
    "\n",
    "    # Check dataframe name\n",
    "    exp_name = \"{}, with {}\".format(derived_mol.name, phos.name)\n",
    "    if not check_df_name(appended, exp_name):\n",
    "        PASS = False\n",
    "\n",
    "    # Check dataframe shape\n",
    "    exp_num_rows = len(derived_mol.index.intersection(phos.index))\n",
    "    exp_num_cols = len(phos.columns) + len(derived_mol.columns)\n",
    "    exp_shape = (exp_num_rows, exp_num_cols)\n",
    "    if not check_df_shape(appended, exp_shape):\n",
    "        PASS = False\n",
    "\n",
    "    # Check column values\n",
    "    if not check_appended_columns(derived_mol, appended, derived_mol.columns, derived_mol.columns):\n",
    "        PASS = False\n",
    "\n",
    "    if not check_appended_columns(phos, appended, phos.columns):\n",
    "        PASS = False\n",
    "\n",
    "    # Print whether the test passed\n",
    "    print_test_result(PASS)\n",
    "\n",
    "\n",
    "# All valid dfs (omics or meta)\n",
    "def test_append_metadata_all_dfs():\n",
    "    \"\"\"Test that append_metadata_to_omics works with all dataframes that are valid for the function.\"\"\"\n",
    "\n",
    "    # Load our dataframes to test, and set the keys we'll use.\n",
    "    clin = en.get_clinical()\n",
    "    derived_mol = en.get_derived_molecular()\n",
    "    exp_setup = en.get_exp_setup()\n",
    "\n",
    "    acet = en.get_acetylproteomics()\n",
    "    CNA = en.get_CNA()\n",
    "    phosg = en.get_phosphoproteomics_gene()\n",
    "    phoss = en.get_phosphoproteomics()\n",
    "    prot = en.get_proteomics()\n",
    "    tran = en.get_transcriptomics()\n",
    "\n",
    "    # Call append_metadata_to_omics on the dataframes\n",
    "\n",
    "    # Check the return values\n",
    "\n",
    "    # Print whether the test passed\n",
    "\n",
    "# Invalid dfs\n",
    "\n",
    "# Invalid keys\n",
    "\n",
    "# Invalid key types\n",
    "\n",
    "# ADD ALL DFS TEST FOR MUT\n",
    "\n",
    "\n",
    "print(\"\\nRunning tests:\\n\")\n",
    " \n",
    "print(\"Testing getters...\")\n",
    "test_get_clinical()\n",
    "test_get_derived_molecular()\n",
    "test_get_experimental_setup()\n",
    "test_get_acetylproteomics()\n",
    "test_get_proteomics()\n",
    "test_get_transcriptomics()\n",
    "test_get_circular_RNA()\n",
    "test_get_miRNA()\n",
    "test_get_CNA()\n",
    "test_get_phosphoproteomics()\n",
    "test_get_phosphoproteomics_gene()\n",
    "test_get_phosphosites()\n",
    "test_get_mutations()\n",
    "test_get_mutations_binary()\n",
    "\n",
    "print(\"\\nTesting compare and append functions...\")\n",
    "test_compare_omics_source_preservation()\n",
    "test_compare_omics_default_parameters()\n",
    "test_compare_omics_one_gene()\n",
    "test_compare_omics_multiple_genes()\n",
    "test_compare_omics_all_dfs()\n",
    "test_compare_omics_invalid_dfs()\n",
    "test_compare_omics_one_invalid_key()\n",
    "test_compare_omics_both_invalid_keys()\n",
    "test_compare_omics_one_list_with_invalid_key()\n",
    "test_compare_omics_both_list_with_invalid_key()\n",
    "test_compare_omics_invalid_key_types() \n",
    "\n",
    "test_append_metadata_source_preservation()\n",
    "test_append_metadata_one_meta_one_omics()\n",
    "test_append_metadata_one_meta_three_omics()\n",
    "test_append_metadata_one_meta_all_omics()\n",
    "test_append_metadata_three_meta_one_omics()\n",
    "test_append_metadata_three_meta_three_omics()\n",
    "test_append_metadata_three_meta_all_omics()\n",
    "test_append_metadata_all_meta_one_omics()\n",
    "test_append_metadata_all_meta_three_omics()\n",
    "test_append_metadata_default_parameters()\n",
    "\n",
    "test_append_mutations_source_preservation()\n",
    "test_append_mutations_one_mut_one_omics()\n",
    "test_append_mutations_one_mut_three_omics()\n",
    "test_append_mutations_one_mut_all_omics()\n",
    "test_append_mutations_three_mut_one_omics()\n",
    "test_append_mutations_three_mut_three_omics()\n",
    "test_append_mutations_three_mut_all_omics()\n",
    "test_append_mutations_one_mut_all_omics_no_location()\n",
    "test_append_mutations_one_mut_one_omics_no_location()\n",
    "test_append_mutations_one_mut_three_omics_no_location()\n",
    "test_append_mutations_three_mut_one_omics_no_location()\n",
    "test_append_mutations_three_mut_three_omics_no_location()\n",
    "test_append_mutations_three_mut_all_omics_no_location()\n",
    "test_append_mutations_invalid_key()\n",
    "\n",
    "print(\"Version:\",en.version())\n",
    "© 2019 GitHub, Inc.\n",
    "Terms\n",
    "Privacy\n",
    "Security\n",
    "Status\n",
    "Help\n",
    "Contact GitHub\n",
    "Pricing\n",
    "API\n",
    "Training\n",
    "Blog\n",
    "About\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
