{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# FSIP2 Mutation Trans Effect on Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook analyzes the trans effect of FSIP2 mutation on interacting and other proteins Transcriptomics, in Endometrial, Colon, and Ovarian cancer."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Library Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "import cptac\n",
    "import cptac.algorithms as al\n",
    "\n",
    "en = cptac.Endometrial()\n",
    "co = cptac.Colon()\n",
    "ov = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = \"FSIP2\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Investigate Proteomics, Phosphoproteomics, Acetylproteomics, or Transcriptomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#omics = \"proteomics\"\n",
    "omics = \"transcriptomics\"\n",
    "#omics = \"phosphoproteomics\"\n",
    "#omics = \"acetylproteomics\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Track all significant comparisons in Dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_significant_comparisons = pd.DataFrame(columns=['Cancer_Type', 'Gene', 'Comparison','Interacting_Protein','P_Value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_to_all_significant_comparisons(df, cancer, interacting, all_sig_comp):\n",
    "    expanded = df\n",
    "    expanded['Gene'] = gene\n",
    "    expanded['Cancer_Type'] = cancer\n",
    "    expanded['Interacting_Protein'] = interacting\n",
    "    \n",
    "    updated_all_comparisons = pd.concat([all_sig_comp, expanded], sort=False)\n",
    "    \n",
    "    return updated_all_comparisons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Interacting Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate interacting protein list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a call to the cptac.algorithms get interacting proteins method, which interacts with the uniprot and string databases to generate a list of known interacting partners with the given gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Interacting Proteins:\n",
      "AKAP3\n",
      "ROPN1\n",
      "FSIP1\n",
      "ZPBP\n",
      "CMBL\n",
      "TMED2\n",
      "DTYMK\n",
      "ABCC12\n",
      "ZMIZ1\n",
      "MGAT4B\n",
      "PDZRN3\n",
      "FSIP2\n",
      "NPFFR2\n",
      "METAP1D\n",
      "CCDC168\n",
      "NPY2R\n",
      "DLK2\n",
      "AKAP4\n",
      "TMEM241\n",
      "ABCC11\n",
      "SNTB1\n",
      "POTEI\n",
      "ABCA13\n",
      "ZNF445\n",
      "SPATS2\n",
      "FLJ22184\n"
     ]
    }
   ],
   "source": [
    "'''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "'''Show interacting protein list'''\n",
    "print(\"Interacting Proteins:\")\n",
    "for interacting_protein in interacting_proteins:\n",
    "    print(interacting_protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FLJ22184 did not match any columns in transcriptomics dataframe. FLJ22184_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "              Comparison   P_Value\n",
      "0  ROPN1_transcriptomics  0.001278\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", True, all_significant_comparisons)\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AKAP3 did not match any columns in transcriptomics dataframe. AKAP3_transcriptomics column inserted, but filled with NaN.\n",
      "ROPN1 did not match any columns in transcriptomics dataframe. ROPN1_transcriptomics column inserted, but filled with NaN.\n",
      "FSIP1 did not match any columns in transcriptomics dataframe. FSIP1_transcriptomics column inserted, but filled with NaN.\n",
      "ZPBP did not match any columns in transcriptomics dataframe. ZPBP_transcriptomics column inserted, but filled with NaN.\n",
      "ABCC12 did not match any columns in transcriptomics dataframe. ABCC12_transcriptomics column inserted, but filled with NaN.\n",
      "FSIP2 did not match any columns in transcriptomics dataframe. FSIP2_transcriptomics column inserted, but filled with NaN.\n",
      "NPFFR2 did not match any columns in transcriptomics dataframe. NPFFR2_transcriptomics column inserted, but filled with NaN.\n",
      "CCDC168 did not match any columns in transcriptomics dataframe. CCDC168_transcriptomics column inserted, but filled with NaN.\n",
      "NPY2R did not match any columns in transcriptomics dataframe. NPY2R_transcriptomics column inserted, but filled with NaN.\n",
      "DLK2 did not match any columns in transcriptomics dataframe. DLK2_transcriptomics column inserted, but filled with NaN.\n",
      "AKAP4 did not match any columns in transcriptomics dataframe. AKAP4_transcriptomics column inserted, but filled with NaN.\n",
      "ABCC11 did not match any columns in transcriptomics dataframe. ABCC11_transcriptomics column inserted, but filled with NaN.\n",
      "POTEI did not match any columns in transcriptomics dataframe. POTEI_transcriptomics column inserted, but filled with NaN.\n",
      "ABCA13 did not match any columns in transcriptomics dataframe. ABCA13_transcriptomics column inserted, but filled with NaN.\n",
      "FLJ22184 did not match any columns in transcriptomics dataframe. FLJ22184_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "                Comparison   P_Value\n",
      "0  METAP1D_transcriptomics  0.000302\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Expand our search for significant comparisons to all proteins in our dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  FSIP2\n",
      "Doing t-test comparisons\n",
      "\n",
      "              Comparison   P_Value\n",
      "0  LTBP4_transcriptomics  0.000001\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  FSIP2\n",
      "Doing t-test comparisons\n",
      "\n",
      "                   Comparison       P_Value\n",
      "0       NR1I2_transcriptomics  1.003627e-08\n",
      "1      CXCL14_transcriptomics  1.083763e-08\n",
      "2        A1CF_transcriptomics  1.228545e-08\n",
      "3      MOGAT3_transcriptomics  3.569568e-08\n",
      "4      PLA2R1_transcriptomics  4.227186e-08\n",
      "5      CYP2B6_transcriptomics  7.609309e-08\n",
      "6     SLC39A5_transcriptomics  8.356187e-08\n",
      "7       PRAP1_transcriptomics  9.341432e-08\n",
      "8        TCF7_transcriptomics  1.597423e-07\n",
      "9        SGK2_transcriptomics  1.647837e-07\n",
      "10      JADE1_transcriptomics  2.013728e-07\n",
      "11       TYMS_transcriptomics  2.079074e-07\n",
      "12     SLC3A1_transcriptomics  3.397795e-07\n",
      "13     LY6G6D_transcriptomics  3.861056e-07\n",
      "14    PLA2G4F_transcriptomics  4.678450e-07\n",
      "15  LINC01560_transcriptomics  4.918297e-07\n",
      "16      DUSP4_transcriptomics  6.523570e-07\n",
      "17    CTTNBP2_transcriptomics  6.909269e-07\n",
      "18   PPP1R14D_transcriptomics  7.413238e-07\n",
      "19       NASP_transcriptomics  8.309954e-07\n",
      "20     RUBCNL_transcriptomics  9.578890e-07\n",
      "21       ACE2_transcriptomics  1.894979e-06\n",
      "22       RTTN_transcriptomics  2.101476e-06\n",
      "23      SSUH2_transcriptomics  2.117385e-06\n",
      "24      TNNC2_transcriptomics  2.293959e-06\n",
      "25  LINC00526_transcriptomics  2.359268e-06\n",
      "26        IYD_transcriptomics  2.930960e-06\n",
      "27     MAPK12_transcriptomics  2.947136e-06\n",
      "28      ADNP2_transcriptomics  3.279001e-06\n",
      "29       RBFA_transcriptomics  3.462061e-06\n",
      "30   PHOSPHO2_transcriptomics  3.499230e-06\n",
      "31      TINAG_transcriptomics  3.699220e-06\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  FSIP2\n",
      "Doing t-test comparisons\n",
      "\n",
      "                Comparison       P_Value\n",
      "0    RASD1_transcriptomics  2.811091e-07\n",
      "1  C5orf27_transcriptomics  8.568228e-07\n",
      "2   MTNR1B_transcriptomics  1.928825e-06\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(\"\\nGene: \", gene)\n",
    "\n",
    "'''Use all proteins'''\n",
    "\n",
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "protdf = protdf.loc[:,~protdf.columns.duplicated()]\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the datafram correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", False, all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Print All Signififant Comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Interacting_Protein</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>ROPN1_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>1.277986e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>METAP1D_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>3.018148e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>LTBP4_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.245444e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>NR1I2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.003627e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>CXCL14_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.083763e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>A1CF_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.228545e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>MOGAT3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.569568e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>PLA2R1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.227186e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>CYP2B6_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.609309e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>SLC39A5_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.356187e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>PRAP1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.341432e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>TCF7_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.597423e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>SGK2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.647837e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>JADE1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.013728e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>TYMS_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.079074e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>SLC3A1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.397795e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>LY6G6D_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.861056e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>PLA2G4F_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.678450e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>LINC01560_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.918297e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>DUSP4_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.523570e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>CTTNBP2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.909269e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>PPP1R14D_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.413238e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>NASP_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.309954e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>RUBCNL_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.578890e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>ACE2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.894979e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>RTTN_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.101476e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>SSUH2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.117385e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>TNNC2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.293959e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>LINC00526_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.359268e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>IYD_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.930960e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>MAPK12_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.947136e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>ADNP2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.279001e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>RBFA_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.462061e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>PHOSPHO2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.499230e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Colon</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>TINAG_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.699220e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>RASD1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.811091e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>C5orf27_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.568228e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>FSIP2</td>\n",
       "      <td>MTNR1B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.928825e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Cancer_Type   Gene                 Comparison Interacting_Protein  \\\n",
       "0   Endometrial  FSIP2      ROPN1_transcriptomics                True   \n",
       "0         Colon  FSIP2    METAP1D_transcriptomics                True   \n",
       "0   Endometrial  FSIP2      LTBP4_transcriptomics               False   \n",
       "0         Colon  FSIP2      NR1I2_transcriptomics               False   \n",
       "1         Colon  FSIP2     CXCL14_transcriptomics               False   \n",
       "2         Colon  FSIP2       A1CF_transcriptomics               False   \n",
       "3         Colon  FSIP2     MOGAT3_transcriptomics               False   \n",
       "4         Colon  FSIP2     PLA2R1_transcriptomics               False   \n",
       "5         Colon  FSIP2     CYP2B6_transcriptomics               False   \n",
       "6         Colon  FSIP2    SLC39A5_transcriptomics               False   \n",
       "7         Colon  FSIP2      PRAP1_transcriptomics               False   \n",
       "8         Colon  FSIP2       TCF7_transcriptomics               False   \n",
       "9         Colon  FSIP2       SGK2_transcriptomics               False   \n",
       "10        Colon  FSIP2      JADE1_transcriptomics               False   \n",
       "11        Colon  FSIP2       TYMS_transcriptomics               False   \n",
       "12        Colon  FSIP2     SLC3A1_transcriptomics               False   \n",
       "13        Colon  FSIP2     LY6G6D_transcriptomics               False   \n",
       "14        Colon  FSIP2    PLA2G4F_transcriptomics               False   \n",
       "15        Colon  FSIP2  LINC01560_transcriptomics               False   \n",
       "16        Colon  FSIP2      DUSP4_transcriptomics               False   \n",
       "17        Colon  FSIP2    CTTNBP2_transcriptomics               False   \n",
       "18        Colon  FSIP2   PPP1R14D_transcriptomics               False   \n",
       "19        Colon  FSIP2       NASP_transcriptomics               False   \n",
       "20        Colon  FSIP2     RUBCNL_transcriptomics               False   \n",
       "21        Colon  FSIP2       ACE2_transcriptomics               False   \n",
       "22        Colon  FSIP2       RTTN_transcriptomics               False   \n",
       "23        Colon  FSIP2      SSUH2_transcriptomics               False   \n",
       "24        Colon  FSIP2      TNNC2_transcriptomics               False   \n",
       "25        Colon  FSIP2  LINC00526_transcriptomics               False   \n",
       "26        Colon  FSIP2        IYD_transcriptomics               False   \n",
       "27        Colon  FSIP2     MAPK12_transcriptomics               False   \n",
       "28        Colon  FSIP2      ADNP2_transcriptomics               False   \n",
       "29        Colon  FSIP2       RBFA_transcriptomics               False   \n",
       "30        Colon  FSIP2   PHOSPHO2_transcriptomics               False   \n",
       "31        Colon  FSIP2      TINAG_transcriptomics               False   \n",
       "0       Ovarian  FSIP2      RASD1_transcriptomics               False   \n",
       "1       Ovarian  FSIP2    C5orf27_transcriptomics               False   \n",
       "2       Ovarian  FSIP2     MTNR1B_transcriptomics               False   \n",
       "\n",
       "         P_Value  \n",
       "0   1.277986e-03  \n",
       "0   3.018148e-04  \n",
       "0   1.245444e-06  \n",
       "0   1.003627e-08  \n",
       "1   1.083763e-08  \n",
       "2   1.228545e-08  \n",
       "3   3.569568e-08  \n",
       "4   4.227186e-08  \n",
       "5   7.609309e-08  \n",
       "6   8.356187e-08  \n",
       "7   9.341432e-08  \n",
       "8   1.597423e-07  \n",
       "9   1.647837e-07  \n",
       "10  2.013728e-07  \n",
       "11  2.079074e-07  \n",
       "12  3.397795e-07  \n",
       "13  3.861056e-07  \n",
       "14  4.678450e-07  \n",
       "15  4.918297e-07  \n",
       "16  6.523570e-07  \n",
       "17  6.909269e-07  \n",
       "18  7.413238e-07  \n",
       "19  8.309954e-07  \n",
       "20  9.578890e-07  \n",
       "21  1.894979e-06  \n",
       "22  2.101476e-06  \n",
       "23  2.117385e-06  \n",
       "24  2.293959e-06  \n",
       "25  2.359268e-06  \n",
       "26  2.930960e-06  \n",
       "27  2.947136e-06  \n",
       "28  3.279001e-06  \n",
       "29  3.462061e-06  \n",
       "30  3.499230e-06  \n",
       "31  3.699220e-06  \n",
       "0   2.811091e-07  \n",
       "1   8.568228e-07  \n",
       "2   1.928825e-06  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if len(all_significant_comparisons) == 0:\n",
    "    print('No Significant Comparisons!')\n",
    "    \n",
    "else:\n",
    "    display(all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Write Significant Comparisons (if any) to Shared CSV file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "existing_results = pd.read_csv(gene+'_Trans_Results.csv')\n",
    "\n",
    "updated_results = pd.concat([existing_results, all_significant_comparisons], sort=False)\n",
    "\n",
    "updated_results.to_csv(path_or_buf = gene + '_Trans_Results.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
