{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ARID1A Mutation Trans Effect on Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook analyzes the trans effect of ARID1A mutation on interacting and other proteins Transcriptomics, in Endometrial, Colon, and Ovarian cancer."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Library Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "import cptac\n",
    "import cptac.algorithms as al\n",
    "\n",
    "en = cptac.Endometrial()\n",
    "co = cptac.Colon()\n",
    "ov = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = \"ARID1A\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Investigate Proteomics, Phosphoproteomics, Acetylproteomics, or Transcriptomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#omics = \"proteomics\"\n",
    "omics = \"transcriptomics\"\n",
    "#omics = \"phosphoproteomics\"\n",
    "#omics = \"acetylproteomics\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Track all significant comparisons in Dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_significant_comparisons = pd.DataFrame(columns=['Cancer_Type', 'Gene', 'Comparison','Interacting_Protein','P_Value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_to_all_significant_comparisons(df, cancer, interacting, all_sig_comp):\n",
    "    expanded = df\n",
    "    expanded['Gene'] = gene\n",
    "    expanded['Cancer_Type'] = cancer\n",
    "    expanded['Interacting_Protein'] = interacting\n",
    "    \n",
    "    updated_all_comparisons = pd.concat([all_sig_comp, expanded], sort=False)\n",
    "    \n",
    "    return updated_all_comparisons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Interacting Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate interacting protein list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a call to the cptac.algorithms get interacting proteins method, which interacts with the uniprot and string databases to generate a list of known interacting partners with the given gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Interacting Proteins:\n",
      "SMARCC1\n",
      "SUPT16H\n",
      "CCND1\n",
      "SMARCD3\n",
      "CREBBP\n",
      "SMARCB1\n",
      "KAT2B\n",
      "DNMT3A\n",
      "SMARCA2\n",
      "SMARCC2\n",
      "KMT2D\n",
      "ARID1A\n",
      "SMARCE1\n",
      "PHF10\n",
      "BAZ1B\n",
      "ARID1B\n",
      "NF1\n",
      "CDC5L\n",
      "SMARCD1\n",
      "SMARCD2\n",
      "SMARCA4\n",
      "ACTL6A\n",
      "BCL7B\n",
      "SS18\n",
      "DPF2\n",
      "DPF3\n",
      "KLF1\n",
      "GATA1\n",
      "ING1\n",
      "PGR\n",
      "BCL7C\n",
      "HIC1\n"
     ]
    }
   ],
   "source": [
    "'''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "'''Show interacting protein list'''\n",
    "print(\"Interacting Proteins:\")\n",
    "for interacting_protein in interacting_proteins:\n",
    "    print(interacting_protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", True, all_significant_comparisons)\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DPF3 did not match any columns in transcriptomics dataframe. DPF3_transcriptomics column inserted, but filled with NaN.\n",
      "KLF1 did not match any columns in transcriptomics dataframe. KLF1_transcriptomics column inserted, but filled with NaN.\n",
      "GATA1 did not match any columns in transcriptomics dataframe. GATA1_transcriptomics column inserted, but filled with NaN.\n",
      "PGR did not match any columns in transcriptomics dataframe. PGR_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BCL7C did not match any columns in transcriptomics dataframe. BCL7C_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "            Comparison       P_Value\n",
      "0  PGR_transcriptomics  6.009689e-08\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Expand our search for significant comparisons to all proteins in our dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  ARID1A\n",
      "Doing t-test comparisons\n",
      "\n",
      "                 Comparison       P_Value\n",
      "0      PRNP_transcriptomics  7.546881e-09\n",
      "1    SAPCD2_transcriptomics  7.544245e-08\n",
      "2      CDH1_transcriptomics  1.324795e-07\n",
      "3      EYA4_transcriptomics  1.683441e-07\n",
      "4    FAM57A_transcriptomics  3.737113e-07\n",
      "5     PCLAF_transcriptomics  3.860434e-07\n",
      "6      LRBA_transcriptomics  8.352878e-07\n",
      "7    TDRD10_transcriptomics  9.668603e-07\n",
      "8     WDR34_transcriptomics  1.308609e-06\n",
      "9     LHCGR_transcriptomics  1.368058e-06\n",
      "10  FAM110B_transcriptomics  1.567902e-06\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  ARID1A\n",
      "Doing t-test comparisons\n",
      "\n",
      "                Comparison       P_Value\n",
      "0    TDGF1_transcriptomics  3.068549e-08\n",
      "1    AXIN2_transcriptomics  6.590190e-07\n",
      "2    APOL2_transcriptomics  6.897007e-07\n",
      "3    RNF43_transcriptomics  1.216561e-06\n",
      "4  POU5F1B_transcriptomics  1.400440e-06\n",
      "5     IL15_transcriptomics  1.529432e-06\n",
      "6  FAM71E1_transcriptomics  2.573590e-06\n",
      "7    ROCK2_transcriptomics  3.370621e-06\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  ARID1A\n",
      "Doing t-test comparisons\n",
      "\n",
      "                       Comparison       P_Value\n",
      "0          MTNR1B_transcriptomics  4.806922e-16\n",
      "1           CDK20_transcriptomics  2.998791e-15\n",
      "2            HMX3_transcriptomics  2.826724e-14\n",
      "3           RXFP1_transcriptomics  4.168296e-13\n",
      "4           GPR97_transcriptomics  5.061682e-13\n",
      "5           TUNAR_transcriptomics  5.713628e-13\n",
      "6         SLC26A2_transcriptomics  1.865577e-12\n",
      "7           PGAP2_transcriptomics  3.420800e-12\n",
      "8     ANKRD20A12P_transcriptomics  9.261376e-12\n",
      "9        TNFRSF18_transcriptomics  9.514260e-12\n",
      "10            HGD_transcriptomics  1.070983e-11\n",
      "11       C10orf71_transcriptomics  3.187757e-11\n",
      "12   LOC100506526_transcriptomics  3.325323e-11\n",
      "13         C2CD4B_transcriptomics  3.860608e-11\n",
      "14         MFAP3L_transcriptomics  1.417773e-10\n",
      "15      LINC00645_transcriptomics  2.321706e-10\n",
      "16          OLFM4_transcriptomics  2.840319e-10\n",
      "17           HES2_transcriptomics  3.318104e-10\n",
      "18         RNF186_transcriptomics  3.846150e-10\n",
      "19         SLC5A9_transcriptomics  4.499720e-10\n",
      "20          CCAT1_transcriptomics  4.591002e-10\n",
      "21          FGF14_transcriptomics  5.719453e-10\n",
      "22      LOC158434_transcriptomics  6.466856e-10\n",
      "23          GABRP_transcriptomics  8.434191e-10\n",
      "24           ERMN_transcriptomics  9.957977e-10\n",
      "25        SCGB2A1_transcriptomics  1.085246e-09\n",
      "26        SERINC2_transcriptomics  1.192787e-09\n",
      "27           HMX2_transcriptomics  2.306227e-09\n",
      "28          CAPN6_transcriptomics  2.674776e-09\n",
      "29           ELP3_transcriptomics  2.962609e-09\n",
      "..                            ...           ...\n",
      "179          FUT6_transcriptomics  6.739185e-07\n",
      "180       HSD17B2_transcriptomics  7.309743e-07\n",
      "181         TPRX1_transcriptomics  7.323498e-07\n",
      "182        TFAP2B_transcriptomics  7.441152e-07\n",
      "183      MTRNR2L6_transcriptomics  7.453821e-07\n",
      "184            F3_transcriptomics  8.296980e-07\n",
      "185       TRIM43B_transcriptomics  8.299754e-07\n",
      "186       PPP1R3B_transcriptomics  8.330275e-07\n",
      "187         RFPL1_transcriptomics  8.986442e-07\n",
      "188         GDF15_transcriptomics  9.257919e-07\n",
      "189       CEACAM1_transcriptomics  9.589350e-07\n",
      "190     MTUS2-AS1_transcriptomics  1.063140e-06\n",
      "191         MFSD4_transcriptomics  1.078302e-06\n",
      "192         NCALD_transcriptomics  1.286969e-06\n",
      "193        AVPR1A_transcriptomics  1.313588e-06\n",
      "194         MESP2_transcriptomics  1.353499e-06\n",
      "195       SLC43A1_transcriptomics  1.403135e-06\n",
      "196           PZP_transcriptomics  1.422187e-06\n",
      "197  LOC100505817_transcriptomics  1.437350e-06\n",
      "198     LOC643486_transcriptomics  1.446004e-06\n",
      "199     LINC01245_transcriptomics  1.531206e-06\n",
      "200         INSM1_transcriptomics  1.536219e-06\n",
      "201  LOC102724484_transcriptomics  1.648421e-06\n",
      "202        KIF13B_transcriptomics  1.650590e-06\n",
      "203         KCNH4_transcriptomics  1.666373e-06\n",
      "204        KHDC1L_transcriptomics  1.673755e-06\n",
      "205        HS3ST1_transcriptomics  1.696851e-06\n",
      "206     LINC00462_transcriptomics  1.753059e-06\n",
      "207        SEMA3G_transcriptomics  2.101998e-06\n",
      "208          ANO1_transcriptomics  2.154795e-06\n",
      "\n",
      "[209 rows x 2 columns]\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(\"\\nGene: \", gene)\n",
    "\n",
    "'''Use all proteins'''\n",
    "\n",
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "protdf = protdf.loc[:,~protdf.columns.duplicated()]\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the datafram correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", False, all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Print All Signififant Comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Interacting_Protein</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>PGR_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>6.009689e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>PRNP_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.546881e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>SAPCD2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.544245e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>CDH1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.324795e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>EYA4_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.683441e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>FAM57A_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.737113e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>PCLAF_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.860434e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>LRBA_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.352878e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>TDRD10_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.668603e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>WDR34_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.308609e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>LHCGR_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.368058e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>FAM110B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.567902e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>TDGF1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.068549e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Colon</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>AXIN2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.590190e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Colon</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>APOL2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.897007e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Colon</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>RNF43_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.216561e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Colon</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>POU5F1B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.400440e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Colon</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>IL15_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.529432e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Colon</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>FAM71E1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.573590e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Colon</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ROCK2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.370621e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>MTNR1B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.806922e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>CDK20_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.998791e-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HMX3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.826724e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>RXFP1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.168296e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>GPR97_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.061682e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>TUNAR_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.713628e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>SLC26A2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.865577e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>PGAP2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.420800e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ANKRD20A12P_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.261376e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>TNFRSF18_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.514260e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>FUT6_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.739185e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HSD17B2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.309743e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>TPRX1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.323498e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>TFAP2B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.441152e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>MTRNR2L6_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.453821e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>F3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.296980e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>TRIM43B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.299754e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>PPP1R3B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.330275e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>RFPL1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.986442e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>GDF15_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.257919e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>CEACAM1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.589350e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>MTUS2-AS1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.063140e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>MFSD4_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.078302e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>NCALD_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.286969e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>AVPR1A_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.313588e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>MESP2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.353499e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>SLC43A1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.403135e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>PZP_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.422187e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>LOC100505817_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.437350e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>LOC643486_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.446004e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>LINC01245_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.531206e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>INSM1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.536219e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>LOC102724484_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.648421e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>KIF13B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.650590e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>KCNH4_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.666373e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>KHDC1L_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.673755e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>HS3ST1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.696851e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>LINC00462_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.753059e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>SEMA3G_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.101998e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ANO1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.154795e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>229 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Cancer_Type    Gene                    Comparison Interacting_Protein  \\\n",
       "0        Ovarian  ARID1A           PGR_transcriptomics                True   \n",
       "0    Endometrial  ARID1A          PRNP_transcriptomics               False   \n",
       "1    Endometrial  ARID1A        SAPCD2_transcriptomics               False   \n",
       "2    Endometrial  ARID1A          CDH1_transcriptomics               False   \n",
       "3    Endometrial  ARID1A          EYA4_transcriptomics               False   \n",
       "4    Endometrial  ARID1A        FAM57A_transcriptomics               False   \n",
       "5    Endometrial  ARID1A         PCLAF_transcriptomics               False   \n",
       "6    Endometrial  ARID1A          LRBA_transcriptomics               False   \n",
       "7    Endometrial  ARID1A        TDRD10_transcriptomics               False   \n",
       "8    Endometrial  ARID1A         WDR34_transcriptomics               False   \n",
       "9    Endometrial  ARID1A         LHCGR_transcriptomics               False   \n",
       "10   Endometrial  ARID1A       FAM110B_transcriptomics               False   \n",
       "0          Colon  ARID1A         TDGF1_transcriptomics               False   \n",
       "1          Colon  ARID1A         AXIN2_transcriptomics               False   \n",
       "2          Colon  ARID1A         APOL2_transcriptomics               False   \n",
       "3          Colon  ARID1A         RNF43_transcriptomics               False   \n",
       "4          Colon  ARID1A       POU5F1B_transcriptomics               False   \n",
       "5          Colon  ARID1A          IL15_transcriptomics               False   \n",
       "6          Colon  ARID1A       FAM71E1_transcriptomics               False   \n",
       "7          Colon  ARID1A         ROCK2_transcriptomics               False   \n",
       "0        Ovarian  ARID1A        MTNR1B_transcriptomics               False   \n",
       "1        Ovarian  ARID1A         CDK20_transcriptomics               False   \n",
       "2        Ovarian  ARID1A          HMX3_transcriptomics               False   \n",
       "3        Ovarian  ARID1A         RXFP1_transcriptomics               False   \n",
       "4        Ovarian  ARID1A         GPR97_transcriptomics               False   \n",
       "5        Ovarian  ARID1A         TUNAR_transcriptomics               False   \n",
       "6        Ovarian  ARID1A       SLC26A2_transcriptomics               False   \n",
       "7        Ovarian  ARID1A         PGAP2_transcriptomics               False   \n",
       "8        Ovarian  ARID1A   ANKRD20A12P_transcriptomics               False   \n",
       "9        Ovarian  ARID1A      TNFRSF18_transcriptomics               False   \n",
       "..           ...     ...                           ...                 ...   \n",
       "179      Ovarian  ARID1A          FUT6_transcriptomics               False   \n",
       "180      Ovarian  ARID1A       HSD17B2_transcriptomics               False   \n",
       "181      Ovarian  ARID1A         TPRX1_transcriptomics               False   \n",
       "182      Ovarian  ARID1A        TFAP2B_transcriptomics               False   \n",
       "183      Ovarian  ARID1A      MTRNR2L6_transcriptomics               False   \n",
       "184      Ovarian  ARID1A            F3_transcriptomics               False   \n",
       "185      Ovarian  ARID1A       TRIM43B_transcriptomics               False   \n",
       "186      Ovarian  ARID1A       PPP1R3B_transcriptomics               False   \n",
       "187      Ovarian  ARID1A         RFPL1_transcriptomics               False   \n",
       "188      Ovarian  ARID1A         GDF15_transcriptomics               False   \n",
       "189      Ovarian  ARID1A       CEACAM1_transcriptomics               False   \n",
       "190      Ovarian  ARID1A     MTUS2-AS1_transcriptomics               False   \n",
       "191      Ovarian  ARID1A         MFSD4_transcriptomics               False   \n",
       "192      Ovarian  ARID1A         NCALD_transcriptomics               False   \n",
       "193      Ovarian  ARID1A        AVPR1A_transcriptomics               False   \n",
       "194      Ovarian  ARID1A         MESP2_transcriptomics               False   \n",
       "195      Ovarian  ARID1A       SLC43A1_transcriptomics               False   \n",
       "196      Ovarian  ARID1A           PZP_transcriptomics               False   \n",
       "197      Ovarian  ARID1A  LOC100505817_transcriptomics               False   \n",
       "198      Ovarian  ARID1A     LOC643486_transcriptomics               False   \n",
       "199      Ovarian  ARID1A     LINC01245_transcriptomics               False   \n",
       "200      Ovarian  ARID1A         INSM1_transcriptomics               False   \n",
       "201      Ovarian  ARID1A  LOC102724484_transcriptomics               False   \n",
       "202      Ovarian  ARID1A        KIF13B_transcriptomics               False   \n",
       "203      Ovarian  ARID1A         KCNH4_transcriptomics               False   \n",
       "204      Ovarian  ARID1A        KHDC1L_transcriptomics               False   \n",
       "205      Ovarian  ARID1A        HS3ST1_transcriptomics               False   \n",
       "206      Ovarian  ARID1A     LINC00462_transcriptomics               False   \n",
       "207      Ovarian  ARID1A        SEMA3G_transcriptomics               False   \n",
       "208      Ovarian  ARID1A          ANO1_transcriptomics               False   \n",
       "\n",
       "          P_Value  \n",
       "0    6.009689e-08  \n",
       "0    7.546881e-09  \n",
       "1    7.544245e-08  \n",
       "2    1.324795e-07  \n",
       "3    1.683441e-07  \n",
       "4    3.737113e-07  \n",
       "5    3.860434e-07  \n",
       "6    8.352878e-07  \n",
       "7    9.668603e-07  \n",
       "8    1.308609e-06  \n",
       "9    1.368058e-06  \n",
       "10   1.567902e-06  \n",
       "0    3.068549e-08  \n",
       "1    6.590190e-07  \n",
       "2    6.897007e-07  \n",
       "3    1.216561e-06  \n",
       "4    1.400440e-06  \n",
       "5    1.529432e-06  \n",
       "6    2.573590e-06  \n",
       "7    3.370621e-06  \n",
       "0    4.806922e-16  \n",
       "1    2.998791e-15  \n",
       "2    2.826724e-14  \n",
       "3    4.168296e-13  \n",
       "4    5.061682e-13  \n",
       "5    5.713628e-13  \n",
       "6    1.865577e-12  \n",
       "7    3.420800e-12  \n",
       "8    9.261376e-12  \n",
       "9    9.514260e-12  \n",
       "..            ...  \n",
       "179  6.739185e-07  \n",
       "180  7.309743e-07  \n",
       "181  7.323498e-07  \n",
       "182  7.441152e-07  \n",
       "183  7.453821e-07  \n",
       "184  8.296980e-07  \n",
       "185  8.299754e-07  \n",
       "186  8.330275e-07  \n",
       "187  8.986442e-07  \n",
       "188  9.257919e-07  \n",
       "189  9.589350e-07  \n",
       "190  1.063140e-06  \n",
       "191  1.078302e-06  \n",
       "192  1.286969e-06  \n",
       "193  1.313588e-06  \n",
       "194  1.353499e-06  \n",
       "195  1.403135e-06  \n",
       "196  1.422187e-06  \n",
       "197  1.437350e-06  \n",
       "198  1.446004e-06  \n",
       "199  1.531206e-06  \n",
       "200  1.536219e-06  \n",
       "201  1.648421e-06  \n",
       "202  1.650590e-06  \n",
       "203  1.666373e-06  \n",
       "204  1.673755e-06  \n",
       "205  1.696851e-06  \n",
       "206  1.753059e-06  \n",
       "207  2.101998e-06  \n",
       "208  2.154795e-06  \n",
       "\n",
       "[229 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if len(all_significant_comparisons) == 0:\n",
    "    print('No Significant Comparisons!')\n",
    "    \n",
    "else:\n",
    "    display(all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Write Significant Comparisons (if any) to Shared CSV file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "existing_results = pd.read_csv(gene+'_Trans_Results.csv')\n",
    "\n",
    "updated_results = pd.concat([existing_results, all_significant_comparisons], sort=False)\n",
    "\n",
    "updated_results.to_csv(path_or_buf = gene + '_Trans_Results.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
