{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# KRAS Mutation Trans Effect on Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook analyzes the trans effect of KRAS mutation on interacting and other proteins Transcriptomics, in Endometrial and Colon cancer."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Library Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that index is up-to-date...Loading acetylproteomics data...Loading clinical data...Loading CNA data...Loading definitions data...Loading miRNA data...Loading phosphoproteomics_gene data...Loading phosphoproteomics_site data...Loading proteomics data...Loading somatic_binary data...Loading somatic data...Loading transcriptomics_circular data...Loading transcriptomics_linear data...Formatting dataframes...Checking that index is up-to-date...Loading clinical data...Loading miRNA data...Loading mutation_binary data...Loading mutation data...Loading phosphoproteomics_normal data...Loading phosphoproteomics_tumor data...Loading proteomics_normal data...Loading proteomics_tumor data...Loading transcriptomics data...Formatting dataframes..."
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "import re\n",
    "import gseapy as gp\n",
    "from gseapy.plot import barplot, dotplot\n",
    "\n",
    "import cptac\n",
    "import cptac.algorithms as al\n",
    "\n",
    "en = cptac.Endometrial()\n",
    "co = cptac.Colon()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Investigate KRAS gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = \"KRAS\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Investigate Proteomics, Phosphoproteomics, Acetylproteomics, or Transcriptomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#omics = \"proteomics\"\n",
    "omics = \"transcriptomics\"\n",
    "#omics = \"phosphoproteomics\"\n",
    "#omics = \"acetylproteomics\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Interacting Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate interacting protein list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generating interacting protein list\n",
      "Interacting Proteins:\n",
      "RAF1\n",
      "MAPK1\n",
      "KRAS\n",
      "MAP2K2\n",
      "MAPK3\n",
      "PIK3CA\n",
      "ERBB3\n",
      "ERBB2\n",
      "EGFR\n",
      "BRAF\n",
      "PIK3CB\n",
      "ARAF\n",
      "MAP2K1\n",
      "PTPN11\n",
      "NF1\n",
      "PIK3CG\n",
      "NRAS\n",
      "SRC\n",
      "GRB2\n",
      "SOS1\n",
      "HRAS\n",
      "CDKN2A\n",
      "RAP1GDS1\n",
      "HNRNPC\n",
      "RASSF2\n",
      "RASSF5\n"
     ]
    }
   ],
   "source": [
    "'''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "omics_object = en.get_proteomics()\n",
    "\n",
    "print(\"Generating interacting protein list\")\n",
    "interacting_proteins_in_omics_df = []\n",
    "\n",
    "'''Only do comparisons on proteins in the omics dataframe'''\n",
    "for ip in interacting_proteins:\n",
    "    if omics == 'phosphoproteomics' or omics == 'acetylproteomics':\n",
    "        col_regex = ip + \"-.*\" # Build a regex to get all columns that match the gene\n",
    "    else:\n",
    "        col_regex = '^{}$'.format(ip)\n",
    "\n",
    "    selected = omics_object.filter(regex=col_regex)\n",
    "\n",
    "    if len(selected.columns) > 0:\n",
    "        interacting_proteins_in_omics_df.append(ip)\n",
    "\n",
    "print(\"Interacting Proteins:\")\n",
    "for interacting_protein in interacting_proteins_in_omics_df:\n",
    "    print(interacting_protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = en.append_mutations_to_omics(mutation_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins_in_omics_df)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate interacting protein list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generating interacting protein list\n",
      "Interacting Proteins:\n",
      "RAF1\n",
      "MAPK1\n",
      "KRAS\n",
      "MAP2K2\n",
      "MAPK3\n",
      "PIK3CA\n",
      "ERBB3\n",
      "ERBB2\n",
      "EGFR\n",
      "BRAF\n",
      "PIK3CB\n",
      "ARAF\n",
      "MAP2K1\n",
      "PTPN11\n",
      "NF1\n",
      "NRAS\n",
      "SRC\n",
      "GRB2\n",
      "SOS1\n",
      "HRAS\n",
      "CDKN2A\n",
      "RAP1GDS1\n",
      "HNRNPC\n",
      "RASSF2\n",
      "RASSF5\n"
     ]
    }
   ],
   "source": [
    "'''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "omics_object = co.get_proteomics()\n",
    "\n",
    "print(\"Generating interacting protein list\")\n",
    "interacting_proteins_in_omics_df = []\n",
    "\n",
    "'''Only do comparisons on proteins in the omics dataframe'''\n",
    "for ip in interacting_proteins:\n",
    "    if omics == 'phosphoproteomics' or omics == 'acetylproteomics':\n",
    "        col_regex = ip + \"-.*\" # Build a regex to get all columns that match the gene\n",
    "    else:\n",
    "        col_regex = '^{}$'.format(ip)\n",
    "\n",
    "    selected = omics_object.filter(regex=col_regex)\n",
    "\n",
    "    if len(selected.columns) > 0:\n",
    "        interacting_proteins_in_omics_df.append(ip)\n",
    "\n",
    "print(\"Interacting Proteins:\")\n",
    "for interacting_protein in interacting_proteins_in_omics_df:\n",
    "    print(interacting_protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = co.append_mutations_to_omics(mutation_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins_in_omics_df)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  KRAS\n",
      "MT-ATP6 did not match any columns in transcriptomics dataframe. MT-ATP6_transcriptomics column inserted, but filled with NaN.\n",
      "MT-ATP8 did not match any columns in transcriptomics dataframe. MT-ATP8_transcriptomics column inserted, but filled with NaN.\n",
      "MT-CO1 did not match any columns in transcriptomics dataframe. MT-CO1_transcriptomics column inserted, but filled with NaN.\n",
      "MT-CO2 did not match any columns in transcriptomics dataframe. MT-CO2_transcriptomics column inserted, but filled with NaN.\n",
      "MT-CO3 did not match any columns in transcriptomics dataframe. MT-CO3_transcriptomics column inserted, but filled with NaN.\n",
      "MT-CYB did not match any columns in transcriptomics dataframe. MT-CYB_transcriptomics column inserted, but filled with NaN.\n",
      "MT-ND1 did not match any columns in transcriptomics dataframe. MT-ND1_transcriptomics column inserted, but filled with NaN.\n",
      "MT-ND2 did not match any columns in transcriptomics dataframe. MT-ND2_transcriptomics column inserted, but filled with NaN.\n",
      "MT-ND3 did not match any columns in transcriptomics dataframe. MT-ND3_transcriptomics column inserted, but filled with NaN.\n",
      "MT-ND4 did not match any columns in transcriptomics dataframe. MT-ND4_transcriptomics column inserted, but filled with NaN.\n",
      "MT-ND4L did not match any columns in transcriptomics dataframe. MT-ND4L_transcriptomics column inserted, but filled with NaN.\n",
      "MT-ND5 did not match any columns in transcriptomics dataframe. MT-ND5_transcriptomics column inserted, but filled with NaN.\n",
      "MT-ND6 did not match any columns in transcriptomics dataframe. MT-ND6_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "              Comparison       P_Value\n",
      "0  SPRY1_transcriptomics  2.500902e-07\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "    proteomics = en.get_proteomics()\n",
    "    all_proteins = proteomics.columns\n",
    "    #all_proteins = all_proteins[:100]\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.append_mutations_to_omics(mutation_genes=[gene], omics_df_name=omics, omics_genes=all_proteins)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  KRAS\n",
      "A1BG did not match any columns in transcriptomics dataframe. A1BG_transcriptomics column inserted, but filled with NaN.\n",
      "ABCA8 did not match any columns in transcriptomics dataframe. ABCA8_transcriptomics column inserted, but filled with NaN.\n",
      "ACAN did not match any columns in transcriptomics dataframe. ACAN_transcriptomics column inserted, but filled with NaN.\n",
      "ACTA1 did not match any columns in transcriptomics dataframe. ACTA1_transcriptomics column inserted, but filled with NaN.\n",
      "ACTBL2 did not match any columns in transcriptomics dataframe. ACTBL2_transcriptomics column inserted, but filled with NaN.\n",
      "ACTL8 did not match any columns in transcriptomics dataframe. ACTL8_transcriptomics column inserted, but filled with NaN.\n",
      "ACTN2 did not match any columns in transcriptomics dataframe. ACTN2_transcriptomics column inserted, but filled with NaN.\n",
      "ACTN3 did not match any columns in transcriptomics dataframe. ACTN3_transcriptomics column inserted, but filled with NaN.\n",
      "ADAMTS16 did not match any columns in transcriptomics dataframe. ADAMTS16_transcriptomics column inserted, but filled with NaN.\n",
      "ADAMTSL4 did not match any columns in transcriptomics dataframe. ADAMTSL4_transcriptomics column inserted, but filled with NaN.\n",
      "ADD2 did not match any columns in transcriptomics dataframe. ADD2_transcriptomics column inserted, but filled with NaN.\n",
      "ADH1A did not match any columns in transcriptomics dataframe. ADH1A_transcriptomics column inserted, but filled with NaN.\n",
      "ADH4 did not match any columns in transcriptomics dataframe. ADH4_transcriptomics column inserted, but filled with NaN.\n",
      "ADIPOQ did not match any columns in transcriptomics dataframe. ADIPOQ_transcriptomics column inserted, but filled with NaN.\n",
      "AFM did not match any columns in transcriptomics dataframe. AFM_transcriptomics column inserted, but filled with NaN.\n",
      "AHSG did not match any columns in transcriptomics dataframe. AHSG_transcriptomics column inserted, but filled with NaN.\n",
      "AHSP did not match any columns in transcriptomics dataframe. AHSP_transcriptomics column inserted, but filled with NaN.\n",
      "AKR1C4 did not match any columns in transcriptomics dataframe. AKR1C4_transcriptomics column inserted, but filled with NaN.\n",
      "AKR1D1 did not match any columns in transcriptomics dataframe. AKR1D1_transcriptomics column inserted, but filled with NaN.\n",
      "ALB did not match any columns in transcriptomics dataframe. ALB_transcriptomics column inserted, but filled with NaN.\n",
      "ALDH1A2 did not match any columns in transcriptomics dataframe. ALDH1A2_transcriptomics column inserted, but filled with NaN.\n",
      "ALOX15 did not match any columns in transcriptomics dataframe. ALOX15_transcriptomics column inserted, but filled with NaN.\n",
      "ALPI did not match any columns in transcriptomics dataframe. ALPI_transcriptomics column inserted, but filled with NaN.\n",
      "AMBP did not match any columns in transcriptomics dataframe. AMBP_transcriptomics column inserted, but filled with NaN.\n",
      "AMY1B did not match any columns in transcriptomics dataframe. AMY1B_transcriptomics column inserted, but filled with NaN.\n",
      "AMY2A did not match any columns in transcriptomics dataframe. AMY2A_transcriptomics column inserted, but filled with NaN.\n",
      "ANK1 did not match any columns in transcriptomics dataframe. ANK1_transcriptomics column inserted, but filled with NaN.\n",
      "AP3B2 did not match any columns in transcriptomics dataframe. AP3B2_transcriptomics column inserted, but filled with NaN.\n",
      "APCS did not match any columns in transcriptomics dataframe. APCS_transcriptomics column inserted, but filled with NaN.\n",
      "APOA1 did not match any columns in transcriptomics dataframe. APOA1_transcriptomics column inserted, but filled with NaN.\n",
      "APOA2 did not match any columns in transcriptomics dataframe. APOA2_transcriptomics column inserted, but filled with NaN.\n",
      "APOA4 did not match any columns in transcriptomics dataframe. APOA4_transcriptomics column inserted, but filled with NaN.\n",
      "APOB did not match any columns in transcriptomics dataframe. APOB_transcriptomics column inserted, but filled with NaN.\n",
      "APOC3 did not match any columns in transcriptomics dataframe. APOC3_transcriptomics column inserted, but filled with NaN.\n",
      "APOC4 did not match any columns in transcriptomics dataframe. APOC4_transcriptomics column inserted, but filled with NaN.\n",
      "APOF did not match any columns in transcriptomics dataframe. APOF_transcriptomics column inserted, but filled with NaN.\n",
      "APOH did not match any columns in transcriptomics dataframe. APOH_transcriptomics column inserted, but filled with NaN.\n",
      "ARG1 did not match any columns in transcriptomics dataframe. ARG1_transcriptomics column inserted, but filled with NaN.\n",
      "ARR3 did not match any columns in transcriptomics dataframe. ARR3_transcriptomics column inserted, but filled with NaN.\n",
      "ASPA did not match any columns in transcriptomics dataframe. ASPA_transcriptomics column inserted, but filled with NaN.\n",
      "ATP12A did not match any columns in transcriptomics dataframe. ATP12A_transcriptomics column inserted, but filled with NaN.\n",
      "ATP1A2 did not match any columns in transcriptomics dataframe. ATP1A2_transcriptomics column inserted, but filled with NaN.\n",
      "ATP1A3 did not match any columns in transcriptomics dataframe. ATP1A3_transcriptomics column inserted, but filled with NaN.\n",
      "ATP1A4 did not match any columns in transcriptomics dataframe. ATP1A4_transcriptomics column inserted, but filled with NaN.\n",
      "ATP1B2 did not match any columns in transcriptomics dataframe. ATP1B2_transcriptomics column inserted, but filled with NaN.\n",
      "ATP2A1 did not match any columns in transcriptomics dataframe. ATP2A1_transcriptomics column inserted, but filled with NaN.\n",
      "ATP2B2 did not match any columns in transcriptomics dataframe. ATP2B2_transcriptomics column inserted, but filled with NaN.\n",
      "ATP2B3 did not match any columns in transcriptomics dataframe. ATP2B3_transcriptomics column inserted, but filled with NaN.\n",
      "ATP4A did not match any columns in transcriptomics dataframe. ATP4A_transcriptomics column inserted, but filled with NaN.\n",
      "ATP6V1G2 did not match any columns in transcriptomics dataframe. ATP6V1G2_transcriptomics column inserted, but filled with NaN.\n",
      "AZU1 did not match any columns in transcriptomics dataframe. AZU1_transcriptomics column inserted, but filled with NaN.\n",
      "B3GAT2 did not match any columns in transcriptomics dataframe. B3GAT2_transcriptomics column inserted, but filled with NaN.\n",
      "B4GALNT2 did not match any columns in transcriptomics dataframe. B4GALNT2_transcriptomics column inserted, but filled with NaN.\n",
      "BCHE did not match any columns in transcriptomics dataframe. BCHE_transcriptomics column inserted, but filled with NaN.\n",
      "BEST4 did not match any columns in transcriptomics dataframe. BEST4_transcriptomics column inserted, but filled with NaN.\n",
      "BPI did not match any columns in transcriptomics dataframe. BPI_transcriptomics column inserted, but filled with NaN.\n",
      "C17orf97 did not match any columns in transcriptomics dataframe. C17orf97_transcriptomics column inserted, but filled with NaN.\n",
      "C1orf194 did not match any columns in transcriptomics dataframe. C1orf194_transcriptomics column inserted, but filled with NaN.\n",
      "C20orf78 did not match any columns in transcriptomics dataframe. C20orf78_transcriptomics column inserted, but filled with NaN.\n",
      "C6 did not match any columns in transcriptomics dataframe. C6_transcriptomics column inserted, but filled with NaN.\n",
      "C6orf15 did not match any columns in transcriptomics dataframe. C6orf15_transcriptomics column inserted, but filled with NaN.\n",
      "C8A did not match any columns in transcriptomics dataframe. C8A_transcriptomics column inserted, but filled with NaN.\n",
      "C8B did not match any columns in transcriptomics dataframe. C8B_transcriptomics column inserted, but filled with NaN.\n",
      "C8G did not match any columns in transcriptomics dataframe. C8G_transcriptomics column inserted, but filled with NaN.\n",
      "C9 did not match any columns in transcriptomics dataframe. C9_transcriptomics column inserted, but filled with NaN.\n",
      "CA1 did not match any columns in transcriptomics dataframe. CA1_transcriptomics column inserted, but filled with NaN.\n",
      "CA3 did not match any columns in transcriptomics dataframe. CA3_transcriptomics column inserted, but filled with NaN.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CA7 did not match any columns in transcriptomics dataframe. CA7_transcriptomics column inserted, but filled with NaN.\n",
      "CADM2 did not match any columns in transcriptomics dataframe. CADM2_transcriptomics column inserted, but filled with NaN.\n",
      "CADM3 did not match any columns in transcriptomics dataframe. CADM3_transcriptomics column inserted, but filled with NaN.\n",
      "CALB1 did not match any columns in transcriptomics dataframe. CALB1_transcriptomics column inserted, but filled with NaN.\n",
      "CALML3 did not match any columns in transcriptomics dataframe. CALML3_transcriptomics column inserted, but filled with NaN.\n",
      "CALML5 did not match any columns in transcriptomics dataframe. CALML5_transcriptomics column inserted, but filled with NaN.\n",
      "CALR3 did not match any columns in transcriptomics dataframe. CALR3_transcriptomics column inserted, but filled with NaN.\n",
      "CAMK2B did not match any columns in transcriptomics dataframe. CAMK2B_transcriptomics column inserted, but filled with NaN.\n",
      "CAMK4 did not match any columns in transcriptomics dataframe. CAMK4_transcriptomics column inserted, but filled with NaN.\n",
      "CAMP did not match any columns in transcriptomics dataframe. CAMP_transcriptomics column inserted, but filled with NaN.\n",
      "CAND2 did not match any columns in transcriptomics dataframe. CAND2_transcriptomics column inserted, but filled with NaN.\n",
      "CASQ2 did not match any columns in transcriptomics dataframe. CASQ2_transcriptomics column inserted, but filled with NaN.\n",
      "CASTOR2 did not match any columns in transcriptomics dataframe. CASTOR2_transcriptomics column inserted, but filled with NaN.\n",
      "CCDC140 did not match any columns in transcriptomics dataframe. CCDC140_transcriptomics column inserted, but filled with NaN.\n",
      "CD5L did not match any columns in transcriptomics dataframe. CD5L_transcriptomics column inserted, but filled with NaN.\n",
      "CDH19 did not match any columns in transcriptomics dataframe. CDH19_transcriptomics column inserted, but filled with NaN.\n",
      "CDH2 did not match any columns in transcriptomics dataframe. CDH2_transcriptomics column inserted, but filled with NaN.\n",
      "CEACAM8 did not match any columns in transcriptomics dataframe. CEACAM8_transcriptomics column inserted, but filled with NaN.\n",
      "CELA2A did not match any columns in transcriptomics dataframe. CELA2A_transcriptomics column inserted, but filled with NaN.\n",
      "CELA3A did not match any columns in transcriptomics dataframe. CELA3A_transcriptomics column inserted, but filled with NaN.\n",
      "CELA3B did not match any columns in transcriptomics dataframe. CELA3B_transcriptomics column inserted, but filled with NaN.\n",
      "CEND1 did not match any columns in transcriptomics dataframe. CEND1_transcriptomics column inserted, but filled with NaN.\n",
      "CFAP65 did not match any columns in transcriptomics dataframe. CFAP65_transcriptomics column inserted, but filled with NaN.\n",
      "CFHR1 did not match any columns in transcriptomics dataframe. CFHR1_transcriptomics column inserted, but filled with NaN.\n",
      "CFHR2 did not match any columns in transcriptomics dataframe. CFHR2_transcriptomics column inserted, but filled with NaN.\n",
      "CFHR3 did not match any columns in transcriptomics dataframe. CFHR3_transcriptomics column inserted, but filled with NaN.\n",
      "CFHR4 did not match any columns in transcriptomics dataframe. CFHR4_transcriptomics column inserted, but filled with NaN.\n",
      "CFHR5 did not match any columns in transcriptomics dataframe. CFHR5_transcriptomics column inserted, but filled with NaN.\n",
      "CHD5 did not match any columns in transcriptomics dataframe. CHD5_transcriptomics column inserted, but filled with NaN.\n",
      "CHGA did not match any columns in transcriptomics dataframe. CHGA_transcriptomics column inserted, but filled with NaN.\n",
      "CHGB did not match any columns in transcriptomics dataframe. CHGB_transcriptomics column inserted, but filled with NaN.\n",
      "CHST5 did not match any columns in transcriptomics dataframe. CHST5_transcriptomics column inserted, but filled with NaN.\n",
      "CKM did not match any columns in transcriptomics dataframe. CKM_transcriptomics column inserted, but filled with NaN.\n",
      "CLDN18 did not match any columns in transcriptomics dataframe. CLDN18_transcriptomics column inserted, but filled with NaN.\n",
      "CLGN did not match any columns in transcriptomics dataframe. CLGN_transcriptomics column inserted, but filled with NaN.\n",
      "CLIC6 did not match any columns in transcriptomics dataframe. CLIC6_transcriptomics column inserted, but filled with NaN.\n",
      "CLTCL1 did not match any columns in transcriptomics dataframe. CLTCL1_transcriptomics column inserted, but filled with NaN.\n",
      "CMA1 did not match any columns in transcriptomics dataframe. CMA1_transcriptomics column inserted, but filled with NaN.\n",
      "CMTM5 did not match any columns in transcriptomics dataframe. CMTM5_transcriptomics column inserted, but filled with NaN.\n",
      "CNTN1 did not match any columns in transcriptomics dataframe. CNTN1_transcriptomics column inserted, but filled with NaN.\n",
      "COL21A1 did not match any columns in transcriptomics dataframe. COL21A1_transcriptomics column inserted, but filled with NaN.\n",
      "COL28A1 did not match any columns in transcriptomics dataframe. COL28A1_transcriptomics column inserted, but filled with NaN.\n",
      "COL2A1 did not match any columns in transcriptomics dataframe. COL2A1_transcriptomics column inserted, but filled with NaN.\n",
      "COL4A5 did not match any columns in transcriptomics dataframe. COL4A5_transcriptomics column inserted, but filled with NaN.\n",
      "COLEC10 did not match any columns in transcriptomics dataframe. COLEC10_transcriptomics column inserted, but filled with NaN.\n",
      "CORO2B did not match any columns in transcriptomics dataframe. CORO2B_transcriptomics column inserted, but filled with NaN.\n",
      "CORO6 did not match any columns in transcriptomics dataframe. CORO6_transcriptomics column inserted, but filled with NaN.\n",
      "CORO7-PAM16 did not match any columns in transcriptomics dataframe. CORO7-PAM16_transcriptomics column inserted, but filled with NaN.\n",
      "CPA1 did not match any columns in transcriptomics dataframe. CPA1_transcriptomics column inserted, but filled with NaN.\n",
      "CPA6 did not match any columns in transcriptomics dataframe. CPA6_transcriptomics column inserted, but filled with NaN.\n",
      "CPB1 did not match any columns in transcriptomics dataframe. CPB1_transcriptomics column inserted, but filled with NaN.\n",
      "CPB2 did not match any columns in transcriptomics dataframe. CPB2_transcriptomics column inserted, but filled with NaN.\n",
      "CPLX2 did not match any columns in transcriptomics dataframe. CPLX2_transcriptomics column inserted, but filled with NaN.\n",
      "CPN1 did not match any columns in transcriptomics dataframe. CPN1_transcriptomics column inserted, but filled with NaN.\n",
      "CPN2 did not match any columns in transcriptomics dataframe. CPN2_transcriptomics column inserted, but filled with NaN.\n",
      "CPO did not match any columns in transcriptomics dataframe. CPO_transcriptomics column inserted, but filled with NaN.\n",
      "CPS1 did not match any columns in transcriptomics dataframe. CPS1_transcriptomics column inserted, but filled with NaN.\n",
      "CR2 did not match any columns in transcriptomics dataframe. CR2_transcriptomics column inserted, but filled with NaN.\n",
      "CRABP1 did not match any columns in transcriptomics dataframe. CRABP1_transcriptomics column inserted, but filled with NaN.\n",
      "CRISP3 did not match any columns in transcriptomics dataframe. CRISP3_transcriptomics column inserted, but filled with NaN.\n",
      "CRMP1 did not match any columns in transcriptomics dataframe. CRMP1_transcriptomics column inserted, but filled with NaN.\n",
      "CRP did not match any columns in transcriptomics dataframe. CRP_transcriptomics column inserted, but filled with NaN.\n",
      "CRTAC1 did not match any columns in transcriptomics dataframe. CRTAC1_transcriptomics column inserted, but filled with NaN.\n",
      "CSPG4 did not match any columns in transcriptomics dataframe. CSPG4_transcriptomics column inserted, but filled with NaN.\n",
      "CSRP3 did not match any columns in transcriptomics dataframe. CSRP3_transcriptomics column inserted, but filled with NaN.\n",
      "CTCFL did not match any columns in transcriptomics dataframe. CTCFL_transcriptomics column inserted, but filled with NaN.\n",
      "CTNNA2 did not match any columns in transcriptomics dataframe. CTNNA2_transcriptomics column inserted, but filled with NaN.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CTNNA3 did not match any columns in transcriptomics dataframe. CTNNA3_transcriptomics column inserted, but filled with NaN.\n",
      "CTRB2 did not match any columns in transcriptomics dataframe. CTRB2_transcriptomics column inserted, but filled with NaN.\n",
      "CTRC did not match any columns in transcriptomics dataframe. CTRC_transcriptomics column inserted, but filled with NaN.\n",
      "CYLC2 did not match any columns in transcriptomics dataframe. CYLC2_transcriptomics column inserted, but filled with NaN.\n",
      "CYP3A4 did not match any columns in transcriptomics dataframe. CYP3A4_transcriptomics column inserted, but filled with NaN.\n",
      "CYP4F8 did not match any columns in transcriptomics dataframe. CYP4F8_transcriptomics column inserted, but filled with NaN.\n",
      "CYS1 did not match any columns in transcriptomics dataframe. CYS1_transcriptomics column inserted, but filled with NaN.\n",
      "DAPL1 did not match any columns in transcriptomics dataframe. DAPL1_transcriptomics column inserted, but filled with NaN.\n",
      "DCLK1 did not match any columns in transcriptomics dataframe. DCLK1_transcriptomics column inserted, but filled with NaN.\n",
      "DDI1 did not match any columns in transcriptomics dataframe. DDI1_transcriptomics column inserted, but filled with NaN.\n",
      "DDX3Y did not match any columns in transcriptomics dataframe. DDX3Y_transcriptomics column inserted, but filled with NaN.\n",
      "DEFA1B did not match any columns in transcriptomics dataframe. DEFA1B_transcriptomics column inserted, but filled with NaN.\n",
      "DEFA3 did not match any columns in transcriptomics dataframe. DEFA3_transcriptomics column inserted, but filled with NaN.\n",
      "DEFA4 did not match any columns in transcriptomics dataframe. DEFA4_transcriptomics column inserted, but filled with NaN.\n",
      "DEFB104B did not match any columns in transcriptomics dataframe. DEFB104B_transcriptomics column inserted, but filled with NaN.\n",
      "DEFB121 did not match any columns in transcriptomics dataframe. DEFB121_transcriptomics column inserted, but filled with NaN.\n",
      "DHRS2 did not match any columns in transcriptomics dataframe. DHRS2_transcriptomics column inserted, but filled with NaN.\n",
      "DKK4 did not match any columns in transcriptomics dataframe. DKK4_transcriptomics column inserted, but filled with NaN.\n",
      "DNM3 did not match any columns in transcriptomics dataframe. DNM3_transcriptomics column inserted, but filled with NaN.\n",
      "DPYSL4 did not match any columns in transcriptomics dataframe. DPYSL4_transcriptomics column inserted, but filled with NaN.\n",
      "DSC3 did not match any columns in transcriptomics dataframe. DSC3_transcriptomics column inserted, but filled with NaN.\n",
      "DSG3 did not match any columns in transcriptomics dataframe. DSG3_transcriptomics column inserted, but filled with NaN.\n",
      "DYNC1I1 did not match any columns in transcriptomics dataframe. DYNC1I1_transcriptomics column inserted, but filled with NaN.\n",
      "ECHDC3 did not match any columns in transcriptomics dataframe. ECHDC3_transcriptomics column inserted, but filled with NaN.\n",
      "EEF1A2 did not match any columns in transcriptomics dataframe. EEF1A2_transcriptomics column inserted, but filled with NaN.\n",
      "EFHD1 did not match any columns in transcriptomics dataframe. EFHD1_transcriptomics column inserted, but filled with NaN.\n",
      "EFNA5 did not match any columns in transcriptomics dataframe. EFNA5_transcriptomics column inserted, but filled with NaN.\n",
      "EIF1AY did not match any columns in transcriptomics dataframe. EIF1AY_transcriptomics column inserted, but filled with NaN.\n",
      "ELANE did not match any columns in transcriptomics dataframe. ELANE_transcriptomics column inserted, but filled with NaN.\n",
      "EMILIN3 did not match any columns in transcriptomics dataframe. EMILIN3_transcriptomics column inserted, but filled with NaN.\n",
      "ENPP7 did not match any columns in transcriptomics dataframe. ENPP7_transcriptomics column inserted, but filled with NaN.\n",
      "EPB42 did not match any columns in transcriptomics dataframe. EPB42_transcriptomics column inserted, but filled with NaN.\n",
      "EPHA6 did not match any columns in transcriptomics dataframe. EPHA6_transcriptomics column inserted, but filled with NaN.\n",
      "EPHA7 did not match any columns in transcriptomics dataframe. EPHA7_transcriptomics column inserted, but filled with NaN.\n",
      "EPX did not match any columns in transcriptomics dataframe. EPX_transcriptomics column inserted, but filled with NaN.\n",
      "EXOC3L2 did not match any columns in transcriptomics dataframe. EXOC3L2_transcriptomics column inserted, but filled with NaN.\n",
      "F11 did not match any columns in transcriptomics dataframe. F11_transcriptomics column inserted, but filled with NaN.\n",
      "F13B did not match any columns in transcriptomics dataframe. F13B_transcriptomics column inserted, but filled with NaN.\n",
      "F2 did not match any columns in transcriptomics dataframe. F2_transcriptomics column inserted, but filled with NaN.\n",
      "F8A2 did not match any columns in transcriptomics dataframe. F8A2_transcriptomics column inserted, but filled with NaN.\n",
      "F9 did not match any columns in transcriptomics dataframe. F9_transcriptomics column inserted, but filled with NaN.\n",
      "FABP7 did not match any columns in transcriptomics dataframe. FABP7_transcriptomics column inserted, but filled with NaN.\n",
      "FAM162B did not match any columns in transcriptomics dataframe. FAM162B_transcriptomics column inserted, but filled with NaN.\n",
      "FBN2 did not match any columns in transcriptomics dataframe. FBN2_transcriptomics column inserted, but filled with NaN.\n",
      "FBN3 did not match any columns in transcriptomics dataframe. FBN3_transcriptomics column inserted, but filled with NaN.\n",
      "FBP2 did not match any columns in transcriptomics dataframe. FBP2_transcriptomics column inserted, but filled with NaN.\n",
      "FCAR did not match any columns in transcriptomics dataframe. FCAR_transcriptomics column inserted, but filled with NaN.\n",
      "FCN1 did not match any columns in transcriptomics dataframe. FCN1_transcriptomics column inserted, but filled with NaN.\n",
      "FCN2 did not match any columns in transcriptomics dataframe. FCN2_transcriptomics column inserted, but filled with NaN.\n",
      "FETUB did not match any columns in transcriptomics dataframe. FETUB_transcriptomics column inserted, but filled with NaN.\n",
      "FGA did not match any columns in transcriptomics dataframe. FGA_transcriptomics column inserted, but filled with NaN.\n",
      "FGB did not match any columns in transcriptomics dataframe. FGB_transcriptomics column inserted, but filled with NaN.\n",
      "FGF1 did not match any columns in transcriptomics dataframe. FGF1_transcriptomics column inserted, but filled with NaN.\n",
      "FGG did not match any columns in transcriptomics dataframe. FGG_transcriptomics column inserted, but filled with NaN.\n",
      "FILIP1 did not match any columns in transcriptomics dataframe. FILIP1_transcriptomics column inserted, but filled with NaN.\n",
      "GAP43 did not match any columns in transcriptomics dataframe. GAP43_transcriptomics column inserted, but filled with NaN.\n",
      "GAPDHS did not match any columns in transcriptomics dataframe. GAPDHS_transcriptomics column inserted, but filled with NaN.\n",
      "GBP7 did not match any columns in transcriptomics dataframe. GBP7_transcriptomics column inserted, but filled with NaN.\n",
      "GC did not match any columns in transcriptomics dataframe. GC_transcriptomics column inserted, but filled with NaN.\n",
      "GCG did not match any columns in transcriptomics dataframe. GCG_transcriptomics column inserted, but filled with NaN.\n",
      "GFAP did not match any columns in transcriptomics dataframe. GFAP_transcriptomics column inserted, but filled with NaN.\n",
      "GFRA1 did not match any columns in transcriptomics dataframe. GFRA1_transcriptomics column inserted, but filled with NaN.\n",
      "GIPC3 did not match any columns in transcriptomics dataframe. GIPC3_transcriptomics column inserted, but filled with NaN.\n",
      "GK2 did not match any columns in transcriptomics dataframe. GK2_transcriptomics column inserted, but filled with NaN.\n",
      "GNAO1 did not match any columns in transcriptomics dataframe. GNAO1_transcriptomics column inserted, but filled with NaN.\n",
      "GNAZ did not match any columns in transcriptomics dataframe. GNAZ_transcriptomics column inserted, but filled with NaN.\n",
      "GNB3 did not match any columns in transcriptomics dataframe. GNB3_transcriptomics column inserted, but filled with NaN.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GNG7 did not match any columns in transcriptomics dataframe. GNG7_transcriptomics column inserted, but filled with NaN.\n",
      "GNRH2 did not match any columns in transcriptomics dataframe. GNRH2_transcriptomics column inserted, but filled with NaN.\n",
      "GP1BB did not match any columns in transcriptomics dataframe. GP1BB_transcriptomics column inserted, but filled with NaN.\n",
      "GP2 did not match any columns in transcriptomics dataframe. GP2_transcriptomics column inserted, but filled with NaN.\n",
      "GP9 did not match any columns in transcriptomics dataframe. GP9_transcriptomics column inserted, but filled with NaN.\n",
      "GPX6 did not match any columns in transcriptomics dataframe. GPX6_transcriptomics column inserted, but filled with NaN.\n",
      "GREM2 did not match any columns in transcriptomics dataframe. GREM2_transcriptomics column inserted, but filled with NaN.\n",
      "GSDMA did not match any columns in transcriptomics dataframe. GSDMA_transcriptomics column inserted, but filled with NaN.\n",
      "GSTT2 did not match any columns in transcriptomics dataframe. GSTT2_transcriptomics column inserted, but filled with NaN.\n",
      "GYPA did not match any columns in transcriptomics dataframe. GYPA_transcriptomics column inserted, but filled with NaN.\n",
      "HABP2 did not match any columns in transcriptomics dataframe. HABP2_transcriptomics column inserted, but filled with NaN.\n",
      "HAPLN1 did not match any columns in transcriptomics dataframe. HAPLN1_transcriptomics column inserted, but filled with NaN.\n",
      "HBD did not match any columns in transcriptomics dataframe. HBD_transcriptomics column inserted, but filled with NaN.\n",
      "HBE1 did not match any columns in transcriptomics dataframe. HBE1_transcriptomics column inserted, but filled with NaN.\n",
      "HBG1 did not match any columns in transcriptomics dataframe. HBG1_transcriptomics column inserted, but filled with NaN.\n",
      "HBG2 did not match any columns in transcriptomics dataframe. HBG2_transcriptomics column inserted, but filled with NaN.\n",
      "HBM did not match any columns in transcriptomics dataframe. HBM_transcriptomics column inserted, but filled with NaN.\n",
      "HBQ1 did not match any columns in transcriptomics dataframe. HBQ1_transcriptomics column inserted, but filled with NaN.\n",
      "HBZ did not match any columns in transcriptomics dataframe. HBZ_transcriptomics column inserted, but filled with NaN.\n",
      "HIST1H1A did not match any columns in transcriptomics dataframe. HIST1H1A_transcriptomics column inserted, but filled with NaN.\n",
      "HIST1H1B did not match any columns in transcriptomics dataframe. HIST1H1B_transcriptomics column inserted, but filled with NaN.\n",
      "HIST1H1D did not match any columns in transcriptomics dataframe. HIST1H1D_transcriptomics column inserted, but filled with NaN.\n",
      "HIST1H1E did not match any columns in transcriptomics dataframe. HIST1H1E_transcriptomics column inserted, but filled with NaN.\n",
      "HIST1H2AA did not match any columns in transcriptomics dataframe. HIST1H2AA_transcriptomics column inserted, but filled with NaN.\n",
      "HIST1H2AB did not match any columns in transcriptomics dataframe. HIST1H2AB_transcriptomics column inserted, but filled with NaN.\n",
      "HIST1H3A did not match any columns in transcriptomics dataframe. HIST1H3A_transcriptomics column inserted, but filled with NaN.\n",
      "HIST1H4G did not match any columns in transcriptomics dataframe. HIST1H4G_transcriptomics column inserted, but filled with NaN.\n",
      "HIST2H2AB did not match any columns in transcriptomics dataframe. HIST2H2AB_transcriptomics column inserted, but filled with NaN.\n",
      "HIST2H3A did not match any columns in transcriptomics dataframe. HIST2H3A_transcriptomics column inserted, but filled with NaN.\n",
      "HIST3H2BB did not match any columns in transcriptomics dataframe. HIST3H2BB_transcriptomics column inserted, but filled with NaN.\n",
      "HIST3H3 did not match any columns in transcriptomics dataframe. HIST3H3_transcriptomics column inserted, but filled with NaN.\n",
      "HLA-DRB3 did not match any columns in transcriptomics dataframe. HLA-DRB3_transcriptomics column inserted, but filled with NaN.\n",
      "HLA-DRB4 did not match any columns in transcriptomics dataframe. HLA-DRB4_transcriptomics column inserted, but filled with NaN.\n",
      "HLA-G did not match any columns in transcriptomics dataframe. HLA-G_transcriptomics column inserted, but filled with NaN.\n",
      "HMCN2 did not match any columns in transcriptomics dataframe. HMCN2_transcriptomics column inserted, but filled with NaN.\n",
      "HMSD did not match any columns in transcriptomics dataframe. HMSD_transcriptomics column inserted, but filled with NaN.\n",
      "HMX3 did not match any columns in transcriptomics dataframe. HMX3_transcriptomics column inserted, but filled with NaN.\n",
      "HOXC9 did not match any columns in transcriptomics dataframe. HOXC9_transcriptomics column inserted, but filled with NaN.\n",
      "HP did not match any columns in transcriptomics dataframe. HP_transcriptomics column inserted, but filled with NaN.\n",
      "HPR did not match any columns in transcriptomics dataframe. HPR_transcriptomics column inserted, but filled with NaN.\n",
      "HPX did not match any columns in transcriptomics dataframe. HPX_transcriptomics column inserted, but filled with NaN.\n",
      "HRG did not match any columns in transcriptomics dataframe. HRG_transcriptomics column inserted, but filled with NaN.\n",
      "HSPA1L did not match any columns in transcriptomics dataframe. HSPA1L_transcriptomics column inserted, but filled with NaN.\n",
      "IFITM5 did not match any columns in transcriptomics dataframe. IFITM5_transcriptomics column inserted, but filled with NaN.\n",
      "IGF2BP1 did not match any columns in transcriptomics dataframe. IGF2BP1_transcriptomics column inserted, but filled with NaN.\n",
      "IGF2BP3 did not match any columns in transcriptomics dataframe. IGF2BP3_transcriptomics column inserted, but filled with NaN.\n",
      "IGFALS did not match any columns in transcriptomics dataframe. IGFALS_transcriptomics column inserted, but filled with NaN.\n",
      "IGLL1 did not match any columns in transcriptomics dataframe. IGLL1_transcriptomics column inserted, but filled with NaN.\n",
      "INA did not match any columns in transcriptomics dataframe. INA_transcriptomics column inserted, but filled with NaN.\n",
      "INSL5 did not match any columns in transcriptomics dataframe. INSL5_transcriptomics column inserted, but filled with NaN.\n",
      "IQCA1 did not match any columns in transcriptomics dataframe. IQCA1_transcriptomics column inserted, but filled with NaN.\n",
      "ITGA2B did not match any columns in transcriptomics dataframe. ITGA2B_transcriptomics column inserted, but filled with NaN.\n",
      "ITIH1 did not match any columns in transcriptomics dataframe. ITIH1_transcriptomics column inserted, but filled with NaN.\n",
      "ITIH2 did not match any columns in transcriptomics dataframe. ITIH2_transcriptomics column inserted, but filled with NaN.\n",
      "ITIH3 did not match any columns in transcriptomics dataframe. ITIH3_transcriptomics column inserted, but filled with NaN.\n",
      "ITIH4 did not match any columns in transcriptomics dataframe. ITIH4_transcriptomics column inserted, but filled with NaN.\n",
      "ITLN2 did not match any columns in transcriptomics dataframe. ITLN2_transcriptomics column inserted, but filled with NaN.\n",
      "JPH2 did not match any columns in transcriptomics dataframe. JPH2_transcriptomics column inserted, but filled with NaN.\n",
      "KANK3 did not match any columns in transcriptomics dataframe. KANK3_transcriptomics column inserted, but filled with NaN.\n",
      "KIF1A did not match any columns in transcriptomics dataframe. KIF1A_transcriptomics column inserted, but filled with NaN.\n",
      "KIF5A did not match any columns in transcriptomics dataframe. KIF5A_transcriptomics column inserted, but filled with NaN.\n",
      "KISS1 did not match any columns in transcriptomics dataframe. KISS1_transcriptomics column inserted, but filled with NaN.\n",
      "KLKB1 did not match any columns in transcriptomics dataframe. KLKB1_transcriptomics column inserted, but filled with NaN.\n",
      "KNG1 did not match any columns in transcriptomics dataframe. KNG1_transcriptomics column inserted, but filled with NaN.\n",
      "KRT1 did not match any columns in transcriptomics dataframe. KRT1_transcriptomics column inserted, but filled with NaN.\n",
      "KRT13 did not match any columns in transcriptomics dataframe. KRT13_transcriptomics column inserted, but filled with NaN.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KRT14 did not match any columns in transcriptomics dataframe. KRT14_transcriptomics column inserted, but filled with NaN.\n",
      "KRT16 did not match any columns in transcriptomics dataframe. KRT16_transcriptomics column inserted, but filled with NaN.\n",
      "KRT2 did not match any columns in transcriptomics dataframe. KRT2_transcriptomics column inserted, but filled with NaN.\n",
      "KRT3 did not match any columns in transcriptomics dataframe. KRT3_transcriptomics column inserted, but filled with NaN.\n",
      "KRT31 did not match any columns in transcriptomics dataframe. KRT31_transcriptomics column inserted, but filled with NaN.\n",
      "KRT34 did not match any columns in transcriptomics dataframe. KRT34_transcriptomics column inserted, but filled with NaN.\n",
      "KRT36 did not match any columns in transcriptomics dataframe. KRT36_transcriptomics column inserted, but filled with NaN.\n",
      "KRT5 did not match any columns in transcriptomics dataframe. KRT5_transcriptomics column inserted, but filled with NaN.\n",
      "KRT6A did not match any columns in transcriptomics dataframe. KRT6A_transcriptomics column inserted, but filled with NaN.\n",
      "KRT73 did not match any columns in transcriptomics dataframe. KRT73_transcriptomics column inserted, but filled with NaN.\n",
      "KRT75 did not match any columns in transcriptomics dataframe. KRT75_transcriptomics column inserted, but filled with NaN.\n",
      "KRT79 did not match any columns in transcriptomics dataframe. KRT79_transcriptomics column inserted, but filled with NaN.\n",
      "KRT82 did not match any columns in transcriptomics dataframe. KRT82_transcriptomics column inserted, but filled with NaN.\n",
      "KRT83 did not match any columns in transcriptomics dataframe. KRT83_transcriptomics column inserted, but filled with NaN.\n",
      "KRT84 did not match any columns in transcriptomics dataframe. KRT84_transcriptomics column inserted, but filled with NaN.\n",
      "KRT85 did not match any columns in transcriptomics dataframe. KRT85_transcriptomics column inserted, but filled with NaN.\n",
      "KRT9 did not match any columns in transcriptomics dataframe. KRT9_transcriptomics column inserted, but filled with NaN.\n",
      "L1CAM did not match any columns in transcriptomics dataframe. L1CAM_transcriptomics column inserted, but filled with NaN.\n",
      "LBP did not match any columns in transcriptomics dataframe. LBP_transcriptomics column inserted, but filled with NaN.\n",
      "LCN15 did not match any columns in transcriptomics dataframe. LCN15_transcriptomics column inserted, but filled with NaN.\n",
      "LDB3 did not match any columns in transcriptomics dataframe. LDB3_transcriptomics column inserted, but filled with NaN.\n",
      "LDHAL6B did not match any columns in transcriptomics dataframe. LDHAL6B_transcriptomics column inserted, but filled with NaN.\n",
      "LGALS16 did not match any columns in transcriptomics dataframe. LGALS16_transcriptomics column inserted, but filled with NaN.\n",
      "LILRA6 did not match any columns in transcriptomics dataframe. LILRA6_transcriptomics column inserted, but filled with NaN.\n",
      "LIN7A did not match any columns in transcriptomics dataframe. LIN7A_transcriptomics column inserted, but filled with NaN.\n",
      "LOC102724652 did not match any columns in transcriptomics dataframe. LOC102724652_transcriptomics column inserted, but filled with NaN.\n",
      "LOC105372280 did not match any columns in transcriptomics dataframe. LOC105372280_transcriptomics column inserted, but filled with NaN.\n",
      "LOC400927-CSNK1E did not match any columns in transcriptomics dataframe. LOC400927-CSNK1E_transcriptomics column inserted, but filled with NaN.\n",
      "LPA did not match any columns in transcriptomics dataframe. LPA_transcriptomics column inserted, but filled with NaN.\n",
      "LPO did not match any columns in transcriptomics dataframe. LPO_transcriptomics column inserted, but filled with NaN.\n",
      "LTC4S did not match any columns in transcriptomics dataframe. LTC4S_transcriptomics column inserted, but filled with NaN.\n",
      "LY6D did not match any columns in transcriptomics dataframe. LY6D_transcriptomics column inserted, but filled with NaN.\n",
      "LY6H did not match any columns in transcriptomics dataframe. LY6H_transcriptomics column inserted, but filled with NaN.\n",
      "LY75-CD302 did not match any columns in transcriptomics dataframe. LY75-CD302_transcriptomics column inserted, but filled with NaN.\n",
      "MACROD2 did not match any columns in transcriptomics dataframe. MACROD2_transcriptomics column inserted, but filled with NaN.\n",
      "MAMDC2 did not match any columns in transcriptomics dataframe. MAMDC2_transcriptomics column inserted, but filled with NaN.\n",
      "MAP1A did not match any columns in transcriptomics dataframe. MAP1A_transcriptomics column inserted, but filled with NaN.\n",
      "MAP1LC3B2 did not match any columns in transcriptomics dataframe. MAP1LC3B2_transcriptomics column inserted, but filled with NaN.\n",
      "MAP6 did not match any columns in transcriptomics dataframe. MAP6_transcriptomics column inserted, but filled with NaN.\n",
      "MAPK4 did not match any columns in transcriptomics dataframe. MAPK4_transcriptomics column inserted, but filled with NaN.\n",
      "MAPT did not match any columns in transcriptomics dataframe. MAPT_transcriptomics column inserted, but filled with NaN.\n",
      "MARK1 did not match any columns in transcriptomics dataframe. MARK1_transcriptomics column inserted, but filled with NaN.\n",
      "MBL2 did not match any columns in transcriptomics dataframe. MBL2_transcriptomics column inserted, but filled with NaN.\n",
      "MEP1B did not match any columns in transcriptomics dataframe. MEP1B_transcriptomics column inserted, but filled with NaN.\n",
      "MGAM did not match any columns in transcriptomics dataframe. MGAM_transcriptomics column inserted, but filled with NaN.\n",
      "MMP23B did not match any columns in transcriptomics dataframe. MMP23B_transcriptomics column inserted, but filled with NaN.\n",
      "MMP8 did not match any columns in transcriptomics dataframe. MMP8_transcriptomics column inserted, but filled with NaN.\n",
      "MPO did not match any columns in transcriptomics dataframe. MPO_transcriptomics column inserted, but filled with NaN.\n",
      "MPZ did not match any columns in transcriptomics dataframe. MPZ_transcriptomics column inserted, but filled with NaN.\n",
      "MT1HL1 did not match any columns in transcriptomics dataframe. MT1HL1_transcriptomics column inserted, but filled with NaN.\n",
      "MT3 did not match any columns in transcriptomics dataframe. MT3_transcriptomics column inserted, but filled with NaN.\n",
      "MTTP did not match any columns in transcriptomics dataframe. MTTP_transcriptomics column inserted, but filled with NaN.\n",
      "MUC5AC did not match any columns in transcriptomics dataframe. MUC5AC_transcriptomics column inserted, but filled with NaN.\n",
      "MUC5B did not match any columns in transcriptomics dataframe. MUC5B_transcriptomics column inserted, but filled with NaN.\n",
      "MUC6 did not match any columns in transcriptomics dataframe. MUC6_transcriptomics column inserted, but filled with NaN.\n",
      "MYL3 did not match any columns in transcriptomics dataframe. MYL3_transcriptomics column inserted, but filled with NaN.\n",
      "MYL4 did not match any columns in transcriptomics dataframe. MYL4_transcriptomics column inserted, but filled with NaN.\n",
      "MYL7 did not match any columns in transcriptomics dataframe. MYL7_transcriptomics column inserted, but filled with NaN.\n",
      "NACA2 did not match any columns in transcriptomics dataframe. NACA2_transcriptomics column inserted, but filled with NaN.\n",
      "NCAM1 did not match any columns in transcriptomics dataframe. NCAM1_transcriptomics column inserted, but filled with NaN.\n",
      "NCAM2 did not match any columns in transcriptomics dataframe. NCAM2_transcriptomics column inserted, but filled with NaN.\n",
      "NCMAP did not match any columns in transcriptomics dataframe. NCMAP_transcriptomics column inserted, but filled with NaN.\n",
      "NEFH did not match any columns in transcriptomics dataframe. NEFH_transcriptomics column inserted, but filled with NaN.\n",
      "NEFL did not match any columns in transcriptomics dataframe. NEFL_transcriptomics column inserted, but filled with NaN.\n",
      "NEFM did not match any columns in transcriptomics dataframe. NEFM_transcriptomics column inserted, but filled with NaN.\n",
      "NKAPL did not match any columns in transcriptomics dataframe. NKAPL_transcriptomics column inserted, but filled with NaN.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NOS1 did not match any columns in transcriptomics dataframe. NOS1_transcriptomics column inserted, but filled with NaN.\n",
      "NOS1AP did not match any columns in transcriptomics dataframe. NOS1AP_transcriptomics column inserted, but filled with NaN.\n",
      "NOVA2 did not match any columns in transcriptomics dataframe. NOVA2_transcriptomics column inserted, but filled with NaN.\n",
      "NPY did not match any columns in transcriptomics dataframe. NPY_transcriptomics column inserted, but filled with NaN.\n",
      "NTN1 did not match any columns in transcriptomics dataframe. NTN1_transcriptomics column inserted, but filled with NaN.\n",
      "OGDHL did not match any columns in transcriptomics dataframe. OGDHL_transcriptomics column inserted, but filled with NaN.\n",
      "ORM1 did not match any columns in transcriptomics dataframe. ORM1_transcriptomics column inserted, but filled with NaN.\n",
      "ORM2 did not match any columns in transcriptomics dataframe. ORM2_transcriptomics column inserted, but filled with NaN.\n",
      "OTC did not match any columns in transcriptomics dataframe. OTC_transcriptomics column inserted, but filled with NaN.\n",
      "PABPC4L did not match any columns in transcriptomics dataframe. PABPC4L_transcriptomics column inserted, but filled with NaN.\n",
      "PADI3 did not match any columns in transcriptomics dataframe. PADI3_transcriptomics column inserted, but filled with NaN.\n",
      "PADI4 did not match any columns in transcriptomics dataframe. PADI4_transcriptomics column inserted, but filled with NaN.\n",
      "PALM2 did not match any columns in transcriptomics dataframe. PALM2_transcriptomics column inserted, but filled with NaN.\n",
      "PCBP3 did not match any columns in transcriptomics dataframe. PCBP3_transcriptomics column inserted, but filled with NaN.\n",
      "PCP4L1 did not match any columns in transcriptomics dataframe. PCP4L1_transcriptomics column inserted, but filled with NaN.\n",
      "PCSK1 did not match any columns in transcriptomics dataframe. PCSK1_transcriptomics column inserted, but filled with NaN.\n",
      "PCSK1N did not match any columns in transcriptomics dataframe. PCSK1N_transcriptomics column inserted, but filled with NaN.\n",
      "PCYT1B did not match any columns in transcriptomics dataframe. PCYT1B_transcriptomics column inserted, but filled with NaN.\n",
      "PDIA2 did not match any columns in transcriptomics dataframe. PDIA2_transcriptomics column inserted, but filled with NaN.\n",
      "PENK did not match any columns in transcriptomics dataframe. PENK_transcriptomics column inserted, but filled with NaN.\n",
      "PGK2 did not match any columns in transcriptomics dataframe. PGK2_transcriptomics column inserted, but filled with NaN.\n",
      "PGLYRP1 did not match any columns in transcriptomics dataframe. PGLYRP1_transcriptomics column inserted, but filled with NaN.\n",
      "PGLYRP2 did not match any columns in transcriptomics dataframe. PGLYRP2_transcriptomics column inserted, but filled with NaN.\n",
      "PI16 did not match any columns in transcriptomics dataframe. PI16_transcriptomics column inserted, but filled with NaN.\n",
      "PKLR did not match any columns in transcriptomics dataframe. PKLR_transcriptomics column inserted, but filled with NaN.\n",
      "PLA2G1B did not match any columns in transcriptomics dataframe. PLA2G1B_transcriptomics column inserted, but filled with NaN.\n",
      "PLG did not match any columns in transcriptomics dataframe. PLG_transcriptomics column inserted, but filled with NaN.\n",
      "PLIN1 did not match any columns in transcriptomics dataframe. PLIN1_transcriptomics column inserted, but filled with NaN.\n",
      "PLIN4 did not match any columns in transcriptomics dataframe. PLIN4_transcriptomics column inserted, but filled with NaN.\n",
      "PLP1 did not match any columns in transcriptomics dataframe. PLP1_transcriptomics column inserted, but filled with NaN.\n",
      "PNLIP did not match any columns in transcriptomics dataframe. PNLIP_transcriptomics column inserted, but filled with NaN.\n",
      "POC1B-GALNT4 did not match any columns in transcriptomics dataframe. POC1B-GALNT4_transcriptomics column inserted, but filled with NaN.\n",
      "PON1 did not match any columns in transcriptomics dataframe. PON1_transcriptomics column inserted, but filled with NaN.\n",
      "POTEF did not match any columns in transcriptomics dataframe. POTEF_transcriptomics column inserted, but filled with NaN.\n",
      "POTEI did not match any columns in transcriptomics dataframe. POTEI_transcriptomics column inserted, but filled with NaN.\n",
      "POTEJ did not match any columns in transcriptomics dataframe. POTEJ_transcriptomics column inserted, but filled with NaN.\n",
      "PPBP did not match any columns in transcriptomics dataframe. PPBP_transcriptomics column inserted, but filled with NaN.\n",
      "PPP4R4 did not match any columns in transcriptomics dataframe. PPP4R4_transcriptomics column inserted, but filled with NaN.\n",
      "PRAM1 did not match any columns in transcriptomics dataframe. PRAM1_transcriptomics column inserted, but filled with NaN.\n",
      "PRG2 did not match any columns in transcriptomics dataframe. PRG2_transcriptomics column inserted, but filled with NaN.\n",
      "PRG3 did not match any columns in transcriptomics dataframe. PRG3_transcriptomics column inserted, but filled with NaN.\n",
      "PRKAA2 did not match any columns in transcriptomics dataframe. PRKAA2_transcriptomics column inserted, but filled with NaN.\n",
      "PRKACG did not match any columns in transcriptomics dataframe. PRKACG_transcriptomics column inserted, but filled with NaN.\n",
      "PRND did not match any columns in transcriptomics dataframe. PRND_transcriptomics column inserted, but filled with NaN.\n",
      "PROC did not match any columns in transcriptomics dataframe. PROC_transcriptomics column inserted, but filled with NaN.\n",
      "PRPH did not match any columns in transcriptomics dataframe. PRPH_transcriptomics column inserted, but filled with NaN.\n",
      "PRR5-ARHGAP8 did not match any columns in transcriptomics dataframe. PRR5-ARHGAP8_transcriptomics column inserted, but filled with NaN.\n",
      "PRR9 did not match any columns in transcriptomics dataframe. PRR9_transcriptomics column inserted, but filled with NaN.\n",
      "PRSS1 did not match any columns in transcriptomics dataframe. PRSS1_transcriptomics column inserted, but filled with NaN.\n",
      "PRTN3 did not match any columns in transcriptomics dataframe. PRTN3_transcriptomics column inserted, but filled with NaN.\n",
      "PSCA did not match any columns in transcriptomics dataframe. PSCA_transcriptomics column inserted, but filled with NaN.\n",
      "PTX3 did not match any columns in transcriptomics dataframe. PTX3_transcriptomics column inserted, but filled with NaN.\n",
      "PYGM did not match any columns in transcriptomics dataframe. PYGM_transcriptomics column inserted, but filled with NaN.\n",
      "PYY did not match any columns in transcriptomics dataframe. PYY_transcriptomics column inserted, but filled with NaN.\n",
      "PZP did not match any columns in transcriptomics dataframe. PZP_transcriptomics column inserted, but filled with NaN.\n",
      "RAB39A did not match any columns in transcriptomics dataframe. RAB39A_transcriptomics column inserted, but filled with NaN.\n",
      "RAB3A did not match any columns in transcriptomics dataframe. RAB3A_transcriptomics column inserted, but filled with NaN.\n",
      "RAB3B did not match any columns in transcriptomics dataframe. RAB3B_transcriptomics column inserted, but filled with NaN.\n",
      "RAB3C did not match any columns in transcriptomics dataframe. RAB3C_transcriptomics column inserted, but filled with NaN.\n",
      "RAB44 did not match any columns in transcriptomics dataframe. RAB44_transcriptomics column inserted, but filled with NaN.\n",
      "RAB6B did not match any columns in transcriptomics dataframe. RAB6B_transcriptomics column inserted, but filled with NaN.\n",
      "RALYL did not match any columns in transcriptomics dataframe. RALYL_transcriptomics column inserted, but filled with NaN.\n",
      "RBPMS2 did not match any columns in transcriptomics dataframe. RBPMS2_transcriptomics column inserted, but filled with NaN.\n",
      "RETN did not match any columns in transcriptomics dataframe. RETN_transcriptomics column inserted, but filled with NaN.\n",
      "RGPD3 did not match any columns in transcriptomics dataframe. RGPD3_transcriptomics column inserted, but filled with NaN.\n",
      "RHD did not match any columns in transcriptomics dataframe. RHD_transcriptomics column inserted, but filled with NaN.\n",
      "RNASE3 did not match any columns in transcriptomics dataframe. RNASE3_transcriptomics column inserted, but filled with NaN.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RPL3L did not match any columns in transcriptomics dataframe. RPL3L_transcriptomics column inserted, but filled with NaN.\n",
      "RPS4Y1 did not match any columns in transcriptomics dataframe. RPS4Y1_transcriptomics column inserted, but filled with NaN.\n",
      "RPS4Y2 did not match any columns in transcriptomics dataframe. RPS4Y2_transcriptomics column inserted, but filled with NaN.\n",
      "RTN1 did not match any columns in transcriptomics dataframe. RTN1_transcriptomics column inserted, but filled with NaN.\n",
      "S100A7 did not match any columns in transcriptomics dataframe. S100A7_transcriptomics column inserted, but filled with NaN.\n",
      "SAA2-SAA4 did not match any columns in transcriptomics dataframe. SAA2-SAA4_transcriptomics column inserted, but filled with NaN.\n",
      "SARDH did not match any columns in transcriptomics dataframe. SARDH_transcriptomics column inserted, but filled with NaN.\n",
      "SCGN did not match any columns in transcriptomics dataframe. SCGN_transcriptomics column inserted, but filled with NaN.\n",
      "SCN7A did not match any columns in transcriptomics dataframe. SCN7A_transcriptomics column inserted, but filled with NaN.\n",
      "SCUBE3 did not match any columns in transcriptomics dataframe. SCUBE3_transcriptomics column inserted, but filled with NaN.\n",
      "SEPT3 did not match any columns in transcriptomics dataframe. SEPT3_transcriptomics column inserted, but filled with NaN.\n",
      "SERPINA10 did not match any columns in transcriptomics dataframe. SERPINA10_transcriptomics column inserted, but filled with NaN.\n",
      "SERPINA4 did not match any columns in transcriptomics dataframe. SERPINA4_transcriptomics column inserted, but filled with NaN.\n",
      "SERPINA5 did not match any columns in transcriptomics dataframe. SERPINA5_transcriptomics column inserted, but filled with NaN.\n",
      "SERPINA6 did not match any columns in transcriptomics dataframe. SERPINA6_transcriptomics column inserted, but filled with NaN.\n",
      "SERPINA7 did not match any columns in transcriptomics dataframe. SERPINA7_transcriptomics column inserted, but filled with NaN.\n",
      "SERPINB10 did not match any columns in transcriptomics dataframe. SERPINB10_transcriptomics column inserted, but filled with NaN.\n",
      "SERPINB2 did not match any columns in transcriptomics dataframe. SERPINB2_transcriptomics column inserted, but filled with NaN.\n",
      "SERPINC1 did not match any columns in transcriptomics dataframe. SERPINC1_transcriptomics column inserted, but filled with NaN.\n",
      "SERPIND1 did not match any columns in transcriptomics dataframe. SERPIND1_transcriptomics column inserted, but filled with NaN.\n",
      "SH3GL2 did not match any columns in transcriptomics dataframe. SH3GL2_transcriptomics column inserted, but filled with NaN.\n",
      "SHBG did not match any columns in transcriptomics dataframe. SHBG_transcriptomics column inserted, but filled with NaN.\n",
      "SIGLEC1 did not match any columns in transcriptomics dataframe. SIGLEC1_transcriptomics column inserted, but filled with NaN.\n",
      "SIGLEC5 did not match any columns in transcriptomics dataframe. SIGLEC5_transcriptomics column inserted, but filled with NaN.\n",
      "SLC14A1 did not match any columns in transcriptomics dataframe. SLC14A1_transcriptomics column inserted, but filled with NaN.\n",
      "SLC16A7 did not match any columns in transcriptomics dataframe. SLC16A7_transcriptomics column inserted, but filled with NaN.\n",
      "SLC2A4 did not match any columns in transcriptomics dataframe. SLC2A4_transcriptomics column inserted, but filled with NaN.\n",
      "SLC4A1 did not match any columns in transcriptomics dataframe. SLC4A1_transcriptomics column inserted, but filled with NaN.\n",
      "SLC4A4 did not match any columns in transcriptomics dataframe. SLC4A4_transcriptomics column inserted, but filled with NaN.\n",
      "SMPX did not match any columns in transcriptomics dataframe. SMPX_transcriptomics column inserted, but filled with NaN.\n",
      "SMYD1 did not match any columns in transcriptomics dataframe. SMYD1_transcriptomics column inserted, but filled with NaN.\n",
      "SNAP25 did not match any columns in transcriptomics dataframe. SNAP25_transcriptomics column inserted, but filled with NaN.\n",
      "SNAP91 did not match any columns in transcriptomics dataframe. SNAP91_transcriptomics column inserted, but filled with NaN.\n",
      "SNCA did not match any columns in transcriptomics dataframe. SNCA_transcriptomics column inserted, but filled with NaN.\n",
      "SPP2 did not match any columns in transcriptomics dataframe. SPP2_transcriptomics column inserted, but filled with NaN.\n",
      "SPRR1A did not match any columns in transcriptomics dataframe. SPRR1A_transcriptomics column inserted, but filled with NaN.\n",
      "SPRR2A did not match any columns in transcriptomics dataframe. SPRR2A_transcriptomics column inserted, but filled with NaN.\n",
      "SPRR2D did not match any columns in transcriptomics dataframe. SPRR2D_transcriptomics column inserted, but filled with NaN.\n",
      "SPRR2F did not match any columns in transcriptomics dataframe. SPRR2F_transcriptomics column inserted, but filled with NaN.\n",
      "SPRR2G did not match any columns in transcriptomics dataframe. SPRR2G_transcriptomics column inserted, but filled with NaN.\n",
      "SPRR3 did not match any columns in transcriptomics dataframe. SPRR3_transcriptomics column inserted, but filled with NaN.\n",
      "SPTA1 did not match any columns in transcriptomics dataframe. SPTA1_transcriptomics column inserted, but filled with NaN.\n",
      "SPTB did not match any columns in transcriptomics dataframe. SPTB_transcriptomics column inserted, but filled with NaN.\n",
      "SSC5D did not match any columns in transcriptomics dataframe. SSC5D_transcriptomics column inserted, but filled with NaN.\n",
      "SST did not match any columns in transcriptomics dataframe. SST_transcriptomics column inserted, but filled with NaN.\n",
      "STON1-GTF2A1L did not match any columns in transcriptomics dataframe. STON1-GTF2A1L_transcriptomics column inserted, but filled with NaN.\n",
      "STUM did not match any columns in transcriptomics dataframe. STUM_transcriptomics column inserted, but filled with NaN.\n",
      "STX11 did not match any columns in transcriptomics dataframe. STX11_transcriptomics column inserted, but filled with NaN.\n",
      "SULT1A2 did not match any columns in transcriptomics dataframe. SULT1A2_transcriptomics column inserted, but filled with NaN.\n",
      "SULT2A1 did not match any columns in transcriptomics dataframe. SULT2A1_transcriptomics column inserted, but filled with NaN.\n",
      "SUSD2 did not match any columns in transcriptomics dataframe. SUSD2_transcriptomics column inserted, but filled with NaN.\n",
      "SYCE3 did not match any columns in transcriptomics dataframe. SYCE3_transcriptomics column inserted, but filled with NaN.\n",
      "SYN1 did not match any columns in transcriptomics dataframe. SYN1_transcriptomics column inserted, but filled with NaN.\n",
      "SYN2 did not match any columns in transcriptomics dataframe. SYN2_transcriptomics column inserted, but filled with NaN.\n",
      "SYNE3 did not match any columns in transcriptomics dataframe. SYNE3_transcriptomics column inserted, but filled with NaN.\n",
      "SYNJ2BP-COX16 did not match any columns in transcriptomics dataframe. SYNJ2BP-COX16_transcriptomics column inserted, but filled with NaN.\n",
      "SYP did not match any columns in transcriptomics dataframe. SYP_transcriptomics column inserted, but filled with NaN.\n",
      "SYT1 did not match any columns in transcriptomics dataframe. SYT1_transcriptomics column inserted, but filled with NaN.\n",
      "TAGLN3 did not match any columns in transcriptomics dataframe. TAGLN3_transcriptomics column inserted, but filled with NaN.\n",
      "TBX20 did not match any columns in transcriptomics dataframe. TBX20_transcriptomics column inserted, but filled with NaN.\n",
      "TCL1A did not match any columns in transcriptomics dataframe. TCL1A_transcriptomics column inserted, but filled with NaN.\n",
      "TF did not match any columns in transcriptomics dataframe. TF_transcriptomics column inserted, but filled with NaN.\n",
      "TGFB2 did not match any columns in transcriptomics dataframe. TGFB2_transcriptomics column inserted, but filled with NaN.\n",
      "TIFAB did not match any columns in transcriptomics dataframe. TIFAB_transcriptomics column inserted, but filled with NaN.\n",
      "TKTL1 did not match any columns in transcriptomics dataframe. TKTL1_transcriptomics column inserted, but filled with NaN.\n",
      "TKTL2 did not match any columns in transcriptomics dataframe. TKTL2_transcriptomics column inserted, but filled with NaN.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TMEM189-UBE2V1 did not match any columns in transcriptomics dataframe. TMEM189-UBE2V1_transcriptomics column inserted, but filled with NaN.\n",
      "TMEM35A did not match any columns in transcriptomics dataframe. TMEM35A_transcriptomics column inserted, but filled with NaN.\n",
      "TMOD1 did not match any columns in transcriptomics dataframe. TMOD1_transcriptomics column inserted, but filled with NaN.\n",
      "TMPPE did not match any columns in transcriptomics dataframe. TMPPE_transcriptomics column inserted, but filled with NaN.\n",
      "TNXB did not match any columns in transcriptomics dataframe. TNXB_transcriptomics column inserted, but filled with NaN.\n",
      "TPPP did not match any columns in transcriptomics dataframe. TPPP_transcriptomics column inserted, but filled with NaN.\n",
      "TRIM72 did not match any columns in transcriptomics dataframe. TRIM72_transcriptomics column inserted, but filled with NaN.\n",
      "TTR did not match any columns in transcriptomics dataframe. TTR_transcriptomics column inserted, but filled with NaN.\n",
      "TUBA3C did not match any columns in transcriptomics dataframe. TUBA3C_transcriptomics column inserted, but filled with NaN.\n",
      "TUBA8 did not match any columns in transcriptomics dataframe. TUBA8_transcriptomics column inserted, but filled with NaN.\n",
      "TUBB1 did not match any columns in transcriptomics dataframe. TUBB1_transcriptomics column inserted, but filled with NaN.\n",
      "TUBB4A did not match any columns in transcriptomics dataframe. TUBB4A_transcriptomics column inserted, but filled with NaN.\n",
      "TUBB8 did not match any columns in transcriptomics dataframe. TUBB8_transcriptomics column inserted, but filled with NaN.\n",
      "TUSC5 did not match any columns in transcriptomics dataframe. TUSC5_transcriptomics column inserted, but filled with NaN.\n",
      "UGT1A8 did not match any columns in transcriptomics dataframe. UGT1A8_transcriptomics column inserted, but filled with NaN.\n",
      "UGT2A2 did not match any columns in transcriptomics dataframe. UGT2A2_transcriptomics column inserted, but filled with NaN.\n",
      "USP9Y did not match any columns in transcriptomics dataframe. USP9Y_transcriptomics column inserted, but filled with NaN.\n",
      "UTS2 did not match any columns in transcriptomics dataframe. UTS2_transcriptomics column inserted, but filled with NaN.\n",
      "VAT1L did not match any columns in transcriptomics dataframe. VAT1L_transcriptomics column inserted, but filled with NaN.\n",
      "VTN did not match any columns in transcriptomics dataframe. VTN_transcriptomics column inserted, but filled with NaN.\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "    proteomics = co.get_proteomics()\n",
    "    all_proteins = proteomics.columns\n",
    "    #all_proteins = all_proteins[:100]\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = co.append_mutations_to_omics(mutation_genes=[gene], omics_df_name=omics, omics_genes=all_proteins)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
