{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# KRAS Mutation Trans Effect on Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook analyzes the trans effect of KRAS mutation on interacting and other proteins Transcriptomics, in Endometrial, Colon, and Ovarian cancer."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Library Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "import cptac\n",
    "import cptac.algorithms as al\n",
    "\n",
    "en = cptac.Endometrial()\n",
    "co = cptac.Colon()\n",
    "ov = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = \"KRAS\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Investigate Proteomics, Phosphoproteomics, Acetylproteomics, or Transcriptomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#omics = \"proteomics\"\n",
    "omics = \"transcriptomics\"\n",
    "#omics = \"phosphoproteomics\"\n",
    "#omics = \"acetylproteomics\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Track all significant comparisons in Dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_significant_comparisons = pd.DataFrame(columns=['Cancer_Type', 'Gene', 'Comparison','Interacting_Protein','P_Value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_to_all_significant_comparisons(df, cancer, interacting, all_sig_comp):\n",
    "    expanded = df\n",
    "    expanded['Gene'] = gene\n",
    "    expanded['Cancer_Type'] = cancer\n",
    "    expanded['Interacting_Protein'] = interacting\n",
    "    \n",
    "    updated_all_comparisons = pd.concat([all_sig_comp, expanded], sort=False)\n",
    "    \n",
    "    return updated_all_comparisons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Interacting Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate interacting protein list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a call to the cptac.algorithms get interacting proteins method, which interacts with the uniprot and string databases to generate a list of known interacting partners with the given gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Interacting Proteins:\n",
      "RAF1\n",
      "MAPK1\n",
      "KRAS\n",
      "MAP2K2\n",
      "MAPK3\n",
      "PIK3CA\n",
      "EGF\n",
      "ERBB3\n",
      "ERBB2\n",
      "EGFR\n",
      "BRAF\n",
      "PIK3CB\n",
      "ARAF\n",
      "MAP2K1\n",
      "PTPN11\n",
      "RET\n",
      "NF1\n",
      "PIK3CG\n",
      "NRAS\n",
      "RALGDS\n",
      "SRC\n",
      "GRB2\n",
      "SOS1\n",
      "HRAS\n",
      "CDKN2A\n",
      "PRRT2\n",
      "RAP1GDS1\n",
      "HNRNPC\n",
      "RASSF2\n",
      "RASSF5\n",
      "CALM1\n"
     ]
    }
   ],
   "source": [
    "'''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "'''Show interacting protein list'''\n",
    "print(\"Interacting Proteins:\")\n",
    "for interacting_protein in interacting_proteins:\n",
    "    print(interacting_protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", True, all_significant_comparisons)\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EGF did not match any columns in transcriptomics dataframe. EGF_transcriptomics column inserted, but filled with NaN.\n",
      "RET did not match any columns in transcriptomics dataframe. RET_transcriptomics column inserted, but filled with NaN.\n",
      "PRRT2 did not match any columns in transcriptomics dataframe. PRRT2_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAPK3 did not match any columns in transcriptomics dataframe. MAPK3_transcriptomics column inserted, but filled with NaN.\n",
      "NRAS did not match any columns in transcriptomics dataframe. NRAS_transcriptomics column inserted, but filled with NaN.\n",
      "PRRT2 did not match any columns in transcriptomics dataframe. PRRT2_transcriptomics column inserted, but filled with NaN.\n",
      "HNRNPC did not match any columns in transcriptomics dataframe. HNRNPC_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Expand our search for significant comparisons to all proteins in our dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  KRAS\n",
      "Doing t-test comparisons\n",
      "\n",
      "              Comparison       P_Value\n",
      "0  DUSP4_transcriptomics  3.002087e-08\n",
      "1  SPRY1_transcriptomics  2.500902e-07\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  KRAS\n",
      "Doing t-test comparisons\n",
      "\n",
      "                Comparison       P_Value\n",
      "0  GALNT10_transcriptomics  6.801284e-09\n",
      "1    TMCC1_transcriptomics  4.559702e-08\n",
      "2    FAIM2_transcriptomics  2.082800e-07\n",
      "3  TMEM211_transcriptomics  4.705642e-07\n",
      "4    TGFBI_transcriptomics  6.646078e-07\n",
      "5     LY6E_transcriptomics  8.340164e-07\n",
      "6   PHLDA1_transcriptomics  1.263308e-06\n",
      "7    DHX58_transcriptomics  1.880427e-06\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  KRAS\n",
      "Doing t-test comparisons\n",
      "\n",
      "                       Comparison       P_Value\n",
      "0           OR1L3_transcriptomics  8.293846e-28\n",
      "1       FAM90A27P_transcriptomics  1.188688e-23\n",
      "2           TRPM6_transcriptomics  7.448553e-18\n",
      "3          VPS13A_transcriptomics  4.204536e-17\n",
      "4       SERPINA11_transcriptomics  8.462675e-17\n",
      "5           GABRP_transcriptomics  9.499937e-17\n",
      "6          MOGAT1_transcriptomics  2.312500e-16\n",
      "7           OR5M1_transcriptomics  2.313998e-16\n",
      "8        PRAMEF17_transcriptomics  2.313998e-16\n",
      "9           VPS52_transcriptomics  2.313998e-16\n",
      "10           AQP2_transcriptomics  2.313998e-16\n",
      "11           BRD2_transcriptomics  2.313998e-16\n",
      "12       C18orf64_transcriptomics  2.313998e-16\n",
      "13        CREB3L4_transcriptomics  2.313998e-16\n",
      "14           DAXX_transcriptomics  2.313998e-16\n",
      "15          GLRA1_transcriptomics  2.313998e-16\n",
      "16          IQCF2_transcriptomics  2.313998e-16\n",
      "17         KCNK18_transcriptomics  2.313998e-16\n",
      "18       KRTAP9-2_transcriptomics  2.313998e-16\n",
      "19       KRTAP9-6_transcriptomics  2.313998e-16\n",
      "20       KRTAP9-7_transcriptomics  2.313998e-16\n",
      "21       KRTAP9-8_transcriptomics  2.313998e-16\n",
      "22   LOC100294145_transcriptomics  2.313998e-16\n",
      "23   LOC101927257_transcriptomics  2.313998e-16\n",
      "24   LOC101928600_transcriptomics  2.313998e-16\n",
      "25          OR9G4_transcriptomics  2.313998e-16\n",
      "26           RGL2_transcriptomics  2.313998e-16\n",
      "27          RPS18_transcriptomics  2.313998e-16\n",
      "28     SLC6A1-AS1_transcriptomics  2.313998e-16\n",
      "29       SNAR-A13_transcriptomics  2.313998e-16\n",
      "..                            ...           ...\n",
      "214         PGAP2_transcriptomics  3.985873e-07\n",
      "215  LOC101927446_transcriptomics  4.657197e-07\n",
      "216        PTCHD4_transcriptomics  4.814934e-07\n",
      "217         GCNT3_transcriptomics  5.190399e-07\n",
      "218        SLC5A9_transcriptomics  5.308594e-07\n",
      "219     LOC646938_transcriptomics  5.491268e-07\n",
      "220         CTRB1_transcriptomics  5.514972e-07\n",
      "221   SNORD115-46_transcriptomics  5.889080e-07\n",
      "222        DUOXA1_transcriptomics  6.054336e-07\n",
      "223         ITGA6_transcriptomics  6.228505e-07\n",
      "224         UHRF1_transcriptomics  6.659445e-07\n",
      "225       TMEM144_transcriptomics  6.823604e-07\n",
      "226         PSMB8_transcriptomics  7.713737e-07\n",
      "227         MYO3A_transcriptomics  7.901197e-07\n",
      "228        CYP2C9_transcriptomics  8.527977e-07\n",
      "229         KCNK6_transcriptomics  9.193175e-07\n",
      "230          DQX1_transcriptomics  1.009286e-06\n",
      "231       SCGB3A1_transcriptomics  1.050847e-06\n",
      "232          ARSA_transcriptomics  1.051925e-06\n",
      "233       RANBP17_transcriptomics  1.054393e-06\n",
      "234       SLC15A2_transcriptomics  1.163659e-06\n",
      "235  LOC101928766_transcriptomics  1.164131e-06\n",
      "236      ARHGAP26_transcriptomics  1.248285e-06\n",
      "237          TRHR_transcriptomics  1.279685e-06\n",
      "238         DEGS2_transcriptomics  1.352818e-06\n",
      "239     METTL21EP_transcriptomics  1.509689e-06\n",
      "240           IHH_transcriptomics  1.584293e-06\n",
      "241         LRP5L_transcriptomics  1.844469e-06\n",
      "242        MTNR1B_transcriptomics  1.873543e-06\n",
      "243          MCM5_transcriptomics  2.096628e-06\n",
      "\n",
      "[244 rows x 2 columns]\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(\"\\nGene: \", gene)\n",
    "\n",
    "'''Use all proteins'''\n",
    "\n",
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "protdf = protdf.loc[:,~protdf.columns.duplicated()]\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the datafram correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", False, all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Print All Signififant Comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Interacting_Protein</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>DUSP4_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.002087e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>SPRY1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.500902e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>GALNT10_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.801284e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Colon</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>TMCC1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.559702e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Colon</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>FAIM2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.082800e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Colon</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>TMEM211_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.705642e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Colon</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>TGFBI_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.646078e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Colon</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>LY6E_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.340164e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Colon</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>PHLDA1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.263308e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Colon</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>DHX58_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.880427e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>OR1L3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.293846e-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>FAM90A27P_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.188688e-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>TRPM6_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.448553e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>VPS13A_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.204536e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>SERPINA11_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.462675e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>GABRP_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.499937e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>MOGAT1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.312500e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>OR5M1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>PRAMEF17_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>VPS52_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>AQP2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>BRD2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>C18orf64_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>CREB3L4_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>DAXX_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>GLRA1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>IQCF2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KCNK18_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KRTAP9-2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KRTAP9-6_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.313998e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>PGAP2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.985873e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>LOC101927446_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.657197e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>PTCHD4_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.814934e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>GCNT3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.190399e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>SLC5A9_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.308594e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>LOC646938_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.491268e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>CTRB1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.514972e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>SNORD115-46_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.889080e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>DUOXA1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.054336e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>ITGA6_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.228505e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>UHRF1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.659445e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>TMEM144_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.823604e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>PSMB8_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.713737e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>MYO3A_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.901197e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>CYP2C9_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.527977e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>KCNK6_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.193175e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>DQX1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.009286e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>SCGB3A1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.050847e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>ARSA_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.051925e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>RANBP17_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.054393e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>SLC15A2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.163659e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>LOC101928766_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.164131e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>ARHGAP26_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.248285e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>TRHR_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.279685e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>DEGS2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.352818e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>METTL21EP_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.509689e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>IHH_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.584293e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>LRP5L_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.844469e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>MTNR1B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.873543e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>Ovarian</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>MCM5_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.096628e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>254 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Cancer_Type  Gene                    Comparison Interacting_Protein  \\\n",
       "0    Endometrial  KRAS         DUSP4_transcriptomics               False   \n",
       "1    Endometrial  KRAS         SPRY1_transcriptomics               False   \n",
       "0          Colon  KRAS       GALNT10_transcriptomics               False   \n",
       "1          Colon  KRAS         TMCC1_transcriptomics               False   \n",
       "2          Colon  KRAS         FAIM2_transcriptomics               False   \n",
       "3          Colon  KRAS       TMEM211_transcriptomics               False   \n",
       "4          Colon  KRAS         TGFBI_transcriptomics               False   \n",
       "5          Colon  KRAS          LY6E_transcriptomics               False   \n",
       "6          Colon  KRAS        PHLDA1_transcriptomics               False   \n",
       "7          Colon  KRAS         DHX58_transcriptomics               False   \n",
       "0        Ovarian  KRAS         OR1L3_transcriptomics               False   \n",
       "1        Ovarian  KRAS     FAM90A27P_transcriptomics               False   \n",
       "2        Ovarian  KRAS         TRPM6_transcriptomics               False   \n",
       "3        Ovarian  KRAS        VPS13A_transcriptomics               False   \n",
       "4        Ovarian  KRAS     SERPINA11_transcriptomics               False   \n",
       "5        Ovarian  KRAS         GABRP_transcriptomics               False   \n",
       "6        Ovarian  KRAS        MOGAT1_transcriptomics               False   \n",
       "7        Ovarian  KRAS         OR5M1_transcriptomics               False   \n",
       "8        Ovarian  KRAS      PRAMEF17_transcriptomics               False   \n",
       "9        Ovarian  KRAS         VPS52_transcriptomics               False   \n",
       "10       Ovarian  KRAS          AQP2_transcriptomics               False   \n",
       "11       Ovarian  KRAS          BRD2_transcriptomics               False   \n",
       "12       Ovarian  KRAS      C18orf64_transcriptomics               False   \n",
       "13       Ovarian  KRAS       CREB3L4_transcriptomics               False   \n",
       "14       Ovarian  KRAS          DAXX_transcriptomics               False   \n",
       "15       Ovarian  KRAS         GLRA1_transcriptomics               False   \n",
       "16       Ovarian  KRAS         IQCF2_transcriptomics               False   \n",
       "17       Ovarian  KRAS        KCNK18_transcriptomics               False   \n",
       "18       Ovarian  KRAS      KRTAP9-2_transcriptomics               False   \n",
       "19       Ovarian  KRAS      KRTAP9-6_transcriptomics               False   \n",
       "..           ...   ...                           ...                 ...   \n",
       "214      Ovarian  KRAS         PGAP2_transcriptomics               False   \n",
       "215      Ovarian  KRAS  LOC101927446_transcriptomics               False   \n",
       "216      Ovarian  KRAS        PTCHD4_transcriptomics               False   \n",
       "217      Ovarian  KRAS         GCNT3_transcriptomics               False   \n",
       "218      Ovarian  KRAS        SLC5A9_transcriptomics               False   \n",
       "219      Ovarian  KRAS     LOC646938_transcriptomics               False   \n",
       "220      Ovarian  KRAS         CTRB1_transcriptomics               False   \n",
       "221      Ovarian  KRAS   SNORD115-46_transcriptomics               False   \n",
       "222      Ovarian  KRAS        DUOXA1_transcriptomics               False   \n",
       "223      Ovarian  KRAS         ITGA6_transcriptomics               False   \n",
       "224      Ovarian  KRAS         UHRF1_transcriptomics               False   \n",
       "225      Ovarian  KRAS       TMEM144_transcriptomics               False   \n",
       "226      Ovarian  KRAS         PSMB8_transcriptomics               False   \n",
       "227      Ovarian  KRAS         MYO3A_transcriptomics               False   \n",
       "228      Ovarian  KRAS        CYP2C9_transcriptomics               False   \n",
       "229      Ovarian  KRAS         KCNK6_transcriptomics               False   \n",
       "230      Ovarian  KRAS          DQX1_transcriptomics               False   \n",
       "231      Ovarian  KRAS       SCGB3A1_transcriptomics               False   \n",
       "232      Ovarian  KRAS          ARSA_transcriptomics               False   \n",
       "233      Ovarian  KRAS       RANBP17_transcriptomics               False   \n",
       "234      Ovarian  KRAS       SLC15A2_transcriptomics               False   \n",
       "235      Ovarian  KRAS  LOC101928766_transcriptomics               False   \n",
       "236      Ovarian  KRAS      ARHGAP26_transcriptomics               False   \n",
       "237      Ovarian  KRAS          TRHR_transcriptomics               False   \n",
       "238      Ovarian  KRAS         DEGS2_transcriptomics               False   \n",
       "239      Ovarian  KRAS     METTL21EP_transcriptomics               False   \n",
       "240      Ovarian  KRAS           IHH_transcriptomics               False   \n",
       "241      Ovarian  KRAS         LRP5L_transcriptomics               False   \n",
       "242      Ovarian  KRAS        MTNR1B_transcriptomics               False   \n",
       "243      Ovarian  KRAS          MCM5_transcriptomics               False   \n",
       "\n",
       "          P_Value  \n",
       "0    3.002087e-08  \n",
       "1    2.500902e-07  \n",
       "0    6.801284e-09  \n",
       "1    4.559702e-08  \n",
       "2    2.082800e-07  \n",
       "3    4.705642e-07  \n",
       "4    6.646078e-07  \n",
       "5    8.340164e-07  \n",
       "6    1.263308e-06  \n",
       "7    1.880427e-06  \n",
       "0    8.293846e-28  \n",
       "1    1.188688e-23  \n",
       "2    7.448553e-18  \n",
       "3    4.204536e-17  \n",
       "4    8.462675e-17  \n",
       "5    9.499937e-17  \n",
       "6    2.312500e-16  \n",
       "7    2.313998e-16  \n",
       "8    2.313998e-16  \n",
       "9    2.313998e-16  \n",
       "10   2.313998e-16  \n",
       "11   2.313998e-16  \n",
       "12   2.313998e-16  \n",
       "13   2.313998e-16  \n",
       "14   2.313998e-16  \n",
       "15   2.313998e-16  \n",
       "16   2.313998e-16  \n",
       "17   2.313998e-16  \n",
       "18   2.313998e-16  \n",
       "19   2.313998e-16  \n",
       "..            ...  \n",
       "214  3.985873e-07  \n",
       "215  4.657197e-07  \n",
       "216  4.814934e-07  \n",
       "217  5.190399e-07  \n",
       "218  5.308594e-07  \n",
       "219  5.491268e-07  \n",
       "220  5.514972e-07  \n",
       "221  5.889080e-07  \n",
       "222  6.054336e-07  \n",
       "223  6.228505e-07  \n",
       "224  6.659445e-07  \n",
       "225  6.823604e-07  \n",
       "226  7.713737e-07  \n",
       "227  7.901197e-07  \n",
       "228  8.527977e-07  \n",
       "229  9.193175e-07  \n",
       "230  1.009286e-06  \n",
       "231  1.050847e-06  \n",
       "232  1.051925e-06  \n",
       "233  1.054393e-06  \n",
       "234  1.163659e-06  \n",
       "235  1.164131e-06  \n",
       "236  1.248285e-06  \n",
       "237  1.279685e-06  \n",
       "238  1.352818e-06  \n",
       "239  1.509689e-06  \n",
       "240  1.584293e-06  \n",
       "241  1.844469e-06  \n",
       "242  1.873543e-06  \n",
       "243  2.096628e-06  \n",
       "\n",
       "[254 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if len(all_significant_comparisons) == 0:\n",
    "    print('No Significant Comparisons!')\n",
    "    \n",
    "else:\n",
    "    display(all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Write Significant Comparisons (if any) to Shared CSV file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "existing_results = pd.read_csv(gene+'_Trans_Results.csv')\n",
    "\n",
    "updated_results = pd.concat([existing_results, all_significant_comparisons], sort=False)\n",
    "\n",
    "updated_results.to_csv(path_or_buf = gene + '_Trans_Results.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
