{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to the CPTAC data service package. Available datasets may be\n",
      "viewed using CPTAC.list_data(). In order to access a specific data\n",
      "set, import a CPTAC subfolder using either 'import CPTAC.Dataset' or\n",
      "'from CPTAC import Dataset'.\n",
      "******\n",
      "Version: 0.3.1\n",
      "******\n",
      "Loading Endometrial CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Acetylation Proteomics Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import CPTAC.Endometrial as en"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Below are the available endometrial data frames contained in this package:\n",
      "\t clinical\n",
      "\t \t Dimensions: (144, 26)\n",
      "\t derived_molecular\n",
      "\t \t Dimensions: (144, 118)\n",
      "\t experimental_setup\n",
      "\t \t Dimensions: (144, 26)\n",
      "\t acetylproteomics\n",
      "\t \t Dimensions: (144, 10862)\n",
      "\t proteomics\n",
      "\t \t Dimensions: (144, 10999)\n",
      "\t transcriptomics\n",
      "\t \t Dimensions: (109, 28057)\n",
      "\t circular_RNA\n",
      "\t \t Dimensions: (109, 4945)\n",
      "\t miRNA\n",
      "\t \t Dimensions: (99, 2337)\n",
      "\t CNA\n",
      "\t \t Dimensions: (95, 28057)\n",
      "\t phosphoproteomics\n",
      "\t \t Dimensions: (144, 73212)\n",
      "\t phosphoproteomics_gene\n",
      "\t \t Dimensions: (144, 8466)\n",
      "\t somatic_mutation_binary\n",
      "\t \t Dimensions: (95, 51559)\n",
      "\t somatic_mutation\n",
      "\t \t Dimensions: (52560, 5)\n"
     ]
    }
   ],
   "source": [
    "en.list_data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on package CPTAC.Endometrial in CPTAC:\n",
      "\n",
      "NAME\n",
      "    CPTAC.Endometrial\n",
      "\n",
      "DESCRIPTION\n",
      "    #   Copyright 2018 Samuel Payne sam_payne@byu.edu\n",
      "    #   Licensed under the Apache License, Version 2.0 (the \"License\");\n",
      "    #   you may not use this file except in compliance with the License.\n",
      "    #   You may obtain a copy of the License at\n",
      "    #       http://www.apache.org/licenses/LICENSE-2.0\n",
      "    #   Unless required by applicable law or agreed to in writing, software\n",
      "    #   distributed under the License is distributed on an \"AS IS\" BASIS,\n",
      "    #   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
      "    #   See the License for the specific language governing permissions and\n",
      "    #   limitations under the License.\n",
      "\n",
      "PACKAGE CONTENTS\n",
      "    dataframe\n",
      "    datatest\n",
      "    fileLoader\n",
      "    utilities\n",
      "\n",
      "FUNCTIONS\n",
      "    append_metadata_to_omics(metadata_df, omics_df, metadata_cols=None, omics_cols=None)\n",
      "        Joins columns from a metadata dataframe (clinical, derived_molecular, or experimental_setup) to part or all of an omics dataframe. Intersection (inner join) of indicies is used.\n",
      "        \n",
      "        Parameters:\n",
      "        metadata_df (pandas DataFrame): Metadata dataframe to select columns from. Either clinical, derived_molecular, or experimental_setup.\n",
      "        omics_df (pandas DataFrame): Omics dataframe to append the metadata columns to.\n",
      "        metadata_cols (str, or list or array-like of str, optional): Column(s) to select from the metadata dataframe. str if one gene, list or array-like of str if multiple. Default is None, which will select the entire metadata dataframe.\n",
      "        omics_cols (str, or list or array-like of str, optional): Column(s) to select from the omics dataframe. str if one gene, list or array-like of str if multiple. Default is None, which will select entire dataframe.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The selected metadata columns, merged with all or part of the omics dataframe.\n",
      "    \n",
      "    append_mutations_to_omics(omics_df, mutation_genes, omics_genes=None, show_location=True)\n",
      "        Select all mutations for specified gene(s), and appends them to all or part of the given omics dataframe. Intersection (inner join) of indicies is used. Each location or mutation cell contains a list, which contains the one or more location or mutation values corresponding to that sample for that gene, or a value indicating that the sample didn't have a mutation in that gene.\n",
      "        \n",
      "        Parameters:\n",
      "        omics_df (pandas DataFrame): Omics dataframe to append the mutation data to.\n",
      "        mutation_genes (str, or list or array-like of str): The gene(s) to get mutation data for. str if one gene, list or array-like of str if multiple.\n",
      "        omics_genes (str, or list or array-like of str, optional): Gene(s) to select from the omics dataframe. str if one gene, list or array-like of str if multiple. Default will select entire dataframe.\n",
      "        show_location (bool, optional): Whether to include the Locations column from the mutation dataframe. Defaults to True.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The mutations for the specified gene, appended to all or part of the omics dataframe. Each location or mutation cell contains a list, which contains the one or more location or mutation values corresponding to that sample for that gene, or a value indicating that the sample didn't have a mutation in that gene.\n",
      "    \n",
      "    compare_omics(omics_df1, omics_df2, cols1=None, cols2=None)\n",
      "        Take specified column(s) from one omics dataframe, and append to specified columns(s) from another omics dataframe. Intersection (inner join) of indicies is used.\n",
      "        \n",
      "        Parameters:\n",
      "        omics_df1 (pandas DataFrame): First omics dataframe to select columns from.\n",
      "        omics_df2 (pandas DataFrame): Second omics dataframe to select columns from.\n",
      "        cols1 (str, or list or array-like of str, optional): Column(s) to select from omics_df1. str if one key, list or array-like of str if multiple. Defaults to None, in which case we'll select the entire dataframe.\n",
      "        cols2 (str, or list or array-like of str, optional): Column(s) to select from omics_df2. str if one key, list or array-like of str if multiple. Defaults to None, in which case we'll select the entire dataframe.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The selected columns from omics_df1 and omics_df2, merged into one dataframe.\n",
      "    \n",
      "    create_patient_ids(clinical)\n",
      "    \n",
      "    define(term)\n",
      "        Parameters\n",
      "        term: string of term to be defined\n",
      "        \n",
      "        Returns\n",
      "        String definition of provided term\n",
      "    \n",
      "    embargo()\n",
      "        Parameters\n",
      "        None\n",
      "        \n",
      "        Opens CPTAC embargo details in web browser\n",
      "        \n",
      "        Returns\n",
      "        None\n",
      "    \n",
      "    get_CNA(unfiltered=False)\n",
      "        Parameters\n",
      "        unfiltered: boolean indicating whether to return unfiltered CNA data\n",
      "        \n",
      "        Returns\n",
      "        CNA dataframe\n",
      "    \n",
      "    get_acetylproteomics(unfiltered=False)\n",
      "        Parameters\n",
      "        unfiltered: boolean indicating whether to return unfiltered acetylproteomics data\n",
      "        \n",
      "        Returns\n",
      "        Acetylproteomics dataframe\n",
      "    \n",
      "    get_circular_RNA(unfiltered=False)\n",
      "        Gets circular_RNA dataframe.\n",
      "        \n",
      "        Parameters:\n",
      "        unfiltered (bool, optional): Whether to include unfiltered samples. Default is false.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The circular_RNA dataframe.\n",
      "    \n",
      "    get_clinical(unfiltered=False)\n",
      "        Parameters\n",
      "        unfiltered: boolean indicating whether to return unfiltered clinical data, aka clinical[\"Case_excluded\"] == \"Yes\"\n",
      "        \n",
      "        Returns\n",
      "        Clinical dataframe\n",
      "    \n",
      "    get_derived_molecular(unfiltered=False)\n",
      "        Parameters\n",
      "        unfiltered: boolean indicating whether to return unfiltered derived molecular data\n",
      "        \n",
      "        Returns\n",
      "        Derived Molecular dataframe\n",
      "    \n",
      "    get_experimental_setup(unfiltered=False)\n",
      "        Parameters\n",
      "        unfiltered: boolean indicating whether to return unfiltered experimental setup data\n",
      "        \n",
      "        Returns\n",
      "        Experimental Setup dataframe\n",
      "    \n",
      "    get_miRNA(unfiltered=False)\n",
      "        Gets miRNA dataframe.\n",
      "        \n",
      "        Parameters:\n",
      "        unfiltered (bool, optional): Whether to include unfiltered samples. Default is false.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The miRNA dataframe.\n",
      "    \n",
      "    get_mutations(unfiltered=False)\n",
      "        Gets the somatic_mutation dataframe.\n",
      "        \n",
      "        Parameters:\n",
      "        unfiltered (bool, optional): Whether to include unfiltered samples. Default is false.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The somatic_mutation dataframe.\n",
      "    \n",
      "    get_mutations_binary(unfiltered=False)\n",
      "        Gets the somatic_mutation_binary dataframe, which has a binary value indicating, for each location on each gene, whether there was a mutation in that gene at that location, for each sample.\n",
      "        \n",
      "        Parameters:\n",
      "        unfiltered (bool, optional): Whether to include unfiltered samples. Default is false.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The somatic_mutation_binary dataframe.\n",
      "    \n",
      "    get_phosphoproteomics(unfiltered=False)\n",
      "        Gets the phosphoproteomics dataframe.\n",
      "        \n",
      "        Parameters:\n",
      "        unfiltered (bool, optional): Whether to include unfiltered samples. Default is false.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The phosphoproteomics dataframe.\n",
      "    \n",
      "    get_phosphoproteomics_gene(unfiltered=False)\n",
      "        Gets the phosphoproteomics_gene dataframe. The gene level phosphorylation measurement is an aggregate metric which potentially averages together individual measurements of different sites. Use get_phosphoproteomics() to view the data for individual sites.\n",
      "        \n",
      "        Parameters:\n",
      "        unfiltered (bool, optional): Whether to include unfiltered samples. Default is false.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The phosphoproteomics_gene dataframe.\n",
      "    \n",
      "    get_phosphosites(genes)\n",
      "        Returns dataframe with all phosphosites of specified gene or list of genes.\n",
      "        \n",
      "        Parameters:\n",
      "        genes (str, or list or array-like of str): gene or list of genes to use to select phosphosites. str if single, list or array-like of str if multiple.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The phosphoproteomics for the specified gene(s).\n",
      "    \n",
      "    get_proteomics(unfiltered=False)\n",
      "        Parameters\n",
      "        unfiltered: boolean indicating whether to return unfiltered proteomics data\n",
      "        \n",
      "        Returns\n",
      "        Proteomics dataframe\n",
      "    \n",
      "    get_transcriptomics(unfiltered=False)\n",
      "        Gets transcriptomics dataframe.\n",
      "        \n",
      "        Parameters:\n",
      "        unfiltered (bool, optional): Whether to include unfiltered samples. Default is false.\n",
      "        \n",
      "        Returns:\n",
      "        pandas DataFrame: The transcriptomics dataframe.\n",
      "    \n",
      "    git_help()\n",
      "        Parameters\n",
      "        None\n",
      "        \n",
      "        Opens github help page\n",
      "        \n",
      "        Returns\n",
      "        None\n",
      "    \n",
      "    link_patient_ids(patient_ids, somatic)\n",
      "    \n",
      "    list_api()\n",
      "        Parameters\n",
      "        None\n",
      "        \n",
      "        Prints docstrings for all accessible functions\n",
      "        \n",
      "        Returns\n",
      "        None\n",
      "    \n",
      "    list_data()\n",
      "        Parameters\n",
      "        None\n",
      "        \n",
      "        Prints list of loaded data frames and dimensions\n",
      "        \n",
      "        Returns\n",
      "        None\n",
      "    \n",
      "    search(term)\n",
      "        Parameters\n",
      "        term: string of term to be searched\n",
      "        \n",
      "        Performs online search of provided term\n",
      "        \n",
      "        Returns\n",
      "        None\n",
      "    \n",
      "    unfiltered_warning()\n",
      "        Parameters\n",
      "        None\n",
      "        \n",
      "        Prints warning to about the unfiltered data\n",
      "        \n",
      "        Returns\n",
      "        None\n",
      "    \n",
      "    version()\n",
      "        Parameters\n",
      "        None\n",
      "        \n",
      "        Prints version number of CPTAC package\n",
      "        \n",
      "        Returns\n",
      "        Version number\n",
      "    \n",
      "    warning()\n",
      "\n",
      "DATA\n",
      "    __warningregistry__ = {'version': 39, (r\"unclosed file <_io.TextIOWrap...\n",
      "    acetylproteomics = idx   A2M-K1168  A2M-K1176  A2M-K135  A2M-K145  ......\n",
      "    acetylproteomics_u = idx   A2M-K1168  A2M-K1176  A2M-K135  A2M-K145  ....\n",
      "    casesToDrop = Index(['S004', 'S013', 'S015', 'S035', 'S043', '...52', ...\n",
      "    clinical =      Proteomics_Participant_ID Proteomics_Tumor_...        ...\n",
      "    clinical_file_data =      Proteomics_Participant_ID Case_excluded  Pr....\n",
      "    clinical_unfiltered =      Proteomics_Participant_ID Case_excluded Pro...\n",
      "    cna = idx   MFSD14A  SASS6  TRMT13  LRRC39   DBT  RTCA....18    -0.18 ...\n",
      "    cna_u = idx   MFSD14A  SASS6  TRMT13  LRRC39   DBT  RTCA...18    -0.18...\n",
      "    data_directory = r'C:\\Users\\Daniel\\AppData\\Local\\conda\\conda\\envs\\Rese...\n",
      "    derived_molecular =          Estrogen_Receptor  Estrogen_Receptor_% .....\n",
      "    derived_molecular_u =          Estrogen_Receptor  Estrogen_Receptor_% ...\n",
      "    dict = {'Age': 'Age of the specified patient.', 'BMI': 'BMI (body mass...\n",
      "    dir_path = r'C:\\Users\\Daniel\\AppData\\Local\\conda\\conda\\envs\\Research\\l...\n",
      "    experimental_setup =       Proteomics_TMT_batch  Proteomics_TMT_plex ....\n",
      "    experimental_setup_u =       Proteomics_TMT_batch  Proteomics_TMT_plex...\n",
      "    file = <_io.TextIOWrapper name='C:\\\\Users\\\\Daniel\\\\AppD...ata\\\\definit...\n",
      "    line = ['Tumor_purity', 'Categorical classification of whether the tum...\n",
      "    miRNA = idx   hsa-let-7a-2-3p  hsa-let-7a-3p  hsa-let-7a... 6.97      ...\n",
      "    miRNA_u = idx   hsa-let-7a-2-3p  hsa-let-7a-3p  hsa-let-7a...6.97     ...\n",
      "    patient_ids = {'C3L-00006': 'S001', 'C3L-00008': 'S002', 'C3L-00032': ...\n",
      "    phosphoproteomics = idx   AAAS-S495  AAAS-S541  AAAS-Y485  AACS-S618.....\n",
      "    phosphoproteomics_gene = idx       AAAS    AACS   AAED1    AAGAB      ...\n",
      "    phosphoproteomics_gene_u = idx       AAAS    AACS   AAED1    AAGAB    ...\n",
      "    phosphoproteomics_u = idx   AAAS-S495  AAAS-S541  AAAS-Y485  AACS-S618...\n",
      "    proteomics = idx     A1BG     A2M    A2ML1  A4GALT      AAAS ...-0.093...\n",
      "    proteomics_u = idx     A1BG     A2M    A2ML1  A4GALT      AAAS ...-0.0...\n",
      "    somatic_binary = idx   A1BG_p.E298K  A1BG_p.S181N  A1CF_p.F487L  ...  ...\n",
      "    somatic_binary_u = idx   A1BG_p.E298K  A1BG_p.S181N  A1CF_p.F487L  ......\n",
      "    somatic_mutation =       Clinical_Patient_Key Patient_Id       Gene......\n",
      "    somatic_mutation_u =                      Patient_Id       Gene      ....\n",
      "    transcriptomics = idx   A1BG  A1BG-AS1  A1CF    A2M  A2M-AS1  A2ML...\n",
      "    ...\n",
      "    transcriptomics_circular = idx   circ_chr10_100260218_100262063_CWF19L...\n",
      "    transcriptomics_circular_u = idx   circ_chr10_100260218_100262063_CWF1...\n",
      "    transcriptomics_u = idx   A1BG  A1BG-AS1  A1CF    A2M  A2M-AS1  A2ML.....\n",
      "\n",
      "FILE\n",
      "    c:\\users\\daniel\\appdata\\local\\conda\\conda\\envs\\research\\lib\\site-packages\\cptac\\endometrial\\__init__.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "en.list_api()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Samples: ['S001', 'S002', 'S003', 'S005', 'S006', 'S007', 'S008', 'S009', 'S010', 'S011', 'S012', 'S014', 'S016', 'S017', 'S018', 'S019', 'S020', 'S021', 'S022', 'S023', 'S024', 'S025', 'S026', 'S027', 'S028', 'S029', 'S030', 'S031', 'S032', 'S033']\n",
      "Proteins: ['A1BG', 'A2M', 'A2ML1', 'A4GALT', 'AAAS', 'AACS', 'AADAT', 'AAED1', 'AAGAB', 'AAK1', 'AAMDC', 'AAMP', 'AAR2', 'AARS', 'AARS2', 'AARSD1', 'AASDHPPT', 'AASS', 'AATF', 'ABAT', 'ABCA8', 'ABCB1', 'ABCB10', 'ABCB6', 'ABCB7', 'ABCB8', 'ABCC1', 'ABCC10', 'ABCC3', 'ABCC4']\n"
     ]
    }
   ],
   "source": [
    "proteomics = en.get_proteomics()\n",
    "samples = proteomics.index\n",
    "proteins = proteomics.columns\n",
    "print(\"Samples:\",samples[0:30].tolist())\n",
    "print(\"Proteins:\",proteins[0:30].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>idx</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>AAK1</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001</th>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.802</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>0.412</td>\n",
       "      <td>-0.664</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.1230</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S002</th>\n",
       "      <td>-0.685</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.684</td>\n",
       "      <td>0.984</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.3340</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>0.1390</td>\n",
       "      <td>1.330</td>\n",
       "      <td>-0.367</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.737</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.0757</td>\n",
       "      <td>-0.4730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S003</th>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.419</td>\n",
       "      <td>-0.500</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.3200</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S005</th>\n",
       "      <td>-1.670</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.443</td>\n",
       "      <td>0.243</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>-0.9290</td>\n",
       "      <td>0.229</td>\n",
       "      <td>-0.223</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0552</td>\n",
       "      <td>-0.0714</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.156</td>\n",
       "      <td>-0.398</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.0301</td>\n",
       "      <td>-0.4670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S006</th>\n",
       "      <td>-0.374</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.537</td>\n",
       "      <td>0.311</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.565</td>\n",
       "      <td>-0.101</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.2200</td>\n",
       "      <td>-0.5620</td>\n",
       "      <td>0.937</td>\n",
       "      <td>-0.646</td>\n",
       "      <td>0.20700</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.1760</td>\n",
       "      <td>0.0513</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 10999 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "idx    A1BG     A2M  A2ML1  A4GALT    AAAS    AACS   AADAT   AAED1  AAGAB  \\\n",
       "S001 -1.180 -0.8630 -0.802   0.222  0.2560  0.6650  1.2800 -0.3390  0.412   \n",
       "S002 -0.685 -1.0700 -0.684   0.984  0.1350  0.3340  1.3000  0.1390  1.330   \n",
       "S003 -0.528 -1.3200  0.435     NaN -0.2400  1.0400 -0.0213 -0.0479  0.419   \n",
       "S005 -1.670 -1.1900 -0.443   0.243 -0.0993  0.7570  0.7400 -0.9290  0.229   \n",
       "S006 -0.374 -0.0206 -0.537   0.311  0.3750  0.0131 -1.1000     NaN  0.565   \n",
       "\n",
       "idx    AAK1  ...   ZSWIM8  ZSWIM9    ZW10  ZWILCH  ZWINT   ZXDC   ZYG11B  \\\n",
       "S001 -0.664  ... -0.08770     NaN  0.0229  0.1090    NaN -0.332 -0.43300   \n",
       "S002 -0.367  ... -0.03560     NaN  0.3630  1.0700  0.737 -0.564 -0.00461   \n",
       "S003 -0.500  ...  0.00112 -0.1450  0.0105 -0.1160    NaN  0.151 -0.07400   \n",
       "S005 -0.223  ...  0.07250 -0.0552 -0.0714  0.0933  0.156 -0.398 -0.07520   \n",
       "S006 -0.101  ... -0.17600     NaN -1.2200 -0.5620  0.937 -0.646  0.20700   \n",
       "\n",
       "idx     ZYX   ZZEF1    ZZZ3  \n",
       "S001 -1.020 -0.1230 -0.0859  \n",
       "S002 -1.130 -0.0757 -0.4730  \n",
       "S003 -0.540  0.3200 -0.4190  \n",
       "S005 -0.797 -0.0301 -0.4670  \n",
       "S006 -1.850 -0.1760  0.0513  \n",
       "\n",
       "[5 rows x 10999 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteomics.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>idx</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2M-AS1</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A2MP1</th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001</th>\n",
       "      <td>4.02</td>\n",
       "      <td>2.16</td>\n",
       "      <td>3.27</td>\n",
       "      <td>13.39</td>\n",
       "      <td>5.88</td>\n",
       "      <td>6.79</td>\n",
       "      <td>1.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>10.34</td>\n",
       "      <td>1.96</td>\n",
       "      <td>...</td>\n",
       "      <td>11.06</td>\n",
       "      <td>10.73</td>\n",
       "      <td>8.40</td>\n",
       "      <td>9.78</td>\n",
       "      <td>10.88</td>\n",
       "      <td>5.93</td>\n",
       "      <td>11.52</td>\n",
       "      <td>10.23</td>\n",
       "      <td>11.50</td>\n",
       "      <td>11.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S002</th>\n",
       "      <td>4.81</td>\n",
       "      <td>2.21</td>\n",
       "      <td>4.86</td>\n",
       "      <td>13.24</td>\n",
       "      <td>5.93</td>\n",
       "      <td>6.33</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.00</td>\n",
       "      <td>10.83</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>10.87</td>\n",
       "      <td>11.43</td>\n",
       "      <td>8.39</td>\n",
       "      <td>9.14</td>\n",
       "      <td>10.38</td>\n",
       "      <td>7.25</td>\n",
       "      <td>11.64</td>\n",
       "      <td>10.64</td>\n",
       "      <td>11.26</td>\n",
       "      <td>11.57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S003</th>\n",
       "      <td>6.24</td>\n",
       "      <td>6.43</td>\n",
       "      <td>3.68</td>\n",
       "      <td>14.32</td>\n",
       "      <td>6.53</td>\n",
       "      <td>9.42</td>\n",
       "      <td>2.79</td>\n",
       "      <td>0.00</td>\n",
       "      <td>10.98</td>\n",
       "      <td>2.13</td>\n",
       "      <td>...</td>\n",
       "      <td>10.06</td>\n",
       "      <td>10.13</td>\n",
       "      <td>8.35</td>\n",
       "      <td>9.27</td>\n",
       "      <td>10.46</td>\n",
       "      <td>6.85</td>\n",
       "      <td>11.60</td>\n",
       "      <td>10.21</td>\n",
       "      <td>11.51</td>\n",
       "      <td>11.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S005</th>\n",
       "      <td>5.31</td>\n",
       "      <td>4.87</td>\n",
       "      <td>5.59</td>\n",
       "      <td>13.77</td>\n",
       "      <td>6.35</td>\n",
       "      <td>4.22</td>\n",
       "      <td>2.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>8.68</td>\n",
       "      <td>1.98</td>\n",
       "      <td>...</td>\n",
       "      <td>10.29</td>\n",
       "      <td>10.41</td>\n",
       "      <td>9.10</td>\n",
       "      <td>9.59</td>\n",
       "      <td>10.15</td>\n",
       "      <td>7.89</td>\n",
       "      <td>11.90</td>\n",
       "      <td>10.21</td>\n",
       "      <td>11.34</td>\n",
       "      <td>11.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S006</th>\n",
       "      <td>9.84</td>\n",
       "      <td>8.83</td>\n",
       "      <td>7.00</td>\n",
       "      <td>13.12</td>\n",
       "      <td>6.49</td>\n",
       "      <td>6.83</td>\n",
       "      <td>1.80</td>\n",
       "      <td>0.00</td>\n",
       "      <td>11.42</td>\n",
       "      <td>3.28</td>\n",
       "      <td>...</td>\n",
       "      <td>10.36</td>\n",
       "      <td>11.24</td>\n",
       "      <td>8.60</td>\n",
       "      <td>9.44</td>\n",
       "      <td>11.80</td>\n",
       "      <td>9.32</td>\n",
       "      <td>11.97</td>\n",
       "      <td>9.77</td>\n",
       "      <td>11.37</td>\n",
       "      <td>12.35</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 28057 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "idx   A1BG  A1BG-AS1  A1CF    A2M  A2M-AS1  A2ML1  A2MP1  A3GALT2  A4GALT  \\\n",
       "S001  4.02      2.16  3.27  13.39     5.88   6.79   1.55     0.97   10.34   \n",
       "S002  4.81      2.21  4.86  13.24     5.93   6.33   0.93     0.00   10.83   \n",
       "S003  6.24      6.43  3.68  14.32     6.53   9.42   2.79     0.00   10.98   \n",
       "S005  5.31      4.87  5.59  13.77     6.35   4.22   2.97     0.00    8.68   \n",
       "S006  9.84      8.83  7.00  13.12     6.49   6.83   1.80     0.00   11.42   \n",
       "\n",
       "idx   A4GNT  ...  ZWILCH  ZWINT  ZXDA  ZXDB   ZXDC  ZYG11A  ZYG11B    ZYX  \\\n",
       "S001   1.96  ...   11.06  10.73  8.40  9.78  10.88    5.93   11.52  10.23   \n",
       "S002   0.00  ...   10.87  11.43  8.39  9.14  10.38    7.25   11.64  10.64   \n",
       "S003   2.13  ...   10.06  10.13  8.35  9.27  10.46    6.85   11.60  10.21   \n",
       "S005   1.98  ...   10.29  10.41  9.10  9.59  10.15    7.89   11.90  10.21   \n",
       "S006   3.28  ...   10.36  11.24  8.60  9.44  11.80    9.32   11.97   9.77   \n",
       "\n",
       "idx   ZZEF1   ZZZ3  \n",
       "S001  11.50  11.47  \n",
       "S002  11.26  11.57  \n",
       "S003  11.51  11.09  \n",
       "S005  11.34  11.51  \n",
       "S006  11.37  12.35  \n",
       "\n",
       "[5 rows x 28057 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transcriptomics = en.get_transcriptomics()\n",
    "transcriptomics.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics_Participant_ID</th>\n",
       "      <th>Proteomics_Tumor_Normal</th>\n",
       "      <th>Country</th>\n",
       "      <th>Histologic_Grade_FIGO</th>\n",
       "      <th>Myometrial_invasion_Specify</th>\n",
       "      <th>Histologic_type</th>\n",
       "      <th>Treatment_naive</th>\n",
       "      <th>Tumor_purity</th>\n",
       "      <th>Path_Stage_Primary_Tumor-pT</th>\n",
       "      <th>Path_Stage_Reg_Lymph_Nodes-pN</th>\n",
       "      <th>...</th>\n",
       "      <th>Age</th>\n",
       "      <th>Diabetes</th>\n",
       "      <th>Race</th>\n",
       "      <th>Ethnicity</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Tumor_Site</th>\n",
       "      <th>Tumor_Site_Other</th>\n",
       "      <th>Tumor_Focality</th>\n",
       "      <th>Tumor_Size_cm</th>\n",
       "      <th>Num_full_term_pregnancies</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>idx</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001</th>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>United States</td>\n",
       "      <td>FIGO grade 1</td>\n",
       "      <td>under 50 %</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pN0</td>\n",
       "      <td>...</td>\n",
       "      <td>64.0</td>\n",
       "      <td>No</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Female</td>\n",
       "      <td>Anterior endometrium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>2.9</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S002</th>\n",
       "      <td>C3L-00008</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>United States</td>\n",
       "      <td>FIGO grade 1</td>\n",
       "      <td>under 50 %</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pNX</td>\n",
       "      <td>...</td>\n",
       "      <td>58.0</td>\n",
       "      <td>No</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Female</td>\n",
       "      <td>Posterior endometrium</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>3.5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S003</th>\n",
       "      <td>C3L-00032</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>United States</td>\n",
       "      <td>FIGO grade 2</td>\n",
       "      <td>under 50 %</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pN0</td>\n",
       "      <td>...</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Female</td>\n",
       "      <td>Other, specify</td>\n",
       "      <td>Anterior and Posterior endometrium</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>4.5</td>\n",
       "      <td>4 or more</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S005</th>\n",
       "      <td>C3L-00090</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>United States</td>\n",
       "      <td>FIGO grade 2</td>\n",
       "      <td>under 50 %</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pNX</td>\n",
       "      <td>...</td>\n",
       "      <td>75.0</td>\n",
       "      <td>No</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Female</td>\n",
       "      <td>Other, specify</td>\n",
       "      <td>Anterior and Posterior endometrium</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>3.5</td>\n",
       "      <td>4 or more</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S006</th>\n",
       "      <td>C3L-00098</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>United States</td>\n",
       "      <td>NaN</td>\n",
       "      <td>under 50 %</td>\n",
       "      <td>Serous</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pNX</td>\n",
       "      <td>...</td>\n",
       "      <td>63.0</td>\n",
       "      <td>No</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Female</td>\n",
       "      <td>Other, specify</td>\n",
       "      <td>Anterior  and Posterior endometrium</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Proteomics_Participant_ID Proteomics_Tumor_Normal        Country  \\\n",
       "idx                                                                     \n",
       "S001                 C3L-00006                   Tumor  United States   \n",
       "S002                 C3L-00008                   Tumor  United States   \n",
       "S003                 C3L-00032                   Tumor  United States   \n",
       "S005                 C3L-00090                   Tumor  United States   \n",
       "S006                 C3L-00098                   Tumor  United States   \n",
       "\n",
       "     Histologic_Grade_FIGO Myometrial_invasion_Specify Histologic_type  \\\n",
       "idx                                                                      \n",
       "S001          FIGO grade 1                  under 50 %    Endometrioid   \n",
       "S002          FIGO grade 1                  under 50 %    Endometrioid   \n",
       "S003          FIGO grade 2                  under 50 %    Endometrioid   \n",
       "S005          FIGO grade 2                  under 50 %    Endometrioid   \n",
       "S006                   NaN                  under 50 %          Serous   \n",
       "\n",
       "     Treatment_naive Tumor_purity Path_Stage_Primary_Tumor-pT  \\\n",
       "idx                                                             \n",
       "S001             YES       Normal              pT1a (FIGO IA)   \n",
       "S002             YES       Normal              pT1a (FIGO IA)   \n",
       "S003             YES       Normal              pT1a (FIGO IA)   \n",
       "S005             YES       Normal              pT1a (FIGO IA)   \n",
       "S006             YES       Normal              pT1a (FIGO IA)   \n",
       "\n",
       "     Path_Stage_Reg_Lymph_Nodes-pN  ...   Age Diabetes   Race  \\\n",
       "idx                                 ...                         \n",
       "S001                           pN0  ...  64.0       No  White   \n",
       "S002                           pNX  ...  58.0       No  White   \n",
       "S003                           pN0  ...  50.0      Yes  White   \n",
       "S005                           pNX  ...  75.0       No  White   \n",
       "S006                           pNX  ...  63.0       No  White   \n",
       "\n",
       "                   Ethnicity  Gender             Tumor_Site  \\\n",
       "idx                                                           \n",
       "S001  Not-Hispanic or Latino  Female   Anterior endometrium   \n",
       "S002  Not-Hispanic or Latino  Female  Posterior endometrium   \n",
       "S003  Not-Hispanic or Latino  Female         Other, specify   \n",
       "S005  Not-Hispanic or Latino  Female         Other, specify   \n",
       "S006  Not-Hispanic or Latino  Female         Other, specify   \n",
       "\n",
       "                         Tumor_Site_Other Tumor_Focality Tumor_Size_cm  \\\n",
       "idx                                                                      \n",
       "S001                                  NaN       Unifocal           2.9   \n",
       "S002                                  NaN       Unifocal           3.5   \n",
       "S003   Anterior and Posterior endometrium       Unifocal           4.5   \n",
       "S005   Anterior and Posterior endometrium       Unifocal           3.5   \n",
       "S006  Anterior  and Posterior endometrium       Unifocal           6.0   \n",
       "\n",
       "     Num_full_term_pregnancies  \n",
       "idx                             \n",
       "S001                         1  \n",
       "S002                         1  \n",
       "S003                 4 or more  \n",
       "S005                 4 or more  \n",
       "S006                         2  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical = en.get_clinical()\n",
    "clinical.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAVIAAAFgCAYAAADpZ/FJAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBo\ndHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAFexJREFUeJzt3X2QXXV9x/HPJ7sbswvRLHDXoSRp\nXFtZG1pQt4wIbiNQB8GJI9WKo7RabDIdR8CR+jDTiZrptLVWBxm1TYpPVMQqD52MHVNQWFZGRJcn\nIWSluqCJoneRBRY2IZvdb/+4J5Ksyd67/d17z72H92tm59z93XPPfs/uvZ/9nd95ckQIAPD/tyTv\nAgCg3RGkAJCIIAWARAQpACQiSAEgEUEKAIkIUgBIRJACQCKCFAASdTZioeecc05s3769EYsGgGZy\nLTM1pEf66KOPNmKxANCS2LQHgEQEKQAkIkgBIBFBCgCJCFIASESQAkAighQAEhGkAJCIIAWARDUF\nqe1LbN9ve4ftSxtdFAC0k6rn2ts+SdJfSzpV0j5J223/d0T8b6OLK5LhsbK2jIxr1+S0VvX2aONQ\nv9YN9OVdFoA6qKVH+lJJ34uI6YjYL+lWSW9sbFnFMjxW1qZtO1Se2qsV3V0qT+3Vpm07NDxWzrs0\nAHVQS5DeL2nI9rG2eySdK2lVY8sqli0j4+rqsHqWdsquTLs6rC0j43mXBqAOqm7aR8RO2x+TdJOk\npyTdK2n//Plsb5C0QZJWr15d5zLb267Jaa3o7jqkrburQ7snp3OqCEA91bSzKSI+FxEvj4ghSY9J\n+q3x0YjYGhGDETFYKpXqXWdbW9Xboz0zs4e07ZmZ1crenpwqAlBPte6178umqyWdL+maRhZVNBuH\n+jUzG5ret18RlenMbGjjUH/epQGog1qvkH+d7WMlzUh6d0RMNrCmwlk30KfNqoyV7p6c1kr22gOF\nUlOQRsSrG11I0a0b6CM4gYLizCYASESQAkAighQAEhGkAJCIIAWARAQpACQiSAEgEUEKAIkIUgBI\nRJACQCKCFAASEaQAkIggBYBEBCkAJCJIASARQQoAiQhSAEhEkAJAIoIUABIRpACQiCAFgEQEKQAk\nIkgBIBFBCgCJCFIASESQAkAighQAEtUUpLbfa3uH7fttX2N7WaMLA4B2UTVIbZ8g6WJJgxFxkqQO\nSRc0ujAAaBe1btp3Suq23SmpR9IvGlcSALSXqkEaET+X9C+SfibpEUlPRMSN8+ezvcH2qO3RiYmJ\n+lcKAC2qlk37XklvkPQiSb8j6Sjbb58/X0RsjYjBiBgslUr1rxQAWlQtm/ZnS3ooIiYiYkbS9ZJe\n1diyAKB91BKkP5P0Sts9ti3pLEk7G1sWALSPWsZI75B0raS7JN2XvWZrg+sCgLbRWctMEfFhSR9u\ncC0A0JY4swkAEhGkAJCIIAWARAQpACQiSAEgEUEKAIkIUgBIRJACQCKCFAASEaQAkIggBYBEBCkA\nJCJIASARQQoAiQhSAEhEkAJAIoIUABIRpACQiCAFgEQEKQAkIkgBIBFBCgCJCFIASESQAkAighQA\nEhGkAJCoapDaPtH2PQd9PWn70mYUBwDtoLPaDBHxI0mnSJLtDkk/l3RDg+sCgLax2E37syT9JCJ+\n2ohiAKAdLTZIL5B0TSMKAYB2VXOQ2l4qab2krx/h+Q22R22PTkxM1Ks+AGh5i+mRvk7SXRHxq8M9\nGRFbI2IwIgZLpVJ9qgOANrCYIH2r2KwHgN9SU5Da7pH0p5Kub2w5ANB+qh7+JEkRMS3p2AbXAgBt\niTObACARQQoAiQhSAEhEkAJAIoIUABIRpACQiCAFgEQEKQAkIkgBIBFBCgCJCFIASESQAkAighQA\nEhGkAJCIIAWARAQpACQiSAEgEUEKAIkIUgBIRJACQCKCFAASEaQAkIggBYBEBCkAJCJIASARQQoA\niQhSAEhUU5DaXmH7WttjtnfaPq3RhQFAu+iscb5PSdoeEW+yvVRSTwNrAoC2UjVIbT9f0pCkd0hS\nROyTtK+xZQFA+6hl075f0oSkL9i+2/aVto9qcF0A0DZqCdJOSS+X9K8R8TJJT0v64PyZbG+wPWp7\ndGJios5lAkDrqiVId0vaHRF3ZN9fq0qwHiIitkbEYEQMlkqletYIAC2tapBGxC8l7bJ9YtZ0lqQH\nGloVALSRWvfav0fS1dke+3FJ72xcSQDQXmoK0oi4R9Jgg2sBgLbEmU0AkIggBYBEBCkAJCJIASAR\nQQoAiQhSAEhEkAJAIoIUABIRpACQiCAFgEQEKQAkIkgBIBFBCgCJCFIASESQAkAighQAEhGkAJCI\nIAWARAQpACQiSAEgUa13EW2K4bGytoyMa9fktFb19mjjUL/WDfTlXRYALKhleqTDY2Vt2rZD5am9\nWtHdpfLUXm3atkPDY+W8SwOABbVMkG4ZGVdXh9WztFN2ZdrVYW0ZGc+7NABYUMsE6a7JaXV3dRzS\n1t3Vod2T0zlVBAC1aZkgXdXboz0zs4e07ZmZ1crenpwqAoDatEyQbhzq18xsaHrffkVUpjOzoY1D\n/XmXBgALapkgXTfQp83r16pv+TI9sWdGfcuXafP6tey1B9Dyajr8yfbDkqYkzUraHxGDjShm3UBf\nYYOTQ7uA4lpMj/Q1EXFKo0K0yDi0Cyi2ltm0LzIO7QKKrdYgDUk32r7T9obDzWB7g+1R26MTExP1\nq7AAOLQLKLZag/T0iHi5pNdJerftofkzRMTWiBiMiMFSqVTXItsdh3YBxVZTkEbEL7JpWdINkk5t\nZFFFw6FdQLFVDVLbR9lefuCxpNdKur/RhRUJh3YBxVbL4U8vlHSD7QPzfyUitje0qgIq8qFdwHNd\n1SCNiHFJJzehFgBoSxz+BACJCFIASESQAkAighQAEhGkAJCIIAWARAQpACQiSAEgEUEKAIkIUgBI\nRJACQCKCFAASEaQAkIggBYBEBCkAJCJIASBRLVfIRx0Mj5W1ZWRcuyantaq3RxuH+rliPlAQ9Eib\nYHisrE3bdqg8tVcrurtUntqrTdt2aHisnHdpAOqAIG2CLSPj6uqwepZ2yq5MuzqsLSPjeZcGoA4I\n0ibYNTmt7q6OQ9q6uzq0e3I6p4oA1BNB2gSrenu0Z2b2kLY9M7Na2duTU0UA6okgbYKNQ/2amQ1N\n79uviMp0Zja0cag/79IA1AFB2gTrBvq0ef1a9S1fpif2zKhv+TJtXr+WvfZAQXD4U5OsG+gjOIGC\nokcKAIkIUgBIVHOQ2u6wfbftbzSyIABoN4vpkV4iaWejCgGAdlVTkNpeKek8SVc2thwAaD+19kgv\nl/R+SXMNrAUA2lLVILX9eknliLizynwbbI/aHp2YmKhbgQDQ6mrpkZ4uab3thyV9VdKZtr88f6aI\n2BoRgxExWCqV6lwmALSuqkEaER+KiJURsUbSBZJujoi3N7wyAGgTHEcKAIkWdYpoRAxLGm5IJQBQ\nB3ncjYIeKYDCyOtuFAQpgMLI624UBCmAwsjrbhRcRg/JuEMqWsWq3h6Vp/aqZ+mz0daMu1HQI0US\n7pCKVpLX3SgIUiThDqloJXndjYJNeyTZNTmtFd1dh7Rxh1TkKY+7UdAjRRLukAoQpEjEHVIBghSJ\nuEMqQJCijiLvAoCcEKRIwuFPAEGKRBz+BBCkSJTXKXlAKyFIkYTDnwCCFIk4/AkgSJGIw58AThFF\nHeRxSh7QSuiRAkAighQAEhGkAJCIIAWARAQpACQiSAEgEUEKAIkIUgBIRJACQKKqQWp7me3v277X\n9g7bH21GYQDQLmo5RfQZSWdGxFO2uyTdZvubEfG9ehczPFbWlpFx7Zqc1qreHm0c6i/MqYdXfOtB\nXXnbQ3p636yOWtqhd53xIl189kvyLgvPUUX+rOWhao80Kp7Kvu3Kvup+V4kiX2n9im89qE/d/GPt\nmZlV55LKZeY+dfOPdcW3Hsy7NDwHFfmzlpeaxkhtd9i+R1JZ0k0RcUe9CynyldavvO0hLbHUuWSJ\nlnhJNq20A81W5M9aXmoK0oiYjYhTJK2UdKrtk+bPY3uD7VHboxMTE4supMhXWn9636yW+NC2Ja60\nA81W5M9aXha11z4iHpc0LOmcwzy3NSIGI2KwVCotupAiX2n9qKUdmps3GDIXlXag2Yr8WctLLXvt\nS7ZXZI+7JZ0taazehRT5SuvvOuNFmgtp/9yc5mIum1bagWYr8mctL7XstT9e0pdsd6gSvF+LiG/U\nu5B1A33arMr4ze7Jaa0s0J7EA3vn2WuPVlDkz1peHFH3HfAaHByM0dHRui8XAJrM1WfhzCYASEaQ\nAkAibn7XJJxJAhQXPdIm4EwSoNgI0ibgTBKg2AjSJuBMEqDYCNIm4EwSoNgI0ibgTBKg2AjSJlg3\n0KfN69eqb/kyPbFnRn3Ll2nz+rXstQcKgsOfmmTdQB/BCRQUPVIASESQAkAighQAEhGkAJCIIAWA\nRAQpACTi8CdgAVy1C7WgRwocAVftQq0IUuAIuGoXakWQAkfAVbtQK4IUOAKu2oVaEaTAEXDVLtSK\nIAWOgKt2oVYc/gQsgKt2oRb0SAEgEUEKAImqBqntVbZvsb3T9g7blzSjMABoF7WMke6X9L6IuMv2\nckl32r4pIh5ocG0A0Baq9kgj4pGIuCt7PCVpp6QTGl0YALSLRY2R2l4j6WWS7mhEMQDQjmoOUttH\nS7pO0qUR8eRhnt9ge9T26MTERD1rBICWVlOQ2u5SJUSvjojrDzdPRGyNiMGIGCyVSvWsEQBaWi17\n7S3pc5J2RsQnG18SALSXWnqkp0u6UNKZtu/Jvs5tcF0A0DaqHv4UEbdJchNq4WrkANpSy5zZxNXI\nAbSrlglSrkYOoF21TJByNXIA7aplLqO3qrdHD//6KT25Z7/2zc5paccSPb+7U2uOPTrv0gBgQS3T\nIz2t/xiVp/Zp3+yclljaNzun8tQ+ndZ/TN6lAcCCWiZIbx9/TKWjl2ppxxLNhbS0Y4lKRy/V7eOP\n5V0aACyoZTbtd01O67ijn6fS8mW/aYsIxkgBtLyW6ZFyx0YA7aplgpQ7NgJoVy0TpNyxEUC7apkx\nUok7NgJoTy3TIwWAdkWQAkAighQAEhGkAJCIIAWARAQpACQiSAEgEUEKAIkIUgBIRJACQCKCFAAS\nEaQAkIggBYBEBCkAJCJIASARQQoAiaoGqe3P2y7bvr8ZBQFAu6mlR/pFSec0uA4AaFtVgzQiRiRx\nc3kAOALGSAEgUd1ufmd7g6QNkrR69ep6LbYwhsfK2jIyrl2T01rV26ONQ/3c6A8oiLr1SCNia0QM\nRsRgqVSq12ILYXisrE3bdqg8tVcrurtUntqrTdt2aHisnHdpAOqATfsm2DIyrq4Oq2dpp+zKtKvD\n2jIynndpAOqglsOfrpF0u6QTbe+2fVHjyyqWXZPT6u7qOKStu6tDuyenc6oIQD1VHSONiLc2o5Ai\nW9Xbo/LUXvUsffbXvWdmVit7e3KsCkC9sGnfBBuH+jUzG5ret18RlenMbGjjUH/epQGoA4K0CdYN\n9Gnz+rXqW75MT+yZUd/yZdq8fi177YGCqNvhT1jYuoE+ghMoKHqkAJCIIAWARAQpACQiSAEgEUEK\nAIkIUgBIRJACQCKCFAASEaQAkIggBYBEjoj6L9SekPTThEUcJ+nROpXTali39lTUdSvqekn1WbdH\nI6LqzT8bEqSpbI9GxGDedTQC69aeirpuRV0vqbnrxqY9ACQiSAEgUasG6da8C2gg1q09FXXdirpe\nUhPXrSXHSAGgnbRqjxQA2kZLBantVbZvsb3T9g7bl+RYS4ftu21/I68a2oXtZba/b/ve7O/20Zzq\nWGP7/nltH7F92QKveYftTze+uuawHbY/cdD3l9n+SI4lNZ3tz9suz38vNFJLBamk/ZLeFxEvlfRK\nSe+2/Qc51XKJpJ0pC7DdsFu52O6oPlfTPCPpzIg4WdIpks6x/cqca3quekbS+baPy7uQHH1RUtVj\nP+uppYI0Ih6JiLuyx1OqBNkJza7D9kpJ50m6coF5TrH9Pds/tH2D7d6sfdj2P9i+VZUwPvg1H7H9\nJds32n7Y9vm2/9n2fba32+7K5jsr6w3fl/13fV7W/rDtTbZvk/Rm2y/OXnen7e/YHmjU72QhUfFU\n9m1X9tVSg+/Z3+VjWc/5QduvPsw859m+3fZxtr9o+wrb37U9bvtN2Ty2/XHb92d/n7dk7Z+1vT57\nfIPtz2ePL7L991lveaftf8967Tfa7m7Aqu5XZSfLew+zfr9r+9vZe/bbtldn7Ydd1+y5v7X9g+w1\nuWxpLFZEjEh6rJk/s6WC9GC210h6maQ7cvjxl0t6v6S5Bea5StIHIuKPJN0n6cMHPbciIv4kIj5x\nmNe9WJWQfoOkL0u6JSL+UNIeSefZXqbKf9S3ZO2dkv7moNfvjYgzIuKrqnxg3hMRr5B0maTPLn5V\n6yMbCrlHUlnSTRGRx9+tms6IOFXSpTr07yXbb5T0QUnnRsSBs2GOl3SGpNdL+qes7XxVet0nSzpb\n0sdtHy9pRNKBcD5B0oEtqTMkfSd7/PuSPhMRayU9LunP6rp2z/qMpLfZfsG89k9Luip7z14t6YqD\nnvutdbX92qzmU1VZ51fYHmpQzW2tJYPU9tGSrpN0aUQ82eSf/XpJ5Yi4c4F5XqBKWN6aNX1J0sFv\nsP9c4Ed8MyJmVAnfDknbs/b7JK2RdKKkhyLiwYWWnf2OXiXp61mAbVHlw5CLiJiNiFMkrZR0qu2T\n8iijSvv12fROVX7XB7xG0gcknRcRkwe1/1dEzEXEA5JemLWdIemabH1/JelWSX+sSli+OhuKekDS\nr7KAPU3Sd7PXPhQR9xyhhrrJPjNXSbp43lOnSfpK9vg/snU54HDr+trs625Jd0kaUCVYMU/L3Y45\n27y9TtLVEXF9tfkb4HRJ622fK2mZpOfb/nJEvH0Ry3h6geeekaSImLM9E88efzanyt/DNS57iaTH\ns/BqGRHxuO1hVcaomjbYn/m1pN55bcdIeih7/Ew2ndWh7/1xSf2SXiJp9KD2Zw567HnTQ0TEz7Ph\nnXNU6Z0eI+nPJT0VEVO2j523vFlJjdi0P+ByVcLvCwvMc/A/niOt6z9GxJY611Y4LdUjtW1Jn5O0\nMyI+mUcNEfGhiFgZEWskXSDp5vkhGhFPSJo8aJztQlV6JvUwJmmN7d9baNlZr+Mh22+WfjN2d3Kd\nalgU2yXbK7LH3aps8o41u45snPYR22dltRyjSrDdVuWlP1Vlk/0q22urzDsi6S3ZUEZJla2F72fP\n3a7KsMGIKj3Uy/TsZn1TRcRjkr4m6aKDmr+ryntakt6m6r+X/5H0V9nWj2yfYLuv3rUWQUsFqSq9\nwQslnWn7nuzr3LyLOoK/VGV87IeqjB9trsdCI2KvpHeqssl+nyo91X87wuxvk3SR7Xsl7VBl3DUP\nx0u6Jftd/ECVMdK8Dhv7C0l/lw133CzpoxHxk2oviogfqfL7/LrtFy8w6w2Sfijp3mz574+IX2bP\nfUeVcdgfq9IbPEY5BWnmE6pcAemAiyW9M/s7Xah5O0Pni4gbVRkKuD17L14raXmDaq0b29eo8k/t\nRNu7bV9U7TXJP5MzmwAgTav1SAGg7RCkAJCIIAWARAQpACQiSAEgEUEKAIkIUgBIRJACQKL/A3vA\nfC1GAlKJAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 360x360 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "diabetic = clinical.loc[clinical['Diabetes'] == 'Yes']\n",
    "retired_diabetic = diabetic.loc[diabetic['Age'] > 65]\n",
    "retired_diabetic\n",
    "sns.lmplot(x='Age', y='Tumor_Size_cm', data = retired_diabetic)\n",
    "plt.ylim(0,10)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical_Patient_Key</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>p.R301Gfs*107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.R314C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>RPL22</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.V72M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>CASZ1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.R233Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>PRAMEF9</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.L30M</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Clinical_Patient_Key Patient_Id     Gene           Mutation       Location\n",
       "0                 S001  C3L-00006    MXRA8    Frame_Shift_Del  p.R301Gfs*107\n",
       "1                 S001  C3L-00006     GNB1  Missense_Mutation        p.R314C\n",
       "2                 S001  C3L-00006    RPL22  Missense_Mutation         p.V72M\n",
       "3                 S001  C3L-00006    CASZ1  Missense_Mutation        p.R233Q\n",
       "4                 S001  C3L-00006  PRAMEF9  Missense_Mutation         p.L30M"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "somatic_mutations = en.get_mutations()\n",
    "somatic_mutations.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
