{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib import pylab\n",
    "import seaborn as sns\n",
    "import cptac\n",
    "import collections"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_df = pd.read_csv(r'C:\\Users\\joncj\\OneDrive\\Documents\\Research\\GitHub\\ccrcc.somatic.consensus.gdc.umichigan.wu.112918.maf.3D_Proximity.pairwise.recurrence.l0.r10.clusters', sep = '\\t')\n",
    "colon_clusters = pd.read_csv('DanHeader_MyData.maf.3D_Proximity.pairwise.recurrence.l0.r10.clusters', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All files already downloaded and correct.\n",
      "                                    \r"
     ]
    }
   ],
   "source": [
    "cptac.download(dataset='Colon')\n",
    "colon = cptac.Colon()\n",
    "proteomics = colon.get_proteomics()\n",
    "transcriptomics = colon.get_transcriptomics()\n",
    "clinical = colon.get_clinical()\n",
    "mutations = colon.get_mutations()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_hotspot_1(cluster_df, mut_df):\n",
    "    '''\n",
    "    @Param cluster_df:\n",
    "        The dataframe containing hotspot cluster information for the desired cancer type\n",
    "                - gathered as a csv from the supercomputer hotspot output\n",
    "                - imported into the desired notebook as a dataframe using pd.read_csv()\n",
    "\n",
    "    @Param mut_df:\n",
    "        The dataframe containing the mutations that are present in the corresponding cancer cohort\n",
    "                - gathered from cptac.cancer_type.get_mutations()\n",
    "\n",
    "    @Return:\n",
    "\n",
    "\n",
    "    This function will take two dataframes as parameters to parse the hotspot. It uses the cluster dataframe\n",
    "    in conjuction with the mutations dataframe to determine, for each patient, if they contain mutations within \n",
    "    the identified hotspots. It will return two dataframes and a dictionary that can be used to further explore the \n",
    "    effects of hotspot mutations. It will return a small dataframe containing two columns: 'Hotspot_ID' and \n",
    "    'Patients_Within.' This will allow a quick user visualiztion of which hotspots would be of value to pursue. \n",
    "    The second dataframe is larger and contains a 'Patient_ID' column as well as a column for each identified hotspot. \n",
    "    It is populated with 'No', 'Yes', or 'Yes_HS' values to denote the patient's relationship with every hotspot.\n",
    "    '''\n",
    "    check = 'no'\n",
    "    # Ensuring that the cluster_df has been properly imported\n",
    "    if isinstance(cluster_df, pd.DataFrame):\n",
    "        check = 'yes'\n",
    "    \n",
    "    if check == 'no':\n",
    "        print ('Incorrect parameter! Please pass through a dataframe.')\n",
    "        return None\n",
    "    \n",
    "    check = 'no'\n",
    "    \n",
    "    column_names = cluster_df.columns.tolist()\n",
    "    if 'Cluster' in column_names:\n",
    "        check = 'yes'\n",
    "        \n",
    "    if check == 'no':\n",
    "        print ('Incorrectly formatted dataframe! Dataframe must contain a Cluster column.')\n",
    "        return None\n",
    "    \n",
    "    cluster_list_initial = (cluster_df.Cluster.unique()).tolist()\n",
    "    cluster_list = list()\n",
    "    \n",
    "    for value in cluster_list_initial:\n",
    "        length = len(cluster_df[cluster_df['Cluster'] == value])\n",
    "        if length >= 2:\n",
    "            cluster_list.append(value)\n",
    "        \n",
    "    cluster_list.sort()\n",
    "\n",
    "    if len(cluster_list) == 0:\n",
    "        print('There are no hotspot clusters that contain more than one mutation.')\n",
    "        return None\n",
    "\n",
    "    gene_dict = {}\n",
    "    mut_dict = {}\n",
    "    \n",
    "    for value in cluster_list:\n",
    "        gene_dict[value] = cluster_df.loc[cluster_df['Cluster'] == value, 'Gene/Drug'].values[0]\n",
    "        mut_list = cluster_df[cluster_df['Cluster'] == value]['Mutation/Gene'].values.tolist()\n",
    "        if str(value).endswith('0'):\n",
    "            mut_dict[gene_dict[value]] = mut_list\n",
    "        else:\n",
    "            mut_dict[str(gene_dict[value]) + '_' + str(value)[-1]] = mut_list\n",
    "   \n",
    "    #Creating dataframe 1 of 2 that will be returned at the end of the function\n",
    "    vis_hs_df = pd.DataFrame()\n",
    "    vis_hs_df['Hotspot_ID'] = mut_dict.keys()\n",
    "    \n",
    "    #Creating dataframe 2 of 2 that will be returned at the end of the function\n",
    "    bin_hs_df = pd.DataFrame()\n",
    "    bin_hs_df['Sample_ID'] = mut_df.index.unique()\n",
    "    \n",
    "    for hs in mut_dict.keys():\n",
    "        #Defaulting each patient to not in the hostpot\n",
    "        bin_hs_df[hs] = 'No'\n",
    "        #Creating the list of mutations in the hotspot\n",
    "        hs_locations = mut_dict[hs]\n",
    "        #Creating a list to be populated with Sample IDs of patients in the hotspot\n",
    "        hs_patients = list()\n",
    "        \n",
    "        #Removes the '_#' that was added to the hotspot if multiple hotspots in the same gene'\n",
    "        if hs[-2] == '_':\n",
    "            hs_gene = hs[:-2]\n",
    "        else:\n",
    "            hs_gene = hs\n",
    "            \n",
    "        for row in mut_df.iterrows():\n",
    "            #Creating variables containing the following: gene, mutation location, and patient ID\n",
    "            info = list(row[1])\n",
    "            gene = info[0]\n",
    "            location = info[2]\n",
    "            Sample_ID = row[0]\n",
    "            \n",
    "            if gene == hs_gene:\n",
    "                #Checking if the location of the mutation is within the hotspot\n",
    "                if location in hs_locations:\n",
    "                    #if the location is in the hotspot, the patients gets labeled yes for mutation and _HS for hotspot\n",
    "                    bin_hs_df.loc[bin_hs_df['Sample_ID'] == Sample_ID, hs] = 'Yes_HS'\n",
    "                    hs_patients.append(Sample_ID)\n",
    "                else:\n",
    "                    #The mutation is on a gene that contains a hotspot, but it is not within the hotspot, so the patient only gets labeled yes\n",
    "                    bin_hs_df.loc[bin_hs_df['Sample_ID'] == Sample_ID, hs] = 'Yes'\n",
    "        \n",
    "        hs_count = len(set(hs_patients))\n",
    "        \n",
    "        vis_hs_df.loc[vis_hs_df['Hotspot_ID'] == hs, 'Patients_Within'] = hs_count\n",
    "        \n",
    "    return(vis_hs_df, bin_hs_df, mut_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_hotspot(path, cancer_type):\n",
    "    '''\n",
    "    @Param path:\n",
    "        (String) The path to the cluster output file that is on your computer after running the Hotspot analysis\n",
    "    \n",
    "    @Param cancer_type:\n",
    "        (String) The cancer type that the hotspot analysis was run on. Please use one of the following: 'Ovarian', 'Endometrial', 'Colon', or 'Renal'.\n",
    "        \n",
    "    @Return:\n",
    "        There will be four outputs for this function:\n",
    "        \n",
    "        vis_hs_df: \n",
    "            visualize hotspot dataframe\n",
    "            \n",
    "            A small dataframe which will allow quick visualization regarding the number of cancer patients that contain hotspot mutations\n",
    "        \n",
    "        bin_hs_df: \n",
    "            binary hotspot dataframe\n",
    "            \n",
    "            A larger dataframe that contains boolean values for each patient and their relationship with the hotspot(True = patient has a hotspot mutation, False = patient does not have a hotspot mutation)\n",
    "        \n",
    "        det_hs_df: \n",
    "            detailed hotspot dataframe\n",
    "            \n",
    "            A larger dataframe that contains nonbinary values for each patient and their relationship with the hotspot(No = no mutation, Yes = mutation but not in the hotspot, Yes_HS = mutation in the hotspot)\n",
    "        \n",
    "        mut_dict:\n",
    "            mutations dictionary\n",
    "            \n",
    "            A dictionary that contains the hotspot gene as the key, and a list of mutations that make up that hotspot\n",
    "            \n",
    "    This function will take two parameters (cluster file path and cancer type) and use them to parse the Hotspot3D program output. It creates a cluster dataframe from the Hotspot3D output, and identifies the patients who contain hotspot mutations. The outputs of this function can be used to run further statistical analysis and exploration on the cancer datasets.\n",
    "    '''\n",
    "    #Importing the desired cluster file from the specified path on the computer\n",
    "    cluster_df = pd.read_csv(path, sep = '\\t')\n",
    "    \n",
    "    #Importing the desired cancer type mutation dataframe from cptac\n",
    "    cptac.download(dataset=cancer_type)\n",
    "    \n",
    "    #Checking what cancer_type variable was passed into the function to get the corresponding cancer data. \n",
    "    if cancer_type == 'Colon':\n",
    "        cancer = cptac.Colon()\n",
    "    if cancer_type == 'Renal':\n",
    "        cancer = cptac.RenalCcrcc()\n",
    "    if cancer_type == 'Endometrial':\n",
    "        cancer = cptac.Endometrial()\n",
    "    if cancer_type == 'Ovarian':\n",
    "        cancer = cptac.Ovarian()\n",
    "\n",
    "    #Importing the somatic mutations data for the correct cancer\n",
    "    mut_df = cancer.get_mutations()\n",
    "    \n",
    "    #Creating a list of all the identified hotspot clusters\n",
    "    cluster_list_initial = (cluster_df.Cluster.unique()).tolist()\n",
    "    cluster_list = list()\n",
    "    \n",
    "    #Checking each cluster to make sure that only clusters containing 2 or more mutations are looked at ('clusters' with only 1 mutation are technically just frequently mutated)\n",
    "    for value in cluster_list_initial:\n",
    "        length = len(cluster_df[cluster_df['Cluster'] == value])\n",
    "        if length >= 2:\n",
    "            cluster_list.append(value)\n",
    "    \n",
    "    #Sorting the list numerically\n",
    "    cluster_list.sort()\n",
    "    \n",
    "    #If there are no clusters that have more than one mutation, the function ends and returns the statement below\n",
    "    if len(cluster_list) == 0:\n",
    "        print('There are no hotspot clusters that contain more than one mutation.')\n",
    "        return None\n",
    "    \n",
    "    #creating the multiple dictionaries that are used to compile hotspots and corresponding mutations\n",
    "    gene_dict = {}\n",
    "    mut_dict = {}\n",
    "    rev_mut_dict = {}\n",
    "    hs_count = {}\n",
    "    \n",
    "    #This loop contructs a reverse dictionary to be used to classify patients' mutations as well as the mutation dictionary output\n",
    "    for value in cluster_list:\n",
    "        gene_dict[value] = cluster_df.loc[cluster_df['Cluster'] == value, 'Gene/Drug'].values[0]\n",
    "        mut_list = cluster_df[cluster_df['Cluster'] == value]['Mutation/Gene'].values.tolist()\n",
    "        if str(value).endswith('0'):\n",
    "            mut_dict[gene_dict[value]] = mut_list\n",
    "            hs_count[gene_dict[value]] = 0\n",
    "        else:\n",
    "            mut_dict[str(gene_dict[value]) + '_' + str(value)[-1]] = mut_list\n",
    "            hs_count[str(gene_dict[value]) + '_' + str(value)[-1]] = 0\n",
    "    \n",
    "    #This loop finalizes the reverse dictionary\n",
    "    for hs in mut_dict.keys():\n",
    "        for mutation in mut_dict[hs]:\n",
    "            rev_mut_dict[mutation] = hs\n",
    "    \n",
    "    #The three dataframe outputs are initialized\n",
    "    vis_hs_df = pd.DataFrame()\n",
    "    vis_hs_df['hotspot_id'] = mut_dict.keys()\n",
    "    \n",
    "    bin_hs_df = pd.DataFrame()\n",
    "    bin_hs_df['sample_id'] = mut_df.index.unique()\n",
    "    \n",
    "    det_hs_df = pd.DataFrame()\n",
    "    det_hs_df['sample_id'] = mut_df.index.unique()\n",
    "    \n",
    "    #This loop populates default values for each patient and hotspot\n",
    "    for hs in mut_dict.keys():\n",
    "        bin_hs_df[hs] = False\n",
    "        det_hs_df[hs] = 'No'\n",
    "    \n",
    "    #This loop iterates through each individual mutation and then properly identifies the mutation in the different dataframes\n",
    "    for row in mut_df.iterrows():\n",
    "        info = list(row[1])\n",
    "        gene = info[0]\n",
    "        location = info[2]\n",
    "        location = 'p.'+str(location)\n",
    "        sample_id = row[0]\n",
    "        \n",
    "        #This statement checks to see if the mutation is one of the hotspot mutations\n",
    "        if location in rev_mut_dict.keys():\n",
    "            hs = rev_mut_dict[location]\n",
    "            hs_count[hs] += 1\n",
    "            \n",
    "            bin_hs_df.loc[bin_hs_df['sample_id'] == sample_id, hs] = True\n",
    "            det_hs_df.loc[det_hs_df['sample_id'] == sample_id, hs] = 'Yes_HS'\n",
    "        \n",
    "        #This statement is used if the mutation is not a hotspot mutation, but if it still on one of the proteins that contains a hotspot\n",
    "        elif gene in mut_dict.keys():\n",
    "            det_hs_df.loc[det_hs_df['sample_id'] == sample_id, hs] = 'Yes'\n",
    "    \n",
    "    #This loop adds the patient count for each hotspot to the small visualize hotspot dataframe\n",
    "    for hs in hs_count.keys():\n",
    "        vis_hs_df.loc[vis_hs_df['hotspot_id'] == hs, 'patients_within'] = hs_count[hs]\n",
    "    \n",
    "    #Return of the three dataframes and mutation dictionary\n",
    "    return (vis_hs_df, bin_hs_df, det_hs_df, mut_dict)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All files already downloaded and correct.\n",
      "                                    \r"
     ]
    }
   ],
   "source": [
    "(vis_hs_df, bin_hs_df, det_hs_df, mut_dict) = parse_hotspot_2('DanHeader_MyData.maf.3D_Proximity.pairwise.recurrence.l0.r10.clusters', 'Colon')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_type = 'Colon'\n",
    "path = 'DanHeader_MyData.maf.3D_Proximity.pairwise.recurrence.l0.r10.clusters'\n",
    "#Importing the desired cluster file from the specified path on the computer\n",
    "cluster_df = pd.read_csv(path, sep = '\\t')\n",
    "\n",
    "#Importing the desired cancer type mutation dataframe from the cptac\n",
    "cptac.download(dataset=cancer_type)\n",
    "if cancer_type == 'Colon':\n",
    "    cancer = cptac.Colon()\n",
    "mut_df = cancer.get_mutations()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "cluster_list_initial = (cluster_df.Cluster.unique()).tolist()\n",
    "cluster_list = list()\n",
    "\n",
    "for value in cluster_list_initial:\n",
    "    length = len(cluster_df[cluster_df['Cluster'] == value])\n",
    "    if length >= 2:\n",
    "        cluster_list.append(value)\n",
    "\n",
    "cluster_list.sort()\n",
    "\n",
    "if len(cluster_list) == 0:\n",
    "    print('There are no hotspot clusters that contain more than one mutation.')\n",
    "    #return None\n",
    "\n",
    "#creating the two dictionaries to be used\n",
    "gene_dict = {}\n",
    "mut_dict = {}\n",
    "rev_mut_dict = {}\n",
    "hs_count = {}\n",
    "\n",
    "for value in cluster_list:\n",
    "    gene_dict[value] = cluster_df.loc[cluster_df['Cluster'] == value, 'Gene/Drug'].values[0]\n",
    "    mut_list = cluster_df[cluster_df['Cluster'] == value]['Mutation/Gene'].values.tolist()\n",
    "    if str(value).endswith('0'):\n",
    "        mut_dict[gene_dict[value]] = mut_list\n",
    "        hs_count[gene_dict[value]] = 0\n",
    "    else:\n",
    "        mut_dict[str(gene_dict[value]) + '_' + str(value)[-1]] = mut_list\n",
    "        hs_count[str(gene_dict[value]) + '_' + str(value)[-1]] = 0\n",
    "\n",
    "for hs in mut_dict.keys():\n",
    "    for mutation in mut_dict[hs]:\n",
    "        rev_mut_dict[mutation] = hs\n",
    "\n",
    "vis_hs_df = pd.DataFrame()\n",
    "vis_hs_df['hotspot_id'] = mut_dict.keys()\n",
    "\n",
    "bin_hs_df = pd.DataFrame()\n",
    "bin_hs_df['sample_id'] = mut_df.index.unique()\n",
    "\n",
    "det_hs_df = pd.DataFrame()\n",
    "det_hs_df['sample_id'] = mut_df.index.unique()\n",
    "\n",
    "for hs in mut_dict.keys():\n",
    "    bin_hs_df[hs] = False\n",
    "    det_hs_df[hs] = 'No'\n",
    "\n",
    "for row in mut_df.iterrows():\n",
    "    info = list(row[1])\n",
    "    gene = info[0]\n",
    "    location = info[2]\n",
    "    location = 'p.'+str(location)\n",
    "    sample_id = row[0]\n",
    "\n",
    "    if location in rev_mut_dict.keys():\n",
    "        hs = rev_mut_dict[location]\n",
    "        hs_count[hs] += 1\n",
    "\n",
    "        bin_hs_df.loc[bin_hs_df['sample_id'] == sample_id, hs] = True\n",
    "        det_hs_df.loc[det_hs_df['sample_id'] == sample_id, hs] = 'Yes_HS'\n",
    "\n",
    "    elif gene in mut_dict.keys():\n",
    "        det_hs_df.loc[det_hs_df['sample_id'] == sample_id, hs] = 'Yes'\n",
    "\n",
    "for hs in hs_count.keys():\n",
    "    vis_hs_df.loc[vis_hs_df['hotspot_id'] == hs, 'patients_within'] = hs_count[hs]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hotspot_id</th>\n",
       "      <th>patients_within</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCB6</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACADSB</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ACAP1</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ACBD3</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ACLY</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ACP5</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ACVRL1</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ADAMTS5</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ADCY10</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ADH7</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ADIPOR2</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AGO1</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AGO2</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AGXT</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>B2M</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ALDH2</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ALDH3A1</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>APC</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ANK1</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>APC_1</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>APIP</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>APPL1</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ARHGEF11</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ARHGEF2</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ARSA</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ATXN1</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>BAG4</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>TAF1</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>TEAD2</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>TF</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>TFAM</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>TGFBR2</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288</th>\n",
       "      <td>TH</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>TIGAR</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>TJP3</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>TNKS2</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>TNKS</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>TNPO1</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>TP53</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>TP53_1</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>TP63</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>TPST2</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>TRMT6</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>TRPA1</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>TTK</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>TTN</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>TUBA1A</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>TXNDC17</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>TXNL4A</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>UPP2</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>VAV1</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>VPS33A</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>WAPL</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>WDR5</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>YARS</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311</th>\n",
       "      <td>ZMPSTE24</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>ZMYM5</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>313 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    hotspot_id  patients_within\n",
       "0          A2M              6.0\n",
       "1        ABCB6              6.0\n",
       "2        ACAA1             11.0\n",
       "3        ACACB              2.0\n",
       "4       ACADSB              5.0\n",
       "5        ACAP1              4.0\n",
       "6        ACBD3              3.0\n",
       "7         ACLY              6.0\n",
       "8         ACP5             20.0\n",
       "9       ACVRL1             19.0\n",
       "10     ADAMTS5             13.0\n",
       "11      ADCY10              2.0\n",
       "12        ADH7              3.0\n",
       "13     ADIPOR2             11.0\n",
       "14        AGO1             16.0\n",
       "15        AGO2              6.0\n",
       "16        AGXT              4.0\n",
       "17         B2M              5.0\n",
       "18       ALDH2              3.0\n",
       "19     ALDH3A1              9.0\n",
       "20         APC             18.0\n",
       "21        ANK1              2.0\n",
       "22       APC_1              4.0\n",
       "23        APIP              3.0\n",
       "24       APPL1              4.0\n",
       "25    ARHGEF11             11.0\n",
       "26     ARHGEF2              5.0\n",
       "27        ARSA             17.0\n",
       "28       ATXN1              4.0\n",
       "29        BAG4              5.0\n",
       "..         ...              ...\n",
       "283       TAF1              4.0\n",
       "284      TEAD2             16.0\n",
       "285         TF              3.0\n",
       "286       TFAM             13.0\n",
       "287     TGFBR2             20.0\n",
       "288         TH             14.0\n",
       "289      TIGAR             15.0\n",
       "290       TJP3              6.0\n",
       "291      TNKS2             11.0\n",
       "292       TNKS              2.0\n",
       "293      TNPO1              3.0\n",
       "294       TP53              9.0\n",
       "295     TP53_1             13.0\n",
       "296       TP63             17.0\n",
       "297      TPST2              4.0\n",
       "298      TRMT6              5.0\n",
       "299      TRPA1              7.0\n",
       "300        TTK              4.0\n",
       "301        TTN              2.0\n",
       "302     TUBA1A             16.0\n",
       "303    TXNDC17              4.0\n",
       "304     TXNL4A              2.0\n",
       "305       UPP2              3.0\n",
       "306       VAV1             22.0\n",
       "307     VPS33A              5.0\n",
       "308       WAPL              3.0\n",
       "309       WDR5              4.0\n",
       "310       YARS              2.0\n",
       "311   ZMPSTE24              4.0\n",
       "312      ZMYM5             13.0\n",
       "\n",
       "[313 rows x 2 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vis_hs_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hotspot_id</th>\n",
       "      <th>patients_within</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACAA1</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ACP5</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ACVRL1</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ADAMTS5</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ADIPOR2</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AGO1</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>APC</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ARHGEF11</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ARSA</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>BCORL1</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>C11orf54</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CADM1</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>CDC23</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>HIST1H3E_1</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>CKMT1A</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>CKMT2</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>CNOT2</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>CRK</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>CRYGS</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>CWC27</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>DCLK1</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>DHPS</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>DNPH1</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>DPAGT1</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>ECE1</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>EGLN1</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>ENO3</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>ERBB4</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>MME</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>MMP9</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>NAPEPLD</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>NFKB2</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>NLGN4X</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>NLRP12</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>PAX5</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>PCDH7</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>PLA2G4A</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>PPP3CA</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>PREP</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>PROZ</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>RBP1</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>RNF144A</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>RRAD</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>RUVBL1</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>SERPINB2</td>\n",
       "      <td>18.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>STARD13</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>STYX</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>TEAD2</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>TFAM</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>TGFBR2</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288</th>\n",
       "      <td>TH</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>TIGAR</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>TNKS2</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>TP53_1</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>TP63</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>TUBA1A</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>VAV1</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>ZMYM5</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>86 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     hotspot_id  patients_within\n",
       "2         ACAA1             11.0\n",
       "8          ACP5             20.0\n",
       "9        ACVRL1             19.0\n",
       "10      ADAMTS5             13.0\n",
       "13      ADIPOR2             11.0\n",
       "14         AGO1             16.0\n",
       "20          APC             18.0\n",
       "25     ARHGEF11             11.0\n",
       "27         ARSA             17.0\n",
       "31       BCORL1             12.0\n",
       "36     C11orf54             20.0\n",
       "45        CADM1             13.0\n",
       "49        CDC23             14.0\n",
       "51       CDKN1A             15.0\n",
       "54   HIST1H3E_1             28.0\n",
       "57       CKMT1A             11.0\n",
       "59        CKMT2             11.0\n",
       "61        CNOT2             16.0\n",
       "67          CRK             16.0\n",
       "68        CRYGS             12.0\n",
       "72        CWC27             27.0\n",
       "73       CYP2A6             11.0\n",
       "75        DCLK1             15.0\n",
       "77         DHPS             11.0\n",
       "82        DNPH1             12.0\n",
       "83       DPAGT1             11.0\n",
       "87         ECE1             11.0\n",
       "88        EGLN1             27.0\n",
       "90         ENO3             17.0\n",
       "93        ERBB4             12.0\n",
       "..          ...              ...\n",
       "174         MME             12.0\n",
       "179        MMP9             24.0\n",
       "184     NAPEPLD             11.0\n",
       "186       NFKB2             18.0\n",
       "188      NLGN4X             16.0\n",
       "189      NLRP12             12.0\n",
       "209        PAX5             26.0\n",
       "210       PCDH7             16.0\n",
       "219     PLA2G4A             20.0\n",
       "230      PPP3CA             11.0\n",
       "232        PREP             14.0\n",
       "235        PROZ             36.0\n",
       "251        RBP1             23.0\n",
       "254     RNF144A             24.0\n",
       "259        RRAD             15.0\n",
       "260      RUVBL1             34.0\n",
       "265    SERPINB2             18.0\n",
       "275     STARD13             22.0\n",
       "278        STYX             11.0\n",
       "284       TEAD2             16.0\n",
       "286        TFAM             13.0\n",
       "287      TGFBR2             20.0\n",
       "288          TH             14.0\n",
       "289       TIGAR             15.0\n",
       "291       TNKS2             11.0\n",
       "295      TP53_1             13.0\n",
       "296        TP63             17.0\n",
       "302      TUBA1A             16.0\n",
       "306        VAV1             22.0\n",
       "312       ZMYM5             13.0\n",
       "\n",
       "[86 rows x 2 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vis_hs_df.loc[vis_hs_df['patients_within'] > 10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
