{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading Endometrial CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Acetylation Proteomics Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import CPTAC.Endometrial as Endometrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import division\n",
    "import pandas as pd, numpy as np, matplotlib.pyplot as plt, seaborn as sns\n",
    "import matplotlib.patches as mpatches\n",
    "import scipy.stats\n",
    "import sys\n",
    "import argparse\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fileToList(group_list):\n",
    "    with open(group_list, 'r') as fh:\n",
    "        return [line.strip() for line in fh.readlines()]\n",
    "\n",
    "def fileToDict(tsv_map_file_name):\n",
    "    with open(tsv_map_file_name, 'r') as fh:\n",
    "        return {line.split()[0]:line.split()[1] for line in fh.readlines()}\n",
    "\n",
    "def correct_pvalues_for_multiple_testing(pvalues, correction_type = \"Benjamini-Hochberg\"):\n",
    "\n",
    "    from numpy import array, empty\n",
    "    pvalues = array(pvalues)\n",
    "    n = sum(~np.isnan(pvalues))\n",
    "    new_pvalues = empty(len(pvalues))\n",
    "\n",
    "    if correction_type == \"Bonferroni\":\n",
    "        new_pvalues = n * pvalues\n",
    "\n",
    "    elif correction_type == \"Bonferroni-Holm\":\n",
    "        values = [ (pvalue, i) for i, pvalue in enumerate(pvalues) ]\n",
    "        values = [x for x in values if ~np.isnan(x[0])]\n",
    "        values.sort()\n",
    "        for rank, vals in enumerate(values):\n",
    "            pvalue, i = vals\n",
    "            new_pvalues[i] = (n-rank) * pvalue\n",
    "    elif correction_type == \"Benjamini-Hochberg\":\n",
    "\n",
    "        values = [ (pvalue, i) for i, pvalue in enumerate(pvalues) ]\n",
    "        values = [x for x in values if ~np.isnan(x[0])]\n",
    "        values.sort()\n",
    "        values.reverse()\n",
    "        new_values = []\n",
    "        for i, vals in enumerate(values):\n",
    "            rank = n - i\n",
    "            pvalue, index = vals\n",
    "            new_values.append((n/rank) * pvalue)\n",
    "        for i in range(0, int(n)-1):\n",
    "            if new_values[i] < new_values[i+1]:\n",
    "                new_values[i+1] = new_values[i]\n",
    "        for i, vals in enumerate(values):\n",
    "            pvalue, index = vals\n",
    "            new_pvalues[index] = new_values[i]\n",
    "\n",
    "    new_pvalues[np.isnan(pvalues)] = np.nan\n",
    "    for i, val in enumerate(new_pvalues):\n",
    "        if ~np.isnan(val):\n",
    "            new_pvalues[i] = min(1, val)\n",
    "\n",
    "    return new_pvalues\n",
    "\n",
    "def assignColors(group_colors, group1_label, group2_label, group1, group2):\n",
    "    if group_colors is not None:\n",
    "        group_color_map = fileToDict(group_colors)\n",
    "\n",
    "        groups_dict = {sample:group1_label for sample in group1}\n",
    "        groups_dict2 = {sample:group2_label for sample in group2}\n",
    "        groups_dict.update(groups_dict2)\n",
    "\n",
    "        sample_color_map = {sample:group_color_map[groups_dict[sample]] for sample in group1+group2}\n",
    "\n",
    "    else:\n",
    "        sample_color_map = {sample:'#571D41' for sample in group1}\n",
    "        sample_color_map.update({sample:'#F5F5F5' for sample in group2})\n",
    "\n",
    "        group_color_map = {group1_label:'#571D41', group2_label:'#F5F5F5'}\n",
    "    return group_color_map, sample_color_map\n",
    "\n",
    "def filterOutliers(df, group1_list, group2_list, gene_column_name, frac_filter):\n",
    "\n",
    "    group1_outliers = [x+'_outliers' for x in group1_list]\n",
    "    group1_notOutliers = [x+'_notOutliers' for x in group1_list]\n",
    "    group2_outliers = [x+'_outliers' for x in group2_list]\n",
    "    group2_notOutliers = [x+'_notOutliers' for x in group2_list]\n",
    "\n",
    "    if frac_filter!=None:\n",
    "        min_num_outlier_samps = len(group1_list)*frac_filter\n",
    "        # Filter for 30% of samples having outliers in group1\n",
    "        num_outlier_samps = (df[group1_outliers] > 0).sum(axis=1)\n",
    "        df = df.loc[num_outlier_samps >= min_num_outlier_samps, :].reset_index(drop=True)\n",
    "\n",
    "    # Filter for higher proportion of outliers in group1 than group2\n",
    "    group1_outlier_rate = df[group1_outliers].sum(axis=1).divide(df[group1_outliers+group1_notOutliers].sum(axis=1), axis=0)\n",
    "    group2_outlier_rate = df[group2_outliers].sum(axis=1).divide(df[group1_outliers+group2_notOutliers].sum(axis=1), axis=0)\n",
    "\n",
    "    df = df.loc[group1_outlier_rate>group2_outlier_rate, :].reset_index(drop=True)\n",
    "\n",
    "    with np.errstate(divide='ignore',invalid='ignore'):\n",
    "        frac_outliers = pd.concat([pd.DataFrame(df[group1_outliers].values/(df[group1_outliers].values+df[group1_notOutliers].values),\n",
    "                                                columns=group1_list),\n",
    "                                   pd.DataFrame(df[group2_outliers].values/(df[group2_outliers].values+df[group2_notOutliers].values),\n",
    "                                                columns=group2_list)],\n",
    "                                   axis=1).reset_index(drop=True)\n",
    "\n",
    "    frac_outliers[gene_column_name] = df[gene_column_name]\n",
    "\n",
    "    return df, frac_outliers\n",
    "\n",
    "def testDifferentGroupsOutliers(group1_list, group2_list, outlier_table):\n",
    "\n",
    "    outliers_group1_list = [x+'_outliers' for x in group1_list]\n",
    "    notOutliers_group1_list = [x+'_notOutliers' for x in group1_list]\n",
    "    outliers_group2_list = [x+'_outliers' for x in group2_list]\n",
    "    notOutliers_group2_list = [x+'_notOutliers' for x in group2_list]\n",
    "\n",
    "    outlier_table['Outlier1'] = outlier_table[outliers_group1_list].sum(axis=1)\n",
    "    outlier_table['NotOutlier1'] = outlier_table[notOutliers_group1_list].sum(axis=1)\n",
    "\n",
    "    outlier_table['Outlier2'] = outlier_table[outliers_group2_list].sum(axis=1)\n",
    "    outlier_table['NotOutlier2'] = outlier_table[notOutliers_group2_list].sum(axis=1)\n",
    "\n",
    "    outlier_table['fisherp'] = outlier_table.apply((lambda r: scipy.stats.fisher_exact([[r['Outlier1'],\n",
    "                                                                                     r['Outlier2']],\n",
    "                                                                                    [r['NotOutlier1'],\n",
    "                                                                                     r['NotOutlier2']]])[1]),axis=1)\n",
    "\n",
    "    outlier_table['fisherFDR'] = correct_pvalues_for_multiple_testing(list(outlier_table['fisherp']),\n",
    "                                     correction_type = \"Benjamini-Hochberg\")\n",
    "    return outlier_table['fisherFDR']\n",
    "\n",
    "def makeHeatMap(heatmap_table, group_color_map,sample_color_map, group1,\n",
    "                output_prefix, blue_or_red, genes_to_highlight=None):\n",
    "\n",
    "    # Set up colors for heatmap\n",
    "    sns.set(font = 'arial', style = 'white', color_codes=True, font_scale = 1)\n",
    "    if blue_or_red == 'red':\n",
    "        cmap = sns.cubehelix_palette(start=0.857, rot=0.00, gamma=1.5, hue=1,\n",
    "                                    light=1, dark=0.2, reverse=False, as_cmap=True)\n",
    "    elif blue_or_red=='blue':\n",
    "        cmap=sns.cubehelix_palette(start=3, rot=0.00, gamma=1.5, hue=1,\n",
    "                                    light=1, dark=0.2, reverse=False, as_cmap=True)\n",
    "    cmap.set_bad('#F5F5F5')\n",
    "\n",
    "    # Set up colors for columns\n",
    "    heatmap_table.columns = [x.split('_')[0] for x in heatmap_table.columns]\n",
    "    group = heatmap_table.columns\n",
    "    column_colors = group.map(sample_color_map)\n",
    "\n",
    "    # Sort the table\n",
    "    heatmap_table['mean_group1'] = heatmap_table.loc[:, group1].mean(axis=1)\n",
    "    heatmap_table = heatmap_table.sort_values('mean_group1')\n",
    "    heatmap_table = heatmap_table.drop('mean_group1', axis=1)\n",
    "\n",
    "    # Make heatmap\n",
    "    g = sns.clustermap(heatmap_table,\n",
    "                       cmap=cmap,\n",
    "                       col_cluster = False,\n",
    "                       row_cluster = False,\n",
    "                       col_colors = column_colors,\n",
    "                       xticklabels=False,\n",
    "                       vmin=0,\n",
    "                       vmax=1,\n",
    "                       cbar_kws={'label':'fraction outliers'},\n",
    "                       # figsize=(0.1*len(list(heatmap_table)), 0.05*len(heatmap_table))\n",
    "                          )\n",
    "    g.ax_row_dendrogram.set_visible(False)\n",
    "    plt.setp(g.ax_heatmap.yaxis.get_majorticklabels(), rotation=0)\n",
    "\n",
    "    g.cax.set_position((0.15,0.12,0.03,0.6)) #move colorbar to right\n",
    "    ax = g.ax_heatmap\n",
    "    ax.set_ylabel('') #change the gene label\n",
    "\n",
    "    # Make the legend the describes the different sample groups\n",
    "    handles = [mpatches.Patch(color=color, label=group) for group, color in group_color_map.items()]\n",
    "\n",
    "    fig = plt.gcf()\n",
    "    fig.legend(handles=handles, bbox_to_anchor=(0.6, 0.10))\n",
    "\n",
    "    # Star genes to highlight on heatmap\n",
    "    if genes_to_highlight != None:\n",
    "        new_labels = []\n",
    "        for gene in ax.get_yticklabels():\n",
    "            gene = gene.get_text()\n",
    "            if gene in genes_to_highlight:\n",
    "                new_labels.append(str(gene)+'*')\n",
    "            else:\n",
    "                new_labels.append(gene)\n",
    "        ax.set_yticklabels(new_labels)\n",
    "\n",
    "    # Save and show the plot\n",
    "    plt.savefig('%s.pdf' %output_prefix, dpi=500, bbox_inches='tight', pad_inches=0.5)\n",
    "    return g\n",
    "\n",
    "def runComparison(outliers, group1, group2,gene_column_name,\n",
    "                  fdr_cut_off=0.05, frac_filter=0.3):\n",
    "# Filter for multiple samples with outliers in group1_list\n",
    "    outliers, frac_outliers = filterOutliers(outliers, group1, group2, gene_column_name, frac_filter)\n",
    "    if len(outliers) == 0:\n",
    "        print(\"No rows have outliers in %s of group1 samples\" % (frac_filter))\n",
    "        sys.exit()\n",
    "    print('Testing %s genes for enrichment in group1' %(len(outliers)))\n",
    "\n",
    "# Doing statistical test on different groups\n",
    "    outliers['FDR'] = testDifferentGroupsOutliers(group1, group2, outliers)\n",
    "\n",
    "    outliers['significant'] = (outliers['FDR'] < fdr_cut_off)\n",
    "    sig_diff_count = sum(outliers['significant'])\n",
    "    print('%s signficantly differential genes at FDR %s' % (sig_diff_count, fdr_cut_off))\n",
    "    return outliers[[gene_column_name, 'FDR']], frac_outliers, sig_diff_count\n",
    "\n",
    "def writeSigGenesToFile(outliers, gene_column_name, output_prefix, group1_label):\n",
    "    sig_genes = outliers.loc[(outliers['significant']==True), gene_column_name]\n",
    "    with open('%s.outlier_sites_in_%s.txt' %(output_prefix, group1_label), 'w') as fh:\n",
    "        for gene in sig_genes:\n",
    "            fh.write('%s\\n'%gene)\n",
    "\n",
    "if __name__==\"__main__\":\n",
    "    parser = argparse.ArgumentParser(description=\"Input samples in two groups and df\")\n",
    "    parser.add_argument('--outliers_table', type=str,\n",
    "                        help='tsv with proteins as rows and samples as columns. Phospho data also needs a \"counts\" column')\n",
    "    parser.add_argument('--gene_column_name', type=str, default='geneSymbol',\n",
    "                        help='Specify column name for protein IDs')\n",
    "    parser.add_argument('--fdr_cut_off', type=float, default=0.05,\n",
    "                        help='Set FDR cut off for which genes are considered sig diff')\n",
    "    parser.add_argument('--output_prefix', type=str, default=str(datetime.now().date()),\n",
    "                        help='Default is current date. Set for prefix for gene lists and heatmap')\n",
    "    parser.add_argument('--group1_label', type=str, default='group1',\n",
    "                        help='Label for group1 for heatmap')\n",
    "    parser.add_argument('--group1_list', type=str,\n",
    "                        help='List of samples from group 1. Samples separated by new lines, no header')\n",
    "    parser.add_argument('--group2_label', type=str, default='group2',\n",
    "                        help='Label for group2 for heatmap')\n",
    "    parser.add_argument('--group2_list', type=str,\n",
    "                        help='List of samples from group 2. Samples separated by new lines, no header')\n",
    "    parser.add_argument('--group_colors', type=str,\n",
    "                        help='tsv, no header, with group names in 1st column and colors in 2nd column. Alternatively can map colors directly to samples. Group labels must match group labels given above.')\n",
    "    parser.add_argument('--genes_to_highlight', type=str, default=None,\n",
    "                        help='List of genes to highlight with * in y tick labels')\n",
    "    parser.add_argument('--blue_or_red', type=str, default='red',\n",
    "                        help='Color scale for heatmap')\n",
    "    parser.add_argument('--output_qvals', type=str, choices=['True', 'False'], default=False,\n",
    "                        help='Save q-values to a table?')\n",
    "    parser.add_argument('--frac_filter', type=str, default=0.3,\n",
    "                        help='None or fraction (bn 0-1) of outlier samples in group1 needed per gene')\n",
    "\n",
    "    args = parser.parse_args()\n",
    "\n",
    "    outliers = pd.read_csv(args.outliers_table, sep='\\t')\n",
    "    gene_column_name = args.gene_column_name\n",
    "    fdr_cut_off = args.fdr_cut_off\n",
    "    output_prefix = args.output_prefix\n",
    "    group1_label = args.group1_label\n",
    "    group2_label = args.group2_label\n",
    "    group1 = [x for x in fileToList(args.group1_list) if x+'_outliers' in outliers.columns]\n",
    "    group2 = [x for x in fileToList(args.group2_list) if x+'_outliers' in outliers.columns]\n",
    "    genes_to_highlight = args.genes_to_highlight\n",
    "    if genes_to_highlight != None:\n",
    "        genes_to_highlight = fileToList(genes_to_highlight)\n",
    "    blue_or_red = args.blue_or_red\n",
    "    output_qvals = args.output_qvals == 'True'\n",
    "    frac_filter = args.frac_filter\n",
    "\n",
    "    if frac_filter=='None':\n",
    "        frac_filter = None\n",
    "    else:\n",
    "        try:\n",
    "            frac_filter = float(frac_filter)\n",
    "            if frac_filter > 1:\n",
    "                print('Non-valid frac_filter value')\n",
    "                sys.exit()\n",
    "            if frac_filter < 0:\n",
    "                print('Non-valid frac_filter value')\n",
    "                sys.exit()\n",
    "        except:\n",
    "            print('Non-valid frac_filter value')\n",
    "            sys.exit()\n",
    "\n",
    "\n",
    "# Assigning colors to samples\n",
    "    group_color_map, sample_color_map = assignColors(args.group_colors, group1_label, group2_label, group1, group2)\n",
    "\n",
    "# Filter for multiple samples with outliers in group1_list\n",
    "    outliers, frac_outliers, sig_diff_count = runComparison(outliers, group1, group2, gene_column_name, fdr_cut_off, frac_filter)\n",
    "\n",
    "    if output_qvals == True:\n",
    "        outliers.to_csv('%s_comparison_qvals.txt'%output_prefix, sep='\\t', index=False)\n",
    "\n",
    "#If enough genes, make heatmap\n",
    "    if sig_diff_count >= 1:\n",
    "        heatmap_table = frac_outliers.loc[(outliers['significant'] == True), [gene_column_name] + group1 + group2]\n",
    "        heatmap_table = heatmap_table.set_index(gene_column_name)\n",
    "\n",
    "        makeHeatMap(heatmap_table, group_color_map, sample_color_map, group1,\n",
    "                    output_prefix, blue_or_red, genes_to_highlight)\n",
    "\n",
    "#Write significantly different genes to a file\n",
    "        writeSigGenesToFile(outliers, gene_column_name, output_prefix, group1_label)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
