{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to the CPTAC data service package. Available datasets may be\n",
      "viewed using CPTAC.list_data(). In order to access a specific data\n",
      "set, import a CPTAC subfolder using either 'import CPTAC.Dataset' or\n",
      "'from CPTAC import Dataset'.\n",
      "******\n",
      "Version: 0.3.1\n",
      "******\n",
      "Loading Endometrial CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Acetylation Proteomics Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "import re\n",
    "\n",
    "import CPTAC.Endometrial as en\n",
    "import CPTAC.Algorithms as al"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ovarian\n",
      "['PTEN', 'PIK3CA', 'ARID1A', 'PIK3R1', 'KRAS', 'CTNNB1', 'CTCF', 'KMT2B', 'TP53', 'ZFHX3', 'ZFHX4', 'DOCK3', 'INPPL1', 'FBXW7', 'MUC16', 'KMT2D', 'HERC2', 'RPL22', 'RBM27', 'NSD1', 'SYNE1', 'SCAF4', 'PCLO', 'JAK1', 'CCDC168', 'ZNF469', 'LMAN1', 'OBSCN', 'AHNAK', 'HUWE1', 'DNAH17']\n"
     ]
    }
   ],
   "source": [
    "somatic = en.get_mutations()\n",
    "clinical = en.get_clinical()\n",
    "prot = en.get_proteomics()\n",
    "gene = 'PTEN'\n",
    "omics_mutations = en.append_mutations_to_omics(mutation_genes=gene, omics_df=prot, omics_genes=gene)\n",
    "omics_mutations = omics_mutations.loc[omics_mutations['Sample_Status'] == 'Tumor']\n",
    "\n",
    "fm = al.get_frequently_mutated(somatic, omics_mutations, .15, show_percentage=False)\n",
    "print(fm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PTEN\n",
      "PIK3CA\n",
      "ARID1A\n",
      "PIK3R1\n",
      "KRAS\n",
      "CTNNB1\n",
      "CTCF\n",
      "KMT2B\n",
      "TP53\n",
      "ZFHX3\n",
      "ZFHX4\n",
      "DOCK3\n",
      "INPPL1\n",
      "FBXW7\n",
      "MUC16\n",
      "KMT2D\n",
      "HERC2\n",
      "RPL22\n",
      "RBM27\n",
      "NSD1\n",
      "SYNE1\n",
      "SCAF4\n",
      "PCLO\n",
      "JAK1\n",
      "CCDC168\n",
      "ZNF469\n",
      "LMAN1\n",
      "OBSCN\n",
      "AHNAK\n",
      "HUWE1\n",
      "DNAH17\n"
     ]
    }
   ],
   "source": [
    "for entry in fm:\n",
    "    print(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  PTEN\n",
      "           Comparison   P_Value\n",
      "0   PIK3CA_proteomics  0.000017\n",
      "1  CSNK2A1_proteomics  0.000102\n",
      "2     TP53_proteomics  0.000935\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  PIK3CA\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  ARID1A\n",
      "           Comparison       P_Value\n",
      "0   ARID1A_proteomics  1.025018e-10\n",
      "1     DPF2_proteomics  2.166609e-07\n",
      "2  SMARCB1_proteomics  6.885135e-05\n",
      "3  SMARCC2_proteomics  8.270473e-05\n",
      "4  SMARCE1_proteomics  1.498027e-04\n",
      "5  SMARCD1_proteomics  1.744915e-04\n",
      "6    BCL7C_proteomics  4.379537e-04\n",
      "7    CCND1_proteomics  1.499832e-03\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  PIK3R1\n",
      "Error with:  PIK3R1 !\n",
      "\n",
      "Gene:  KRAS\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  CTNNB1\n",
      "            Comparison       P_Value\n",
      "0    CTNNB1_proteomics  1.374820e-11\n",
      "1    CTNNA2_proteomics  3.221686e-09\n",
      "2      LEF1_proteomics  1.083350e-07\n",
      "3  CTNNBIP1_proteomics  2.340440e-06\n",
      "4      CDH2_proteomics  1.718146e-04\n",
      "5     AXIN1_proteomics  1.402113e-03\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  CTCF\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  KMT2B\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  TP53\n",
      "          Comparison       P_Value\n",
      "0    TP53_proteomics  3.991072e-07\n",
      "1   AURKA_proteomics  1.583437e-05\n",
      "2  CDKN2A_proteomics  5.088182e-04\n",
      "3     BLM_proteomics  1.324065e-03\n",
      "4    CDK2_proteomics  1.489990e-03\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  ZFHX3\n",
      "         Comparison   P_Value\n",
      "0  ZFHX3_proteomics  0.001988\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  ZFHX4\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  DOCK3\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  INPPL1\n",
      "          Comparison       P_Value\n",
      "0  INPPL1_proteomics  9.820989e-08\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  FBXW7\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  MUC16\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  KMT2D\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  HERC2\n",
      "          Comparison   P_Value\n",
      "0  TRIM36_proteomics  0.000339\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  RPL22\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  RBM27\n",
      "         Comparison       P_Value\n",
      "0  RBM27_proteomics  2.698889e-10\n",
      "1  ZC3H4_proteomics  3.875150e-04\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  NSD1\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  SYNE1\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  SCAF4\n",
      "         Comparison   P_Value\n",
      "0  SCAF4_proteomics  0.000097\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  PCLO\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  JAK1\n",
      "        Comparison   P_Value\n",
      "0  JAK1_proteomics  0.000017\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  CCDC168\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  ZNF469\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  LMAN1\n",
      "          Comparison  P_Value\n",
      "0  SEC24A_proteomics   0.0017\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  OBSCN\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  AHNAK\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  HUWE1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda3/lib/python3.6/site-packages/numpy/core/fromnumeric.py:3157: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "/anaconda3/lib/python3.6/site-packages/numpy/core/_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No significant comparisons.\n",
      "\n",
      "Gene:  DNAH17\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "proteomics = en.get_proteomics()\n",
    "for protein in fm:\n",
    "    try:\n",
    "        print(\"\\nGene: \", protein)\n",
    "        interacting_proteins = al.get_interacting_proteins(protein)\n",
    "        ips = []\n",
    "        for ip in interacting_proteins:\n",
    "            if ip in proteomics.columns:\n",
    "                ips.append(ip)\n",
    "\n",
    "        protdf = en.append_mutations_to_omics(mutation_genes=[protein], omics_df=proteomics, omics_genes=ips)\n",
    "        protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "        #print(protdf)\n",
    "\n",
    "        for ind, row in protdf.iterrows():\n",
    "            if 'Wildtype_Tumor' not in row[protein+\"_Mutation\"]:\n",
    "                protdf.at[ind,'Label'] = 'Mutated'\n",
    "            else:\n",
    "                protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "        #print(protdf)\n",
    "        protdf = protdf.drop(protein+\"_Mutation\",axis=1)\n",
    "        protdf = protdf.drop(protein+\"_Location\",axis=1)\n",
    "        protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "        columns_to_comp = []\n",
    "        for ip in ips:\n",
    "            columns_to_comp.append(ip+\"_proteomics\")\n",
    "\n",
    "        wrap_results = al.wrap_ttest(protdf, 'Label', columns_to_comp)\n",
    "        \n",
    "        if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "        \n",
    "    except:\n",
    "        print(\"Error with: \", protein, \"!\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
