{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to the CPTAC data service package. Available datasets may be\n",
      "viewed using CPTAC.list_data(). In order to access a specific data\n",
      "set, import a CPTAC subfolder using either 'import CPTAC.Dataset' or\n",
      "'from CPTAC import Dataset'.\n",
      "******\n",
      "Version: 0.3.1\n",
      "******\n",
      "Loading Endometrial CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Acetylation Proteomics Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "import re\n",
    "\n",
    "import CPTAC.Endometrial as en\n",
    "import CPTAC.Algorithms as al"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ovarian\n",
      "['PTEN', 'PIK3CA', 'ARID1A', 'PIK3R1', 'KRAS', 'CTNNB1', 'CTCF', 'KMT2B', 'TP53', 'ZFHX3', 'ZFHX4', 'DOCK3', 'INPPL1', 'FBXW7', 'MUC16', 'KMT2D', 'HERC2', 'RPL22', 'RBM27', 'NSD1', 'SYNE1', 'SCAF4', 'PCLO', 'JAK1', 'CCDC168', 'ZNF469', 'LMAN1', 'OBSCN', 'AHNAK', 'HUWE1', 'DNAH17']\n"
     ]
    }
   ],
   "source": [
    "somatic = en.get_mutations()\n",
    "clinical = en.get_clinical()\n",
    "prot = en.get_proteomics()\n",
    "gene = 'PTEN'\n",
    "omics_mutations = en.append_mutations_to_omics(mutation_genes=gene, omics_df=prot, omics_genes=gene)\n",
    "omics_mutations = omics_mutations.loc[omics_mutations['Sample_Status'] == 'Tumor']\n",
    "\n",
    "fm = al.get_frequently_mutated(somatic, omics_mutations, .15, show_percentage=False)\n",
    "print(fm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PTEN\n",
      "PIK3CA\n",
      "ARID1A\n",
      "PIK3R1\n",
      "KRAS\n",
      "CTNNB1\n",
      "CTCF\n",
      "KMT2B\n",
      "TP53\n",
      "ZFHX3\n",
      "ZFHX4\n",
      "DOCK3\n",
      "INPPL1\n",
      "FBXW7\n",
      "MUC16\n",
      "KMT2D\n",
      "HERC2\n",
      "RPL22\n",
      "RBM27\n",
      "NSD1\n",
      "SYNE1\n",
      "SCAF4\n",
      "PCLO\n",
      "JAK1\n",
      "CCDC168\n",
      "ZNF469\n",
      "LMAN1\n",
      "OBSCN\n",
      "AHNAK\n",
      "HUWE1\n",
      "DNAH17\n"
     ]
    }
   ],
   "source": [
    "for entry in fm:\n",
    "    print(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  PTEN\n",
      "           Comparison   P_Value\n",
      "0   PIK3CA_proteomics  0.000017\n",
      "1  CSNK2A1_proteomics  0.000102\n",
      "2     TP53_proteomics  0.000935\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  PIK3CA\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  ARID1A\n",
      "           Comparison       P_Value\n",
      "0   ARID1A_proteomics  1.025018e-10\n",
      "1     DPF2_proteomics  2.166609e-07\n",
      "2  SMARCB1_proteomics  6.885135e-05\n",
      "3  SMARCC2_proteomics  8.270473e-05\n",
      "4  SMARCE1_proteomics  1.498027e-04\n",
      "5  SMARCD1_proteomics  1.744915e-04\n",
      "6    BCL7C_proteomics  4.379537e-04\n",
      "7    CCND1_proteomics  1.499832e-03\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  PIK3R1\n",
      "Error with:  PIK3R1 !\n",
      "\n",
      "Gene:  KRAS\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  CTNNB1\n",
      "            Comparison       P_Value\n",
      "0    CTNNB1_proteomics  1.374820e-11\n",
      "1    CTNNA2_proteomics  3.221686e-09\n",
      "2      LEF1_proteomics  1.083350e-07\n",
      "3  CTNNBIP1_proteomics  2.340440e-06\n",
      "4      CDH2_proteomics  1.718146e-04\n",
      "5     AXIN1_proteomics  1.402113e-03\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  CTCF\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  KMT2B\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  TP53\n",
      "          Comparison       P_Value\n",
      "0    TP53_proteomics  3.991072e-07\n",
      "1   AURKA_proteomics  1.583437e-05\n",
      "2  CDKN2A_proteomics  5.088182e-04\n",
      "3     BLM_proteomics  1.324065e-03\n",
      "4    CDK2_proteomics  1.489990e-03\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  ZFHX3\n",
      "         Comparison   P_Value\n",
      "0  ZFHX3_proteomics  0.001988\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  ZFHX4\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  DOCK3\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  INPPL1\n",
      "          Comparison       P_Value\n",
      "0  INPPL1_proteomics  9.820989e-08\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  FBXW7\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  MUC16\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  KMT2D\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  HERC2\n",
      "          Comparison   P_Value\n",
      "0  TRIM36_proteomics  0.000339\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  RPL22\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  RBM27\n",
      "         Comparison       P_Value\n",
      "0  RBM27_proteomics  2.698889e-10\n",
      "1  ZC3H4_proteomics  3.875150e-04\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  NSD1\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  SYNE1\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  SCAF4\n",
      "         Comparison   P_Value\n",
      "0  SCAF4_proteomics  0.000097\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  PCLO\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  JAK1\n",
      "        Comparison   P_Value\n",
      "0  JAK1_proteomics  0.000017\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  CCDC168\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  ZNF469\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  LMAN1\n",
      "          Comparison  P_Value\n",
      "0  SEC24A_proteomics   0.0017\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Gene:  OBSCN\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  AHNAK\n",
      "No significant comparisons.\n",
      "\n",
      "Gene:  HUWE1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda3/lib/python3.6/site-packages/numpy/core/fromnumeric.py:3157: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  **kwargs)\n",
      "/anaconda3/lib/python3.6/site-packages/numpy/core/_methods.py:132: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No significant comparisons.\n",
      "\n",
      "Gene:  DNAH17\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "proteomics = en.get_proteomics()\n",
    "for protein in fm:\n",
    "    try:\n",
    "        print(\"\\nGene: \", protein)\n",
    "        interacting_proteins = al.get_interacting_proteins(protein)\n",
    "        ips = []\n",
    "        for ip in interacting_proteins:\n",
    "            if ip in proteomics.columns:\n",
    "                ips.append(ip)\n",
    "\n",
    "        protdf = en.append_mutations_to_omics(mutation_genes=[protein], omics_df=proteomics, omics_genes=ips)\n",
    "        protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "        #print(protdf)\n",
    "\n",
    "        for ind, row in protdf.iterrows():\n",
    "            if 'Wildtype_Tumor' not in row[protein+\"_Mutation\"]:\n",
    "                protdf.at[ind,'Label'] = 'Mutated'\n",
    "            else:\n",
    "                protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "        #print(protdf)\n",
    "        protdf = protdf.drop(protein+\"_Mutation\",axis=1)\n",
    "        protdf = protdf.drop(protein+\"_Location\",axis=1)\n",
    "        protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "        columns_to_comp = []\n",
    "        for ip in ips:\n",
    "            columns_to_comp.append(ip+\"_proteomics\")\n",
    "\n",
    "        wrap_results = al.wrap_ttest(protdf, 'Label', columns_to_comp)\n",
    "        \n",
    "        if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "        \n",
    "    except:\n",
    "        print(\"Error with: \", protein, \"!\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'float' object has no attribute 'tolist'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-11-d316fb15e719>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mphos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0men\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_phosphoproteomics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mtestdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0men\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend_mutations_to_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmutation_genes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'PIK3R1'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mphos\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_genes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'PIK3R1'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtestdf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/CPTAC/Endometrial/__init__.py\u001b[0m in \u001b[0;36mappend_mutations_to_omics\u001b[0;34m(omics_df, mutation_genes, omics_genes, show_location)\u001b[0m\n\u001b[1;32m    484\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    485\u001b[0m     \u001b[0;31m# Return the merge.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 486\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mUtilities\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend_mutations_to_omics\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmutation_genes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_genes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mshow_location\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    487\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    488\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mdefine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mterm\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/CPTAC/Endometrial/utilities.py\u001b[0m in \u001b[0;36mappend_mutations_to_omics\u001b[0;34m(self, somatic_mutation, omics_df, mutation_genes, omics_genes, show_location)\u001b[0m\n\u001b[1;32m    286\u001b[0m         \"\"\"\n\u001b[1;32m    287\u001b[0m         \u001b[0momics\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_omics_from_str_or_list\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0momics_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0momics_genes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 288\u001b[0;31m         \u001b[0mmutations\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_mutations_from_str_or_list\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmutation_genes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    289\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    290\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0momics\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmutations\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# If either selector returned None, then there were gene(s) that didn't match anything, and an error message was printed. We'll return None.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/CPTAC/Endometrial/utilities.py\u001b[0m in \u001b[0;36mget_mutations_from_str_or_list\u001b[0;34m(self, somatic_mutation, genes)\u001b[0m\n\u001b[1;32m    268\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_mutations_for_gene\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    269\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseries\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindexes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbase\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mIndex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# If it's a str, or list or array-like of str of genes, feed it to the proper function\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 270\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_mutations_for_genes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    271\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# If it's neither of those, they done messed up. Tell 'em.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    272\u001b[0m             \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Genes parameter {} is of invalid type {}. Valid types: str, or list or array-like of str.\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/CPTAC/Endometrial/utilities.py\u001b[0m in \u001b[0;36mget_mutations_for_genes\u001b[0;34m(self, somatic_mutation, genes)\u001b[0m\n\u001b[1;32m    248\u001b[0m         \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdrop_duplicates\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Create an empty dataframe, which we'll fill with the columns we select using our genes, and then return.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    249\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mgene\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgenes\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 250\u001b[0;31m             \u001b[0mselected\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_mutations_for_gene\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msomatic_mutation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Get the mutations for our gene\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    251\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mselected\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# If there's no mutation data for that gene, get_mutations_for_gene will have printed an error message. Return None.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    252\u001b[0m                 \u001b[0;32mreturn\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/CPTAC/Endometrial/utilities.py\u001b[0m in \u001b[0;36mget_mutations_for_gene\u001b[0;34m(self, mutations, gene)\u001b[0m\n\u001b[1;32m    226\u001b[0m                 \u001b[0msample_locations_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0msample_locations\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    227\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# It's a pandas Series\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 228\u001b[0;31m                 \u001b[0msample_locations_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msample_locations\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    229\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    230\u001b[0m             \u001b[0;31m# Put in our template dataframe\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'float' object has no attribute 'tolist'"
     ]
    }
   ],
   "source": [
    "testdf = en.append_mutations_to_omics(mutation_genes=['PIK3R1'], omics_df=proteomics, omics_genes=['PIK3R1'])\n",
    "print(testdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOC110117498-PIK3R3\n",
      "PIK3AP1\n",
      "PIK3C2A\n",
      "PIK3C2B\n",
      "PIK3C3\n",
      "PIK3CA\n",
      "PIK3CB\n",
      "PIK3CD\n",
      "PIK3CG\n",
      "PIK3IP1\n",
      "PIK3R1\n",
      "PIK3R2\n",
      "PIK3R4\n",
      "PIKFYVE\n"
     ]
    }
   ],
   "source": [
    "for col in proteomics.columns:\n",
    "    if 'PIK' in col:\n",
    "        print(col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
