{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Omics Cookbook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Library Imports"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run this cell to import the necessary libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to the cptac data service package. Available datasets may be\n",
      "viewed using cptac.list_data(). In order to access a specific data\n",
      "set, import a cptac subfolder using either 'import cptac.dataset' or\n",
      "'from cptac import dataset'.\n",
      "******\n",
      "Version: 0.4.0\n",
      "******\n",
      "You have loaded the cptac endometrial dataset. To view available\n",
      "dataframes, use cptac.endometrial.list_data(). To view available\n",
      "functions for accessing and manipulating the dataframes, use\n",
      "cptac.endometrial.list_api().\n",
      "endometrial data version: 2.1\n",
      "\n",
      "Loading Dictionary...\n",
      "Loading cptac endometrial data:\n",
      "Loading proteomics data...\n",
      "Loading clinical data...\n",
      "Loading acetylproteomics data...\n",
      "Loading phosphoproteomics_site data...\n",
      "Loading somatic data...\n",
      "Loading transcriptomics_circular data...\n",
      "Loading phosphoproteomics_gene data...\n",
      "Loading transcriptomics_linear data...\n",
      "Loading miRNA data...\n",
      "Loading somatic_binary data...\n",
      "Loading CNA data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "import re\n",
    "\n",
    "import cptac.endometrial as en\n",
    "import cptac.algorithms as al"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get frequently mutated genes list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Enter the type of cancer and the cutoff for mutation frequency that you would like to use."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Gene  Percent Mutated\n",
      "0     RPL22         0.168421\n",
      "1    ARID1A         0.452632\n",
      "2    PIK3CA         0.494737\n",
      "3     FBXW7         0.189474\n",
      "4     RBM27         0.168421\n",
      "5      NSD1         0.168421\n",
      "6     SYNE1         0.168421\n",
      "7     ZFHX4         0.200000\n",
      "8      PTEN         0.789474\n",
      "9      CTCF         0.284211\n",
      "10   ZNF469         0.157895\n",
      "11     TP53         0.221053\n",
      "12    KMT2B         0.242105\n",
      "13     KRAS         0.326316\n",
      "14    LMAN1         0.157895\n",
      "15    SCAF4         0.168421\n",
      "16   PIK3R1         0.389474\n",
      "17   CTNNB1         0.305263\n",
      "18     PCLO         0.168421\n",
      "19    KMT2D         0.178947\n",
      "20    HERC2         0.178947\n",
      "21    ZFHX3         0.221053\n",
      "22     JAK1         0.168421\n",
      "23    OBSCN         0.157895\n",
      "24    AHNAK         0.157895\n",
      "25  CCDC168         0.168421\n",
      "26    DOCK3         0.200000\n",
      "27    MUC16         0.189474\n",
      "28    HUWE1         0.157895\n",
      "29   DNAH17         0.157895\n",
      "30   INPPL1         0.200000\n"
     ]
    }
   ],
   "source": [
    "cancer_type = \"Endometrial\"\n",
    "desired_cutoff = .15\n",
    "\n",
    "fm = al.get_frequently_mutated(cancer_type, cutoff=desired_cutoff)\n",
    "print(fm)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Select a gene from the list of frequently mutated genes above"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set the gene to which of the above frequently mutated genes you want to examine. For example, if you want to look at the gene PTEN, change the cell below to say \"gene = 'PTEN'\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = 'ARID1A'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Select desired omics comparison"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Select which type of omics you want to compare. If you want to compare proteomics data for the interacting proteins, uncomment the \"omics = en.get_proteomics()\" line, etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "#omics = en.get_proteomics()\n",
    "#omics = en.get_transcriptomics()\n",
    "omics = en.get_phosphoproteomics()\n",
    "#omics = en.get_acetylproteomics()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate interacting proteins and test omics comparisons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Simply run this cell after following the instructions above to see the results of the comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  ZFHX3\n",
      "Generating interacting protein list\n",
      "Doing t-test comparisons\n",
      "\n",
      "         Comparison   P_Value\n",
      "0  ZFHX3_proteomics  0.001988\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    '''Prevent slice degrees of freedom warning'''\n",
    "    import warnings\n",
    "    warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "    interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "    print(\"Generating interacting protein list\")\n",
    "    interacting_proteins_in_omics_df = []\n",
    "\n",
    "    '''Only do comparisons on proteins in the omics dataframe'''\n",
    "    for ip in interacting_proteins:\n",
    "        if omics.name == 'phosphoproteomics' or omics.name == 'acetylproteomics':\n",
    "            col_regex = ip + \"-.*\" # Build a regex to get all columns that match the gene\n",
    "        else:\n",
    "            col_regex = '^{}$'.format(ip)\n",
    "\n",
    "        selected = omics.filter(regex=col_regex)\n",
    "\n",
    "        if len(selected.columns) > 0:\n",
    "            interacting_proteins_in_omics_df.append(ip)\n",
    "\n",
    "    '''Ceate dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.append_mutations_to_omics(mutation_genes=[gene], omics_df=omics, omics_genes=interacting_proteins_in_omics_df)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "\n",
    "except:\n",
    "    print(\"Error in Comparison\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Look at effect on all proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This will look at the effect of the selected gene mutation on all proteins, and report any significant results."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  ARID1A\n",
      "Generating protein list\n",
      "Error in Comparison\n",
      "columns overlap but no suffix specified: Index(['ACADM-S240_phosphoproteomics'], dtype='object')\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    '''Prevent slice degrees of freedom warning'''\n",
    "    #import warnings\n",
    "    #warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "    proteomics = en.get_proteomics()\n",
    "    proteins = proteomics.columns\n",
    "    proteins = proteins[:300]\n",
    "\n",
    "    print(\"Generating protein list\")\n",
    "    interacting_proteins_in_omics_df = []\n",
    "\n",
    "    '''Only do comparisons on proteins in the omics dataframe'''\n",
    "    for ip in proteins:\n",
    "        if omics.name == 'phosphoproteomics' or omics.name == 'acetylproteomics':\n",
    "            col_regex = ip + \"-.*\" # Build a regex to get all columns that match the gene\n",
    "        else:\n",
    "            col_regex = '^{}$'.format(ip)\n",
    "\n",
    "        selected = omics.filter(regex=col_regex)\n",
    "\n",
    "        if len(selected.columns) > 0:\n",
    "            interacting_proteins_in_omics_df.append(ip)\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.append_mutations_to_omics(mutation_genes=[gene], omics_df=omics, omics_genes=interacting_proteins_in_omics_df)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    #print(protdf)\n",
    "\n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           A1BG_transcriptomics  A2M_transcriptomics  A2ML1_transcriptomics  \\\n",
      "Sample_ID                                                                     \n",
      "S001                       4.02                13.39                   6.79   \n",
      "S002                       4.81                13.24                   6.33   \n",
      "S003                       6.24                14.32                   9.42   \n",
      "S005                       5.31                13.77                   4.22   \n",
      "S006                       9.84                13.12                   6.83   \n",
      "S007                       5.03                14.13                   7.75   \n",
      "S008                       3.17                13.49                   5.88   \n",
      "S009                       6.03                13.67                   6.69   \n",
      "S010                       6.02                14.30                   8.47   \n",
      "S011                       5.71                14.97                   9.96   \n",
      "S012                       5.17                13.40                   8.69   \n",
      "S014                       5.38                14.13                  11.45   \n",
      "S016                       5.86                13.15                   3.65   \n",
      "S017                       3.27                12.28                   6.97   \n",
      "S018                       4.65                13.42                  10.75   \n",
      "S019                       4.33                13.06                  13.60   \n",
      "S020                       4.58                15.05                   8.90   \n",
      "S021                       4.60                12.69                   9.56   \n",
      "S022                       4.90                14.13                   5.44   \n",
      "S023                       4.03                13.07                   4.46   \n",
      "S024                       4.62                14.12                  11.77   \n",
      "S025                       5.46                13.98                   3.94   \n",
      "S026                       5.73                14.70                   8.07   \n",
      "S027                       4.13                13.13                   4.68   \n",
      "S028                       6.08                15.56                   5.92   \n",
      "S029                       5.24                14.48                   5.87   \n",
      "S030                       7.36                14.15                   8.97   \n",
      "S031                       5.73                13.36                   5.60   \n",
      "S032                       5.31                14.65                   5.02   \n",
      "S033                       4.00                14.71                  11.18   \n",
      "...                         ...                  ...                    ...   \n",
      "S073                       5.81                12.53                   5.67   \n",
      "S074                       5.63                14.82                   6.47   \n",
      "S075                       4.28                14.31                  10.19   \n",
      "S076                       5.52                13.79                   9.97   \n",
      "S077                       7.17                13.57                   6.05   \n",
      "S078                       6.11                13.75                   6.29   \n",
      "S079                       5.75                13.96                   4.20   \n",
      "S080                       5.55                14.00                   6.36   \n",
      "S081                       6.09                13.94                   5.76   \n",
      "S082                       5.82                14.34                   3.82   \n",
      "S083                       4.21                13.18                   8.37   \n",
      "S084                       4.62                14.03                   8.78   \n",
      "S085                       5.83                13.78                   6.26   \n",
      "S086                       5.11                13.74                  10.26   \n",
      "S087                       5.41                13.50                   7.49   \n",
      "S088                       6.65                13.37                   5.12   \n",
      "S090                       7.44                14.90                   5.22   \n",
      "S091                       3.79                15.16                   9.16   \n",
      "S092                       6.94                13.08                   6.25   \n",
      "S093                       4.95                14.86                   9.54   \n",
      "S094                       5.34                13.84                   7.95   \n",
      "S095                       5.22                14.73                   7.26   \n",
      "S096                       5.63                13.96                  11.41   \n",
      "S097                       7.63                13.49                   6.43   \n",
      "S098                       4.85                12.88                   8.45   \n",
      "S099                       4.84                13.37                   5.90   \n",
      "S100                       4.80                13.40                   5.12   \n",
      "S101                       5.95                14.23                  11.96   \n",
      "S102                       5.25                13.75                   2.06   \n",
      "S103                       6.79                14.96                   7.50   \n",
      "\n",
      "           A4GALT_transcriptomics  AAAS_transcriptomics  AACS_transcriptomics  \\\n",
      "Sample_ID                                                                       \n",
      "S001                        10.34                 10.26                 10.53   \n",
      "S002                        10.83                 10.44                 10.45   \n",
      "S003                        10.98                  9.24                 11.29   \n",
      "S005                         8.68                  9.67                  9.85   \n",
      "S006                        11.42                  9.87                  9.98   \n",
      "S007                         9.61                 10.17                 10.48   \n",
      "S008                         9.71                 10.23                 10.74   \n",
      "S009                         8.78                 10.21                  9.66   \n",
      "S010                         9.98                  9.65                 11.00   \n",
      "S011                        10.22                 10.30                 10.71   \n",
      "S012                        11.42                  9.58                 10.71   \n",
      "S014                        10.20                 10.11                 10.21   \n",
      "S016                         8.37                 10.33                  9.17   \n",
      "S017                        10.04                 10.11                 12.82   \n",
      "S018                        10.05                 10.51                 10.58   \n",
      "S019                        10.04                 10.16                 10.71   \n",
      "S020                        10.71                  9.74                 10.27   \n",
      "S021                         9.60                 10.48                 11.59   \n",
      "S022                         9.49                 10.37                 10.52   \n",
      "S023                         7.10                 10.69                 10.37   \n",
      "S024                        10.29                 10.27                 10.55   \n",
      "S025                        10.04                 10.34                 10.53   \n",
      "S026                         9.90                 10.52                 10.56   \n",
      "S027                        10.74                 10.06                 11.01   \n",
      "S028                        10.17                 10.24                  9.84   \n",
      "S029                        10.02                 10.00                 10.53   \n",
      "S030                        10.23                 10.33                 10.77   \n",
      "S031                        10.06                  9.63                 10.02   \n",
      "S032                        10.22                  9.87                 10.56   \n",
      "S033                        10.31                  9.84                 10.02   \n",
      "...                           ...                   ...                   ...   \n",
      "S073                        10.80                  9.90                  9.65   \n",
      "S074                         9.79                 10.07                 10.52   \n",
      "S075                        10.64                 10.06                 11.03   \n",
      "S076                        10.36                 10.52                 10.50   \n",
      "S077                        10.77                 10.55                 10.96   \n",
      "S078                         8.92                 10.21                 10.42   \n",
      "S079                        10.74                 10.04                 10.73   \n",
      "S080                         9.98                 10.35                 10.56   \n",
      "S081                         9.82                 10.40                 10.76   \n",
      "S082                         8.77                  9.93                  9.10   \n",
      "S083                        10.74                  9.82                 11.10   \n",
      "S084                        10.43                 10.23                 11.20   \n",
      "S085                        10.09                  9.88                 11.11   \n",
      "S086                         9.21                 10.39                 11.24   \n",
      "S087                        11.24                 10.08                 10.51   \n",
      "S088                        10.17                 10.18                 10.56   \n",
      "S090                         9.63                 10.73                  9.64   \n",
      "S091                         8.98                 10.67                  9.66   \n",
      "S092                        10.28                 10.90                 10.96   \n",
      "S093                        10.81                 10.15                 10.97   \n",
      "S094                         9.17                 10.10                  9.77   \n",
      "S095                         8.77                 10.41                 10.31   \n",
      "S096                         7.80                 10.44                 10.96   \n",
      "S097                         8.99                 10.48                 10.05   \n",
      "S098                         9.64                 10.20                  9.74   \n",
      "S099                         9.84                 10.76                 10.97   \n",
      "S100                        10.64                 10.35                 11.02   \n",
      "S101                         8.05                 10.03                 10.13   \n",
      "S102                         9.87                 10.08                  9.34   \n",
      "S103                         8.83                 10.29                 10.14   \n",
      "\n",
      "           AADAT_transcriptomics  AAED1_transcriptomics  \\\n",
      "Sample_ID                                                 \n",
      "S001                        9.23                   7.68   \n",
      "S002                       10.45                   7.30   \n",
      "S003                        6.66                   8.51   \n",
      "S005                       10.05                   8.42   \n",
      "S006                        5.95                   8.82   \n",
      "S007                        8.67                   8.58   \n",
      "S008                        8.65                   7.79   \n",
      "S009                        8.61                   7.34   \n",
      "S010                        8.38                   8.44   \n",
      "S011                        9.19                   8.14   \n",
      "S012                        9.06                   7.99   \n",
      "S014                        7.75                   7.29   \n",
      "S016                        6.68                   7.94   \n",
      "S017                        9.95                   7.47   \n",
      "S018                        9.02                   7.64   \n",
      "S019                        9.09                   8.08   \n",
      "S020                        8.05                   8.21   \n",
      "S021                        8.47                   7.33   \n",
      "S022                        8.79                   8.24   \n",
      "S023                        9.13                   6.43   \n",
      "S024                        8.75                   7.82   \n",
      "S025                       10.26                   7.87   \n",
      "S026                        8.98                   8.28   \n",
      "S027                        9.00                   7.33   \n",
      "S028                        8.54                   8.50   \n",
      "S029                        8.23                   8.45   \n",
      "S030                        9.53                   7.87   \n",
      "S031                        9.40                   7.75   \n",
      "S032                        8.36                   8.61   \n",
      "S033                        9.95                   7.63   \n",
      "...                          ...                    ...   \n",
      "S073                        8.88                   7.82   \n",
      "S074                        9.43                   8.46   \n",
      "S075                        8.94                   7.98   \n",
      "S076                        8.92                   8.71   \n",
      "S077                        8.94                   7.46   \n",
      "S078                        9.52                   6.35   \n",
      "S079                        8.92                   8.53   \n",
      "S080                        8.98                   8.12   \n",
      "S081                       10.38                   7.09   \n",
      "S082                        7.63                   7.21   \n",
      "S083                        8.26                   8.15   \n",
      "S084                        9.32                   7.36   \n",
      "S085                        9.25                   7.98   \n",
      "S086                       10.13                   7.91   \n",
      "S087                        8.69                   7.31   \n",
      "S088                       10.84                   8.75   \n",
      "S090                        8.91                   8.38   \n",
      "S091                        9.37                   7.59   \n",
      "S092                        8.77                   8.28   \n",
      "S093                        8.78                   8.06   \n",
      "S094                        8.58                   7.35   \n",
      "S095                        8.34                   8.43   \n",
      "S096                        7.49                   7.51   \n",
      "S097                        8.25                   7.75   \n",
      "S098                        7.44                   7.11   \n",
      "S099                        9.96                   7.11   \n",
      "S100                        8.81                   7.34   \n",
      "S101                        6.93                   7.67   \n",
      "S102                        6.04                   8.91   \n",
      "S103                        8.14                   8.15   \n",
      "\n",
      "           AAGAB_transcriptomics  AAK1_transcriptomics  ...  \\\n",
      "Sample_ID                                               ...   \n",
      "S001                       11.42                 10.91  ...   \n",
      "S002                       12.22                 10.48  ...   \n",
      "S003                       11.58                 10.71  ...   \n",
      "S005                       11.07                 11.06  ...   \n",
      "S006                       11.23                 12.08  ...   \n",
      "S007                       11.01                 12.47  ...   \n",
      "S008                       11.00                 11.68  ...   \n",
      "S009                       11.28                 12.13  ...   \n",
      "S010                       11.06                 12.37  ...   \n",
      "S011                       10.70                 11.80  ...   \n",
      "S012                       11.62                 11.42  ...   \n",
      "S014                       11.02                 11.86  ...   \n",
      "S016                       10.96                 12.31  ...   \n",
      "S017                       11.07                 11.47  ...   \n",
      "S018                       11.37                 11.98  ...   \n",
      "S019                       11.51                 12.07  ...   \n",
      "S020                       11.44                 11.82  ...   \n",
      "S021                       10.67                 11.26  ...   \n",
      "S022                       11.39                 11.73  ...   \n",
      "S023                       11.08                 11.83  ...   \n",
      "S024                       11.18                 11.25  ...   \n",
      "S025                       11.03                 12.07  ...   \n",
      "S026                       11.04                 11.41  ...   \n",
      "S027                       11.41                 11.29  ...   \n",
      "S028                       10.75                 12.20  ...   \n",
      "S029                       11.39                 11.93  ...   \n",
      "S030                       11.52                 11.83  ...   \n",
      "S031                       11.09                 12.06  ...   \n",
      "S032                       11.25                 11.28  ...   \n",
      "S033                       10.86                 12.26  ...   \n",
      "...                          ...                   ...  ...   \n",
      "S073                       11.50                 10.49  ...   \n",
      "S074                       10.93                 12.14  ...   \n",
      "S075                       11.34                 12.20  ...   \n",
      "S076                       10.68                 12.04  ...   \n",
      "S077                       11.39                 11.16  ...   \n",
      "S078                       10.90                 12.01  ...   \n",
      "S079                       11.19                 11.79  ...   \n",
      "S080                       11.42                 11.06  ...   \n",
      "S081                       10.49                 11.86  ...   \n",
      "S082                       10.74                 12.31  ...   \n",
      "S083                       10.74                 11.66  ...   \n",
      "S084                       11.02                 11.45  ...   \n",
      "S085                       10.80                 11.81  ...   \n",
      "S086                       10.51                 11.08  ...   \n",
      "S087                       10.83                 11.51  ...   \n",
      "S088                       11.29                 11.22  ...   \n",
      "S090                       10.80                 11.21  ...   \n",
      "S091                       10.89                 11.40  ...   \n",
      "S092                       10.97                 11.03  ...   \n",
      "S093                       10.91                 11.66  ...   \n",
      "S094                       10.95                 11.77  ...   \n",
      "S095                       10.55                 12.17  ...   \n",
      "S096                       11.21                 12.30  ...   \n",
      "S097                       10.80                 11.75  ...   \n",
      "S098                       10.95                 11.67  ...   \n",
      "S099                       11.42                 11.46  ...   \n",
      "S100                       10.85                 11.24  ...   \n",
      "S101                       10.98                 12.26  ...   \n",
      "S102                       11.50                 12.49  ...   \n",
      "S103                       10.83                 12.74  ...   \n",
      "\n",
      "           AKAP10_transcriptomics  AKAP11_transcriptomics  \\\n",
      "Sample_ID                                                   \n",
      "S001                        10.64                   12.51   \n",
      "S002                        10.77                   10.66   \n",
      "S003                        10.83                   12.16   \n",
      "S005                        11.23                   12.24   \n",
      "S006                        10.84                   11.43   \n",
      "S007                        10.66                   12.94   \n",
      "S008                        10.83                   12.42   \n",
      "S009                        11.04                   12.57   \n",
      "S010                        11.01                   13.01   \n",
      "S011                        10.63                   12.29   \n",
      "S012                        11.17                   12.28   \n",
      "S014                        10.66                   12.46   \n",
      "S016                        11.52                   11.59   \n",
      "S017                        10.91                   12.37   \n",
      "S018                        10.67                   12.66   \n",
      "S019                        10.72                   12.14   \n",
      "S020                        10.83                   12.09   \n",
      "S021                        10.74                   12.51   \n",
      "S022                        10.75                   12.31   \n",
      "S023                        10.86                   12.13   \n",
      "S024                        10.73                   11.83   \n",
      "S025                        10.67                   12.49   \n",
      "S026                        10.82                   12.16   \n",
      "S027                        11.12                   12.89   \n",
      "S028                        10.73                   12.46   \n",
      "S029                        10.76                   12.09   \n",
      "S030                        10.74                   12.09   \n",
      "S031                        10.42                   11.10   \n",
      "S032                        10.72                   11.88   \n",
      "S033                        10.87                   12.93   \n",
      "...                           ...                     ...   \n",
      "S073                        11.07                   12.70   \n",
      "S074                        10.68                   12.46   \n",
      "S075                        10.84                   13.12   \n",
      "S076                        10.90                   12.63   \n",
      "S077                        10.74                   11.85   \n",
      "S078                        10.79                   12.69   \n",
      "S079                        10.83                   12.29   \n",
      "S080                        10.91                   12.29   \n",
      "S081                        10.83                   12.48   \n",
      "S082                        11.26                   11.99   \n",
      "S083                        10.55                   12.25   \n",
      "S084                        10.56                   12.31   \n",
      "S085                        10.79                   12.22   \n",
      "S086                        10.57                   11.85   \n",
      "S087                        10.89                   12.19   \n",
      "S088                        10.69                   11.81   \n",
      "S090                        10.99                   12.34   \n",
      "S091                        10.82                   12.30   \n",
      "S092                        10.65                   11.58   \n",
      "S093                        10.81                   12.54   \n",
      "S094                        11.18                   12.71   \n",
      "S095                        10.50                   12.29   \n",
      "S096                        10.35                   12.40   \n",
      "S097                        10.97                   12.10   \n",
      "S098                        11.16                   12.41   \n",
      "S099                        10.57                   11.92   \n",
      "S100                        10.34                   11.01   \n",
      "S101                        10.85                   12.73   \n",
      "S102                        10.23                   11.72   \n",
      "S103                        11.22                   12.32   \n",
      "\n",
      "           AKAP12_transcriptomics  AKAP13_transcriptomics  \\\n",
      "Sample_ID                                                   \n",
      "S001                        11.06                   13.26   \n",
      "S002                        10.05                   13.23   \n",
      "S003                        10.65                   13.39   \n",
      "S005                        10.18                   13.24   \n",
      "S006                         8.21                   13.06   \n",
      "S007                        12.28                   13.41   \n",
      "S008                        10.71                   13.68   \n",
      "S009                        12.59                   13.26   \n",
      "S010                        10.07                   14.13   \n",
      "S011                        12.11                   13.72   \n",
      "S012                         9.81                   13.62   \n",
      "S014                        13.01                   14.55   \n",
      "S016                        13.23                   12.94   \n",
      "S017                         9.19                   12.92   \n",
      "S018                        10.75                   14.05   \n",
      "S019                        12.27                   13.97   \n",
      "S020                        12.09                   14.08   \n",
      "S021                         8.83                   13.54   \n",
      "S022                        11.30                   14.21   \n",
      "S023                        14.64                   13.08   \n",
      "S024                        10.63                   13.70   \n",
      "S025                        10.78                   13.66   \n",
      "S026                        11.97                   13.79   \n",
      "S027                        11.60                   13.43   \n",
      "S028                        13.25                   14.62   \n",
      "S029                        10.61                   14.01   \n",
      "S030                        11.95                   13.81   \n",
      "S031                        12.48                   13.69   \n",
      "S032                        10.85                   13.65   \n",
      "S033                        12.17                   14.09   \n",
      "...                           ...                     ...   \n",
      "S073                        11.14                   13.86   \n",
      "S074                        11.51                   13.58   \n",
      "S075                        10.91                   13.77   \n",
      "S076                        12.62                   15.20   \n",
      "S077                        13.69                   13.36   \n",
      "S078                        11.39                   14.24   \n",
      "S079                        11.64                   13.49   \n",
      "S080                        11.17                   13.33   \n",
      "S081                        11.93                   14.24   \n",
      "S082                        13.21                   13.61   \n",
      "S083                        11.92                   13.53   \n",
      "S084                        11.30                   13.72   \n",
      "S085                        10.69                   13.32   \n",
      "S086                        10.69                   13.50   \n",
      "S087                         9.72                   13.64   \n",
      "S088                        10.28                   13.18   \n",
      "S090                        11.43                   13.69   \n",
      "S091                        10.67                   13.67   \n",
      "S092                        12.36                   13.79   \n",
      "S093                        11.90                   13.64   \n",
      "S094                        10.55                   13.63   \n",
      "S095                        12.19                   13.68   \n",
      "S096                        11.15                   13.91   \n",
      "S097                        13.22                   13.75   \n",
      "S098                        13.28                   14.05   \n",
      "S099                        12.20                   13.78   \n",
      "S100                        10.85                   12.76   \n",
      "S101                        11.21                   13.70   \n",
      "S102                        15.11                   14.49   \n",
      "S103                        12.59                   13.81   \n",
      "\n",
      "           AKAP17A_transcriptomics  AKAP5_transcriptomics  \\\n",
      "Sample_ID                                                   \n",
      "S001                         10.33                   8.89   \n",
      "S002                         10.88                   8.62   \n",
      "S003                         10.58                   8.28   \n",
      "S005                         10.00                   8.41   \n",
      "S006                         11.29                   9.15   \n",
      "S007                         10.46                   9.61   \n",
      "S008                         11.13                   8.90   \n",
      "S009                         10.19                   9.53   \n",
      "S010                         10.62                  10.29   \n",
      "S011                         10.92                   8.95   \n",
      "S012                         10.90                   8.73   \n",
      "S014                         10.89                   8.70   \n",
      "S016                         10.40                   9.05   \n",
      "S017                         11.19                   9.09   \n",
      "S018                         10.83                   8.67   \n",
      "S019                         10.35                   8.84   \n",
      "S020                         10.54                   8.55   \n",
      "S021                         11.48                   8.85   \n",
      "S022                         11.07                   9.26   \n",
      "S023                         10.81                   8.95   \n",
      "S024                         10.88                   8.61   \n",
      "S025                         11.21                   8.97   \n",
      "S026                         10.74                   8.78   \n",
      "S027                         11.11                   8.71   \n",
      "S028                         10.73                   9.17   \n",
      "S029                         11.26                   9.06   \n",
      "S030                         10.52                   9.14   \n",
      "S031                         10.91                   8.33   \n",
      "S032                         10.79                   8.55   \n",
      "S033                         10.36                   9.67   \n",
      "...                            ...                    ...   \n",
      "S073                         10.34                   8.58   \n",
      "S074                         10.60                   9.14   \n",
      "S075                         10.82                   8.61   \n",
      "S076                         11.13                   8.74   \n",
      "S077                         11.13                   8.62   \n",
      "S078                         11.18                   9.73   \n",
      "S079                         10.72                   8.47   \n",
      "S080                         10.70                   8.41   \n",
      "S081                         10.85                   9.17   \n",
      "S082                         10.34                  10.13   \n",
      "S083                         11.53                   8.85   \n",
      "S084                         11.01                   8.74   \n",
      "S085                         11.10                   9.32   \n",
      "S086                         11.12                   8.76   \n",
      "S087                         11.25                   9.04   \n",
      "S088                         10.64                   9.03   \n",
      "S090                         10.85                   8.77   \n",
      "S091                         10.81                   9.14   \n",
      "S092                         10.86                   9.37   \n",
      "S093                         10.85                   8.82   \n",
      "S094                         11.06                   9.41   \n",
      "S095                         11.36                   8.71   \n",
      "S096                         11.02                   9.56   \n",
      "S097                         10.77                   8.94   \n",
      "S098                         11.50                   9.01   \n",
      "S099                         11.26                   8.85   \n",
      "S100                         10.86                   8.57   \n",
      "S101                         10.53                   9.47   \n",
      "S102                         11.06                   9.80   \n",
      "S103                         10.72                   9.30   \n",
      "\n",
      "           AKAP7_transcriptomics  AKAP8_transcriptomics  \\\n",
      "Sample_ID                                                 \n",
      "S001                        9.59                   9.94   \n",
      "S002                       11.04                  10.67   \n",
      "S003                        8.56                  10.16   \n",
      "S005                        9.46                  10.56   \n",
      "S006                        9.42                  11.31   \n",
      "S007                        7.58                  10.19   \n",
      "S008                        9.26                  10.40   \n",
      "S009                        9.50                  11.49   \n",
      "S010                        8.71                  10.55   \n",
      "S011                        8.67                  10.30   \n",
      "S012                       10.52                  10.36   \n",
      "S014                        9.26                  10.38   \n",
      "S016                        8.44                  10.79   \n",
      "S017                        9.98                  10.82   \n",
      "S018                        7.71                  10.26   \n",
      "S019                        7.46                  10.71   \n",
      "S020                        8.02                  10.33   \n",
      "S021                        8.88                  10.74   \n",
      "S022                        9.09                  10.25   \n",
      "S023                        8.08                  10.10   \n",
      "S024                        8.91                  10.56   \n",
      "S025                        8.76                  10.59   \n",
      "S026                        9.21                  10.24   \n",
      "S027                        9.66                  10.10   \n",
      "S028                        8.14                  10.57   \n",
      "S029                        8.54                  10.41   \n",
      "S030                       10.19                  10.40   \n",
      "S031                        9.50                  10.66   \n",
      "S032                        8.73                  10.35   \n",
      "S033                        9.82                  10.56   \n",
      "...                          ...                    ...   \n",
      "S073                        9.70                  10.25   \n",
      "S074                        8.87                  10.42   \n",
      "S075                       10.12                  10.21   \n",
      "S076                        8.24                  10.31   \n",
      "S077                        9.89                  10.59   \n",
      "S078                       10.43                  10.55   \n",
      "S079                       10.01                  10.59   \n",
      "S080                        9.45                  10.67   \n",
      "S081                        8.11                  10.72   \n",
      "S082                        8.29                  10.91   \n",
      "S083                        9.80                  10.58   \n",
      "S084                        9.04                  10.53   \n",
      "S085                        9.85                  10.47   \n",
      "S086                       10.09                  10.44   \n",
      "S087                        9.86                  10.34   \n",
      "S088                        9.37                  10.10   \n",
      "S090                        8.86                  10.60   \n",
      "S091                        8.77                  10.74   \n",
      "S092                        9.43                  10.72   \n",
      "S093                        9.93                  10.41   \n",
      "S094                        9.71                  10.42   \n",
      "S095                        8.57                  10.38   \n",
      "S096                        8.30                  10.37   \n",
      "S097                        8.14                   9.68   \n",
      "S098                        8.54                  10.16   \n",
      "S099                        9.65                  10.43   \n",
      "S100                        8.52                  10.33   \n",
      "S101                        9.05                  10.74   \n",
      "S102                        7.61                  10.62   \n",
      "S103                        8.71                  10.36   \n",
      "\n",
      "           AKAP8L_transcriptomics     Label  \n",
      "Sample_ID                                    \n",
      "S001                        10.72   Mutated  \n",
      "S002                        11.04   Mutated  \n",
      "S003                        10.45  Wildtype  \n",
      "S005                        10.49  Wildtype  \n",
      "S006                        11.52  Wildtype  \n",
      "S007                        10.73  Wildtype  \n",
      "S008                        11.14   Mutated  \n",
      "S009                        12.17  Wildtype  \n",
      "S010                        10.95  Wildtype  \n",
      "S011                        10.83  Wildtype  \n",
      "S012                        11.50   Mutated  \n",
      "S014                        10.60   Mutated  \n",
      "S016                        11.42  Wildtype  \n",
      "S017                        11.09  Wildtype  \n",
      "S018                        10.27   Mutated  \n",
      "S019                        10.88   Mutated  \n",
      "S020                        10.70  Wildtype  \n",
      "S021                        11.37  Wildtype  \n",
      "S022                        11.28   Mutated  \n",
      "S023                        11.48   Mutated  \n",
      "S024                        10.92   Mutated  \n",
      "S025                        11.32  Wildtype  \n",
      "S026                        10.89   Mutated  \n",
      "S027                        10.81   Mutated  \n",
      "S028                        10.90  Wildtype  \n",
      "S029                        10.91  Wildtype  \n",
      "S030                        10.95   Mutated  \n",
      "S031                        11.63  Wildtype  \n",
      "S032                        10.95  Wildtype  \n",
      "S033                        10.99  Wildtype  \n",
      "...                           ...       ...  \n",
      "S073                        10.62   Mutated  \n",
      "S074                        10.78  Wildtype  \n",
      "S075                        11.09  Wildtype  \n",
      "S076                        10.82  Wildtype  \n",
      "S077                        10.92   Mutated  \n",
      "S078                        10.99   Mutated  \n",
      "S079                        10.99   Mutated  \n",
      "S080                        11.12   Mutated  \n",
      "S081                        11.28  Wildtype  \n",
      "S082                        11.46  Wildtype  \n",
      "S083                        11.52   Mutated  \n",
      "S084                        10.98   Mutated  \n",
      "S085                        11.22   Mutated  \n",
      "S086                        11.82  Wildtype  \n",
      "S087                        11.43   Mutated  \n",
      "S088                        11.13  Wildtype  \n",
      "S090                        10.79   Mutated  \n",
      "S091                        10.84   Mutated  \n",
      "S092                        10.48   Mutated  \n",
      "S093                        10.83   Mutated  \n",
      "S094                        11.39   Mutated  \n",
      "S095                        10.76  Wildtype  \n",
      "S096                        11.32  Wildtype  \n",
      "S097                        10.66  Wildtype  \n",
      "S098                        11.01   Mutated  \n",
      "S099                        10.97   Mutated  \n",
      "S100                        11.12   Mutated  \n",
      "S101                        11.38  Wildtype  \n",
      "S102                        10.73  Wildtype  \n",
      "S103                        10.92  Wildtype  \n",
      "\n",
      "[95 rows x 301 columns]\n"
     ]
    }
   ],
   "source": [
    "print(protdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['A1BG', 'A2M', 'A2ML1', 'A4GALT', 'AAAS', 'AACS', 'AADAT', 'AAED1',\n",
      "       'AAGAB', 'AAK1',\n",
      "       ...\n",
      "       'ARHGAP21', 'ARHGAP23', 'ARHGAP24', 'ARHGAP25', 'ARHGAP26', 'ARHGAP27',\n",
      "       'ARHGAP28', 'ARHGAP29', 'ARHGAP30', 'ARHGAP31'],\n",
      "      dtype='object', name='idx', length=600)\n",
      "600\n",
      "['A1BG', 'A2M', 'A2ML1', 'A4GALT', 'AAAS', 'AACS', 'AADAT', 'AAED1', 'AAGAB', 'AAK1', 'AAMDC', 'AAMP', 'AAR2', 'AARS', 'AARS2', 'AARSD1', 'AASDHPPT', 'AASS', 'AATF', 'ABAT', 'ABCA8', 'ABCB1', 'ABCB10', 'ABCB6', 'ABCB7', 'ABCB8', 'ABCC1', 'ABCC10', 'ABCC3', 'ABCC4', 'ABCD1', 'ABCD3', 'ABCD4', 'ABCE1', 'ABCF1', 'ABCF2', 'ABCF3', 'ABCG1', 'ABHD10', 'ABHD11', 'ABHD12', 'ABHD13', 'ABHD14A', 'ABHD14A-ACY1', 'ABHD14B', 'ABHD16A', 'ABHD17A', 'ABHD17B', 'ABHD17C', 'ABHD2', 'ABHD3', 'ABHD4', 'ABHD5', 'ABHD6', 'ABI1', 'ABI2', 'ABI3', 'ABI3BP', 'ABL1', 'ABL2', 'ABLIM1', 'ABLIM2', 'ABLIM3', 'ABO', 'ABR', 'ABRACL', 'ABRAXAS1', 'ABRAXAS2', 'ABT1', 'ABTB2', 'ACAA1', 'ACAA2', 'ACACA', 'ACACB', 'ACAD10', 'ACAD11', 'ACAD8', 'ACAD9', 'ACADL', 'ACADM', 'ACADS', 'ACADSB', 'ACADVL', 'ACAP1', 'ACAP2', 'ACAP3', 'ACAT1', 'ACAT2', 'ACBD3', 'ACBD5', 'ACBD6', 'ACBD7', 'ACD', 'ACE', 'ACIN1', 'ACLY', 'ACO1', 'ACO2', 'ACOT1', 'ACOT11', 'ACOT13', 'ACOT2', 'ACOT4', 'ACOT7', 'ACOT8', 'ACOT9', 'ACOX1', 'ACOX2', 'ACOX3', 'ACP1', 'ACP2', 'ACP5', 'ACP6', 'ACPP', 'ACSF2', 'ACSF3', 'ACSL1', 'ACSL3', 'ACSL4', 'ACSL5', 'ACSM3', 'ACSS1', 'ACSS2', 'ACSS3', 'ACTA1', 'ACTA2', 'ACTB', 'ACTBL2', 'ACTG1', 'ACTG2', 'ACTL6A', 'ACTL6B', 'ACTL8', 'ACTN1', 'ACTN2', 'ACTN3', 'ACTN4', 'ACTR10', 'ACTR1A', 'ACTR1B', 'ACTR2', 'ACTR3', 'ACTR3B', 'ACTR5', 'ACTR6', 'ACTR8', 'ACTRT2', 'ACVR1', 'ACY3', 'ACYP1', 'ACYP2', 'ADA', 'ADA2', 'ADAL', 'ADAM10', 'ADAM15', 'ADAM17', 'ADAM28', 'ADAM9', 'ADAMDEC1', 'ADAMTS1', 'ADAMTS15', 'ADAMTS2', 'ADAMTS5', 'ADAMTS8', 'ADAMTS9', 'ADAMTSL1', 'ADAMTSL2', 'ADAMTSL3', 'ADAMTSL4', 'ADAMTSL5', 'ADAP1', 'ADAP2', 'ADAR', 'ADARB1', 'ADAT1', 'ADAT2', 'ADAT3', 'ADCK1', 'ADCK5', 'ADCY3', 'ADCY5', 'ADCY6', 'ADD1', 'ADD2', 'ADD3', 'ADGRA3', 'ADGRE2', 'ADGRE5', 'ADGRF5', 'ADGRG1', 'ADGRG2', 'ADGRG3', 'ADGRL1', 'ADGRL2', 'ADGRL3', 'ADGRL4', 'ADH1A', 'ADH1B', 'ADH1C', 'ADH4', 'ADH5', 'ADH6', 'ADH7', 'ADHFE1', 'ADI1', 'ADIPOQ', 'ADIPOR1', 'ADIRF', 'ADK', 'ADM', 'ADNP', 'ADO', 'ADPGK', 'ADPRH', 'ADPRHL2', 'ADPRM', 'ADRA2A', 'ADRM1', 'ADSL', 'ADSS', 'ADSSL1', 'AEBP1', 'AEBP2', 'AES', 'AFAP1', 'AFAP1L1', 'AFAP1L2', 'AFDN', 'AFF1', 'AFF4', 'AFG1L', 'AFG3L2', 'AFM', 'AFMID', 'AFTPH', 'AGA', 'AGAP1', 'AGAP2', 'AGAP3', 'AGFG1', 'AGFG2', 'AGGF1', 'AGK', 'AGL', 'AGMAT', 'AGO1', 'AGO2', 'AGO3', 'AGO4', 'AGPAT1', 'AGPAT2', 'AGPAT3', 'AGPAT4', 'AGPAT5', 'AGPS', 'AGR2', 'AGR3', 'AGRN', 'AGT', 'AGTPBP1', 'AGTRAP', 'AHCTF1', 'AHCY', 'AHCYL1', 'AHCYL2', 'AHDC1', 'AHNAK', 'AHNAK2', 'AHR', 'AHSA1', 'AHSG', 'AHSP', 'AIDA', 'AIF1', 'AIF1L', 'AIFM1', 'AIFM2', 'AIG1', 'AIMP1', 'AIMP2', 'AIP', 'AJUBA', 'AK1', 'AK2', 'AK3', 'AK4', 'AK6', 'AK7', 'AK8', 'AKAP1', 'AKAP10', 'AKAP11', 'AKAP12', 'AKAP13', 'AKAP17A', 'AKAP5', 'AKAP7', 'AKAP8', 'AKAP8L', 'AKAP9', 'AKIP1', 'AKIRIN2', 'AKNA', 'AKR1A1', 'AKR1B1', 'AKR1B10', 'AKR1B15', 'AKR1C1', 'AKR1C2', 'AKR1C3', 'AKR1C4', 'AKR1D1', 'AKR7A2', 'AKR7A3', 'AKT1', 'AKT1S1', 'AKT2', 'AKT3', 'AKTIP', 'ALAD', 'ALAS1', 'ALB', 'ALCAM', 'ALDH16A1', 'ALDH18A1', 'ALDH1A1', 'ALDH1A2', 'ALDH1A3', 'ALDH1B1', 'ALDH1L1', 'ALDH1L2', 'ALDH2', 'ALDH3A1', 'ALDH3A2', 'ALDH3B1', 'ALDH4A1', 'ALDH5A1', 'ALDH6A1', 'ALDH7A1', 'ALDH9A1', 'ALDOA', 'ALDOB', 'ALDOC', 'ALG1', 'ALG10', 'ALG11', 'ALG12', 'ALG13', 'ALG14', 'ALG2', 'ALG5', 'ALG6', 'ALG8', 'ALG9', 'ALKBH1', 'ALKBH2', 'ALKBH3', 'ALKBH4', 'ALKBH5', 'ALKBH6', 'ALKBH7', 'ALKBH8', 'ALMS1', 'ALOX15', 'ALOX15B', 'ALOX5', 'ALOX5AP', 'ALPG', 'ALPK1', 'ALPL', 'ALPP', 'ALS2', 'ALYREF', 'AMACR', 'AMBP', 'AMBRA1', 'AMD1', 'AMDHD2', 'AMFR', 'AMMECR1', 'AMMECR1L', 'AMN1', 'AMOT', 'AMOTL1', 'AMOTL2', 'AMPD1', 'AMPD2', 'AMPD3', 'AMPH', 'AMT', 'AMY1B', 'AMZ2', 'ANAPC1', 'ANAPC10', 'ANAPC11', 'ANAPC13', 'ANAPC15', 'ANAPC16', 'ANAPC2', 'ANAPC4', 'ANAPC5', 'ANAPC7', 'ANG', 'ANGEL2', 'ANGPT2', 'ANGPT4', 'ANGPTL1', 'ANGPTL2', 'ANGPTL4', 'ANK1', 'ANK2', 'ANK3', 'ANKFY1', 'ANKHD1', 'ANKHD1-EIF4EBP3', 'ANKIB1', 'ANKLE2', 'ANKMY2', 'ANKRD10', 'ANKRD11', 'ANKRD12', 'ANKRD13A', 'ANKRD13D', 'ANKRD16', 'ANKRD17', 'ANKRD2', 'ANKRD22', 'ANKRD26', 'ANKRD27', 'ANKRD28', 'ANKRD35', 'ANKRD39', 'ANKRD40', 'ANKRD42', 'ANKRD44', 'ANKRD45', 'ANKRD46', 'ANKRD49', 'ANKRD50', 'ANKRD52', 'ANKRD54', 'ANKS1A', 'ANKS3', 'ANKS4B', 'ANKS6', 'ANKZF1', 'ANLN', 'ANO1', 'ANO10', 'ANO6', 'ANO9', 'ANOS1', 'ANP32A', 'ANP32B', 'ANP32C', 'ANP32E', 'ANPEP', 'ANTXR1', 'ANTXR2', 'ANXA1', 'ANXA11', 'ANXA13', 'ANXA2', 'ANXA3', 'ANXA4', 'ANXA5', 'ANXA6', 'ANXA7', 'ANXA8', 'ANXA8L1', 'ANXA9', 'AOAH', 'AOC1', 'AOC3', 'AOX1', 'AP1AR', 'AP1B1', 'AP1G1', 'AP1G2', 'AP1M1', 'AP1M2', 'AP1S1', 'AP1S2', 'AP1S3', 'AP2A1', 'AP2A2', 'AP2B1', 'AP2M1', 'AP2S1', 'AP3B1', 'AP3B2', 'AP3D1', 'AP3M1', 'AP3M2', 'AP3S1', 'AP3S2', 'AP4B1', 'AP4E1', 'AP4M1', 'AP4S1', 'AP5B1', 'AP5M1', 'AP5S1', 'AP5Z1', 'APAF1', 'APBA1', 'APBA2', 'APBA3', 'APBB1', 'APBB1IP', 'APBB2', 'APC', 'APCDD1', 'APCS', 'APEH', 'APEX1', 'APEX2', 'APH1A', 'API5', 'APIP', 'APLF', 'APLP2', 'APMAP', 'APOA1', 'APOA2', 'APOA4', 'APOA5', 'APOB', 'APOBEC3A', 'APOBEC3B', 'APOBEC3C', 'APOBEC3D', 'APOBEC3F', 'APOBEC3G', 'APOBR', 'APOC1', 'APOC2', 'APOC3', 'APOC4', 'APOD', 'APOE', 'APOF', 'APOH', 'APOL1', 'APOL2', 'APOL3', 'APOL6', 'APOM', 'APOO', 'APOOL', 'APOPT1', 'APP', 'APPBP2', 'APPL1', 'APPL2', 'APRT', 'APTX', 'AQP1', 'AQP5', 'AQR', 'AR', 'ARAF', 'ARAP1', 'ARAP2', 'ARAP3', 'ARCN1', 'ARF1', 'ARF3', 'ARF4', 'ARF5', 'ARF6', 'ARFGAP1', 'ARFGAP2', 'ARFGAP3', 'ARFGEF1', 'ARFGEF2', 'ARFGEF3', 'ARFIP1', 'ARFIP2', 'ARFRP1', 'ARG1', 'ARG2', 'ARGLU1', 'ARHGAP1', 'ARHGAP10', 'ARHGAP12', 'ARHGAP15', 'ARHGAP17', 'ARHGAP18', 'ARHGAP21', 'ARHGAP23', 'ARHGAP24', 'ARHGAP25', 'ARHGAP26', 'ARHGAP27', 'ARHGAP28', 'ARHGAP29', 'ARHGAP30', 'ARHGAP31']\n",
      "600\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A1BG_proteomics</th>\n",
       "      <th>A2M_proteomics</th>\n",
       "      <th>A2ML1_proteomics</th>\n",
       "      <th>A4GALT_proteomics</th>\n",
       "      <th>AAAS_proteomics</th>\n",
       "      <th>AACS_proteomics</th>\n",
       "      <th>AADAT_proteomics</th>\n",
       "      <th>AAED1_proteomics</th>\n",
       "      <th>AAGAB_proteomics</th>\n",
       "      <th>AAK1_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>ARHGAP26_proteomics</th>\n",
       "      <th>ARHGAP27_proteomics</th>\n",
       "      <th>ARHGAP28_proteomics</th>\n",
       "      <th>ARHGAP29_proteomics</th>\n",
       "      <th>ARHGAP30_proteomics</th>\n",
       "      <th>ARHGAP31_proteomics</th>\n",
       "      <th>ZFHX3_Mutation</th>\n",
       "      <th>ZFHX3_Location</th>\n",
       "      <th>ZFHX3_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001</th>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.802</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>0.412</td>\n",
       "      <td>-0.664</td>\n",
       "      <td>...</td>\n",
       "      <td>0.46400</td>\n",
       "      <td>-0.1390</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.182</td>\n",
       "      <td>-0.326</td>\n",
       "      <td>-0.595</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S002</th>\n",
       "      <td>-0.685</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.684</td>\n",
       "      <td>0.984</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.3340</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>0.1390</td>\n",
       "      <td>1.330</td>\n",
       "      <td>-0.367</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.06410</td>\n",
       "      <td>0.7010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.557</td>\n",
       "      <td>-1.070</td>\n",
       "      <td>-0.699</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S003</th>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.419</td>\n",
       "      <td>-0.500</td>\n",
       "      <td>...</td>\n",
       "      <td>0.79300</td>\n",
       "      <td>0.0586</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.237</td>\n",
       "      <td>0.475</td>\n",
       "      <td>-0.396</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S005</th>\n",
       "      <td>-1.670</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.443</td>\n",
       "      <td>0.243</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>-0.9290</td>\n",
       "      <td>0.229</td>\n",
       "      <td>-0.223</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.00251</td>\n",
       "      <td>-0.0359</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.238</td>\n",
       "      <td>-0.299</td>\n",
       "      <td>-0.742</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S006</th>\n",
       "      <td>-0.374</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.537</td>\n",
       "      <td>0.311</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.565</td>\n",
       "      <td>-0.101</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.66700</td>\n",
       "      <td>0.1900</td>\n",
       "      <td>-0.616</td>\n",
       "      <td>0.856</td>\n",
       "      <td>-0.121</td>\n",
       "      <td>0.321</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  604 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           A1BG_proteomics  A2M_proteomics  A2ML1_proteomics  \\\n",
       "Sample_ID                                                      \n",
       "S001                -1.180         -0.8630            -0.802   \n",
       "S002                -0.685         -1.0700            -0.684   \n",
       "S003                -0.528         -1.3200             0.435   \n",
       "S005                -1.670         -1.1900            -0.443   \n",
       "S006                -0.374         -0.0206            -0.537   \n",
       "\n",
       "           A4GALT_proteomics  AAAS_proteomics  AACS_proteomics  \\\n",
       "Sample_ID                                                        \n",
       "S001                   0.222           0.2560           0.6650   \n",
       "S002                   0.984           0.1350           0.3340   \n",
       "S003                     NaN          -0.2400           1.0400   \n",
       "S005                   0.243          -0.0993           0.7570   \n",
       "S006                   0.311           0.3750           0.0131   \n",
       "\n",
       "           AADAT_proteomics  AAED1_proteomics  AAGAB_proteomics  \\\n",
       "Sample_ID                                                         \n",
       "S001                 1.2800           -0.3390             0.412   \n",
       "S002                 1.3000            0.1390             1.330   \n",
       "S003                -0.0213           -0.0479             0.419   \n",
       "S005                 0.7400           -0.9290             0.229   \n",
       "S006                -1.1000               NaN             0.565   \n",
       "\n",
       "           AAK1_proteomics  ...  ARHGAP26_proteomics  ARHGAP27_proteomics  \\\n",
       "Sample_ID                   ...                                             \n",
       "S001                -0.664  ...              0.46400              -0.1390   \n",
       "S002                -0.367  ...             -0.06410               0.7010   \n",
       "S003                -0.500  ...              0.79300               0.0586   \n",
       "S005                -0.223  ...             -0.00251              -0.0359   \n",
       "S006                -0.101  ...             -0.66700               0.1900   \n",
       "\n",
       "           ARHGAP28_proteomics  ARHGAP29_proteomics  ARHGAP30_proteomics  \\\n",
       "Sample_ID                                                                  \n",
       "S001                       NaN               -0.182               -0.326   \n",
       "S002                       NaN                0.557               -1.070   \n",
       "S003                       NaN               -0.237                0.475   \n",
       "S005                       NaN               -0.238               -0.299   \n",
       "S006                    -0.616                0.856               -0.121   \n",
       "\n",
       "           ARHGAP31_proteomics    ZFHX3_Mutation  ZFHX3_Location  \\\n",
       "Sample_ID                                                          \n",
       "S001                    -0.595  [Wildtype_Tumor]   [No_mutation]   \n",
       "S002                    -0.699  [Wildtype_Tumor]   [No_mutation]   \n",
       "S003                    -0.396  [Wildtype_Tumor]   [No_mutation]   \n",
       "S005                    -0.742  [Wildtype_Tumor]   [No_mutation]   \n",
       "S006                     0.321  [Wildtype_Tumor]   [No_mutation]   \n",
       "\n",
       "           ZFHX3_Mutation_Status  Sample_Status  \n",
       "Sample_ID                                        \n",
       "S001              Wildtype_Tumor          Tumor  \n",
       "S002              Wildtype_Tumor          Tumor  \n",
       "S003              Wildtype_Tumor          Tumor  \n",
       "S005              Wildtype_Tumor          Tumor  \n",
       "S006              Wildtype_Tumor          Tumor  \n",
       "\n",
       "[5 rows x 604 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteomics = en.get_proteomics()\n",
    "proteins = proteomics.columns\n",
    "proteins = proteins[:600]\n",
    "\n",
    "print(proteins)\n",
    "print(len(proteins))\n",
    "\n",
    "interacting_proteins_in_omics_df = []\n",
    "\n",
    "'''Only do comparisons on proteins in the omics dataframe'''\n",
    "for ip in proteins:\n",
    "    if omics.name == 'phosphoproteomics' or omics.name == 'acetylproteomics':\n",
    "        col_regex = ip + \"-.*\" # Build a regex to get all columns that match the gene\n",
    "    else:\n",
    "        col_regex = '^{}$'.format(ip)\n",
    "\n",
    "    selected = omics.filter(regex=col_regex)\n",
    "\n",
    "    if len(selected.columns) > 0:\n",
    "        interacting_proteins_in_omics_df.append(ip)\n",
    "\n",
    "print(interacting_proteins_in_omics_df)\n",
    "print(len(interacting_proteins_in_omics_df))\n",
    "\n",
    "protdf = en.append_mutations_to_omics(mutation_genes=[gene], omics_df=omics, omics_genes=interacting_proteins_in_omics_df)\n",
    "protdf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
