{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting cptac\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/42/ee/e2652fefdd7d4bd52f3623f4cae4e171bddf092476312bb66d9eadb17cea/cptac-0.6.5-py3-none-any.whl (4.9MB)\n",
      "\u001b[K     |████████████████████████████████| 4.9MB 986kB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied, skipping upgrade: scipy>=1.2.1 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (1.3.1)\n",
      "Requirement already satisfied, skipping upgrade: numpy>=1.16.3 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (1.16.4)\n",
      "Requirement already satisfied, skipping upgrade: beautifulsoup4>=4.7.1 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (4.8.0)\n",
      "Requirement already satisfied, skipping upgrade: requests>=2.21.0 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (2.22.0)\n",
      "Requirement already satisfied, skipping upgrade: pandas==0.25.* in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (0.25.1)\n",
      "Requirement already satisfied, skipping upgrade: urllib3>=1.24.2 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (1.24.2)\n",
      "Requirement already satisfied, skipping upgrade: soupsieve>=1.2 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from beautifulsoup4>=4.7.1->cptac) (1.9.2)\n",
      "Requirement already satisfied, skipping upgrade: idna<2.9,>=2.5 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from requests>=2.21.0->cptac) (2.8)\n",
      "Requirement already satisfied, skipping upgrade: certifi>=2017.4.17 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from requests>=2.21.0->cptac) (2019.6.16)\n",
      "Requirement already satisfied, skipping upgrade: chardet<3.1.0,>=3.0.2 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from requests>=2.21.0->cptac) (3.0.4)\n",
      "Requirement already satisfied, skipping upgrade: python-dateutil>=2.6.1 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from pandas==0.25.*->cptac) (2.8.0)\n",
      "Requirement already satisfied, skipping upgrade: pytz>=2017.2 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from pandas==0.25.*->cptac) (2019.2)\n",
      "Requirement already satisfied, skipping upgrade: six>=1.5 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from python-dateutil>=2.6.1->pandas==0.25.*->cptac) (1.12.0)\n",
      "Installing collected packages: cptac\n",
      "  Found existing installation: cptac 0.6.4\n",
      "    Uninstalling cptac-0.6.4:\n",
      "      Successfully uninstalled cptac-0.6.4\n",
      "Successfully installed cptac-0.6.5\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install --upgrade cptac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import sys\n",
    "import operator\n",
    "import collections\n",
    "import os\n",
    "import cptac\n",
    "from datetime import date\n",
    "#from xml.etree import ElementTree\n",
    "from xml.dom import minidom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fixNewLines(s):\n",
    "    if \"\\n\" in s:\n",
    "        nList = s.split(\"\\n\")\n",
    "    else:\n",
    "        nList = []\n",
    "        nList.append(s)\n",
    "    return nList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fixTabs(s):\n",
    "    if \"\\t\" in s:\n",
    "        tList = s.split(\"\\t\")\n",
    "    else:\n",
    "        tList = []\n",
    "        tList.append(s)\n",
    "    return tList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fixParen(s):\n",
    "    if s.startswith(\"(\") and s.endswith(\")\"):\n",
    "        s = s[1:(len(s) - 1)]\n",
    "    return s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [],
   "source": [
    "#function to split each item with a newline into two items\n",
    "\n",
    "#fix if genes are in parenthesis\n",
    "\n",
    "def fixParsingErrors(interactingList):\n",
    "    fixedList = []\n",
    "    for item in interactingList:\n",
    "        item = item.strip()\n",
    "        newlineFreeList = fixNewLines(item)\n",
    "        for n in newlineFreeList:\n",
    "            tabFreeList = fixTabs(n)\n",
    "            for t in tabFreeList:\n",
    "                s = fixParen(t)\n",
    "                s = s.strip()\n",
    "                if not s.isspace():\n",
    "                    fixedList.append(s)\n",
    "    return list(set(fixedList))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def getUniprotGeneList():\n",
    "    filePath = '/Users/coribushman/github/cptac/cptac/utils/Uniprot_Proteome.tsv'\n",
    "    bioplex_interactions = pd.read_csv(filePath, sep='\\t')\n",
    "    uniprotList = []\n",
    "    for ind, row in bioplex_interactions.iterrows():\n",
    "        prots = str(row['Gene names'])\n",
    "        #print(prots)\n",
    "        prot_arr = prots.split(' ')\n",
    "        #print(prot_arr)\n",
    "        for prot in prot_arr:\n",
    "            if prot not in uniprotList:\n",
    "                uniprotList.append(prot)\n",
    "    return uniprotList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [],
   "source": [
    "def intersectWithUniprot(interactingGenes):\n",
    "    #get uniprot genes from file\n",
    "    uniprotGenes = getUniprotGeneList()\n",
    "    \n",
    "    #intersect\n",
    "    geneSet = set(interactingGenes)\n",
    "    uniprotSet = set(uniprotGenes)   #convert to set\n",
    "    intersectSet = geneSet.intersection(uniprotSet)\n",
    "    \n",
    "    today = date.today()\n",
    "    todayDate = today.strftime(\"%m/%d/%y\")\n",
    "    \n",
    "    print(\"WikiPathways Interacting Proteins as of \" + str(todayDate))\n",
    "    return intersectSet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getGenesPerPathwayDict():\n",
    "    path = '/Users/coribushman/Desktop/wikipathways-20191010-gpml-Homo_sapiens'\n",
    "\n",
    "    pathwaysPerGene = {}\n",
    "    genesPerPathway = {}\n",
    "\n",
    "    for fileName in os.listdir(path):\n",
    "        if fileName == \".DS_Store\":\n",
    "            continue\n",
    "        filePath = path + \"/\" + fileName\n",
    "        pathwayGenes = []\n",
    "\n",
    "        xmldoc = minidom.parse(filePath)\n",
    "        pathway = xmldoc.getElementsByTagName(\"Pathway\")[0]\n",
    "        pathwayName = pathway.getAttribute(\"Name\")\n",
    "\n",
    "        dataNodes = pathway.getElementsByTagName(\"DataNode\")\n",
    "        for node in dataNodes:\n",
    "            if node.getAttribute(\"Type\") == \"GeneProduct\":\n",
    "                geneName = node.getAttribute(\"TextLabel\")\n",
    "                pathwayGenes.append(geneName)\n",
    "\n",
    "        genesPerPathway[pathwayName] = pathwayGenes\n",
    "    return genesPerPathway"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getAllPathways(genesPerPathway):\n",
    "    allPathways = genesPerPathway.keys()\n",
    "    allPathways = list(allPathways)\n",
    "    return allPathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getAllGenes(genesPerPathway):\n",
    "    allGenes = []\n",
    "    for geneList in genesPerPathway.values():\n",
    "        allGenes.extend(geneList)\n",
    "\n",
    "    allGenes = fixParsingErrors(allGenes)\n",
    "    return allGenes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Target Of Rapamycin (TOR) Signaling', 'NRF2-ARE regulation',\n",
       "       'PDGF Pathway', 'Eukaryotic Transcription Initiation',\n",
       "       'Interleukin-11 Signaling Pathway', 'Vitamin D Receptor Pathway',\n",
       "       'T-Cell antigen Receptor (TCR) pathway during Staphylococcus aureus infection',\n",
       "       'Phosphatidylcholine catabolism',\n",
       "       'SRF and miRs in Smooth Muscle Differentiation and Proliferation',\n",
       "       'Glucocorticoid and Mineralcorticoid Metabolism',\n",
       "       ...\n",
       "       'Serotonin Transporter Activity',\n",
       "       'TCA Cycle (aka Krebs or citric acid cycle)', 'Serotonin and anxiety',\n",
       "       'Sudden Infant Death Syndrome (SIDS) Susceptibility Pathways',\n",
       "       'Canonical NF-KB pathway',\n",
       "       'Photodynamic therapy-induced HIF-1 survival signaling',\n",
       "       'Arrhythmogenic Right Ventricular Cardiomyopathy',\n",
       "       'Ethanol metabolism resulting in production of ROS by CYP2E1',\n",
       "       'Interactions between immune cells and microRNAs in tumor microenvironment',\n",
       "       'Metabolism of Tetrahydrocannabinol (THC)'],\n",
       "      dtype='object', length=537)"
      ]
     },
     "execution_count": 227,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genesPerPathway = getGenesPerPathwayDict()\n",
    "allPathways = getAllPathways(genesPerPathway)\n",
    "allGenes = getAllGenes(genesPerPathway)\n",
    "\n",
    "#df = pd.DataFrame(genesPerPathway, columns = allPathways, index = allGenes)\n",
    "df = pd.DataFrame(columns = allPathways, index = allGenes)\n",
    "df = df.fillna(False)\n",
    "\n",
    "for column in df.colu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['a', '']\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "OLD FUNCTIONS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WikiPathways Interacting Proteins as of 10/25/19\n",
      "{'NOXA1', 'SIRT1', 'CHK1', 'PTEN', 'ATR', 'EP300', 'IL2', 'MDM2', 'CCND1', 'CDKN1A', 'HIPK2', 'DCAKD', 'PRKDC', 'SFN', 'PIAS1', 'RELA', 'SIAH1', 'GADD45A', 'HDAC1', 'ATM', 'CHK2', 'CBP', 'BRCA1', 'MDM4', 'CSNK1D', 'IL6', 'PML', 'AURKA', 'FADD', 'PUMA'}\n"
     ]
    }
   ],
   "source": [
    "wikiPathwaysInterGenes = findInteractions(\"p53\")\n",
    "filteredInterGenes = fixParsingErrors(wikiPathwaysInterGenes)\n",
    "finalGeneSet = intersectWithUniprot(filteredInterGenes)\n",
    "print(finalGeneSet)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#size=-1 to get all genes, but takes a long time\n",
    "#if statements might need to be added in case keys not in dict when parsing\n",
    "\n",
    "#this function will query wikipathways for all interactions of a given gene for the Homo Sapiens species\n",
    "#some interactions won't be genes\n",
    "\n",
    "import urllib3\n",
    "import json\n",
    "\n",
    "def findInteractions(geneName): #retuns list of interacting elements\n",
    "    interactionList = []\n",
    "    \n",
    "    try:\n",
    "        requestURL = \"http://webservice.wikipathways.org/findInteractions?query=\" + geneName + \"&format=json\" \n",
    "        http = urllib3.PoolManager()\n",
    "        response = http.request('GET', requestURL)\n",
    "\n",
    "        if response.status == 200: #if success, get data from API response\n",
    "            responseData = response.data\n",
    "            #print(type(responseData))\n",
    "            responseDataDict = json.loads(responseData)\n",
    "            #print(responseDataDict)\n",
    "            resultList = list(responseDataDict.values())\n",
    "            interactionsList = resultList[0]\n",
    "\n",
    "            for interactionDict in interactionsList:\n",
    "\n",
    "                if interactionDict[\"species\"] == \"Homo sapiens\":\n",
    "                    rightValuesList = interactionDict[\"fields\"][\"right\"][\"values\"]\n",
    "                    for value in rightValuesList:\n",
    "                        if (value not in interactionList) and (geneName.casefold() not in value.casefold()):\n",
    "                            interactionList.append(value)\n",
    "\n",
    "                    leftValuesList = interactionDict[\"fields\"][\"left\"][\"values\"]\n",
    "                    for value in leftValuesList:\n",
    "                        if (value not in interactionList) and (geneName.casefold() not in value.casefold()):\n",
    "                            interactionList.append(value)\n",
    "                else:\n",
    "                    continue       \n",
    "        else:\n",
    "            print(\"Something went wrong with the response status\")\n",
    "\n",
    "        return interactionList\n",
    "    except:\n",
    "        print(\"There was a problem getting a response from the WikiPathways API\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[]\n",
      "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n",
      "<Pathway xmlns=\"http://pathvisio.org/GPML/2013a\" Name=\"4-hydroxytamoxifen, Dexamethasone, and Retinoic Acids Regulation of p27 Expression\" Version=\"20160816\" Organism=\"Homo sapiens\">\n",
      "  <Comment Source=\"WikiPathways-description\">This pathway is modeled after Figure 7 in the article ''Upstream molecular signaling pathways of p27(Kip1) expression: effects of 4-hydroxytamoxifen, dexamethasone, and retinoic acids'' (See Bibliography). Several anticancer agents up-regulate expression of p27, a cyclin-dpendent kinase inhibitor that blocks the transition from G1 to S phase. The pathways used by the anticancer agents (retinoic acids, 4-hydroxytamoxifen, dexamethasone) are shown above from human breast cancer cells.</Comment>\n",
      "  <BiopaxRef>f5c</BiopaxRef>\n",
      "  <Graphics BoardWidth=\"1659.0\" BoardHeight=\"820.0\"/>\n",
      "  <DataNode TextLabel=\"Triciribine\" GraphId=\"a188c\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"311.0\" CenterY=\"262.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"65399\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MNK1\" GraphId=\"a2a00\" Type=\"GeneProduct\" GroupRef=\"bc75a\">\n",
      "    <Graphics CenterX=\"1138.0\" CenterY=\"517.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000079277\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"eIF4E\" GraphId=\"a641b\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"1138.0\" CenterY=\"604.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000151247\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"AMPK\" GraphId=\"ad286\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"464.0\" CenterY=\"237.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Uniprot-TrEMBL\" ID=\"Q13131\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"LY 294, 002\" GraphId=\"ae1da\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"303.0\" CenterY=\"183.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"3973\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MKK6\" GraphId=\"b2ed3\" Type=\"GeneProduct\" GroupRef=\"f6aca\">\n",
      "    <Graphics CenterX=\"1261.75\" CenterY=\"294.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000108984\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Raf\" GraphId=\"b5118\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"204.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000132155\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"4-Hydroxytamoxifen\" GraphId=\"bd676\" Type=\"Metabolite\" GroupRef=\"c74ea\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"88.0\" Width=\"104.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"449459\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"SB202190\" GraphId=\"bda16\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1339.75\" CenterY=\"463.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"5353940\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Rotenone\" GraphId=\"c3556\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"583.0\" CenterY=\"140.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"6758\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Mlk3\" GraphId=\"cbc42\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"1262.0\" CenterY=\"199.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000173327\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MEK2\" GraphId=\"ccec7\" Type=\"GeneProduct\" GroupRef=\"c8a45\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"294.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000126934\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"AKT\" GraphId=\"cd60e\" Type=\"GeneProduct\" GroupRef=\"aac1e\">\n",
      "    <Graphics CenterX=\"186.0\" CenterY=\"292.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000142208\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"p38MAPK\" GraphId=\"ce646\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"1261.0\" CenterY=\"527.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000197442\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Dexamethasone\" GraphId=\"d037b\" Type=\"Metabolite\" GroupRef=\"c74ea\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"108.0\" Width=\"104.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"5743\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"PD98059\" GraphId=\"d6571\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1060.0\" CenterY=\"356.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"4713\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"EIF4EBP1\" GraphId=\"db08f\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"973.75\" CenterY=\"415.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000187840\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"AICA Riboside Compound C\" GraphId=\"dc985\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"621.0\" CenterY=\"211.0\" Width=\"150.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"17513\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Cap-Independent Translation Initiation of p27 mRNA Through its 5'UTR\" GraphId=\"de848\" Type=\"Pathway\">\n",
      "    <Comment>Identifier not found</Comment>\n",
      "    <Graphics CenterX=\"1354.5\" CenterY=\"689.0\" Width=\"463.0\" Height=\"20.0\" ZOrder=\"32768\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\" ShapeType=\"None\" Color=\"14961e\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MEK1\" GraphId=\"e049b\" Type=\"GeneProduct\" GroupRef=\"c8a45\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"274.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000169032\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"ERK Inhibitor I\" GraphId=\"e3241\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1007.75\" CenterY=\"501.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"3377241\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"PBK\" GraphId=\"e775d\" Type=\"GeneProduct\" GroupRef=\"aac1e\">\n",
      "    <Graphics CenterX=\"186.0\" CenterY=\"312.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000168078\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MNK2\" GraphId=\"ec796\" Type=\"GeneProduct\" GroupRef=\"bc75a\">\n",
      "    <Graphics CenterX=\"1138.0\" CenterY=\"537.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000099875\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"ERK2\" GraphId=\"eca2b\" Type=\"GeneProduct\" GroupRef=\"cd15c\">\n",
      "    <Graphics CenterX=\"1139.0\" CenterY=\"425.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000100030\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"TSC\" GraphId=\"ed6bc\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"463.0\" CenterY=\"365.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000165699\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"ERK1\" GraphId=\"ede91\" Type=\"GeneProduct\" GroupRef=\"cd15c\">\n",
      "    <Graphics CenterX=\"1139.0\" CenterY=\"405.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000102882\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Retinoic Acids?\" GraphId=\"eec0d\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1461.0\" CenterY=\"482.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"HMDB\" ID=\"HMDB0001852\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MTOR Protein Kinase Signaling Pathway\" GraphId=\"f06e5\" Type=\"Pathway\">\n",
      "    <Graphics CenterX=\"973.0\" CenterY=\"202.0\" Width=\"282.0\" Height=\"20.0\" ZOrder=\"32768\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\" ShapeType=\"None\" Color=\"14961e\"/>\n",
      "    <Xref Database=\"WikiPathways\" ID=\"WP3318\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"NSC119889\" GraphId=\"f4490\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1573.0\" CenterY=\"482.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"274104\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MKK3\" GraphId=\"f99be\" Type=\"GeneProduct\" GroupRef=\"f6aca\">\n",
      "    <Graphics CenterX=\"1261.75\" CenterY=\"274.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000034152\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"mTOR\" GraphId=\"fd553\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"463.0\" CenterY=\"427.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000198793\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"4-Hydroxytamoxifen\" GraphId=\"e28f9\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"226.0\" CenterY=\"86.0\" Width=\"104.0\" Height=\"20.0\" ZOrder=\"32769\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"449459\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Dexamethasone\" GraphId=\"b2270\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"464.0\" CenterY=\"96.0\" Width=\"104.0\" Height=\"20.0\" ZOrder=\"32770\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"5743\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"EIF4EBP1\" GraphId=\"a416b\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"463.75\" CenterY=\"524.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32773\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000187840\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Cap-Independent Translation Initiation of p27 mRNA Through its 5'UTR\" GraphId=\"bc30f\" Type=\"Pathway\">\n",
      "    <Comment>Identifier not found</Comment>\n",
      "    <Graphics CenterX=\"462.5\" CenterY=\"637.0\" Width=\"463.0\" Height=\"20.0\" ZOrder=\"32775\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\" ShapeType=\"None\" Color=\"14961e\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </DataNode>\n",
      "  <State GraphRef=\"ad286\" TextLabel=\"P\" GraphId=\"af1a5\">\n",
      "    <Graphics RelX=\"0.821570182394926\" RelY=\"-1.0\" Width=\"15.0\" Height=\"15.0\" ShapeType=\"Oval\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </State>\n",
      "  <State GraphRef=\"cd60e\" TextLabel=\"P\" GraphId=\"cc511\">\n",
      "    <Graphics RelX=\"-0.925\" RelY=\"-0.9\" Width=\"15.0\" Height=\"15.0\" ShapeType=\"Oval\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </State>\n",
      "  <State GraphRef=\"fd553\" TextLabel=\"P\" GraphId=\"d450e\">\n",
      "    <Graphics RelX=\"1.0\" RelY=\"1.0\" Width=\"15.0\" Height=\"15.0\" ShapeType=\"Oval\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </State>\n",
      "  <State GraphRef=\"a416b\" TextLabel=\"P\" GraphId=\"d72f8\">\n",
      "    <Graphics RelX=\"1.0\" RelY=\"1.0\" Width=\"15.0\" Height=\"15.0\" ShapeType=\"Oval\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </State>\n",
      "  <Interaction GraphId=\"id128f6b9b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"263.0\" Y=\"183.0\" GraphRef=\"ae1da\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"208.5\" Y=\"217.5\" GraphRef=\"d48c9\" RelX=\"0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id14955cbf\">\n",
      "    <Graphics ZOrder=\"12288\" LineStyle=\"Broken\" LineThickness=\"1.0\">\n",
      "      <Point X=\"226.0\" Y=\"96.0\" GraphRef=\"e28f9\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"186.0\" Y=\"137.5\" GraphRef=\"f55ba\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id1573b86d\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1007.75\" Y=\"491.25\" GraphRef=\"e3241\" RelX=\"0.0\" RelY=\"-1.0\"/>\n",
      "      <Point X=\"1087.0\" Y=\"431.0\" GraphRef=\"e78c4\" RelX=\"-1.0\" RelY=\"0.5\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id1de6f372\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1261.75\" Y=\"316.25\" GraphRef=\"a2b86\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1261.0\" Y=\"517.0\" GraphRef=\"ce646\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id1e3811ea\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1140.0\" Y=\"214.0\" GraphRef=\"b5118\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1140.0\" Y=\"252.0\" GraphRef=\"dcd72\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id2357bcc4\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1138.0\" Y=\"614.0\" GraphRef=\"a641b\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1238.75\" Y=\"679.0\" GraphRef=\"de848\" RelX=\"-0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id338fd47b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1060.0\" Y=\"346.0\" GraphRef=\"d6571\" RelX=\"0.0\" RelY=\"-1.0\"/>\n",
      "      <Point X=\"1088.0\" Y=\"284.0\" GraphRef=\"dcd72\" RelX=\"-1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id33efe69e\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1339.75\" Y=\"473.25\" GraphRef=\"bda16\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1281.0\" Y=\"517.0\" GraphRef=\"ce646\" RelX=\"0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id37d1ac01\">\n",
      "    <Graphics ConnectorType=\"Elbow\" ZOrder=\"12288\" LineStyle=\"Broken\" LineThickness=\"1.0\">\n",
      "      <Point X=\"464.0\" Y=\"106.0\" GraphRef=\"b2270\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"424.0\" Y=\"237.0\" GraphRef=\"ad286\" RelX=\"-1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id3b3d47e0\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"463.0\" Y=\"375.0\" GraphRef=\"ed6bc\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"463.0\" Y=\"417.0\" GraphRef=\"fd553\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id3ecb520d\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1140.0\" Y=\"126.0\" GraphRef=\"d80b2\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1140.0\" Y=\"194.0\" GraphRef=\"b5118\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id40ee33da\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"464.5\" Y=\"174.0\" GraphRef=\"eae0b\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"464.0\" Y=\"227.0\" GraphRef=\"ad286\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id41d4985b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"271.0\" Y=\"262.0\" GraphRef=\"a188c\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"238.0\" Y=\"286.0\" GraphRef=\"b5fa0\" RelX=\"1.0\" RelY=\"-0.5\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id4259e563\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1140.0\" Y=\"316.0\" GraphRef=\"dcd72\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1139.0\" Y=\"383.0\" GraphRef=\"e78c4\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id56c56f1b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"463.75\" Y=\"534.25\" GraphRef=\"a416b\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"462.5\" Y=\"627.0\" GraphRef=\"bc30f\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id5d38e51e\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1139.0\" Y=\"447.0\" GraphRef=\"e78c4\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1138.0\" Y=\"495.0\" GraphRef=\"a2eb1\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id61467640\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1013.75\" Y=\"415.25\" GraphRef=\"db08f\" RelX=\"1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"1087.0\" Y=\"415.0\" GraphRef=\"e78c4\" RelX=\"-1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id67ad93e4\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"464.0\" Y=\"247.0\" GraphRef=\"ad286\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"463.0\" Y=\"355.0\" GraphRef=\"ed6bc\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id74d16a08\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1221.0\" Y=\"527.0\" GraphRef=\"ce646\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"1190.0\" Y=\"527.0\" GraphRef=\"a2eb1\" RelX=\"1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id7a969ee4\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"973.0\" Y=\"212.0\" GraphRef=\"f06e5\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"973.75\" Y=\"405.25\" GraphRef=\"db08f\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id7e2249f0\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"186.0\" Y=\"242.5\" GraphRef=\"d48c9\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"186.0\" Y=\"270.0\" GraphRef=\"b5fa0\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id810511fc\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"546.0\" Y=\"211.0\" GraphRef=\"dc985\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"504.0\" Y=\"237.0\" GraphRef=\"ad286\" RelX=\"1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idac4a0ace\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1110.0\" Y=\"126.0\" GraphRef=\"d80b2\" RelX=\"-0.5\" RelY=\"1.0\"/>\n",
      "      <Point X=\"973.0\" Y=\"192.0\" GraphRef=\"f06e5\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idb363dd12\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1461.0\" Y=\"492.0\" GraphRef=\"eec0d\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1461.75\" Y=\"550.0\" GraphRef=\"a697f\" RelX=\"-0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idbe383038\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"186.0\" Y=\"162.5\" GraphRef=\"f55ba\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"186.0\" Y=\"217.5\" GraphRef=\"d48c9\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idbf97bab3\">\n",
      "    <Graphics ZOrder=\"12288\" LineStyle=\"Broken\" LineThickness=\"1.0\">\n",
      "      <Point X=\"464.0\" Y=\"106.0\" GraphRef=\"b2270\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"464.5\" Y=\"154.0\" GraphRef=\"eae0b\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idc34f3c45\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1354.5\" Y=\"699.0\" GraphRef=\"de848\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1354.5\" Y=\"770.0\" GraphRef=\"f3f7d\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idc70ddd2d\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"423.0\" Y=\"365.0\" GraphRef=\"ed6bc\" RelX=\"-1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "      <Point X=\"238.0\" Y=\"302.0\" GraphRef=\"b5fa0\" RelX=\"1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idca79bbc8\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"462.5\" Y=\"647.0\" GraphRef=\"bc30f\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"462.5\" Y=\"701.0\" GraphRef=\"d7ec4\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idd3085a06\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1170.0\" Y=\"126.0\" GraphRef=\"d80b2\" RelX=\"0.5\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1262.0\" Y=\"189.0\" GraphRef=\"cbc42\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idd4e700d8\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1262.0\" Y=\"209.0\" GraphRef=\"cbc42\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1261.75\" Y=\"264.25\" GraphRef=\"f99be\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"ide08a3a0b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1573.0\" Y=\"492.0\" GraphRef=\"f4490\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1573.25\" Y=\"550.0\" GraphRef=\"a697f\" RelX=\"0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"ided308b41\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1138.0\" Y=\"559.0\" GraphRef=\"a2eb1\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1138.0\" Y=\"594.0\" GraphRef=\"a641b\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idef415bbb\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"543.0\" Y=\"140.0\" GraphRef=\"c3556\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"511.0\" Y=\"164.0\" GraphRef=\"eae0b\" RelX=\"1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idf9929b96\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"463.0\" Y=\"437.0\" GraphRef=\"fd553\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"463.75\" Y=\"514.25\" GraphRef=\"a416b\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idfda40a6c\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1517.5\" Y=\"579.0\" GraphRef=\"a697f\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1470.25\" Y=\"679.0\" GraphRef=\"de848\" RelX=\"0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Label TextLabel=\"A\" GraphId=\"a5592\">\n",
      "    <Graphics CenterX=\"110.0\" CenterY=\"88.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"Hypomethylation of 5' mRNA cap\" GraphId=\"a697f\">\n",
      "    <Graphics CenterX=\"1517.5\" CenterY=\"564.5\" Width=\"223.0\" Height=\"29.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"PI3K\" GraphId=\"d48c9\">\n",
      "    <Graphics CenterX=\"186.0\" CenterY=\"230.0\" Width=\"90.0\" Height=\"25.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"AMP/ATP Ratio\" GraphId=\"eae0b\">\n",
      "    <Graphics CenterX=\"464.5\" CenterY=\"164.0\" Width=\"93.0\" Height=\"20.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"Prevention of Cancer\" GraphId=\"f3f7d\">\n",
      "    <Graphics CenterX=\"1354.5\" CenterY=\"780.0\" Width=\"167.0\" Height=\"20.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"Prevention of Cancer\" GraphId=\"d7ec4\">\n",
      "    <Graphics CenterX=\"462.5\" CenterY=\"711.0\" Width=\"167.0\" Height=\"20.0\" ZOrder=\"32776\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"B\" GraphId=\"f0fce\">\n",
      "    <Graphics CenterX=\"877.0\" CenterY=\"107.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32777\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"Receptor Tyrosine Kinases\" GraphId=\"f55ba\">\n",
      "    <Graphics CenterX=\"186.0\" CenterY=\"150.0\" Width=\"162.0\" Height=\"25.0\" ZOrder=\"32778\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Group GroupId=\"f6aca\" GraphId=\"a2b86\" Style=\"Complex\"/>\n",
      "  <Group GroupId=\"bc75a\" GraphId=\"a2eb1\" Style=\"Complex\"/>\n",
      "  <Group GroupId=\"aac1e\" GraphId=\"b5fa0\" Style=\"Complex\"/>\n",
      "  <Group GroupId=\"c74ea\" GraphId=\"d80b2\" Style=\"Group\"/>\n",
      "  <Group GroupId=\"c8a45\" GraphId=\"dcd72\" Style=\"Complex\"/>\n",
      "  <Group GroupId=\"cd15c\" GraphId=\"e78c4\" Style=\"Complex\"/>\n",
      "  <InfoBox CenterX=\"0.0\" CenterY=\"0.0\"/>\n",
      "  <Biopax>\n",
      "    <bp:openControlledVocabulary xmlns:bp=\"http://www.biopax.org/release/biopax-level3.owl#\">\n",
      "      <bp:TERM xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\" rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">breast cancer pathway</bp:TERM>\n",
      "      <bp:ID xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\" rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">PW:0000624</bp:ID>\n",
      "      <bp:Ontology xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\" rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">Pathway Ontology</bp:Ontology>\n",
      "    </bp:openControlledVocabulary>\n",
      "    <bp:PublicationXref xmlns:bp=\"http://www.biopax.org/release/biopax-level3.owl#\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\" rdf:id=\"f5c\">\n",
      "      <bp:ID rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">20170512</bp:ID>\n",
      "      <bp:DB rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">PubMed</bp:DB>\n",
      "      <bp:TITLE rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">Upstream molecular signaling pathways of p27(Kip1) expression: effects of 4-hydroxytamoxifen, dexamethasone, and retinoic acids.</bp:TITLE>\n",
      "      <bp:SOURCE rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">Cancer Cell Int</bp:SOURCE>\n",
      "      <bp:YEAR rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">2010</bp:YEAR>\n",
      "      <bp:AUTHORS rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">Eto I</bp:AUTHORS>\n",
      "    </bp:PublicationXref>\n",
      "  <bp:openControlledVocabulary xmlns:bp=\"http://www.biopax.org/release/biopax-level3.owl#\"><bp:TERM>breast cancer</bp:TERM><bp:ID>DOID:1612</bp:ID><bp:Ontology>Disease</bp:Ontology></bp:openControlledVocabulary></Biopax>\n",
      "</Pathway>\n"
     ]
    },
    {
     "ename": "ParseError",
     "evalue": "not well-formed (invalid token): line 1, column 0 (<string>)",
     "output_type": "error",
     "traceback": [
      "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
      "  File \u001b[1;32m\"/Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m3326\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \u001b[1;32m\"<ipython-input-94-413cdc4cb9fc>\"\u001b[0m, line \u001b[1;32m26\u001b[0m, in \u001b[1;35m<module>\u001b[0m\n    parseWikipathways()\n",
      "  File \u001b[1;32m\"<ipython-input-94-413cdc4cb9fc>\"\u001b[0m, line \u001b[1;32m11\u001b[0m, in \u001b[1;35mparseWikipathways\u001b[0m\n    dom = ElementTree.parse(filePath)\n",
      "  File \u001b[1;32m\"/Users/coribushman/anaconda3/envs/Research/lib/python3.7/xml/etree/ElementTree.py\"\u001b[0m, line \u001b[1;32m1197\u001b[0m, in \u001b[1;35mparse\u001b[0m\n    tree.parse(source, parser)\n",
      "\u001b[0;36m  File \u001b[0;32m\"/Users/coribushman/anaconda3/envs/Research/lib/python3.7/xml/etree/ElementTree.py\"\u001b[0;36m, line \u001b[0;32m598\u001b[0;36m, in \u001b[0;35mparse\u001b[0;36m\u001b[0m\n\u001b[0;31m    self._root = parser._parse_whole(source)\u001b[0m\n",
      "\u001b[0;36m  File \u001b[0;32m\"<string>\"\u001b[0;36m, line \u001b[0;32munknown\u001b[0m\n\u001b[0;31mParseError\u001b[0m\u001b[0;31m:\u001b[0m not well-formed (invalid token): line 1, column 0\n"
     ]
    }
   ],
   "source": [
    "#NEW FUNCTION WITH XML FILES\n",
    "\n",
    "def parseWikipathways():\n",
    "    pathwaysPerGene = {}\n",
    "    genesPerPathway = {}\n",
    "    \n",
    "    path = '/Users/coribushman/Desktop/wikipathways_homo_sapiens'\n",
    "    for fileName in os.listdir(path):\n",
    "        filePath = path + \"/\" + fileName\n",
    "        \n",
    "        dom = ElementTree.parse(filePath)\n",
    "        pathwayName = dom.findall(\"Pathway/Name\")\n",
    "        print(pathwayName)\n",
    "        \n",
    "        with open(filePath, 'r') as file:\n",
    "            fileContents = file.read()\n",
    "            fileContents = fileContents.rstrip(\"\\n\")\n",
    "            pattern = \"Name=*\\n* Version=\"\n",
    "            print(fileContents)\n",
    "            if re.search(pattern, fileContents, re.MULTILINE):\n",
    "                \n",
    "                pathwayName = re.findall(r\"Name=*\\n* Version=\", fileContents, re.MULTILINE)\n",
    "                print(pathwayName)\n",
    "                print(\"done\")\n",
    "\n",
    "parseWikipathways()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#METHOD 2: GO THROUGH EACH GENE IN MY UNFILTERED LIST AND SEE IF IT IS IN UNIPROT\n",
    "        \n",
    "import certifi\n",
    "import json\n",
    "import urllib3\n",
    "#import requests, sys\n",
    "\n",
    "def geneInUniprot(geneName):  #pass in the interacting list from findInteractions\n",
    "    try:\n",
    "        requestURL = \"https://www.ebi.ac.uk/proteins/api/proteins?offset=0&size=-1&gene=\" + geneName + \"&organism=homo%20sapiens&format=json\"\n",
    "        http = urllib3.PoolManager(cert_reqs='CERT_REQUIRED', ca_certs=certifi.where())\n",
    "        response = http.request('GET', requestURL)\n",
    "    except:\n",
    "        return \"n\"\n",
    "\n",
    "    if response.status == 200: #if success, get data from API response\n",
    "        responseData = response.data\n",
    "        responseDataList = json.loads(responseData)\n",
    "        \n",
    "        \n",
    "        \n",
    "        if (len(responseDataList) != 0) and (\"accession\" in responseDataList[0].keys()) and (geneName != \"DNA\"):\n",
    "            return geneName\n",
    "        else:\n",
    "            return \"n\"\n",
    "    else:\n",
    "        #print(\"Something went wrong with the response status\")\n",
    "        return \"n\"\n",
    "        \n",
    "#p = geneInUniprot(\"TP53\")\n",
    "#print(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#NEW\n",
    "#REMOVE DNA LATER\n",
    "\n",
    "import certifi\n",
    "import json\n",
    "import urllib3\n",
    "#import requests, sys\n",
    "\n",
    "def geneInUniprotNew(geneList):  #pass in the interacting list from findInteractions\n",
    "    try:\n",
    "        requestURL = \"https://www.ebi.ac.uk/proteins/api/proteins?offset=0&size=-1&exact_gene=\" + geneList + \"&organism=homo%20sapiens&format=json\"\n",
    "        http = urllib3.PoolManager(cert_reqs='CERT_REQUIRED', ca_certs=certifi.where())\n",
    "        response = http.request('GET', requestURL)\n",
    "    except:\n",
    "        return [\"API request didn't work\"]\n",
    "\n",
    "    if response.status == 200: #if success, get data from API response\n",
    "        responseData = response.data\n",
    "        responseDataList = json.loads(responseData)\n",
    "        \n",
    "        filteredList = []\n",
    "        for data in responseDataList:\n",
    "            filteredList.append(data[\"gene\"][0][\"name\"][\"value\"])\n",
    "            #if \"accession\" in data.keys():\n",
    "                #filteredList.append(data[\"gene\"][\"name\"][\"value\"])\n",
    "                \n",
    "        filteredSet = set(filteredList)\n",
    "        #print(\"geneList: \" + geneList)\n",
    "        #print(\"filteredSet: \" + str(filteredSet))\n",
    "        return filteredSet\n",
    "        \n",
    "    else:\n",
    "        #print(\"Something went wrong with the response status\")\n",
    "        return [\"response status is not 200\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Ubiquitin-mediated', 'degradation', 'SNAI2', 'Calcium', 'Substrate &eacute;', 'ATM', 'ATR', 'BLM', 'CDC25C', 'MDM2', 'del17p', '', 'MIR15A', 'MIR16-1', 'Chk2', 'DNA', 'CREBBP', 'ARF', '(CDKN2A)', 'MTA2', 'Accumulation DNA damage', 'SREBF1', 'Nuclear abnormalities', 'RCHY1', 'DNA damage', 'DNA Damage', 'MYC', 'HMGB1', 'APC', 'TGFB', 'OTUD5', 'RFC4', 'CTNNB1', 'SNURF', 'PLAC8', 'CHEK2', 'Hypoxia', 'nitric oxide', 'TP73 Delta Np73', 'SUMO-1', 'OTX2', 'TNFSF10', 'MAPK14', 'MAPK13', 'MAPK12', 'MAPK11', 'Chk1', 'UCHL1', 'MEG3', 'P85A_HUMAN', 'CDC42', 'CAPN1', 'CAPN2', 'MIR1285-2', 'GRIN1', 'RHOC', 'RHOA', 'ROCK2', 'RHOB', 'cenersen sodium', 'PRKAB1', 'PRKAG1', 'PRKAA1', 'IFI16', 'GSK3B', 'ING1', 'BCL2', 'ROS', 'EP300', 'MDM4', 'p14 ARF ', '(CDKN2A)', 'HPV E6', 'PCNA', 'MIR34B', 'MIR34A', 'MIR34C', 'Senescence', 'p21', 'Apoptosis', 'Cyclin B', 'CDK1', 'GADD45A', 'p53 pathway', 'BAX', 'CDKN1A', 'GADD45G', 'BAK1', 'POLK', 'DDB2', 'GADD45B', 'uc339', '(Entraper1)', 'NANOG', 'hsa-mir-143/145 cluster', 'hsa-mir-34b/34c cluster', 'PMAIP1', 'BBC3', 'BOK', 'THBS1', 'BCL2L11', 'CDKN1C', 'CDKN1B', 'Ubiquitin-mediated', 'proteasome degradation', 'ATAD2', 'RAD17', 'UBE2T', 'COL9A3', 'COL9A1', 'MTDH', 'LBR', 'EGFR', 'RFC3', 'UBE2C', 'FANCI', 'CEBPZ', 'RNF144B', 'S100A6', 'Grown arrest and apoptosis', 'G1/S transition of mitotic cell cycle', 'Increase in ', 'proliferation', 'PIDD', 'CDK5R1', 'Neuron death', 'SLC11A2', 'Cell Survival', 'ING2', 'GAD45A', 'SERPINE1', 'SESN1', 'RRM2B', 'FAS', 'TNFRSF10B', 'DNA damage ATM dependent', 'CCNB3', 'CCNB2', 'CCNB1', 'SFN', 'hsa-mir-143/145', 'hsa-mir-34b/34c', 'miR-143/145 polycistron', 'G2/M Arrest', 'p53 signaling pathway', 'SAT2', 'SAT1', 'CASP8', 'SESN2', 'DDIT4', 'Proliferation', 'Differentiation', 'CDC2', 'ATF2', 'ELK1', 'p38', 'MAPK8', 'JNK', 'MAPK9', 'MAPK10', 'Bcl-xl', 'Cell proliferation', 'Pro-apoptosis', 'BMF', 'BID', 'Apoptosis', 'BNIP3L', 'BNIP3', 'HIF1A', 'ARNT', 'RB1', 'HBEGF', 'CCNA2', 'CDKN2A', 'CCNE1', 'BCL3', 'FASLG', 'FGF7', 'BCL2L1', 'CCND1', 'JUN']\n",
      "inputStr: Ubiquitin-mediated,degradation,SNAI2,Calcium,Substrate &eacute;,ATM,ATR,BLM,CDC25C,MDM2,del17p,,MIR15A,MIR16-1,Chk2,DNA,CREBBP,ARF,(CDKN2A),MTA2\n",
      "inputStr: Accumulation DNA damage,SREBF1,Nuclear abnormalities,RCHY1,DNA damage,DNA Damage,MYC,HMGB1,APC,TGFB,OTUD5,RFC4,CTNNB1,SNURF,PLAC8,CHEK2,Hypoxia,nitric oxide,TP73 Delta Np73,SUMO-1\n",
      "inputStr: OTX2,TNFSF10,MAPK14,MAPK13,MAPK12,MAPK11,Chk1,UCHL1,MEG3,P85A_HUMAN,CDC42,CAPN1,CAPN2,MIR1285-2,GRIN1,RHOC,RHOA,ROCK2,RHOB,cenersen sodium\n",
      "inputStr: PRKAB1,PRKAG1,PRKAA1,IFI16,GSK3B,ING1,BCL2,ROS,EP300,MDM4,p14 ARF ,(CDKN2A),HPV E6,PCNA,MIR34B,MIR34A,MIR34C,Senescence,p21,Apoptosis\n",
      "inputStr: Cyclin B,CDK1,GADD45A,p53 pathway,BAX,CDKN1A,GADD45G,BAK1,POLK,DDB2,GADD45B,uc339,(Entraper1),NANOG,hsa-mir-143/145 cluster,hsa-mir-34b/34c cluster,PMAIP1,BBC3,BOK,THBS1\n",
      "inputStr: BCL2L11,CDKN1C,CDKN1B,Ubiquitin-mediated,proteasome degradation,ATAD2,RAD17,UBE2T,COL9A3,COL9A1,MTDH,LBR,EGFR,RFC3,UBE2C,FANCI,CEBPZ,RNF144B,S100A6,Grown arrest and apoptosis\n",
      "inputStr: G1/S transition of mitotic cell cycle,Increase in ,proliferation,PIDD,CDK5R1,Neuron death,SLC11A2,Cell Survival,ING2,GAD45A,SERPINE1,SESN1,RRM2B,FAS,TNFRSF10B,DNA damage ATM dependent,CCNB3,CCNB2,CCNB1,SFN\n",
      "inputStr: hsa-mir-143/145,hsa-mir-34b/34c,miR-143/145 polycistron,G2/M Arrest,p53 signaling pathway,SAT2,SAT1,CASP8,SESN2,DDIT4,Proliferation,Differentiation,CDC2,ATF2,ELK1,p38,MAPK8,JNK,MAPK9,MAPK10\n",
      "inputStr: Bcl-xl,Cell proliferation,Pro-apoptosis,BMF,BID,Apoptosis,BNIP3L,BNIP3,HIF1A,ARNT,RB1,HBEGF,CCNA2,CDKN2A,CCNE1,BCL3,FASLG,FGF7,BCL2L1,CCND1\n",
      "[\"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", 'JUN']\n",
      "--- 31.514771938323975 seconds ---\n"
     ]
    }
   ],
   "source": [
    "import time\n",
    "start_time = time.time()\n",
    "\n",
    "interactingList = findInteractions(\"TP53\")\n",
    "\n",
    "fixedInteractingList = fixNewlines(interactingList)\n",
    "#print(sorted(fixedInteractingList))\n",
    "#print(\"----------------------------------------------------------------\")\n",
    "\n",
    "goodList = []    \n",
    "listForFunct = []\n",
    "print(fixedInteractingList)\n",
    "\n",
    "for gene in fixedInteractingList:\n",
    "    listForFunct.append(gene)\n",
    "    if len(listForFunct) == 20:\n",
    "        inputStr = \",\".join(listForFunct)\n",
    "        print(\"inputStr: \" + inputStr)\n",
    "        goodList += (geneInUniprotNew(inputStr))\n",
    "        listForFunct = []\n",
    "\n",
    "if len(listForFunct) > 0:\n",
    "    inputStr = \",\".join(listForFunct)\n",
    "    goodList += (geneInUniprotNew(inputStr))\n",
    "\n",
    "print(goodList)\n",
    "\n",
    "print(\"--- %s seconds ---\" % (time.time() - start_time))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "123\n",
      "['SNAI2', 'ATM', 'ATR', 'BLM', 'CDC25C', 'MDM2', 'Chk2', 'CREBBP', 'ARF', 'MTA2', 'SREBF1', 'RCHY1', 'MYC', 'HMGB1', 'APC', 'TGFB', 'OTUD5', 'RFC4', 'CTNNB1', 'SNURF', 'PLAC8', 'CHEK2', 'OTX2', 'TNFSF10', 'MAPK14', 'MAPK13', 'MAPK12', 'MAPK11', 'Chk1', 'UCHL1', 'CDC42', 'CAPN1', 'CAPN2', 'GRIN1', 'RHOC', 'RHOA', 'ROCK2', 'RHOB', 'PRKAB1', 'PRKAG1', 'PRKAA1', 'IFI16', 'GSK3B', 'ING1', 'BCL2', 'ROS', 'EP300', 'MDM4', 'PCNA', 'CDK1', 'GADD45A', 'BAX', 'CDKN1A', 'GADD45G', 'BAK1', 'POLK', 'DDB2', 'GADD45B', 'NANOG', 'PMAIP1', 'BBC3', 'BOK', 'THBS1', 'BCL2L11', 'CDKN1C', 'CDKN1B', 'ATAD2', 'RAD17', 'UBE2T', 'COL9A3', 'COL9A1', 'MTDH', 'LBR', 'EGFR', 'RFC3', 'UBE2C', 'FANCI', 'CEBPZ', 'RNF144B', 'S100A6', 'PIDD', 'CDK5R1', 'SLC11A2', 'ING2', 'SERPINE1', 'SESN1', 'RRM2B', 'FAS', 'TNFRSF10B', 'CCNB3', 'CCNB2', 'CCNB1', 'SFN', 'SAT2', 'SAT1', 'CASP8', 'SESN2', 'DDIT4', 'CDC2', 'ATF2', 'ELK1', 'p38', 'MAPK8', 'JNK', 'MAPK9', 'MAPK10', 'BMF', 'BID', 'BNIP3L', 'BNIP3', 'HIF1A', 'ARNT', 'RB1', 'HBEGF', 'CCNA2', 'CDKN2A', 'CCNE1', 'BCL3', 'FASLG', 'FGF7', 'BCL2L1', 'CCND1', 'JUN']\n",
      "--- 398.3327190876007 minutes ---\n"
     ]
    }
   ],
   "source": [
    "#Method 2: check individually if each gene in interacting list is a uniprot gene\n",
    "#FASTER\n",
    "\n",
    "import time\n",
    "start_time = time.time()\n",
    "\n",
    "interactingList = findInteractions(\"TP53\")\n",
    "\n",
    "#filteredGenes = [geneInUniprot(x) for x in interactingList if geneInUniprot(x) != \"n\"]\n",
    "\n",
    "fixedInteractingList = fixNewlines(interactingList)\n",
    "\n",
    "filteredGenes = []\n",
    "for gene in fixedInteractingList:\n",
    "    if geneInUniprot(gene) != \"n\":\n",
    "        filteredGenes.append(gene)\n",
    "print(len(filteredGenes))\n",
    "print(filteredGenes)\n",
    "\n",
    "print(\"--- %s seconds ---\" % ((time.time() - start_time)))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
