{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already up-to-date: cptac in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (0.6.5)\n",
      "Requirement already satisfied, skipping upgrade: urllib3>=1.24.2 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (1.24.2)\n",
      "Requirement already satisfied, skipping upgrade: requests>=2.21.0 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (2.22.0)\n",
      "Requirement already satisfied, skipping upgrade: scipy>=1.2.1 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (1.3.1)\n",
      "Requirement already satisfied, skipping upgrade: beautifulsoup4>=4.7.1 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (4.8.0)\n",
      "Requirement already satisfied, skipping upgrade: numpy>=1.16.3 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (1.16.4)\n",
      "Requirement already satisfied, skipping upgrade: pandas==0.25.* in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from cptac) (0.25.1)\n",
      "Requirement already satisfied, skipping upgrade: certifi>=2017.4.17 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from requests>=2.21.0->cptac) (2019.6.16)\n",
      "Requirement already satisfied, skipping upgrade: idna<2.9,>=2.5 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from requests>=2.21.0->cptac) (2.8)\n",
      "Requirement already satisfied, skipping upgrade: chardet<3.1.0,>=3.0.2 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from requests>=2.21.0->cptac) (3.0.4)\n",
      "Requirement already satisfied, skipping upgrade: soupsieve>=1.2 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from beautifulsoup4>=4.7.1->cptac) (1.9.2)\n",
      "Requirement already satisfied, skipping upgrade: pytz>=2017.2 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from pandas==0.25.*->cptac) (2019.2)\n",
      "Requirement already satisfied, skipping upgrade: python-dateutil>=2.6.1 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from pandas==0.25.*->cptac) (2.8.0)\n",
      "Requirement already satisfied, skipping upgrade: six>=1.5 in /Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages (from python-dateutil>=2.6.1->pandas==0.25.*->cptac) (1.12.0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install --upgrade cptac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import sys\n",
    "import operator\n",
    "import collections\n",
    "import os\n",
    "import cptac\n",
    "from datetime import date\n",
    "#from xml.etree import ElementTree\n",
    "from xml.dom import minidom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fixNewLines(s):\n",
    "    if \"\\n\" in s:\n",
    "        nList = s.split(\"\\n\")\n",
    "    else:\n",
    "        nList = []\n",
    "        nList.append(s)\n",
    "    return nList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fixTabs(s):\n",
    "    if \"\\t\" in s:\n",
    "        tList = s.split(\"\\t\")\n",
    "    else:\n",
    "        tList = []\n",
    "        tList.append(s)\n",
    "    return tList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['dsf']"
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def fixParen(q):\n",
    "    fixedList = []\n",
    "    list1 = q.split(\")\")\n",
    "    for i in list1:\n",
    "        i = i.strip()\n",
    "        list2 = i.split(\"(\")\n",
    "        for j in list2:\n",
    "            j = j.strip()\n",
    "            if j != \"\":\n",
    "                fixedList.append(j)\n",
    "    return fixedList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [],
   "source": [
    "#function to split each item with a newline into two items\n",
    "\n",
    "#fix if genes are in parenthesis\n",
    "\n",
    "def fixParsingErrors(interactingList):\n",
    "    fixedList = []\n",
    "    for i in interactingList:\n",
    "        i = i.strip()\n",
    "        noSpaces = i.split(\" \")\n",
    "        for item in noSpaces:\n",
    "            newlineFreeList = fixNewLines(item)\n",
    "            for n in newlineFreeList:\n",
    "                tabFreeList = fixTabs(n)\n",
    "                for t in tabFreeList:\n",
    "                    noParen = fixParen(t)\n",
    "                    fixedList.extend(noParen)\n",
    "    return list(set(fixedList))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def getUniprotGeneList():\n",
    "    filePath = '/Users/coribushman/github/cptac/cptac/utils/Uniprot_Proteome.tsv'\n",
    "    bioplex_interactions = pd.read_csv(filePath, sep='\\t')\n",
    "    uniprotList = []\n",
    "    for ind, row in bioplex_interactions.iterrows():\n",
    "        prots = str(row['Gene names'])\n",
    "        #print(prots)\n",
    "        prot_arr = prots.split(' ')\n",
    "        #print(prot_arr)\n",
    "        for prot in prot_arr:\n",
    "            if prot not in uniprotList:\n",
    "                uniprotList.append(prot)\n",
    "    return uniprotList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [],
   "source": [
    "def intersectWithUniprot(interactingGenes):\n",
    "    #get uniprot genes from file\n",
    "    uniprotGenes = getUniprotGeneList()\n",
    "    \n",
    "    #intersect\n",
    "    geneSet = set(interactingGenes)\n",
    "    uniprotSet = set(uniprotGenes)   #convert to set\n",
    "    intersectSet = geneSet.intersection(uniprotSet)\n",
    "    \n",
    "    #today = date.today()\n",
    "    #todayDate = today.strftime(\"%m/%d/%y\")\n",
    "    \n",
    "    #print(\"WikiPathways Interacting Proteins as of \" + str(todayDate))  #change from date to version of download\n",
    "    return list(intersectSet)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getGenesPerPathwayDict():\n",
    "    path = '/Users/coribushman/Desktop/wikipathways-20191010-gpml-Homo_sapiens'\n",
    "\n",
    "    pathwaysPerGene = {}\n",
    "    genesPerPathway = {}\n",
    "\n",
    "    for fileName in os.listdir(path):\n",
    "        if fileName == \".DS_Store\":\n",
    "            continue\n",
    "        filePath = path + \"/\" + fileName\n",
    "        pathwayGenes = []\n",
    "\n",
    "        xmldoc = minidom.parse(filePath)\n",
    "        pathway = xmldoc.getElementsByTagName(\"Pathway\")[0]\n",
    "        pathwayName = pathway.getAttribute(\"Name\")\n",
    "\n",
    "        dataNodes = pathway.getElementsByTagName(\"DataNode\")\n",
    "        for node in dataNodes:\n",
    "            if node.getAttribute(\"Type\") == \"GeneProduct\":\n",
    "                geneName = node.getAttribute(\"TextLabel\")\n",
    "                pathwayGenes.append(geneName)\n",
    "\n",
    "        pathwayGenes = fixParsingErrors(pathwayGenes)\n",
    "        pathwayGenes = intersectWithUniprot(pathwayGenes)\n",
    "        #pathwayGenes = intersectWithUniprot(pathwayGenes)\n",
    "        genesPerPathway[pathwayName] = pathwayGenes\n",
    "    return genesPerPathway"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getAllPathways(genesPerPathway):\n",
    "    allPathways = genesPerPathway.keys()\n",
    "    allPathways = list(allPathways)\n",
    "    return allPathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getAllGenes(genesPerPathway):\n",
    "    allGenes = []\n",
    "    for geneList in genesPerPathway.values():\n",
    "        allGenes.extend(geneList)\n",
    "    allGenes = list(set(allGenes))\n",
    "    allGenes = fixParsingErrors(allGenes)\n",
    "    return allGenes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": [
    "#gene, list of genes, pathways, list of pathways\n",
    "#have people test it to make sure it works\n",
    "\n",
    "#filter dict genes (they have newlines), intersect w/ uniprot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "genesPerPathway = getGenesPerPathwayDict()\n",
    "allPathways = getAllPathways(genesPerPathway)\n",
    "allGenes = getAllGenes(genesPerPathway)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target Of Rapamycin (TOR) Signaling</th>\n",
       "      <th>NRF2-ARE regulation</th>\n",
       "      <th>PDGF Pathway</th>\n",
       "      <th>Eukaryotic Transcription Initiation</th>\n",
       "      <th>Interleukin-11 Signaling Pathway</th>\n",
       "      <th>Vitamin D Receptor Pathway</th>\n",
       "      <th>T-Cell antigen Receptor (TCR) pathway during Staphylococcus aureus infection</th>\n",
       "      <th>Phosphatidylcholine catabolism</th>\n",
       "      <th>SRF and miRs in Smooth Muscle Differentiation and Proliferation</th>\n",
       "      <th>Glucocorticoid and Mineralcorticoid Metabolism</th>\n",
       "      <th>...</th>\n",
       "      <th>Serotonin Transporter Activity</th>\n",
       "      <th>TCA Cycle (aka Krebs or citric acid cycle)</th>\n",
       "      <th>Serotonin and anxiety</th>\n",
       "      <th>Sudden Infant Death Syndrome (SIDS) Susceptibility Pathways</th>\n",
       "      <th>Canonical NF-KB pathway</th>\n",
       "      <th>Photodynamic therapy-induced HIF-1 survival signaling</th>\n",
       "      <th>Arrhythmogenic Right Ventricular Cardiomyopathy</th>\n",
       "      <th>Ethanol metabolism resulting in production of ROS by CYP2E1</th>\n",
       "      <th>Interactions between immune cells and microRNAs in tumor microenvironment</th>\n",
       "      <th>Metabolism of Tetrahydrocannabinol (THC)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>FBXW2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CEP83</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TBX2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>FOXC1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AMBRA1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>RPA2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>LOXL3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NFIB</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PTPN6</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>SMC1L1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8039 rows Ã— 537 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Target Of Rapamycin (TOR) Signaling  NRF2-ARE regulation  \\\n",
       "FBXW2                                 False                False   \n",
       "CEP83                                 False                False   \n",
       "TBX2                                  False                False   \n",
       "FOXC1                                 False                False   \n",
       "AMBRA1                                False                False   \n",
       "...                                     ...                  ...   \n",
       "RPA2                                  False                False   \n",
       "LOXL3                                 False                False   \n",
       "NFIB                                  False                False   \n",
       "PTPN6                                 False                False   \n",
       "SMC1L1                                False                False   \n",
       "\n",
       "        PDGF Pathway  Eukaryotic Transcription Initiation  \\\n",
       "FBXW2          False                                False   \n",
       "CEP83          False                                False   \n",
       "TBX2           False                                False   \n",
       "FOXC1          False                                False   \n",
       "AMBRA1         False                                False   \n",
       "...              ...                                  ...   \n",
       "RPA2           False                                False   \n",
       "LOXL3          False                                False   \n",
       "NFIB           False                                False   \n",
       "PTPN6          False                                False   \n",
       "SMC1L1         False                                False   \n",
       "\n",
       "        Interleukin-11 Signaling Pathway  Vitamin D Receptor Pathway  \\\n",
       "FBXW2                              False                       False   \n",
       "CEP83                              False                       False   \n",
       "TBX2                               False                       False   \n",
       "FOXC1                              False                       False   \n",
       "AMBRA1                             False                       False   \n",
       "...                                  ...                         ...   \n",
       "RPA2                               False                       False   \n",
       "LOXL3                              False                       False   \n",
       "NFIB                               False                       False   \n",
       "PTPN6                              False                       False   \n",
       "SMC1L1                             False                       False   \n",
       "\n",
       "        T-Cell antigen Receptor (TCR) pathway during Staphylococcus aureus infection  \\\n",
       "FBXW2                                               False                              \n",
       "CEP83                                               False                              \n",
       "TBX2                                                False                              \n",
       "FOXC1                                               False                              \n",
       "AMBRA1                                              False                              \n",
       "...                                                   ...                              \n",
       "RPA2                                                False                              \n",
       "LOXL3                                               False                              \n",
       "NFIB                                                False                              \n",
       "PTPN6                                               False                              \n",
       "SMC1L1                                              False                              \n",
       "\n",
       "        Phosphatidylcholine catabolism  \\\n",
       "FBXW2                            False   \n",
       "CEP83                            False   \n",
       "TBX2                             False   \n",
       "FOXC1                            False   \n",
       "AMBRA1                           False   \n",
       "...                                ...   \n",
       "RPA2                             False   \n",
       "LOXL3                            False   \n",
       "NFIB                             False   \n",
       "PTPN6                            False   \n",
       "SMC1L1                           False   \n",
       "\n",
       "        SRF and miRs in Smooth Muscle Differentiation and Proliferation  \\\n",
       "FBXW2                                               False                 \n",
       "CEP83                                               False                 \n",
       "TBX2                                                False                 \n",
       "FOXC1                                               False                 \n",
       "AMBRA1                                              False                 \n",
       "...                                                   ...                 \n",
       "RPA2                                                False                 \n",
       "LOXL3                                               False                 \n",
       "NFIB                                                False                 \n",
       "PTPN6                                               False                 \n",
       "SMC1L1                                              False                 \n",
       "\n",
       "        Glucocorticoid and Mineralcorticoid Metabolism  ...  \\\n",
       "FBXW2                                            False  ...   \n",
       "CEP83                                            False  ...   \n",
       "TBX2                                             False  ...   \n",
       "FOXC1                                            False  ...   \n",
       "AMBRA1                                           False  ...   \n",
       "...                                                ...  ...   \n",
       "RPA2                                             False  ...   \n",
       "LOXL3                                            False  ...   \n",
       "NFIB                                             False  ...   \n",
       "PTPN6                                            False  ...   \n",
       "SMC1L1                                           False  ...   \n",
       "\n",
       "        Serotonin Transporter Activity  \\\n",
       "FBXW2                            False   \n",
       "CEP83                            False   \n",
       "TBX2                             False   \n",
       "FOXC1                            False   \n",
       "AMBRA1                           False   \n",
       "...                                ...   \n",
       "RPA2                             False   \n",
       "LOXL3                            False   \n",
       "NFIB                             False   \n",
       "PTPN6                            False   \n",
       "SMC1L1                           False   \n",
       "\n",
       "        TCA Cycle (aka Krebs or citric acid cycle)  Serotonin and anxiety  \\\n",
       "FBXW2                                        False                  False   \n",
       "CEP83                                        False                  False   \n",
       "TBX2                                         False                  False   \n",
       "FOXC1                                        False                  False   \n",
       "AMBRA1                                       False                  False   \n",
       "...                                            ...                    ...   \n",
       "RPA2                                         False                  False   \n",
       "LOXL3                                        False                  False   \n",
       "NFIB                                         False                  False   \n",
       "PTPN6                                        False                  False   \n",
       "SMC1L1                                       False                  False   \n",
       "\n",
       "        Sudden Infant Death Syndrome (SIDS) Susceptibility Pathways  \\\n",
       "FBXW2                                               False             \n",
       "CEP83                                               False             \n",
       "TBX2                                                False             \n",
       "FOXC1                                               False             \n",
       "AMBRA1                                              False             \n",
       "...                                                   ...             \n",
       "RPA2                                                False             \n",
       "LOXL3                                               False             \n",
       "NFIB                                                False             \n",
       "PTPN6                                               False             \n",
       "SMC1L1                                              False             \n",
       "\n",
       "        Canonical NF-KB pathway  \\\n",
       "FBXW2                     False   \n",
       "CEP83                     False   \n",
       "TBX2                      False   \n",
       "FOXC1                     False   \n",
       "AMBRA1                    False   \n",
       "...                         ...   \n",
       "RPA2                      False   \n",
       "LOXL3                     False   \n",
       "NFIB                      False   \n",
       "PTPN6                     False   \n",
       "SMC1L1                    False   \n",
       "\n",
       "        Photodynamic therapy-induced HIF-1 survival signaling  \\\n",
       "FBXW2                                               False       \n",
       "CEP83                                               False       \n",
       "TBX2                                                False       \n",
       "FOXC1                                               False       \n",
       "AMBRA1                                              False       \n",
       "...                                                   ...       \n",
       "RPA2                                                False       \n",
       "LOXL3                                               False       \n",
       "NFIB                                                False       \n",
       "PTPN6                                               False       \n",
       "SMC1L1                                              False       \n",
       "\n",
       "        Arrhythmogenic Right Ventricular Cardiomyopathy  \\\n",
       "FBXW2                                             False   \n",
       "CEP83                                             False   \n",
       "TBX2                                              False   \n",
       "FOXC1                                             False   \n",
       "AMBRA1                                            False   \n",
       "...                                                 ...   \n",
       "RPA2                                              False   \n",
       "LOXL3                                             False   \n",
       "NFIB                                              False   \n",
       "PTPN6                                             False   \n",
       "SMC1L1                                            False   \n",
       "\n",
       "        Ethanol metabolism resulting in production of ROS by CYP2E1  \\\n",
       "FBXW2                                               False             \n",
       "CEP83                                               False             \n",
       "TBX2                                                False             \n",
       "FOXC1                                               False             \n",
       "AMBRA1                                              False             \n",
       "...                                                   ...             \n",
       "RPA2                                                False             \n",
       "LOXL3                                               False             \n",
       "NFIB                                                False             \n",
       "PTPN6                                               False             \n",
       "SMC1L1                                              False             \n",
       "\n",
       "        Interactions between immune cells and microRNAs in tumor microenvironment  \\\n",
       "FBXW2                                               False                           \n",
       "CEP83                                               False                           \n",
       "TBX2                                                False                           \n",
       "FOXC1                                               False                           \n",
       "AMBRA1                                              False                           \n",
       "...                                                   ...                           \n",
       "RPA2                                                False                           \n",
       "LOXL3                                               False                           \n",
       "NFIB                                                False                           \n",
       "PTPN6                                               False                           \n",
       "SMC1L1                                              False                           \n",
       "\n",
       "        Metabolism of Tetrahydrocannabinol (THC)  \n",
       "FBXW2                                      False  \n",
       "CEP83                                      False  \n",
       "TBX2                                       False  \n",
       "FOXC1                                      False  \n",
       "AMBRA1                                     False  \n",
       "...                                          ...  \n",
       "RPA2                                       False  \n",
       "LOXL3                                      False  \n",
       "NFIB                                       False  \n",
       "PTPN6                                      False  \n",
       "SMC1L1                                     False  \n",
       "\n",
       "[8039 rows x 537 columns]"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df = pd.DataFrame(genesPerPathway, columns = allPathways, index = allGenes)\n",
    "df = pd.DataFrame(columns = allPathways, index = allGenes)\n",
    "df = df.fillna(False)\n",
    "\n",
    "\n",
    "for pathway in df.columns:\n",
    "    for gene in df.index:\n",
    "        if gene in genesPerPathway[pathway]:\n",
    "            #df.replace(to_replace = False, value = True)\n",
    "            df.at[gene, pathway] = True\n",
    "            \n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PLA2G4A',\n",
       " 'GAP',\n",
       " 'A',\n",
       " 'RASA1',\n",
       " 'alpha',\n",
       " 'SH-PTP2',\n",
       " 'MAPK1',\n",
       " 'PTN11',\n",
       " 'AP-1',\n",
       " 'JNK1',\n",
       " 'P',\n",
       " 'ELK1',\n",
       " 'F',\n",
       " 'PLCG1',\n",
       " 'c-Fos',\n",
       " 'cPLA2',\n",
       " 'MAP3K1',\n",
       " 'MAP2K4',\n",
       " 'SRF',\n",
       " 'G',\n",
       " 'IkB',\n",
       " 'NFkB',\n",
       " 'MP2K1',\n",
       " '1',\n",
       " 'c-Jun',\n",
       " 'D',\n",
       " 'MEKK1',\n",
       " 'PDGFB',\n",
       " 'Ras',\n",
       " 'JNKK1',\n",
       " 'MAPK3',\n",
       " 'H-Ras-1',\n",
       " 'MAPK8',\n",
       " 'PDGFRB',\n",
       " 'MEK',\n",
       " 'RPS6',\n",
       " 'AKT1',\n",
       " 'PIM2',\n",
       " 'PIK3R3',\n",
       " 'EIF4G3',\n",
       " 'PRKCA',\n",
       " 'PIK3R2',\n",
       " 'PIK3CB',\n",
       " 'AKT3',\n",
       " 'EIF4G1',\n",
       " 'EIF4E',\n",
       " 'RPS6KA1',\n",
       " 'RPS6KA2',\n",
       " 'PDK1',\n",
       " 'MAP2K7',\n",
       " 'MAP2K5',\n",
       " 'MAP2K2',\n",
       " 'AKT2',\n",
       " 'PIK3R1',\n",
       " 'RPS6KA4',\n",
       " 'PIM1',\n",
       " 'MAP2K6',\n",
       " 'MAPK10',\n",
       " 'MAP2K1',\n",
       " 'EIF4A1',\n",
       " 'PRKCD',\n",
       " 'RPS6KB1',\n",
       " 'EIF4A2',\n",
       " 'MTOR',\n",
       " 'EIF4EBP1',\n",
       " 'PIK3CA',\n",
       " 'RPS6KA5',\n",
       " 'CSNK2A1',\n",
       " 'RPTOR',\n",
       " 'RPS6KB2',\n",
       " 'MAPK14',\n",
       " 'PDCD4',\n",
       " 'EIF4B',\n",
       " 'PRKACA',\n",
       " 'MAPK9',\n",
       " 'PIK3CG',\n",
       " 'MAP2K3',\n",
       " 'HRAS',\n",
       " 'RB1',\n",
       " 'CBL',\n",
       " 'CDKN2C',\n",
       " 'CDK6',\n",
       " 'IRS1',\n",
       " 'BRAF',\n",
       " 'RAF1',\n",
       " 'BRCA1',\n",
       " 'BRCA2',\n",
       " 'PLCG2',\n",
       " 'FGFR1',\n",
       " 'PRKCG',\n",
       " 'PRKCZ',\n",
       " 'ERRFI1',\n",
       " 'MDM2',\n",
       " 'MDM4',\n",
       " 'PIK3C2G',\n",
       " 'PDGFRA',\n",
       " 'IGF1R',\n",
       " 'CCND1',\n",
       " 'FGFR2',\n",
       " 'CDKN2B',\n",
       " 'CCND2',\n",
       " 'TSC2',\n",
       " 'NRAS',\n",
       " 'FOXO1',\n",
       " 'PTEN',\n",
       " 'CDKN1B',\n",
       " 'CDK2',\n",
       " 'FOXO4',\n",
       " 'CDK4',\n",
       " 'GRB2',\n",
       " 'CCNE1',\n",
       " 'TSC1',\n",
       " 'EGFR',\n",
       " 'NF1',\n",
       " 'ARF',\n",
       " 'PRKCH',\n",
       " 'MET',\n",
       " 'ERBB2',\n",
       " 'FOXO3',\n",
       " 'GAB1',\n",
       " 'PIK3C2A',\n",
       " 'P16',\n",
       " 'SRC',\n",
       " 'MSH6',\n",
       " 'PRKCI',\n",
       " 'KRAS',\n",
       " 'PIK3CD',\n",
       " 'PIK3C2B',\n",
       " 'PRKCB',\n",
       " 'E2F1',\n",
       " 'ATM',\n",
       " 'SPRY2',\n",
       " 'EP300',\n",
       " 'PDPK1',\n",
       " 'TP53',\n",
       " 'ERBB3',\n",
       " 'CDKN1A',\n",
       " 'ARAF',\n",
       " 'PRKCQ',\n",
       " 'RAC1',\n",
       " 'CREB1',\n",
       " 'MAPKAPK2',\n",
       " 'MAP3K7',\n",
       " 'DDIT3',\n",
       " 'RIPK1',\n",
       " 'MAP3K5',\n",
       " 'MYC',\n",
       " 'TGFB2',\n",
       " 'MAX',\n",
       " 'MAPKAPK5',\n",
       " 'MKNK1',\n",
       " 'SHC1',\n",
       " 'MAP3K9',\n",
       " 'RASGRF1',\n",
       " 'HMGN1',\n",
       " 'DAXX',\n",
       " 'MEF2D',\n",
       " 'TRAF2',\n",
       " 'CDC42',\n",
       " 'TRADD',\n",
       " 'TGFBR1',\n",
       " 'STAT1',\n",
       " 'HSPB1',\n",
       " 'ATF2',\n",
       " 'FOXM1',\n",
       " 'CLDN15',\n",
       " 'EED',\n",
       " 'NOTCH3',\n",
       " 'CLDN12',\n",
       " 'WNT7B',\n",
       " 'RBBP4',\n",
       " 'WNT16',\n",
       " 'CLDN5',\n",
       " 'COL4A5',\n",
       " 'LRP6',\n",
       " 'CLDN14',\n",
       " 'CLDN2',\n",
       " 'FOXQ1',\n",
       " 'TGFBR2',\n",
       " 'PAK1',\n",
       " 'CLDN18',\n",
       " 'CLDN6',\n",
       " 'CLDN22',\n",
       " 'FZD2',\n",
       " 'FZD6',\n",
       " 'PKP2',\n",
       " 'FN1',\n",
       " 'TGFB1',\n",
       " 'NRP2',\n",
       " 'FZD5',\n",
       " 'EIF5A2',\n",
       " 'DLK1',\n",
       " 'CDKL2',\n",
       " 'EZH2',\n",
       " 'MMP9',\n",
       " 'CTNNB1',\n",
       " 'CRB3',\n",
       " 'Claudins',\n",
       " 'ZEB1',\n",
       " 'CLDN3',\n",
       " 'FZD9',\n",
       " 'NOTCH4',\n",
       " 'NOTCH1',\n",
       " 'CLDN1',\n",
       " 'CLDN11',\n",
       " 'NUBPL',\n",
       " 'FZD7',\n",
       " 'JUP',\n",
       " 'TRAF6',\n",
       " 'WNT9B',\n",
       " 'COL4A4',\n",
       " 'GSK3B',\n",
       " 'COL4A1',\n",
       " 'SMAD3',\n",
       " 'LRP5',\n",
       " 'TJP1',\n",
       " 'CDH1',\n",
       " 'Cytokeratins',\n",
       " 'CLDN17',\n",
       " 'FOXC2',\n",
       " 'WNT5A',\n",
       " 'SOS1',\n",
       " 'PKD1',\n",
       " 'TGFB3',\n",
       " 'FZD4',\n",
       " 'CLDN4',\n",
       " 'ID1',\n",
       " 'LATS2',\n",
       " 'DSP',\n",
       " 'HIF1A',\n",
       " 'FZD10',\n",
       " 'WNT4',\n",
       " 'NR2C2',\n",
       " 'SMAD2',\n",
       " 'CLDN16',\n",
       " 'VTN',\n",
       " 'WNT7A',\n",
       " 'WNT5B',\n",
       " 'WNT1',\n",
       " 'DLL4',\n",
       " 'FZD3',\n",
       " 'WNT8B',\n",
       " 'MAPK11',\n",
       " 'CLDN9',\n",
       " 'FMNL2',\n",
       " 'COL4A3',\n",
       " 'CLDN20',\n",
       " 'GDF15',\n",
       " 'ZEB2',\n",
       " 'CLDN8',\n",
       " 'CTDSP1',\n",
       " 'CDH2',\n",
       " 'COL4A6',\n",
       " 'COL4A2',\n",
       " 'JAG2',\n",
       " 'TMPRSS4',\n",
       " 'WNT8A',\n",
       " 'PKP1',\n",
       " 'ID2',\n",
       " 'WNT2B',\n",
       " 'NOTCH2',\n",
       " 'WNT10B',\n",
       " 'receptors',\n",
       " 'WNT11',\n",
       " 'CLDN23',\n",
       " 'TWIST1',\n",
       " 'FZD1',\n",
       " 'SPARC',\n",
       " 'WNT6',\n",
       " 'WNT9A',\n",
       " 'PROX1',\n",
       " 'WNT10A',\n",
       " 'MMPs',\n",
       " 'WNT3A',\n",
       " 'ITGA5',\n",
       " 'CLDN7',\n",
       " 'SOS2',\n",
       " 'WNT2',\n",
       " 'WNT3',\n",
       " 'CLDN10',\n",
       " 'Collagen',\n",
       " 'SNAI2',\n",
       " 'CLDN19',\n",
       " 'MMP15',\n",
       " 'JAG1',\n",
       " 'TWIST2',\n",
       " 'pre-miR-9-2',\n",
       " 'CLDN24',\n",
       " 'FZD8',\n",
       " 'OCLN',\n",
       " 'MAPK12',\n",
       " 'DLL1',\n",
       " 'SMAD4',\n",
       " 'TUSC3',\n",
       " 'DLL3',\n",
       " 'kinase',\n",
       " 'RBPJ',\n",
       " 'STRAP',\n",
       " 'SUZ12',\n",
       " 'MMP2',\n",
       " 'SNAI1',\n",
       " 'MPP5',\n",
       " 'MAPK13',\n",
       " 'Tyrosine',\n",
       " 'CHUK',\n",
       " 'HSPG2',\n",
       " 'CASP9',\n",
       " 'CASP12',\n",
       " 'IFNA14',\n",
       " 'NFKB1',\n",
       " 'IFNA21',\n",
       " 'IRAK4',\n",
       " 'CASP10',\n",
       " 'DDB2',\n",
       " 'VDAC3',\n",
       " 'CXCL8',\n",
       " 'FASLG',\n",
       " 'STAT4',\n",
       " 'IFNA4',\n",
       " 'YWHAZ',\n",
       " 'JAK1',\n",
       " 'BAD',\n",
       " 'TLR3',\n",
       " 'DDB1',\n",
       " 'CREB3L4',\n",
       " 'STAT3',\n",
       " 'BIRC5',\n",
       " 'MYD88',\n",
       " 'NFATC4',\n",
       " 'NFATC2',\n",
       " 'EGR3',\n",
       " 'IKBKG',\n",
       " 'CREBBP',\n",
       " 'NFATC1',\n",
       " 'FADD',\n",
       " 'CREB3L2',\n",
       " 'RELA',\n",
       " 'TICAM1',\n",
       " 'CREB3L1',\n",
       " 'IFNA1',\n",
       " 'IKBKE',\n",
       " 'IFNA16',\n",
       " 'IFNAR1',\n",
       " 'TRAF3',\n",
       " 'PTK2B',\n",
       " 'CREB3L3',\n",
       " 'TBK1',\n",
       " 'IFNA2',\n",
       " 'EGR2',\n",
       " 'BAX',\n",
       " 'DDX3X',\n",
       " 'CREB3',\n",
       " 'IFNB1',\n",
       " 'IFNA7',\n",
       " 'CASP8',\n",
       " 'IRF3',\n",
       " 'IFNA5',\n",
       " 'JAK3',\n",
       " 'IFNA17',\n",
       " 'YWHAQ',\n",
       " 'BID',\n",
       " 'JUN',\n",
       " 'APAF1',\n",
       " 'FAS',\n",
       " 'TLR4',\n",
       " 'JAK2',\n",
       " 'TNF',\n",
       " 'IFNA8',\n",
       " 'IRAK1',\n",
       " 'STAT5B',\n",
       " 'CREB5',\n",
       " 'MAVS',\n",
       " 'TAB1',\n",
       " 'IFNA10',\n",
       " 'IFNA6',\n",
       " 'IFNA13',\n",
       " 'CASP3',\n",
       " 'CYCS',\n",
       " 'ATF4',\n",
       " 'STAT5A',\n",
       " 'ATP6AP1',\n",
       " 'DDX58',\n",
       " 'IKBKB',\n",
       " 'TIRAP',\n",
       " 'TLR2',\n",
       " 'BCL2',\n",
       " 'IRF7',\n",
       " 'PCNA',\n",
       " 'NFATC3',\n",
       " 'FOS',\n",
       " 'TYK2',\n",
       " 'IFIH1',\n",
       " 'IL6',\n",
       " 'ATF6B',\n",
       " 'SLC10A1',\n",
       " 'TAB2',\n",
       " 'YWHAB',\n",
       " 'STAT2',\n",
       " 'TICAM2',\n",
       " 'STAT6',\n",
       " 'PRKAA1',\n",
       " 'IRS4',\n",
       " 'RAPGEF1',\n",
       " 'MAP4K2',\n",
       " 'ENPP1',\n",
       " 'RAB4A',\n",
       " 'SGKL',\n",
       " 'IRS1/2',\n",
       " 'GRB10',\n",
       " 'SHC3',\n",
       " 'MAP3K11',\n",
       " 'CRK',\n",
       " 'PSCD3',\n",
       " 'XBP1',\n",
       " 'RHEB',\n",
       " 'MAPK4',\n",
       " 'VAMP2',\n",
       " 'STXBP1',\n",
       " 'SLC2A4',\n",
       " 'STXBP4',\n",
       " 'PRKCB1',\n",
       " 'MAPK7',\n",
       " 'CAP1',\n",
       " 'SORBS1',\n",
       " 'CBLC',\n",
       " 'MAP3K14',\n",
       " 'MAP4K4',\n",
       " 'INSR',\n",
       " 'PPP1R3A',\n",
       " 'STX4A',\n",
       " 'MAP3K12',\n",
       " 'SOCS3',\n",
       " 'ARF1',\n",
       " 'TBC1D4',\n",
       " 'FLOT2',\n",
       " 'MAP4K1',\n",
       " 'ARF6',\n",
       " 'IRS2',\n",
       " 'PRKAA2',\n",
       " 'MAP3K6',\n",
       " 'SNX26',\n",
       " 'MAP3K3',\n",
       " 'INPPL1',\n",
       " 'STXBP3',\n",
       " 'GSK3A',\n",
       " 'FLOT1',\n",
       " 'RAC2',\n",
       " 'EGR1',\n",
       " 'MAP3K4',\n",
       " 'MAPK6',\n",
       " 'PTPN1',\n",
       " 'MINK1',\n",
       " 'MAP4K5',\n",
       " 'STXBP2',\n",
       " 'GYS2',\n",
       " 'EHD1',\n",
       " 'GYG',\n",
       " 'PFKL',\n",
       " 'SLC2A1',\n",
       " 'CBLB',\n",
       " 'SHC2',\n",
       " 'APS',\n",
       " 'GRB14',\n",
       " 'RHOJ',\n",
       " 'FOXO1A',\n",
       " 'PIK3C3',\n",
       " 'PTP',\n",
       " 'MAP3K10',\n",
       " 'RPS6KA6',\n",
       " 'MAP3K2',\n",
       " 'MAP3K8',\n",
       " 'LIPE',\n",
       " 'TRIB3',\n",
       " 'SNAP25',\n",
       " 'RRAD',\n",
       " 'EHD2',\n",
       " 'SNAP23',\n",
       " 'PTPN11',\n",
       " 'SOCS1',\n",
       " 'PTPRF',\n",
       " 'RHOQ',\n",
       " 'FRAP1',\n",
       " 'PIK3',\n",
       " 'PIK3R4',\n",
       " 'KIF5B',\n",
       " 'KIF3A',\n",
       " 'RPS6KA3',\n",
       " 'SGK',\n",
       " 'SGK2',\n",
       " 'PFKM',\n",
       " 'FOXO3A',\n",
       " 'INPP4A',\n",
       " 'MAP3K13',\n",
       " 'MAP4K3',\n",
       " 'MYO1C',\n",
       " 'GYS1',\n",
       " 'CASP6',\n",
       " 'FAP1',\n",
       " 'DFFB',\n",
       " 'ARHGDIB',\n",
       " 'IL1A',\n",
       " 'PRKDC',\n",
       " 'CFLAR',\n",
       " 'PARP1',\n",
       " 'DFFA',\n",
       " 'LMNB2',\n",
       " 'SPTAN1',\n",
       " 'ACTA1',\n",
       " 'ACTG1',\n",
       " 'MAPKAPK3',\n",
       " 'LMNB1',\n",
       " 'FAF1',\n",
       " 'RIPK2',\n",
       " 'LMNA',\n",
       " 'PAK2',\n",
       " 'CASP7',\n",
       " 'ACTB',\n",
       " 'TGFA',\n",
       " 'HBEGF',\n",
       " 'CAMK2G',\n",
       " 'PAK4',\n",
       " 'ABL1',\n",
       " 'Myc',\n",
       " 'p21',\n",
       " 'BTC',\n",
       " 'NRG2',\n",
       " 'NRG4',\n",
       " 'BIM',\n",
       " 'p53',\n",
       " 'EGF',\n",
       " 'CAMK2B',\n",
       " 'eIF-4EBP',\n",
       " 'CAMK2D',\n",
       " 'ABL2',\n",
       " 'EREG',\n",
       " 'ERBB1',\n",
       " 'ELK',\n",
       " 'ERBB4',\n",
       " 'PAK3',\n",
       " 'GSK3',\n",
       " 'PAK5',\n",
       " 'AREG',\n",
       " 'SHC4',\n",
       " 'FAK',\n",
       " 'NRG1',\n",
       " 'NCK1',\n",
       " 'p27',\n",
       " 'CAMK2A',\n",
       " 'PAK6',\n",
       " 'mTOR',\n",
       " 'NRG3',\n",
       " 'BUB1B-PAK6',\n",
       " 'NCK2',\n",
       " 'CRKL',\n",
       " 'RASGRP4',\n",
       " 'RASGRP3',\n",
       " 'PPP3CC',\n",
       " 'FGF21',\n",
       " 'CACNG6',\n",
       " 'CACNA1S',\n",
       " 'CACNG1',\n",
       " 'PDGFA',\n",
       " 'DUSP9',\n",
       " 'NR4A1',\n",
       " 'PPP5C',\n",
       " 'CACNA1G',\n",
       " 'PLA2G4D',\n",
       " 'CACNA1B',\n",
       " 'FGF8',\n",
       " 'MAPK8IP3',\n",
       " 'ELK4',\n",
       " 'FGF17',\n",
       " 'ECSIT',\n",
       " 'IL1B',\n",
       " 'PPP5D1',\n",
       " 'HSPA1L',\n",
       " 'TNFRSF1A',\n",
       " 'RAPGEF2',\n",
       " 'FGF12',\n",
       " 'HSPA1B',\n",
       " 'HSPA6',\n",
       " 'CACNA1A',\n",
       " 'CACNB3',\n",
       " 'CACNG4',\n",
       " 'CACNG2',\n",
       " 'IL1R1',\n",
       " 'ARRB1',\n",
       " 'STK4',\n",
       " 'FGF13',\n",
       " 'PLA2G4B',\n",
       " 'CACNB1',\n",
       " 'DUSP6',\n",
       " 'CACNA2D1',\n",
       " 'DUSP16',\n",
       " 'RASA2',\n",
       " 'FGF23',\n",
       " 'PLA2G4F',\n",
       " 'MAPK8IP2',\n",
       " 'TAOK1',\n",
       " 'FGF16',\n",
       " 'NGF',\n",
       " 'RAC3',\n",
       " 'DUSP2',\n",
       " 'CDC25B',\n",
       " 'FGF19',\n",
       " 'FGF20',\n",
       " 'CACNA1F',\n",
       " 'NTRK2',\n",
       " 'CACNB2',\n",
       " 'NTF3',\n",
       " 'RRAS',\n",
       " 'CACNG3',\n",
       " 'DUSP10',\n",
       " 'ARRB2',\n",
       " 'FLNB',\n",
       " 'PTPN7',\n",
       " 'CACNA1D',\n",
       " 'NTRK1',\n",
       " 'FGF4',\n",
       " 'FGF5',\n",
       " 'PPP3CB',\n",
       " 'FGF3',\n",
       " 'MKNK2',\n",
       " 'HSPA8',\n",
       " 'FGF18',\n",
       " 'ZAK',\n",
       " 'STMN1',\n",
       " 'RASGRP1',\n",
       " 'DUSP8',\n",
       " 'CACNG5',\n",
       " 'HSPA1A',\n",
       " 'STK3',\n",
       " 'FGF7',\n",
       " 'CD14',\n",
       " 'PTPN5',\n",
       " 'PPP3R2',\n",
       " 'RASGRF2',\n",
       " 'DUSP1',\n",
       " 'FGF11',\n",
       " 'RAP1B',\n",
       " 'CACNA1E',\n",
       " 'CACNB4',\n",
       " 'MEF2C',\n",
       " 'GADD45A',\n",
       " 'FGF1',\n",
       " 'MRAS',\n",
       " 'DUSP3',\n",
       " 'RRAS2',\n",
       " 'FGF2',\n",
       " 'FGF9',\n",
       " 'LRRK2',\n",
       " 'NLK',\n",
       " 'FGF6',\n",
       " 'CACNA1C',\n",
       " 'IL1R2',\n",
       " 'FGF22',\n",
       " 'CACNA2D2',\n",
       " 'FGF10',\n",
       " 'PRKACG',\n",
       " 'FLNC',\n",
       " 'MAPK8IP1',\n",
       " 'NTF4',\n",
       " 'TAOK3',\n",
       " 'CACNA1H',\n",
       " 'PPP3CA',\n",
       " 'MP1',\n",
       " 'FGFR4',\n",
       " 'PLA2G4E',\n",
       " 'GNA12',\n",
       " 'GNG12',\n",
       " 'RAP1A',\n",
       " 'RELB',\n",
       " 'BDNF',\n",
       " 'DUSP4',\n",
       " 'HSPA2',\n",
       " 'CACNG8',\n",
       " 'CACNA1I',\n",
       " 'FGF14',\n",
       " 'PTPRR',\n",
       " 'JUND',\n",
       " 'PRKACB',\n",
       " 'PPM1B',\n",
       " 'NFKB2',\n",
       " 'PPP3R1',\n",
       " 'DUSP7',\n",
       " 'CACNA2D4',\n",
       " 'CACNA2D3',\n",
       " 'MAPT',\n",
       " 'FLNA',\n",
       " 'TAOK2',\n",
       " 'PPM1A',\n",
       " 'CACNG7',\n",
       " 'PLA2G4C',\n",
       " 'FGFR3',\n",
       " 'RASGRP2',\n",
       " 'FBXW2',\n",
       " 'PPP2R2A',\n",
       " 'FOSL1',\n",
       " 'CSNK1E',\n",
       " 'LDLR',\n",
       " 'FOXD3',\n",
       " 'SOX2',\n",
       " 'TCF7L1',\n",
       " 'NKD1',\n",
       " 'DVL1',\n",
       " 'PPP2R3B',\n",
       " 'CTBP2',\n",
       " 'PPP2R5C',\n",
       " 'DVL2',\n",
       " 'AXIN2',\n",
       " 'PPP2CB',\n",
       " 'CTNND1',\n",
       " 'FRAT1',\n",
       " 'CBP',\n",
       " 'APC',\n",
       " 'Prkcc',\n",
       " 'CTBP1',\n",
       " 'ESRRB',\n",
       " 'PPP2R4',\n",
       " 'PPP2R5E',\n",
       " 'PPP2R2C',\n",
       " 'TCF7L2',\n",
       " 'PPARD',\n",
       " 'NFYA',\n",
       " 'MMP7',\n",
       " 'PPP2R1A',\n",
       " 'Groucho',\n",
       " 'PRKD1',\n",
       " 'DVL3',\n",
       " 'LEF1',\n",
       " 'PPP2R2B',\n",
       " 'NANOG',\n",
       " 'AXIN1',\n",
       " 'NKD2',\n",
       " 'TCF7',\n",
       " 'POU5F1',\n",
       " 'PRKCE',\n",
       " 'CD44',\n",
       " 'LRRK2-mut',\n",
       " 'RHOA',\n",
       " 'CCND3',\n",
       " 'RACGAP1',\n",
       " 'PPM1J',\n",
       " 'PPP2CA',\n",
       " 'ZBTB33',\n",
       " 'PPP2R3A',\n",
       " 'PLAU',\n",
       " 'P300',\n",
       " 'PAFAH1B1',\n",
       " 'PPP2R1B',\n",
       " 'TNFRSF1B',\n",
       " 'C1R',\n",
       " 'BAK1',\n",
       " 'C5',\n",
       " 'C3AR1',\n",
       " 'CR2',\n",
       " 'C5AR1',\n",
       " 'TNK2',\n",
       " 'C2',\n",
       " 'C1QB',\n",
       " 'C1QC',\n",
       " 'BAG4',\n",
       " 'TRAF1',\n",
       " 'NFKBIE',\n",
       " 'Cyct',\n",
       " 'CDKN1C',\n",
       " 'C1S',\n",
       " 'TDP2',\n",
       " 'C4B',\n",
       " 'C1QA',\n",
       " 'CCL5',\n",
       " 'IL12A',\n",
       " 'IL8',\n",
       " 'NFKBIA',\n",
       " 'CD86',\n",
       " 'CXCL10',\n",
       " 'CYLD',\n",
       " 'TLR6',\n",
       " 'SYK',\n",
       " 'TNFAIP3',\n",
       " 'TLR2s',\n",
       " 'OTUD5',\n",
       " 'CUEDC2',\n",
       " 'FBXW5',\n",
       " 'IRF5',\n",
       " 'CISH',\n",
       " 'hsa-mir-105-1',\n",
       " 'TLR7',\n",
       " 'gene',\n",
       " 'TLR1',\n",
       " 'MIR155',\n",
       " 'NEU1',\n",
       " 'TLR9',\n",
       " 'Tifa',\n",
       " 'MLST8',\n",
       " 'TLR4s',\n",
       " 'RNF41',\n",
       " 'SMAD6',\n",
       " 'SFTPA1',\n",
       " 'CD40',\n",
       " 'SPP1',\n",
       " 'TRAFD1',\n",
       " 'CXCL9',\n",
       " 'TOLLIP',\n",
       " 'MIR146A',\n",
       " 'CTNNAL1',\n",
       " 'CCL3',\n",
       " 'RBCK1',\n",
       " 'IRAK2',\n",
       " 'MAP3K7IP1',\n",
       " 'Trim30',\n",
       " 'SFTPD',\n",
       " 'CD180',\n",
       " 'MYD88s',\n",
       " 'IRAK3',\n",
       " 'IL12B',\n",
       " 'hsa-let-7e',\n",
       " 'CCL4',\n",
       " 'BTK',\n",
       " 'TREM1',\n",
       " 'MBL2',\n",
       " 'SQSTM1',\n",
       " 'PLK1',\n",
       " 'CD80',\n",
       " 'PELI2',\n",
       " 'hsa-mir-98',\n",
       " 'CXCL11',\n",
       " 'TLR8',\n",
       " 'USP7',\n",
       " 'PELI1',\n",
       " 'SIGIRR',\n",
       " 'hsa-let-7i',\n",
       " 'IFNAR2',\n",
       " 'MAP3K7IP2',\n",
       " 'TAG',\n",
       " 'PIK3R5',\n",
       " 'SARM1',\n",
       " 'LY96',\n",
       " 'LBP',\n",
       " 'MAP3K7IP3',\n",
       " 'RNF31',\n",
       " 'PELI3',\n",
       " 'TLR5',\n",
       " 'RNF216',\n",
       " 'PTPN6',\n",
       " 'ZMYND11',\n",
       " 'NFKBIB',\n",
       " 'TAB3',\n",
       " 'ABCG8',\n",
       " 'mlst8',\n",
       " 'CYP7A1',\n",
       " 'ANGPTL8',\n",
       " 'PDK',\n",
       " 'LXR',\n",
       " 'SREBP1a-c',\n",
       " 'THRA',\n",
       " 'ABCG5',\n",
       " 'SLC16A2',\n",
       " 'AMPKy1',\n",
       " 'AMPKy2',\n",
       " 'AMPKy3',\n",
       " 'Rictor',\n",
       " 'CYP3A4',\n",
       " 'SCD',\n",
       " 'RXRA',\n",
       " 'DIO2',\n",
       " 'PEPCK',\n",
       " 'GLUT1',\n",
       " 'PPARalpha',\n",
       " 'AMPKb1',\n",
       " 'GLUT4',\n",
       " 'CIP42',\n",
       " 'THRB',\n",
       " 'RAS',\n",
       " 'CYP2B6',\n",
       " 'FASN',\n",
       " 'FBP',\n",
       " 'PP2A',\n",
       " 'Exo70',\n",
       " 'AMPKa1',\n",
       " 'SIN1',\n",
       " 'SEST3',\n",
       " 'Chrebp',\n",
       " 'SREBP2',\n",
       " 'Raptor',\n",
       " 'SLCO1C1',\n",
       " 'GCK',\n",
       " 'AMPKb2',\n",
       " 'GS',\n",
       " 'AMPKa2',\n",
       " 'G6PC',\n",
       " 'LPL',\n",
       " 'HRK',\n",
       " 'MCL1',\n",
       " 'CASP4',\n",
       " 'IRF1',\n",
       " 'BBC3',\n",
       " 'BOK',\n",
       " 'TP73',\n",
       " 'MIR29B2',\n",
       " 'PRF1',\n",
       " 'IGF1',\n",
       " 'GZMB',\n",
       " 'MIR29B1',\n",
       " 'BCL2L2',\n",
       " 'BIRC4',\n",
       " 'BCL2L11',\n",
       " 'TP63',\n",
       " 'BIRC3',\n",
       " 'TNFRSF21',\n",
       " 'CASP1',\n",
       " 'TNFRSF10B',\n",
       " 'CRADD',\n",
       " 'MIR29A',\n",
       " 'LTA',\n",
       " 'CDKN2A',\n",
       " 'DIABLO',\n",
       " 'BCL2L1',\n",
       " 'BIRC2',\n",
       " 'IRF6',\n",
       " 'CASP2',\n",
       " 'TNFSF10',\n",
       " 'IGF2',\n",
       " 'PMAIP1',\n",
       " 'BNIP3L',\n",
       " 'IRF2',\n",
       " 'IRF4',\n",
       " 'HELLS',\n",
       " 'CASP11',\n",
       " 'TNFRSF25',\n",
       " 'MBP',\n",
       " 'SIPA1',\n",
       " 'MAP2',\n",
       " 'RapGAPII',\n",
       " 'RASA3',\n",
       " 'PLCB3']"
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df.loc[df[\"MAP2K4\"] == True]\n",
    "\n",
    "#t = df[\"MAP2K4\"][df[\"MAP2K4\"]==True]\n",
    "#t\n",
    "\n",
    "pathways=[]\n",
    "row = df.loc[[\"MAP2K4\"]]\n",
    "#print(row)\n",
    "#print(type(row))\n",
    "for pathway in row:\n",
    "    #print(pathway)\n",
    "    if row[pathway][\"MAP2K4\"]==True:\n",
    "        pathways.append(pathway)\n",
    "\n",
    "#print(pathways)\n",
    "        \n",
    "genes_in_pathways = []\n",
    "for pw in pathways:\n",
    "    for g in genesPerPathway[pw]:\n",
    "        if g not in genes_in_pathways:\n",
    "            genes_in_pathways.append(g)\n",
    "            \n",
    "\n",
    "genes_in_pathways\n",
    "            \n",
    "#genes_in_pathways\n",
    "\n",
    "#df['a'][df['a']==0]\n",
    "#temp = df.loc[\"MAP2K4\"]\n",
    "#t2 = temp.index[temp==True]\n",
    "#type(t2)\n",
    "#tdf = df[[t2]]\n",
    "#tdf\n",
    "#temp2 = temp.index[temp==True]\n",
    "\n",
    "#s = df.iloc[0]        \n",
    "#a = s.index[s == 1]\n",
    "#temp2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "OLD FUNCTIONS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'list' object has no attribute 'endswith'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-86-7b3eb4f9c23d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     24\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m \u001b[0mw\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"asd(sdi)\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 26\u001b[0;31m \u001b[0mfixParen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mw\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-86-7b3eb4f9c23d>\u001b[0m in \u001b[0;36mfixParen\u001b[0;34m(s)\u001b[0m\n\u001b[1;32m      9\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m             \u001b[0;31m#print(str(type(i)) + \": \" + str(i))\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 11\u001b[0;31m             \u001b[0mfixParen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     12\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;34m\")\"\u001b[0m \u001b[0;32min\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     13\u001b[0m         \u001b[0ms\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\")\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-86-7b3eb4f9c23d>\u001b[0m in \u001b[0;36mfixParen\u001b[0;34m(s)\u001b[0m\n\u001b[1;32m     16\u001b[0m             \u001b[0;31m#print(str(type(i)) + \": \" + str(i))\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m             \u001b[0mfixParen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 18\u001b[0;31m     \u001b[0;32mif\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendswith\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\")\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     19\u001b[0m         \u001b[0ms\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     20\u001b[0m         \u001b[0ms\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'list' object has no attribute 'endswith'"
     ]
    }
   ],
   "source": [
    "def oldfixParen(s):\n",
    "    stringList = []\n",
    "    if s.startswith(\"(\") and s.endswith(\")\"):\n",
    "        s = s[1:(len(s) - 1)]\n",
    "        s = str(s)\n",
    "    if \"(\" in s:\n",
    "        s = s.split(\"(\")\n",
    "        #print(str(type(s)) + \": \" + str(s))\n",
    "        for i in s:\n",
    "            #print(str(type(i)) + \": \" + str(i))\n",
    "            fixParen(i)\n",
    "    if \")\" in s:\n",
    "        s = s.split(\")\")\n",
    "        #print(str(type(s)) + \": \" + str(s))\n",
    "        for i in s:\n",
    "            #print(str(type(i)) + \": \" + str(i))\n",
    "            fixParen(i)\n",
    "    if s.endswith(\")\"):\n",
    "        s = s[:-1]\n",
    "        s = str(s)\n",
    "        #print(str(type(s)) + \": \" + str(s))\n",
    "    stringList.append(s)\n",
    "    return stringList\n",
    "\n",
    "w = \"asd(sdi)\"\n",
    "fixParen(w)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WikiPathways Interacting Proteins as of 10/25/19\n",
      "{'NOXA1', 'SIRT1', 'CHK1', 'PTEN', 'ATR', 'EP300', 'IL2', 'MDM2', 'CCND1', 'CDKN1A', 'HIPK2', 'DCAKD', 'PRKDC', 'SFN', 'PIAS1', 'RELA', 'SIAH1', 'GADD45A', 'HDAC1', 'ATM', 'CHK2', 'CBP', 'BRCA1', 'MDM4', 'CSNK1D', 'IL6', 'PML', 'AURKA', 'FADD', 'PUMA'}\n"
     ]
    }
   ],
   "source": [
    "wikiPathwaysInterGenes = findInteractions(\"p53\")\n",
    "filteredInterGenes = fixParsingErrors(wikiPathwaysInterGenes)\n",
    "finalGeneSet = intersectWithUniprot(filteredInterGenes)\n",
    "print(finalGeneSet)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#size=-1 to get all genes, but takes a long time\n",
    "#if statements might need to be added in case keys not in dict when parsing\n",
    "\n",
    "#this function will query wikipathways for all interactions of a given gene for the Homo Sapiens species\n",
    "#some interactions won't be genes\n",
    "\n",
    "import urllib3\n",
    "import json\n",
    "\n",
    "def findInteractions(geneName): #retuns list of interacting elements\n",
    "    interactionList = []\n",
    "    \n",
    "    try:\n",
    "        requestURL = \"http://webservice.wikipathways.org/findInteractions?query=\" + geneName + \"&format=json\" \n",
    "        http = urllib3.PoolManager()\n",
    "        response = http.request('GET', requestURL)\n",
    "\n",
    "        if response.status == 200: #if success, get data from API response\n",
    "            responseData = response.data\n",
    "            #print(type(responseData))\n",
    "            responseDataDict = json.loads(responseData)\n",
    "            #print(responseDataDict)\n",
    "            resultList = list(responseDataDict.values())\n",
    "            interactionsList = resultList[0]\n",
    "\n",
    "            for interactionDict in interactionsList:\n",
    "\n",
    "                if interactionDict[\"species\"] == \"Homo sapiens\":\n",
    "                    rightValuesList = interactionDict[\"fields\"][\"right\"][\"values\"]\n",
    "                    for value in rightValuesList:\n",
    "                        if (value not in interactionList) and (geneName.casefold() not in value.casefold()):\n",
    "                            interactionList.append(value)\n",
    "\n",
    "                    leftValuesList = interactionDict[\"fields\"][\"left\"][\"values\"]\n",
    "                    for value in leftValuesList:\n",
    "                        if (value not in interactionList) and (geneName.casefold() not in value.casefold()):\n",
    "                            interactionList.append(value)\n",
    "                else:\n",
    "                    continue       \n",
    "        else:\n",
    "            print(\"Something went wrong with the response status\")\n",
    "\n",
    "        return interactionList\n",
    "    except:\n",
    "        print(\"There was a problem getting a response from the WikiPathways API\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[]\n",
      "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n",
      "<Pathway xmlns=\"http://pathvisio.org/GPML/2013a\" Name=\"4-hydroxytamoxifen, Dexamethasone, and Retinoic Acids Regulation of p27 Expression\" Version=\"20160816\" Organism=\"Homo sapiens\">\n",
      "  <Comment Source=\"WikiPathways-description\">This pathway is modeled after Figure 7 in the article ''Upstream molecular signaling pathways of p27(Kip1) expression: effects of 4-hydroxytamoxifen, dexamethasone, and retinoic acids'' (See Bibliography). Several anticancer agents up-regulate expression of p27, a cyclin-dpendent kinase inhibitor that blocks the transition from G1 to S phase. The pathways used by the anticancer agents (retinoic acids, 4-hydroxytamoxifen, dexamethasone) are shown above from human breast cancer cells.</Comment>\n",
      "  <BiopaxRef>f5c</BiopaxRef>\n",
      "  <Graphics BoardWidth=\"1659.0\" BoardHeight=\"820.0\"/>\n",
      "  <DataNode TextLabel=\"Triciribine\" GraphId=\"a188c\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"311.0\" CenterY=\"262.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"65399\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MNK1\" GraphId=\"a2a00\" Type=\"GeneProduct\" GroupRef=\"bc75a\">\n",
      "    <Graphics CenterX=\"1138.0\" CenterY=\"517.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000079277\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"eIF4E\" GraphId=\"a641b\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"1138.0\" CenterY=\"604.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000151247\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"AMPK\" GraphId=\"ad286\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"464.0\" CenterY=\"237.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Uniprot-TrEMBL\" ID=\"Q13131\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"LY 294, 002\" GraphId=\"ae1da\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"303.0\" CenterY=\"183.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"3973\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MKK6\" GraphId=\"b2ed3\" Type=\"GeneProduct\" GroupRef=\"f6aca\">\n",
      "    <Graphics CenterX=\"1261.75\" CenterY=\"294.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000108984\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Raf\" GraphId=\"b5118\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"204.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000132155\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"4-Hydroxytamoxifen\" GraphId=\"bd676\" Type=\"Metabolite\" GroupRef=\"c74ea\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"88.0\" Width=\"104.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"449459\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"SB202190\" GraphId=\"bda16\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1339.75\" CenterY=\"463.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"5353940\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Rotenone\" GraphId=\"c3556\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"583.0\" CenterY=\"140.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"6758\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Mlk3\" GraphId=\"cbc42\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"1262.0\" CenterY=\"199.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000173327\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MEK2\" GraphId=\"ccec7\" Type=\"GeneProduct\" GroupRef=\"c8a45\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"294.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000126934\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"AKT\" GraphId=\"cd60e\" Type=\"GeneProduct\" GroupRef=\"aac1e\">\n",
      "    <Graphics CenterX=\"186.0\" CenterY=\"292.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000142208\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"p38MAPK\" GraphId=\"ce646\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"1261.0\" CenterY=\"527.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000197442\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Dexamethasone\" GraphId=\"d037b\" Type=\"Metabolite\" GroupRef=\"c74ea\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"108.0\" Width=\"104.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"5743\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"PD98059\" GraphId=\"d6571\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1060.0\" CenterY=\"356.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"4713\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"EIF4EBP1\" GraphId=\"db08f\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"973.75\" CenterY=\"415.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000187840\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"AICA Riboside Compound C\" GraphId=\"dc985\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"621.0\" CenterY=\"211.0\" Width=\"150.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"17513\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Cap-Independent Translation Initiation of p27 mRNA Through its 5'UTR\" GraphId=\"de848\" Type=\"Pathway\">\n",
      "    <Comment>Identifier not found</Comment>\n",
      "    <Graphics CenterX=\"1354.5\" CenterY=\"689.0\" Width=\"463.0\" Height=\"20.0\" ZOrder=\"32768\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\" ShapeType=\"None\" Color=\"14961e\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MEK1\" GraphId=\"e049b\" Type=\"GeneProduct\" GroupRef=\"c8a45\">\n",
      "    <Graphics CenterX=\"1140.0\" CenterY=\"274.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000169032\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"ERK Inhibitor I\" GraphId=\"e3241\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1007.75\" CenterY=\"501.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"3377241\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"PBK\" GraphId=\"e775d\" Type=\"GeneProduct\" GroupRef=\"aac1e\">\n",
      "    <Graphics CenterX=\"186.0\" CenterY=\"312.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000168078\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MNK2\" GraphId=\"ec796\" Type=\"GeneProduct\" GroupRef=\"bc75a\">\n",
      "    <Graphics CenterX=\"1138.0\" CenterY=\"537.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000099875\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"ERK2\" GraphId=\"eca2b\" Type=\"GeneProduct\" GroupRef=\"cd15c\">\n",
      "    <Graphics CenterX=\"1139.0\" CenterY=\"425.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000100030\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"TSC\" GraphId=\"ed6bc\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"463.0\" CenterY=\"365.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000165699\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"ERK1\" GraphId=\"ede91\" Type=\"GeneProduct\" GroupRef=\"cd15c\">\n",
      "    <Graphics CenterX=\"1139.0\" CenterY=\"405.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000102882\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Retinoic Acids?\" GraphId=\"eec0d\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1461.0\" CenterY=\"482.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"HMDB\" ID=\"HMDB0001852\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MTOR Protein Kinase Signaling Pathway\" GraphId=\"f06e5\" Type=\"Pathway\">\n",
      "    <Graphics CenterX=\"973.0\" CenterY=\"202.0\" Width=\"282.0\" Height=\"20.0\" ZOrder=\"32768\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\" ShapeType=\"None\" Color=\"14961e\"/>\n",
      "    <Xref Database=\"WikiPathways\" ID=\"WP3318\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"NSC119889\" GraphId=\"f4490\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"1573.0\" CenterY=\"482.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"274104\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"MKK3\" GraphId=\"f99be\" Type=\"GeneProduct\" GroupRef=\"f6aca\">\n",
      "    <Graphics CenterX=\"1261.75\" CenterY=\"274.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000034152\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"mTOR\" GraphId=\"fd553\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"463.0\" CenterY=\"427.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32768\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000198793\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"4-Hydroxytamoxifen\" GraphId=\"e28f9\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"226.0\" CenterY=\"86.0\" Width=\"104.0\" Height=\"20.0\" ZOrder=\"32769\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"449459\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Dexamethasone\" GraphId=\"b2270\" Type=\"Metabolite\">\n",
      "    <Graphics CenterX=\"464.0\" CenterY=\"96.0\" Width=\"104.0\" Height=\"20.0\" ZOrder=\"32770\" FontSize=\"10\" Valign=\"Middle\" Color=\"0000ff\"/>\n",
      "    <Xref Database=\"PubChem-compound\" ID=\"5743\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"EIF4EBP1\" GraphId=\"a416b\" Type=\"GeneProduct\">\n",
      "    <Graphics CenterX=\"463.75\" CenterY=\"524.25\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32773\" FontSize=\"10\" Valign=\"Middle\"/>\n",
      "    <Xref Database=\"Ensembl\" ID=\"ENSG00000187840\"/>\n",
      "  </DataNode>\n",
      "  <DataNode TextLabel=\"Cap-Independent Translation Initiation of p27 mRNA Through its 5'UTR\" GraphId=\"bc30f\" Type=\"Pathway\">\n",
      "    <Comment>Identifier not found</Comment>\n",
      "    <Graphics CenterX=\"462.5\" CenterY=\"637.0\" Width=\"463.0\" Height=\"20.0\" ZOrder=\"32775\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\" ShapeType=\"None\" Color=\"14961e\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </DataNode>\n",
      "  <State GraphRef=\"ad286\" TextLabel=\"P\" GraphId=\"af1a5\">\n",
      "    <Graphics RelX=\"0.821570182394926\" RelY=\"-1.0\" Width=\"15.0\" Height=\"15.0\" ShapeType=\"Oval\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </State>\n",
      "  <State GraphRef=\"cd60e\" TextLabel=\"P\" GraphId=\"cc511\">\n",
      "    <Graphics RelX=\"-0.925\" RelY=\"-0.9\" Width=\"15.0\" Height=\"15.0\" ShapeType=\"Oval\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </State>\n",
      "  <State GraphRef=\"fd553\" TextLabel=\"P\" GraphId=\"d450e\">\n",
      "    <Graphics RelX=\"1.0\" RelY=\"1.0\" Width=\"15.0\" Height=\"15.0\" ShapeType=\"Oval\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </State>\n",
      "  <State GraphRef=\"a416b\" TextLabel=\"P\" GraphId=\"d72f8\">\n",
      "    <Graphics RelX=\"1.0\" RelY=\"1.0\" Width=\"15.0\" Height=\"15.0\" ShapeType=\"Oval\"/>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </State>\n",
      "  <Interaction GraphId=\"id128f6b9b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"263.0\" Y=\"183.0\" GraphRef=\"ae1da\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"208.5\" Y=\"217.5\" GraphRef=\"d48c9\" RelX=\"0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id14955cbf\">\n",
      "    <Graphics ZOrder=\"12288\" LineStyle=\"Broken\" LineThickness=\"1.0\">\n",
      "      <Point X=\"226.0\" Y=\"96.0\" GraphRef=\"e28f9\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"186.0\" Y=\"137.5\" GraphRef=\"f55ba\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id1573b86d\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1007.75\" Y=\"491.25\" GraphRef=\"e3241\" RelX=\"0.0\" RelY=\"-1.0\"/>\n",
      "      <Point X=\"1087.0\" Y=\"431.0\" GraphRef=\"e78c4\" RelX=\"-1.0\" RelY=\"0.5\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id1de6f372\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1261.75\" Y=\"316.25\" GraphRef=\"a2b86\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1261.0\" Y=\"517.0\" GraphRef=\"ce646\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id1e3811ea\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1140.0\" Y=\"214.0\" GraphRef=\"b5118\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1140.0\" Y=\"252.0\" GraphRef=\"dcd72\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id2357bcc4\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1138.0\" Y=\"614.0\" GraphRef=\"a641b\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1238.75\" Y=\"679.0\" GraphRef=\"de848\" RelX=\"-0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id338fd47b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1060.0\" Y=\"346.0\" GraphRef=\"d6571\" RelX=\"0.0\" RelY=\"-1.0\"/>\n",
      "      <Point X=\"1088.0\" Y=\"284.0\" GraphRef=\"dcd72\" RelX=\"-1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id33efe69e\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1339.75\" Y=\"473.25\" GraphRef=\"bda16\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1281.0\" Y=\"517.0\" GraphRef=\"ce646\" RelX=\"0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id37d1ac01\">\n",
      "    <Graphics ConnectorType=\"Elbow\" ZOrder=\"12288\" LineStyle=\"Broken\" LineThickness=\"1.0\">\n",
      "      <Point X=\"464.0\" Y=\"106.0\" GraphRef=\"b2270\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"424.0\" Y=\"237.0\" GraphRef=\"ad286\" RelX=\"-1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id3b3d47e0\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"463.0\" Y=\"375.0\" GraphRef=\"ed6bc\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"463.0\" Y=\"417.0\" GraphRef=\"fd553\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id3ecb520d\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1140.0\" Y=\"126.0\" GraphRef=\"d80b2\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1140.0\" Y=\"194.0\" GraphRef=\"b5118\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id40ee33da\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"464.5\" Y=\"174.0\" GraphRef=\"eae0b\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"464.0\" Y=\"227.0\" GraphRef=\"ad286\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id41d4985b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"271.0\" Y=\"262.0\" GraphRef=\"a188c\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"238.0\" Y=\"286.0\" GraphRef=\"b5fa0\" RelX=\"1.0\" RelY=\"-0.5\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id4259e563\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1140.0\" Y=\"316.0\" GraphRef=\"dcd72\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1139.0\" Y=\"383.0\" GraphRef=\"e78c4\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id56c56f1b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"463.75\" Y=\"534.25\" GraphRef=\"a416b\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"462.5\" Y=\"627.0\" GraphRef=\"bc30f\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id5d38e51e\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1139.0\" Y=\"447.0\" GraphRef=\"e78c4\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1138.0\" Y=\"495.0\" GraphRef=\"a2eb1\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id61467640\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1013.75\" Y=\"415.25\" GraphRef=\"db08f\" RelX=\"1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"1087.0\" Y=\"415.0\" GraphRef=\"e78c4\" RelX=\"-1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id67ad93e4\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"464.0\" Y=\"247.0\" GraphRef=\"ad286\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"463.0\" Y=\"355.0\" GraphRef=\"ed6bc\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id74d16a08\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1221.0\" Y=\"527.0\" GraphRef=\"ce646\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"1190.0\" Y=\"527.0\" GraphRef=\"a2eb1\" RelX=\"1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id7a969ee4\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"973.0\" Y=\"212.0\" GraphRef=\"f06e5\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"973.75\" Y=\"405.25\" GraphRef=\"db08f\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id7e2249f0\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"186.0\" Y=\"242.5\" GraphRef=\"d48c9\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"186.0\" Y=\"270.0\" GraphRef=\"b5fa0\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"id810511fc\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"546.0\" Y=\"211.0\" GraphRef=\"dc985\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"504.0\" Y=\"237.0\" GraphRef=\"ad286\" RelX=\"1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idac4a0ace\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1110.0\" Y=\"126.0\" GraphRef=\"d80b2\" RelX=\"-0.5\" RelY=\"1.0\"/>\n",
      "      <Point X=\"973.0\" Y=\"192.0\" GraphRef=\"f06e5\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idb363dd12\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1461.0\" Y=\"492.0\" GraphRef=\"eec0d\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1461.75\" Y=\"550.0\" GraphRef=\"a697f\" RelX=\"-0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idbe383038\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"186.0\" Y=\"162.5\" GraphRef=\"f55ba\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"186.0\" Y=\"217.5\" GraphRef=\"d48c9\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idbf97bab3\">\n",
      "    <Graphics ZOrder=\"12288\" LineStyle=\"Broken\" LineThickness=\"1.0\">\n",
      "      <Point X=\"464.0\" Y=\"106.0\" GraphRef=\"b2270\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"464.5\" Y=\"154.0\" GraphRef=\"eae0b\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idc34f3c45\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1354.5\" Y=\"699.0\" GraphRef=\"de848\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1354.5\" Y=\"770.0\" GraphRef=\"f3f7d\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idc70ddd2d\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"423.0\" Y=\"365.0\" GraphRef=\"ed6bc\" RelX=\"-1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "      <Point X=\"238.0\" Y=\"302.0\" GraphRef=\"b5fa0\" RelX=\"1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idca79bbc8\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"462.5\" Y=\"647.0\" GraphRef=\"bc30f\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"462.5\" Y=\"701.0\" GraphRef=\"d7ec4\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idd3085a06\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1170.0\" Y=\"126.0\" GraphRef=\"d80b2\" RelX=\"0.5\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1262.0\" Y=\"189.0\" GraphRef=\"cbc42\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idd4e700d8\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1262.0\" Y=\"209.0\" GraphRef=\"cbc42\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1261.75\" Y=\"264.25\" GraphRef=\"f99be\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"ide08a3a0b\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1573.0\" Y=\"492.0\" GraphRef=\"f4490\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1573.25\" Y=\"550.0\" GraphRef=\"a697f\" RelX=\"0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"ided308b41\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1138.0\" Y=\"559.0\" GraphRef=\"a2eb1\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1138.0\" Y=\"594.0\" GraphRef=\"a641b\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idef415bbb\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"543.0\" Y=\"140.0\" GraphRef=\"c3556\" RelX=\"-1.0\" RelY=\"0.0\"/>\n",
      "      <Point X=\"511.0\" Y=\"164.0\" GraphRef=\"eae0b\" RelX=\"1.0\" RelY=\"0.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idf9929b96\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"463.0\" Y=\"437.0\" GraphRef=\"fd553\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"463.75\" Y=\"514.25\" GraphRef=\"a416b\" RelX=\"0.0\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Interaction GraphId=\"idfda40a6c\">\n",
      "    <Graphics ZOrder=\"12288\" LineThickness=\"1.0\">\n",
      "      <Point X=\"1517.5\" Y=\"579.0\" GraphRef=\"a697f\" RelX=\"0.0\" RelY=\"1.0\"/>\n",
      "      <Point X=\"1470.25\" Y=\"679.0\" GraphRef=\"de848\" RelX=\"0.5\" RelY=\"-1.0\" ArrowHead=\"Arrow\"/>\n",
      "    </Graphics>\n",
      "    <Xref Database=\"\" ID=\"\"/>\n",
      "  </Interaction>\n",
      "  <Label TextLabel=\"A\" GraphId=\"a5592\">\n",
      "    <Graphics CenterX=\"110.0\" CenterY=\"88.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"Hypomethylation of 5' mRNA cap\" GraphId=\"a697f\">\n",
      "    <Graphics CenterX=\"1517.5\" CenterY=\"564.5\" Width=\"223.0\" Height=\"29.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"PI3K\" GraphId=\"d48c9\">\n",
      "    <Graphics CenterX=\"186.0\" CenterY=\"230.0\" Width=\"90.0\" Height=\"25.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"AMP/ATP Ratio\" GraphId=\"eae0b\">\n",
      "    <Graphics CenterX=\"464.5\" CenterY=\"164.0\" Width=\"93.0\" Height=\"20.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"Prevention of Cancer\" GraphId=\"f3f7d\">\n",
      "    <Graphics CenterX=\"1354.5\" CenterY=\"780.0\" Width=\"167.0\" Height=\"20.0\" ZOrder=\"28672\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"Prevention of Cancer\" GraphId=\"d7ec4\">\n",
      "    <Graphics CenterX=\"462.5\" CenterY=\"711.0\" Width=\"167.0\" Height=\"20.0\" ZOrder=\"32776\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"B\" GraphId=\"f0fce\">\n",
      "    <Graphics CenterX=\"877.0\" CenterY=\"107.0\" Width=\"80.0\" Height=\"20.0\" ZOrder=\"32777\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Label TextLabel=\"Receptor Tyrosine Kinases\" GraphId=\"f55ba\">\n",
      "    <Graphics CenterX=\"186.0\" CenterY=\"150.0\" Width=\"162.0\" Height=\"25.0\" ZOrder=\"32778\" FillColor=\"ffffff\" FontWeight=\"Bold\" FontSize=\"12\" Valign=\"Middle\"/>\n",
      "  </Label>\n",
      "  <Group GroupId=\"f6aca\" GraphId=\"a2b86\" Style=\"Complex\"/>\n",
      "  <Group GroupId=\"bc75a\" GraphId=\"a2eb1\" Style=\"Complex\"/>\n",
      "  <Group GroupId=\"aac1e\" GraphId=\"b5fa0\" Style=\"Complex\"/>\n",
      "  <Group GroupId=\"c74ea\" GraphId=\"d80b2\" Style=\"Group\"/>\n",
      "  <Group GroupId=\"c8a45\" GraphId=\"dcd72\" Style=\"Complex\"/>\n",
      "  <Group GroupId=\"cd15c\" GraphId=\"e78c4\" Style=\"Complex\"/>\n",
      "  <InfoBox CenterX=\"0.0\" CenterY=\"0.0\"/>\n",
      "  <Biopax>\n",
      "    <bp:openControlledVocabulary xmlns:bp=\"http://www.biopax.org/release/biopax-level3.owl#\">\n",
      "      <bp:TERM xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\" rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">breast cancer pathway</bp:TERM>\n",
      "      <bp:ID xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\" rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">PW:0000624</bp:ID>\n",
      "      <bp:Ontology xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\" rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">Pathway Ontology</bp:Ontology>\n",
      "    </bp:openControlledVocabulary>\n",
      "    <bp:PublicationXref xmlns:bp=\"http://www.biopax.org/release/biopax-level3.owl#\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\" rdf:id=\"f5c\">\n",
      "      <bp:ID rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">20170512</bp:ID>\n",
      "      <bp:DB rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">PubMed</bp:DB>\n",
      "      <bp:TITLE rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">Upstream molecular signaling pathways of p27(Kip1) expression: effects of 4-hydroxytamoxifen, dexamethasone, and retinoic acids.</bp:TITLE>\n",
      "      <bp:SOURCE rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">Cancer Cell Int</bp:SOURCE>\n",
      "      <bp:YEAR rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">2010</bp:YEAR>\n",
      "      <bp:AUTHORS rdf:datatype=\"http://www.w3.org/2001/XMLSchema#string\">Eto I</bp:AUTHORS>\n",
      "    </bp:PublicationXref>\n",
      "  <bp:openControlledVocabulary xmlns:bp=\"http://www.biopax.org/release/biopax-level3.owl#\"><bp:TERM>breast cancer</bp:TERM><bp:ID>DOID:1612</bp:ID><bp:Ontology>Disease</bp:Ontology></bp:openControlledVocabulary></Biopax>\n",
      "</Pathway>\n"
     ]
    },
    {
     "ename": "ParseError",
     "evalue": "not well-formed (invalid token): line 1, column 0 (<string>)",
     "output_type": "error",
     "traceback": [
      "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
      "  File \u001b[1;32m\"/Users/coribushman/anaconda3/envs/Research/lib/python3.7/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m3326\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \u001b[1;32m\"<ipython-input-94-413cdc4cb9fc>\"\u001b[0m, line \u001b[1;32m26\u001b[0m, in \u001b[1;35m<module>\u001b[0m\n    parseWikipathways()\n",
      "  File \u001b[1;32m\"<ipython-input-94-413cdc4cb9fc>\"\u001b[0m, line \u001b[1;32m11\u001b[0m, in \u001b[1;35mparseWikipathways\u001b[0m\n    dom = ElementTree.parse(filePath)\n",
      "  File \u001b[1;32m\"/Users/coribushman/anaconda3/envs/Research/lib/python3.7/xml/etree/ElementTree.py\"\u001b[0m, line \u001b[1;32m1197\u001b[0m, in \u001b[1;35mparse\u001b[0m\n    tree.parse(source, parser)\n",
      "\u001b[0;36m  File \u001b[0;32m\"/Users/coribushman/anaconda3/envs/Research/lib/python3.7/xml/etree/ElementTree.py\"\u001b[0;36m, line \u001b[0;32m598\u001b[0;36m, in \u001b[0;35mparse\u001b[0;36m\u001b[0m\n\u001b[0;31m    self._root = parser._parse_whole(source)\u001b[0m\n",
      "\u001b[0;36m  File \u001b[0;32m\"<string>\"\u001b[0;36m, line \u001b[0;32munknown\u001b[0m\n\u001b[0;31mParseError\u001b[0m\u001b[0;31m:\u001b[0m not well-formed (invalid token): line 1, column 0\n"
     ]
    }
   ],
   "source": [
    "#NEW FUNCTION WITH XML FILES\n",
    "\n",
    "def parseWikipathways():\n",
    "    pathwaysPerGene = {}\n",
    "    genesPerPathway = {}\n",
    "    \n",
    "    path = '/Users/coribushman/Desktop/wikipathways_homo_sapiens'\n",
    "    for fileName in os.listdir(path):\n",
    "        filePath = path + \"/\" + fileName\n",
    "        \n",
    "        dom = ElementTree.parse(filePath)\n",
    "        pathwayName = dom.findall(\"Pathway/Name\")\n",
    "        print(pathwayName)\n",
    "        \n",
    "        with open(filePath, 'r') as file:\n",
    "            fileContents = file.read()\n",
    "            fileContents = fileContents.rstrip(\"\\n\")\n",
    "            pattern = \"Name=*\\n* Version=\"\n",
    "            print(fileContents)\n",
    "            if re.search(pattern, fileContents, re.MULTILINE):\n",
    "                \n",
    "                pathwayName = re.findall(r\"Name=*\\n* Version=\", fileContents, re.MULTILINE)\n",
    "                print(pathwayName)\n",
    "                print(\"done\")\n",
    "\n",
    "parseWikipathways()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#METHOD 2: GO THROUGH EACH GENE IN MY UNFILTERED LIST AND SEE IF IT IS IN UNIPROT\n",
    "        \n",
    "import certifi\n",
    "import json\n",
    "import urllib3\n",
    "#import requests, sys\n",
    "\n",
    "def geneInUniprot(geneName):  #pass in the interacting list from findInteractions\n",
    "    try:\n",
    "        requestURL = \"https://www.ebi.ac.uk/proteins/api/proteins?offset=0&size=-1&gene=\" + geneName + \"&organism=homo%20sapiens&format=json\"\n",
    "        http = urllib3.PoolManager(cert_reqs='CERT_REQUIRED', ca_certs=certifi.where())\n",
    "        response = http.request('GET', requestURL)\n",
    "    except:\n",
    "        return \"n\"\n",
    "\n",
    "    if response.status == 200: #if success, get data from API response\n",
    "        responseData = response.data\n",
    "        responseDataList = json.loads(responseData)\n",
    "        \n",
    "        \n",
    "        \n",
    "        if (len(responseDataList) != 0) and (\"accession\" in responseDataList[0].keys()) and (geneName != \"DNA\"):\n",
    "            return geneName\n",
    "        else:\n",
    "            return \"n\"\n",
    "    else:\n",
    "        #print(\"Something went wrong with the response status\")\n",
    "        return \"n\"\n",
    "        \n",
    "#p = geneInUniprot(\"TP53\")\n",
    "#print(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#NEW\n",
    "#REMOVE DNA LATER\n",
    "\n",
    "import certifi\n",
    "import json\n",
    "import urllib3\n",
    "#import requests, sys\n",
    "\n",
    "def geneInUniprotNew(geneList):  #pass in the interacting list from findInteractions\n",
    "    try:\n",
    "        requestURL = \"https://www.ebi.ac.uk/proteins/api/proteins?offset=0&size=-1&exact_gene=\" + geneList + \"&organism=homo%20sapiens&format=json\"\n",
    "        http = urllib3.PoolManager(cert_reqs='CERT_REQUIRED', ca_certs=certifi.where())\n",
    "        response = http.request('GET', requestURL)\n",
    "    except:\n",
    "        return [\"API request didn't work\"]\n",
    "\n",
    "    if response.status == 200: #if success, get data from API response\n",
    "        responseData = response.data\n",
    "        responseDataList = json.loads(responseData)\n",
    "        \n",
    "        filteredList = []\n",
    "        for data in responseDataList:\n",
    "            filteredList.append(data[\"gene\"][0][\"name\"][\"value\"])\n",
    "            #if \"accession\" in data.keys():\n",
    "                #filteredList.append(data[\"gene\"][\"name\"][\"value\"])\n",
    "                \n",
    "        filteredSet = set(filteredList)\n",
    "        #print(\"geneList: \" + geneList)\n",
    "        #print(\"filteredSet: \" + str(filteredSet))\n",
    "        return filteredSet\n",
    "        \n",
    "    else:\n",
    "        #print(\"Something went wrong with the response status\")\n",
    "        return [\"response status is not 200\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Ubiquitin-mediated', 'degradation', 'SNAI2', 'Calcium', 'Substrate &eacute;', 'ATM', 'ATR', 'BLM', 'CDC25C', 'MDM2', 'del17p', '', 'MIR15A', 'MIR16-1', 'Chk2', 'DNA', 'CREBBP', 'ARF', '(CDKN2A)', 'MTA2', 'Accumulation DNA damage', 'SREBF1', 'Nuclear abnormalities', 'RCHY1', 'DNA damage', 'DNA Damage', 'MYC', 'HMGB1', 'APC', 'TGFB', 'OTUD5', 'RFC4', 'CTNNB1', 'SNURF', 'PLAC8', 'CHEK2', 'Hypoxia', 'nitric oxide', 'TP73 Delta Np73', 'SUMO-1', 'OTX2', 'TNFSF10', 'MAPK14', 'MAPK13', 'MAPK12', 'MAPK11', 'Chk1', 'UCHL1', 'MEG3', 'P85A_HUMAN', 'CDC42', 'CAPN1', 'CAPN2', 'MIR1285-2', 'GRIN1', 'RHOC', 'RHOA', 'ROCK2', 'RHOB', 'cenersen sodium', 'PRKAB1', 'PRKAG1', 'PRKAA1', 'IFI16', 'GSK3B', 'ING1', 'BCL2', 'ROS', 'EP300', 'MDM4', 'p14 ARF ', '(CDKN2A)', 'HPV E6', 'PCNA', 'MIR34B', 'MIR34A', 'MIR34C', 'Senescence', 'p21', 'Apoptosis', 'Cyclin B', 'CDK1', 'GADD45A', 'p53 pathway', 'BAX', 'CDKN1A', 'GADD45G', 'BAK1', 'POLK', 'DDB2', 'GADD45B', 'uc339', '(Entraper1)', 'NANOG', 'hsa-mir-143/145 cluster', 'hsa-mir-34b/34c cluster', 'PMAIP1', 'BBC3', 'BOK', 'THBS1', 'BCL2L11', 'CDKN1C', 'CDKN1B', 'Ubiquitin-mediated', 'proteasome degradation', 'ATAD2', 'RAD17', 'UBE2T', 'COL9A3', 'COL9A1', 'MTDH', 'LBR', 'EGFR', 'RFC3', 'UBE2C', 'FANCI', 'CEBPZ', 'RNF144B', 'S100A6', 'Grown arrest and apoptosis', 'G1/S transition of mitotic cell cycle', 'Increase in ', 'proliferation', 'PIDD', 'CDK5R1', 'Neuron death', 'SLC11A2', 'Cell Survival', 'ING2', 'GAD45A', 'SERPINE1', 'SESN1', 'RRM2B', 'FAS', 'TNFRSF10B', 'DNA damage ATM dependent', 'CCNB3', 'CCNB2', 'CCNB1', 'SFN', 'hsa-mir-143/145', 'hsa-mir-34b/34c', 'miR-143/145 polycistron', 'G2/M Arrest', 'p53 signaling pathway', 'SAT2', 'SAT1', 'CASP8', 'SESN2', 'DDIT4', 'Proliferation', 'Differentiation', 'CDC2', 'ATF2', 'ELK1', 'p38', 'MAPK8', 'JNK', 'MAPK9', 'MAPK10', 'Bcl-xl', 'Cell proliferation', 'Pro-apoptosis', 'BMF', 'BID', 'Apoptosis', 'BNIP3L', 'BNIP3', 'HIF1A', 'ARNT', 'RB1', 'HBEGF', 'CCNA2', 'CDKN2A', 'CCNE1', 'BCL3', 'FASLG', 'FGF7', 'BCL2L1', 'CCND1', 'JUN']\n",
      "inputStr: Ubiquitin-mediated,degradation,SNAI2,Calcium,Substrate &eacute;,ATM,ATR,BLM,CDC25C,MDM2,del17p,,MIR15A,MIR16-1,Chk2,DNA,CREBBP,ARF,(CDKN2A),MTA2\n",
      "inputStr: Accumulation DNA damage,SREBF1,Nuclear abnormalities,RCHY1,DNA damage,DNA Damage,MYC,HMGB1,APC,TGFB,OTUD5,RFC4,CTNNB1,SNURF,PLAC8,CHEK2,Hypoxia,nitric oxide,TP73 Delta Np73,SUMO-1\n",
      "inputStr: OTX2,TNFSF10,MAPK14,MAPK13,MAPK12,MAPK11,Chk1,UCHL1,MEG3,P85A_HUMAN,CDC42,CAPN1,CAPN2,MIR1285-2,GRIN1,RHOC,RHOA,ROCK2,RHOB,cenersen sodium\n",
      "inputStr: PRKAB1,PRKAG1,PRKAA1,IFI16,GSK3B,ING1,BCL2,ROS,EP300,MDM4,p14 ARF ,(CDKN2A),HPV E6,PCNA,MIR34B,MIR34A,MIR34C,Senescence,p21,Apoptosis\n",
      "inputStr: Cyclin B,CDK1,GADD45A,p53 pathway,BAX,CDKN1A,GADD45G,BAK1,POLK,DDB2,GADD45B,uc339,(Entraper1),NANOG,hsa-mir-143/145 cluster,hsa-mir-34b/34c cluster,PMAIP1,BBC3,BOK,THBS1\n",
      "inputStr: BCL2L11,CDKN1C,CDKN1B,Ubiquitin-mediated,proteasome degradation,ATAD2,RAD17,UBE2T,COL9A3,COL9A1,MTDH,LBR,EGFR,RFC3,UBE2C,FANCI,CEBPZ,RNF144B,S100A6,Grown arrest and apoptosis\n",
      "inputStr: G1/S transition of mitotic cell cycle,Increase in ,proliferation,PIDD,CDK5R1,Neuron death,SLC11A2,Cell Survival,ING2,GAD45A,SERPINE1,SESN1,RRM2B,FAS,TNFRSF10B,DNA damage ATM dependent,CCNB3,CCNB2,CCNB1,SFN\n",
      "inputStr: hsa-mir-143/145,hsa-mir-34b/34c,miR-143/145 polycistron,G2/M Arrest,p53 signaling pathway,SAT2,SAT1,CASP8,SESN2,DDIT4,Proliferation,Differentiation,CDC2,ATF2,ELK1,p38,MAPK8,JNK,MAPK9,MAPK10\n",
      "inputStr: Bcl-xl,Cell proliferation,Pro-apoptosis,BMF,BID,Apoptosis,BNIP3L,BNIP3,HIF1A,ARNT,RB1,HBEGF,CCNA2,CDKN2A,CCNE1,BCL3,FASLG,FGF7,BCL2L1,CCND1\n",
      "[\"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", \"API request didn't work\", 'JUN']\n",
      "--- 31.514771938323975 seconds ---\n"
     ]
    }
   ],
   "source": [
    "import time\n",
    "start_time = time.time()\n",
    "\n",
    "interactingList = findInteractions(\"TP53\")\n",
    "\n",
    "fixedInteractingList = fixNewlines(interactingList)\n",
    "#print(sorted(fixedInteractingList))\n",
    "#print(\"----------------------------------------------------------------\")\n",
    "\n",
    "goodList = []    \n",
    "listForFunct = []\n",
    "print(fixedInteractingList)\n",
    "\n",
    "for gene in fixedInteractingList:\n",
    "    listForFunct.append(gene)\n",
    "    if len(listForFunct) == 20:\n",
    "        inputStr = \",\".join(listForFunct)\n",
    "        print(\"inputStr: \" + inputStr)\n",
    "        goodList += (geneInUniprotNew(inputStr))\n",
    "        listForFunct = []\n",
    "\n",
    "if len(listForFunct) > 0:\n",
    "    inputStr = \",\".join(listForFunct)\n",
    "    goodList += (geneInUniprotNew(inputStr))\n",
    "\n",
    "print(goodList)\n",
    "\n",
    "print(\"--- %s seconds ---\" % (time.time() - start_time))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "123\n",
      "['SNAI2', 'ATM', 'ATR', 'BLM', 'CDC25C', 'MDM2', 'Chk2', 'CREBBP', 'ARF', 'MTA2', 'SREBF1', 'RCHY1', 'MYC', 'HMGB1', 'APC', 'TGFB', 'OTUD5', 'RFC4', 'CTNNB1', 'SNURF', 'PLAC8', 'CHEK2', 'OTX2', 'TNFSF10', 'MAPK14', 'MAPK13', 'MAPK12', 'MAPK11', 'Chk1', 'UCHL1', 'CDC42', 'CAPN1', 'CAPN2', 'GRIN1', 'RHOC', 'RHOA', 'ROCK2', 'RHOB', 'PRKAB1', 'PRKAG1', 'PRKAA1', 'IFI16', 'GSK3B', 'ING1', 'BCL2', 'ROS', 'EP300', 'MDM4', 'PCNA', 'CDK1', 'GADD45A', 'BAX', 'CDKN1A', 'GADD45G', 'BAK1', 'POLK', 'DDB2', 'GADD45B', 'NANOG', 'PMAIP1', 'BBC3', 'BOK', 'THBS1', 'BCL2L11', 'CDKN1C', 'CDKN1B', 'ATAD2', 'RAD17', 'UBE2T', 'COL9A3', 'COL9A1', 'MTDH', 'LBR', 'EGFR', 'RFC3', 'UBE2C', 'FANCI', 'CEBPZ', 'RNF144B', 'S100A6', 'PIDD', 'CDK5R1', 'SLC11A2', 'ING2', 'SERPINE1', 'SESN1', 'RRM2B', 'FAS', 'TNFRSF10B', 'CCNB3', 'CCNB2', 'CCNB1', 'SFN', 'SAT2', 'SAT1', 'CASP8', 'SESN2', 'DDIT4', 'CDC2', 'ATF2', 'ELK1', 'p38', 'MAPK8', 'JNK', 'MAPK9', 'MAPK10', 'BMF', 'BID', 'BNIP3L', 'BNIP3', 'HIF1A', 'ARNT', 'RB1', 'HBEGF', 'CCNA2', 'CDKN2A', 'CCNE1', 'BCL3', 'FASLG', 'FGF7', 'BCL2L1', 'CCND1', 'JUN']\n",
      "--- 398.3327190876007 minutes ---\n"
     ]
    }
   ],
   "source": [
    "#Method 2: check individually if each gene in interacting list is a uniprot gene\n",
    "#FASTER\n",
    "\n",
    "import time\n",
    "start_time = time.time()\n",
    "\n",
    "interactingList = findInteractions(\"TP53\")\n",
    "\n",
    "#filteredGenes = [geneInUniprot(x) for x in interactingList if geneInUniprot(x) != \"n\"]\n",
    "\n",
    "fixedInteractingList = fixNewlines(interactingList)\n",
    "\n",
    "filteredGenes = []\n",
    "for gene in fixedInteractingList:\n",
    "    if geneInUniprot(gene) != \"n\":\n",
    "        filteredGenes.append(gene)\n",
    "print(len(filteredGenes))\n",
    "print(filteredGenes)\n",
    "\n",
    "print(\"--- %s seconds ---\" % ((time.time() - start_time)))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
