{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get_frequently_mutated Final "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_frequently_mutated(cancer_type, cutoff=.1):  \n",
    "    \"\"\"take cancer type to import cptac and find the frequently mutated genes in total tumors compared to the cutoff.\n",
    "        \n",
    "        Parameters:\n",
    "        cancer_type (string): type of  cancer \n",
    "        cutoff (float): used as comparison to determine status of gene mutation frequency\n",
    "        \n",
    "        Returns:\n",
    "        freq_mutated_df (pd.DataFrame): DataFrame of frequently mutated genes passing the cutoff\n",
    "            and percent mutated (mutated genes / total tumors), percent Truncated, percent Missence\n",
    "        \n",
    "        There are many types of mutations catagorized into the columns 'Truncated' and 'Missence'. \n",
    "        The Truncated column includes: Frame_Shift_Del, Frame_Shift_Ins, Splice_Site, Nonsense_Mutation, Nonstop_Mutation\n",
    "        The Missence column includes: In_Frame_Del, In_Frame_Ins, Missense_Mutation\n",
    "        These columns count multiple mutations of one gene in the same sample, so percentages in the last two columns may \n",
    "        exceed the total percent mutated (which only counts if the gene was mutated once)\n",
    "            \"\"\" \n",
    "    \n",
    "    # import CPTAC and pandas\n",
    "    \n",
    "    import pandas as pd\n",
    "    colon = False\n",
    "    if cancer_type == \"endometrial\" or cancer_type == \"Endometrial\":\n",
    "        import cptac.endometrial as cptac\n",
    "        \n",
    "    elif cancer_type == \"colon\" or cancer_type == \"Colon\":\n",
    "        import cptac.colon as cptac\n",
    "        colon = True\n",
    "        \n",
    "    elif cancer_type == \"ovarian\" or cancer_type == \"Ovarian\":\n",
    "        import cptac.ovarian as cptac\n",
    "    \n",
    "    else:\n",
    "        str_cancer_options = '\\n' + 'Options: endometrial, ovarian, colon'\n",
    "        print(\"Please enter a valid cancer type.\", str_cancer_options)\n",
    "        return 0\n",
    "    \n",
    "    # get data frames\n",
    "    somatic_mutations = cptac.get_mutations()\n",
    "    proteomics = cptac.get_proteomics()\n",
    "    sample_status_map = cptac.get_sample_status_map()\n",
    "    merged_mutations = somatic_mutations.join(sample_status_map, how=\"left\") \n",
    "    \n",
    "    # standardize mutation names \n",
    "    if colon == True:\n",
    "        mutation_equivalents = {'frameshift substitution': 'Frame_Shift_Del' , 'frameshift deletion': 'Frame_Shift_Del', \n",
    "            'frameshift insertion': 'Frame_Shift_Ins', 'stopgain': 'Nonsense_Mutation ', 'stoploss':'Nonstop_Mutation',\n",
    "            'nonsynonymous SNV': 'Missense_Mutation','nonframeshift insertion': 'In_Frame_Ins',\n",
    "            'nonframeshift deletion': 'In_Frame_Del', 'nonframeshift substitution': 'Missense_Mutation'}\n",
    "        merged_mutations = merged_mutations.replace(to_replace = mutation_equivalents)\n",
    "        \n",
    "    # get list of unique genes\n",
    "    unique_genes = ['ARID1A']\n",
    "        #somatic_mutations['Gene'].unique()\n",
    "    \n",
    "    # get total tumors/patients\n",
    "    sample_status_series = sample_status_map.value_counts()\n",
    "    total_tumor_patients = sample_status_series[0]\n",
    "        \n",
    "    # find frequently mutated genes and their percent mutated. Create lists for frequently mutated genes and percentage.\n",
    "    freq_mut = ['ARID1A']\n",
    "    total_percent_mutated = []\n",
    "    for gene in unique_genes:\n",
    "        gene_mutated = merged_mutations.loc[merged_mutations['Gene'] == gene]\n",
    "        gene_mutated = gene_mutated.index.unique()\n",
    "        num_gene_mutated = len(gene_mutated)\n",
    "        percentage = (num_gene_mutated / total_tumor_patients)\n",
    "        if percentage > cutoff:\n",
    "            freq_mut.append(gene)\n",
    "            total_percent_mutated.append(percentage)\n",
    "    \n",
    "    # find truncated percentage\n",
    "    truncated = []\n",
    "    for gene in freq_mut:\n",
    "        gene_mutated = merged_mutations.loc[merged_mutations['Gene'] == gene]\n",
    "        truncated_df = gene_mutated.loc[(gene_mutated['Mutation'] != 'In_Frame_Del') & \n",
    "            (gene_mutated['Mutation'] != 'In_Frame_Ins') & (gene_mutated['Mutation'] != 'Missense_Mutation')] \n",
    "        num_gene_mutated = len(truncated_df.index.unique())\n",
    "        percentage = (num_gene_mutated / total_tumor_patients)\n",
    "        truncated.append(percentage)\n",
    "        \n",
    "        print(truncated)\n",
    "        print('trunc:', num_gene_mutated)\n",
    "        \n",
    "        print(truncated_df)\n",
    "        \n",
    "\n",
    "    # find missence percentage \n",
    "    missence = []\n",
    "    for gene in freq_mut:\n",
    "        gene_mutated = merged_mutations.loc[merged_mutations['Gene'] == gene]\n",
    "        \n",
    "        missence_mutations = gene_mutated.loc[(gene_mutated['Mutation'] == 'In_Frame_Ins') \n",
    "                | (gene_mutated['Mutation'] == 'In_Frame_Del') | (gene_mutated['Mutation'] == 'Missense_Mutation')]\n",
    "        num_gene_mutated = len(missence_mutations.index.unique())\n",
    "        print(num_gene_mutated)\n",
    "        print(missence_mutations.index.unique())\n",
    "        #print(missence_mutations)\n",
    "    \n",
    "        percentage = (num_gene_mutated / total_tumor_patients)\n",
    "        missence.append(percentage)\n",
    "        \n",
    "        \n",
    "        \n",
    "    # create dataframe\n",
    "    merged_lists = list(zip(freq_mut, total_percent_mutated, truncated, missence))\n",
    "    freq_mutated_df = pd.DataFrame(merged_lists, columns =['Gene', 'Fraction_Mutated', 'Truncated', 'Missence'])\n",
    "                   \n",
    "    return freq_mutated_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.4]\n",
      "trunc: 38\n",
      "             Gene           Mutation       Location Sample_Status\n",
      "Sample_ID                                                        \n",
      "S002       ARID1A  Nonsense_Mutation        p.Q403*         Tumor\n",
      "S002       ARID1A    Frame_Shift_Del  p.D1850Tfs*33         Tumor\n",
      "S008       ARID1A    Frame_Shift_Del   p.T541Pfs*77         Tumor\n",
      "S012       ARID1A  Nonsense_Mutation       p.R1989*         Tumor\n",
      "S014       ARID1A  Nonsense_Mutation        p.Q557*         Tumor\n",
      "S018       ARID1A  Nonsense_Mutation       p.R1989*         Tumor\n",
      "S019       ARID1A    Frame_Shift_Ins   p.D1850Gfs*4         Tumor\n",
      "S023       ARID1A  Nonsense_Mutation       p.C1827*         Tumor\n",
      "S023       ARID1A  Nonsense_Mutation       p.R1989*         Tumor\n",
      "S024       ARID1A  Nonsense_Mutation        p.Y471*         Tumor\n",
      "S024       ARID1A    Frame_Shift_Ins  p.H1581Qfs*22         Tumor\n",
      "S026       ARID1A  Nonsense_Mutation        p.R750*         Tumor\n",
      "S030       ARID1A    Frame_Shift_Del   p.A339Lfs*24         Tumor\n",
      "S036       ARID1A    Frame_Shift_Del   p.V1561Hfs*8         Tumor\n",
      "S037       ARID1A    Frame_Shift_Ins   p.D1850Gfs*4         Tumor\n",
      "S038       ARID1A  Nonsense_Mutation       p.R1276*         Tumor\n",
      "S040       ARID1A  Nonsense_Mutation       p.R1989*         Tumor\n",
      "S049       ARID1A    Frame_Shift_Del  p.K2045Sfs*53         Tumor\n",
      "S050       ARID1A  Nonsense_Mutation        p.Q543*         Tumor\n",
      "S050       ARID1A    Frame_Shift_Ins  p.Q1327Afs*11         Tumor\n",
      "S051       ARID1A    Frame_Shift_Del   p.G276Efs*87         Tumor\n",
      "S051       ARID1A    Frame_Shift_Del  p.D1850Tfs*33         Tumor\n",
      "S053       ARID1A    Frame_Shift_Del   p.Q521Rfs*98         Tumor\n",
      "S057       ARID1A  Nonsense_Mutation       p.R1276*         Tumor\n",
      "S057       ARID1A    Frame_Shift_Del  p.F2141Sfs*59         Tumor\n",
      "S064       ARID1A  Nonsense_Mutation        p.R693*         Tumor\n",
      "S068       ARID1A    Frame_Shift_Ins   p.D1850Gfs*4         Tumor\n",
      "S069       ARID1A    Frame_Shift_Ins   p.Q355Tfs*45         Tumor\n",
      "S069       ARID1A    Frame_Shift_Del  p.D1850Tfs*33         Tumor\n",
      "S073       ARID1A    Frame_Shift_Del  p.L1160Kfs*32         Tumor\n",
      "S073       ARID1A    Frame_Shift_Ins   p.F2141Lfs*9         Tumor\n",
      "S077       ARID1A    Frame_Shift_Del   p.Q372Sfs*19         Tumor\n",
      "S078       ARID1A    Frame_Shift_Ins    p.A6Gfs*109         Tumor\n",
      "S078       ARID1A    Frame_Shift_Del    p.P427Rfs*6         Tumor\n",
      "S080       ARID1A    Frame_Shift_Del   p.Y551Tfs*68         Tumor\n",
      "S083       ARID1A    Frame_Shift_Del  p.G248Afs*148         Tumor\n",
      "S084       ARID1A    Frame_Shift_Del  p.I1855Tfs*28         Tumor\n",
      "S085       ARID1A    Frame_Shift_Del  p.K1072Nfs*21         Tumor\n",
      "S087       ARID1A  Nonsense_Mutation        p.S607*         Tumor\n",
      "S090       ARID1A    Frame_Shift_Del  p.L1967Cfs*48         Tumor\n",
      "S091       ARID1A    Frame_Shift_Del  p.Q1452Rfs*29         Tumor\n",
      "S091       ARID1A    Frame_Shift_Ins   p.D1850Gfs*4         Tumor\n",
      "S092       ARID1A    Frame_Shift_Del   p.Y551Tfs*68         Tumor\n",
      "S092       ARID1A    Frame_Shift_Del  p.F2141Sfs*59         Tumor\n",
      "S093       ARID1A    Frame_Shift_Del  p.M2231Rfs*41         Tumor\n",
      "S094       ARID1A    Frame_Shift_Del   p.G276Efs*87         Tumor\n",
      "S094       ARID1A  Nonsense_Mutation       p.Q1947*         Tumor\n",
      "S098       ARID1A  Nonsense_Mutation        p.R693*         Tumor\n",
      "S098       ARID1A    Frame_Shift_Ins  p.Q1537Pfs*31         Tumor\n",
      "S099       ARID1A  Nonsense_Mutation       p.R2158*         Tumor\n",
      "S100       ARID1A    Frame_Shift_Ins  p.L1176Ifs*17         Tumor\n",
      "[0.4, 0.4]\n",
      "trunc: 38\n",
      "             Gene           Mutation       Location Sample_Status\n",
      "Sample_ID                                                        \n",
      "S002       ARID1A  Nonsense_Mutation        p.Q403*         Tumor\n",
      "S002       ARID1A    Frame_Shift_Del  p.D1850Tfs*33         Tumor\n",
      "S008       ARID1A    Frame_Shift_Del   p.T541Pfs*77         Tumor\n",
      "S012       ARID1A  Nonsense_Mutation       p.R1989*         Tumor\n",
      "S014       ARID1A  Nonsense_Mutation        p.Q557*         Tumor\n",
      "S018       ARID1A  Nonsense_Mutation       p.R1989*         Tumor\n",
      "S019       ARID1A    Frame_Shift_Ins   p.D1850Gfs*4         Tumor\n",
      "S023       ARID1A  Nonsense_Mutation       p.C1827*         Tumor\n",
      "S023       ARID1A  Nonsense_Mutation       p.R1989*         Tumor\n",
      "S024       ARID1A  Nonsense_Mutation        p.Y471*         Tumor\n",
      "S024       ARID1A    Frame_Shift_Ins  p.H1581Qfs*22         Tumor\n",
      "S026       ARID1A  Nonsense_Mutation        p.R750*         Tumor\n",
      "S030       ARID1A    Frame_Shift_Del   p.A339Lfs*24         Tumor\n",
      "S036       ARID1A    Frame_Shift_Del   p.V1561Hfs*8         Tumor\n",
      "S037       ARID1A    Frame_Shift_Ins   p.D1850Gfs*4         Tumor\n",
      "S038       ARID1A  Nonsense_Mutation       p.R1276*         Tumor\n",
      "S040       ARID1A  Nonsense_Mutation       p.R1989*         Tumor\n",
      "S049       ARID1A    Frame_Shift_Del  p.K2045Sfs*53         Tumor\n",
      "S050       ARID1A  Nonsense_Mutation        p.Q543*         Tumor\n",
      "S050       ARID1A    Frame_Shift_Ins  p.Q1327Afs*11         Tumor\n",
      "S051       ARID1A    Frame_Shift_Del   p.G276Efs*87         Tumor\n",
      "S051       ARID1A    Frame_Shift_Del  p.D1850Tfs*33         Tumor\n",
      "S053       ARID1A    Frame_Shift_Del   p.Q521Rfs*98         Tumor\n",
      "S057       ARID1A  Nonsense_Mutation       p.R1276*         Tumor\n",
      "S057       ARID1A    Frame_Shift_Del  p.F2141Sfs*59         Tumor\n",
      "S064       ARID1A  Nonsense_Mutation        p.R693*         Tumor\n",
      "S068       ARID1A    Frame_Shift_Ins   p.D1850Gfs*4         Tumor\n",
      "S069       ARID1A    Frame_Shift_Ins   p.Q355Tfs*45         Tumor\n",
      "S069       ARID1A    Frame_Shift_Del  p.D1850Tfs*33         Tumor\n",
      "S073       ARID1A    Frame_Shift_Del  p.L1160Kfs*32         Tumor\n",
      "S073       ARID1A    Frame_Shift_Ins   p.F2141Lfs*9         Tumor\n",
      "S077       ARID1A    Frame_Shift_Del   p.Q372Sfs*19         Tumor\n",
      "S078       ARID1A    Frame_Shift_Ins    p.A6Gfs*109         Tumor\n",
      "S078       ARID1A    Frame_Shift_Del    p.P427Rfs*6         Tumor\n",
      "S080       ARID1A    Frame_Shift_Del   p.Y551Tfs*68         Tumor\n",
      "S083       ARID1A    Frame_Shift_Del  p.G248Afs*148         Tumor\n",
      "S084       ARID1A    Frame_Shift_Del  p.I1855Tfs*28         Tumor\n",
      "S085       ARID1A    Frame_Shift_Del  p.K1072Nfs*21         Tumor\n",
      "S087       ARID1A  Nonsense_Mutation        p.S607*         Tumor\n",
      "S090       ARID1A    Frame_Shift_Del  p.L1967Cfs*48         Tumor\n",
      "S091       ARID1A    Frame_Shift_Del  p.Q1452Rfs*29         Tumor\n",
      "S091       ARID1A    Frame_Shift_Ins   p.D1850Gfs*4         Tumor\n",
      "S092       ARID1A    Frame_Shift_Del   p.Y551Tfs*68         Tumor\n",
      "S092       ARID1A    Frame_Shift_Del  p.F2141Sfs*59         Tumor\n",
      "S093       ARID1A    Frame_Shift_Del  p.M2231Rfs*41         Tumor\n",
      "S094       ARID1A    Frame_Shift_Del   p.G276Efs*87         Tumor\n",
      "S094       ARID1A  Nonsense_Mutation       p.Q1947*         Tumor\n",
      "S098       ARID1A  Nonsense_Mutation        p.R693*         Tumor\n",
      "S098       ARID1A    Frame_Shift_Ins  p.Q1537Pfs*31         Tumor\n",
      "S099       ARID1A  Nonsense_Mutation       p.R2158*         Tumor\n",
      "S100       ARID1A    Frame_Shift_Ins  p.L1176Ifs*17         Tumor\n",
      "13\n",
      "Index(['S001', 'S014', 'S022', 'S024', 'S027', 'S037', 'S059', 'S064', 'S073',\n",
      "       'S079', 'S085', 'S094', 'S099'],\n",
      "      dtype='object', name='Sample_ID')\n",
      "13\n",
      "Index(['S001', 'S014', 'S022', 'S024', 'S027', 'S037', 'S059', 'S064', 'S073',\n",
      "       'S079', 'S085', 'S094', 'S099'],\n",
      "      dtype='object', name='Sample_ID')\n"
     ]
    }
   ],
   "source": [
    "endo_freq_mutated_df = get_frequently_mutated(\"endometrial\", .25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Fraction_Mutated</th>\n",
       "      <th>Truncated</th>\n",
       "      <th>Missence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>0.452632</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.136842</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene  Fraction_Mutated  Truncated  Missence\n",
       "0  ARID1A          0.452632        0.4  0.136842"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total_tumor_patients 110\n"
     ]
    }
   ],
   "source": [
    "colon_freq_mutated_df = get_frequently_mutated(\"colon\", .25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Percent Mutated</th>\n",
       "      <th>Truncated</th>\n",
       "      <th>Missence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>APC</td>\n",
       "      <td>0.745455</td>\n",
       "      <td>3.136364</td>\n",
       "      <td>0.381818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.509091</td>\n",
       "      <td>2.145455</td>\n",
       "      <td>4.845455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.354545</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.627273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>0.254545</td>\n",
       "      <td>0.036364</td>\n",
       "      <td>0.381818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SYNE1</td>\n",
       "      <td>0.263636</td>\n",
       "      <td>0.281818</td>\n",
       "      <td>1.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.545455</td>\n",
       "      <td>1.409091</td>\n",
       "      <td>8.318182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>0.318182</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.636364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>OBSCN</td>\n",
       "      <td>0.263636</td>\n",
       "      <td>0.190909</td>\n",
       "      <td>1.127273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CCDC168</td>\n",
       "      <td>0.263636</td>\n",
       "      <td>0.263636</td>\n",
       "      <td>0.336364</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene  Percent Mutated  Truncated  Missence\n",
       "0      APC         0.745455   3.136364  0.381818\n",
       "1     TP53         0.509091   2.145455  4.845455\n",
       "2    MUC16         0.354545   0.100000  0.627273\n",
       "3     FAT3         0.254545   0.036364  0.381818\n",
       "4    SYNE1         0.263636   0.281818  1.018182\n",
       "5      TTN         0.545455   1.409091  8.318182\n",
       "6     KRAS         0.318182   0.000000  0.636364\n",
       "7    OBSCN         0.263636   0.190909  1.127273\n",
       "8  CCDC168         0.263636   0.263636  0.336364"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "colon_freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Ovarian, cutoff = .1 (default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You have loaded the cptac ovarian dataset. To view available\n",
      "dataframes, use cptac.ovarian.list_data(). To view available functions\n",
      "for accessing and manipulating the dataframes, use\n",
      "cptac.ovarian.list_api().\n",
      "ovarian data version: Most recent release\n",
      "\n",
      "Loading cptac ovarian data:\n",
      "Loading clinical data...\n",
      "Loading cnv data...\n",
      "Loading phosphoproteomics data...\n",
      "Loading proteomics data...\n",
      "Loading somatic_38 data...\n",
      "Loading transcriptomics data...\n",
      "Loading treatment data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until June 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until June 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n",
      "total_tumor_patients 111\n"
     ]
    }
   ],
   "source": [
    "ovarian_freq_mutated_df = get_frequently_mutated(\"ovarian\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Percent Mutated</th>\n",
       "      <th>Truncated</th>\n",
       "      <th>Missence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.279279</td>\n",
       "      <td>0.252252</td>\n",
       "      <td>0.468468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.693694</td>\n",
       "      <td>0.243243</td>\n",
       "      <td>0.468468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.144144</td>\n",
       "      <td>0.072072</td>\n",
       "      <td>0.225225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>0.117117</td>\n",
       "      <td>0.027027</td>\n",
       "      <td>0.090090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MT-ND5</td>\n",
       "      <td>0.144144</td>\n",
       "      <td>0.081081</td>\n",
       "      <td>0.063063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>FCGBP</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.054054</td>\n",
       "      <td>0.081081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MUC5B</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.027027</td>\n",
       "      <td>0.099099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MUC4</td>\n",
       "      <td>0.306306</td>\n",
       "      <td>0.153153</td>\n",
       "      <td>0.432432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MUC17</td>\n",
       "      <td>0.126126</td>\n",
       "      <td>0.162162</td>\n",
       "      <td>0.549550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NCOR2</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.090090</td>\n",
       "      <td>0.027027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MUC12</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.090090</td>\n",
       "      <td>0.396396</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene  Percent Mutated  Truncated  Missence\n",
       "0      TTN         0.279279   0.252252  0.468468\n",
       "1     TP53         0.693694   0.243243  0.468468\n",
       "2    MUC16         0.144144   0.072072  0.225225\n",
       "3   MT-CO1         0.117117   0.027027  0.090090\n",
       "4   MT-ND5         0.144144   0.081081  0.063063\n",
       "5    FCGBP         0.108108   0.054054  0.081081\n",
       "6    MUC5B         0.108108   0.027027  0.099099\n",
       "7     MUC4         0.306306   0.153153  0.432432\n",
       "8    MUC17         0.126126   0.162162  0.549550\n",
       "9    NCOR2         0.108108   0.090090  0.027027\n",
       "10   MUC12         0.108108   0.090090  0.396396"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ovarian_freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Debug "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total_tumor_patients: 83\n",
      "num_gene_mutated TP53 : 77\n",
      "num_gene_mutated PTEN : 4\n"
     ]
    }
   ],
   "source": [
    "#find frequently mutated - check Ovarian PTEN percentage = .9277 \n",
    "somatic = c.get_mutations()\n",
    "prot = c.get_proteomics()\n",
    "gene = 'PTEN'\n",
    "omics_mutations = c.append_mutations_to_omics(mutation_genes=gene, omics_df=prot, omics_genes=gene)\n",
    "unique_genes = ['TP53','PTEN']\n",
    "#somatic['Gene'].unique()\n",
    "gene_mutated = somatic.loc[somatic['Gene'] == 'PTEN']\n",
    "gene_and_freq_d = {}\n",
    "tumors = omics_mutations.loc[omics_mutations['Sample_Status'] == 'Tumor']\n",
    "total_tumor_patients = len(tumors)\n",
    "print('total_tumor_patients:', total_tumor_patients)\n",
    "cutoff = .25\n",
    "for gene in unique_genes:\n",
    "    gene_mutated = somatic.loc[somatic['Gene'] == gene].index\n",
    "    num_gene_mutated = len(gene_mutated.unique())\n",
    "    print('num_gene_mutated', gene, ':', num_gene_mutated)\n",
    "   \n",
    "    percentage = (num_gene_mutated / total_tumor_patients)\n",
    "    if percentage > cutoff:\n",
    "        gene_and_freq_d[gene] = percentage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'TP53': 0.927710843373494}"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_and_freq_d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene_mutated:  77.0 / 83 total_tumor_patients\n",
      "percent: 0.927710843373494\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Index(['S002', 'S006', 'S007', 'S009', 'S011', 'S013', 'S014', 'S015', 'S020',\n",
       "       'S021', 'S022', 'S024', 'S025', 'S026', 'S027', 'S029', 'S031', 'S033',\n",
       "       'S035', 'S036', 'S037', 'S038', 'S039', 'S040', 'S041', 'S042', 'S043',\n",
       "       'S044', 'S045', 'S047', 'S048', 'S049', 'S050', 'S051', 'S052', 'S053',\n",
       "       'S054', 'S055', 'S057', 'S058', 'S059', 'S060', 'S061', 'S062', 'S063',\n",
       "       'S064', 'S065', 'S066', 'S067', 'S068', 'S069', 'S073', 'S074', 'S075',\n",
       "       'S076', 'S082', 'S083', 'S086', 'S087', 'S089', 'S090', 'S093', 'S095',\n",
       "       'S096', 'S097', 'S098', 'S100', 'S103', 'S106', 'S107', 'S108', 'S109',\n",
       "       'S111', 'S112', 'S113', 'S115', 'S116'],\n",
       "      dtype='object', name='Sample_ID')"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'TP53'\n",
    "gene_mutated = somatic.loc[somatic['Gene'] == gene].index\n",
    "num_gene_mutated = float(len(gene_mutated.unique()))\n",
    "print('gene_mutated: ', num_gene_mutated, '/', total_tumor_patients, 'total_tumor_patients')\n",
    "percentage = (num_gene_mutated / total_tumor_patients)\n",
    "print('percent:',percentage)\n",
    "gene_mutated.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
