{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MUC5B Mutation Trans Effect on Proteomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook analyzes the trans effect of MUC5B mutation on interacting and other proteins, in Endometrial, Colon, and Ovarian cancer."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Library Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "import cptac\n",
    "import cptac.algorithms as al\n",
    "\n",
    "en = cptac.Endometrial()\n",
    "co = cptac.Colon()\n",
    "ov = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Specify Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = \"MUC5B\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Investigate Proteomics, Phosphoproteomics, Acetylproteomics, or Transcriptomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "omics = \"proteomics\"\n",
    "#omics = \"transcriptomics\"\n",
    "#omics = \"phosphoproteomics\"\n",
    "#omics = \"acetylproteomics\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Track all significant comparisons in dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_significant_comparisons = pd.DataFrame(columns=['Cancer_Type', 'Gene', 'Comparison','Interacting_Protein','P_Value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_to_all_significant_comparisons(df, cancer, interacting, all_sig_comp):\n",
    "    expanded = df\n",
    "    expanded['Gene'] = gene\n",
    "    expanded['Cancer_Type'] = cancer\n",
    "    expanded['Interacting_Protein'] = interacting\n",
    "    \n",
    "    updated_all_comparisons = pd.concat([all_sig_comp, expanded], sort=False)\n",
    "    \n",
    "    return updated_all_comparisons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Interacting Proteins: Proteomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate interacting protein list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a call to the cptac.algorithms get interacting proteins method, which interacts with the uniprot and string databases to generate a list of known interacting partners with the given gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Interacting Proteins:\n",
      "GALNT8\n",
      "ST6GAL1\n",
      "ST6GALNAC2\n",
      "ST3GAL3\n",
      "MUC17\n",
      "GALNT16\n",
      "GALNT12\n",
      "MUC21\n",
      "MUC3A\n",
      "ST3GAL2\n",
      "GCNT3\n",
      "MUC16\n",
      "C1GALT1\n",
      "MUC6\n",
      "MUC7\n",
      "MUC20\n",
      "MUC15\n",
      "MUC4\n",
      "GALNTL6\n",
      "ST3GAL4\n",
      "MUC5B\n",
      "MUC12\n",
      "B3GNT6\n",
      "MUC1\n",
      "MUC13\n",
      "MUC5AC\n",
      "UBC\n",
      "KIR2DS2\n",
      "C7orf25\n",
      "ICE2\n",
      "FRMD1\n",
      "PCK2\n",
      "HBM\n",
      "ARHGAP12\n",
      "TIMM50\n",
      "DDX31\n",
      "ESR2\n",
      "AGR2\n"
     ]
    }
   ],
   "source": [
    "'''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "print(\"Interacting Proteins:\")\n",
    "for interacting_protein in interacting_proteins:\n",
    "    print(interacting_protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GALNT8 did not match any columns in proteomics dataframe. GALNT8_proteomics column inserted, but filled with NaN.\n",
      "ST3GAL3 did not match any columns in proteomics dataframe. ST3GAL3_proteomics column inserted, but filled with NaN.\n",
      "MUC17 did not match any columns in proteomics dataframe. MUC17_proteomics column inserted, but filled with NaN.\n",
      "MUC21 did not match any columns in proteomics dataframe. MUC21_proteomics column inserted, but filled with NaN.\n",
      "MUC3A did not match any columns in proteomics dataframe. MUC3A_proteomics column inserted, but filled with NaN.\n",
      "MUC7 did not match any columns in proteomics dataframe. MUC7_proteomics column inserted, but filled with NaN.\n",
      "MUC20 did not match any columns in proteomics dataframe. MUC20_proteomics column inserted, but filled with NaN.\n",
      "MUC15 did not match any columns in proteomics dataframe. MUC15_proteomics column inserted, but filled with NaN.\n",
      "MUC4 did not match any columns in proteomics dataframe. MUC4_proteomics column inserted, but filled with NaN.\n",
      "GALNTL6 did not match any columns in proteomics dataframe. GALNTL6_proteomics column inserted, but filled with NaN.\n",
      "ST3GAL4 did not match any columns in proteomics dataframe. ST3GAL4_proteomics column inserted, but filled with NaN.\n",
      "MUC12 did not match any columns in proteomics dataframe. MUC12_proteomics column inserted, but filled with NaN.\n",
      "B3GNT6 did not match any columns in proteomics dataframe. B3GNT6_proteomics column inserted, but filled with NaN.\n",
      "UBC did not match any columns in proteomics dataframe. UBC_proteomics column inserted, but filled with NaN.\n",
      "KIR2DS2 did not match any columns in proteomics dataframe. KIR2DS2_proteomics column inserted, but filled with NaN.\n",
      "FRMD1 did not match any columns in proteomics dataframe. FRMD1_proteomics column inserted, but filled with NaN.\n",
      "ESR2 did not match any columns in proteomics dataframe. ESR2_proteomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GALNT8 did not match any columns in proteomics dataframe. GALNT8_proteomics column inserted, but filled with NaN.\n",
      "ST6GALNAC2 did not match any columns in proteomics dataframe. ST6GALNAC2_proteomics column inserted, but filled with NaN.\n",
      "ST3GAL3 did not match any columns in proteomics dataframe. ST3GAL3_proteomics column inserted, but filled with NaN.\n",
      "MUC17 did not match any columns in proteomics dataframe. MUC17_proteomics column inserted, but filled with NaN.\n",
      "GALNT16 did not match any columns in proteomics dataframe. GALNT16_proteomics column inserted, but filled with NaN.\n",
      "MUC21 did not match any columns in proteomics dataframe. MUC21_proteomics column inserted, but filled with NaN.\n",
      "MUC3A did not match any columns in proteomics dataframe. MUC3A_proteomics column inserted, but filled with NaN.\n",
      "ST3GAL2 did not match any columns in proteomics dataframe. ST3GAL2_proteomics column inserted, but filled with NaN.\n",
      "MUC16 did not match any columns in proteomics dataframe. MUC16_proteomics column inserted, but filled with NaN.\n",
      "MUC7 did not match any columns in proteomics dataframe. MUC7_proteomics column inserted, but filled with NaN.\n",
      "MUC20 did not match any columns in proteomics dataframe. MUC20_proteomics column inserted, but filled with NaN.\n",
      "MUC15 did not match any columns in proteomics dataframe. MUC15_proteomics column inserted, but filled with NaN.\n",
      "GALNTL6 did not match any columns in proteomics dataframe. GALNTL6_proteomics column inserted, but filled with NaN.\n",
      "MUC12 did not match any columns in proteomics dataframe. MUC12_proteomics column inserted, but filled with NaN.\n",
      "UBC did not match any columns in proteomics dataframe. UBC_proteomics column inserted, but filled with NaN.\n",
      "KIR2DS2 did not match any columns in proteomics dataframe. KIR2DS2_proteomics column inserted, but filled with NaN.\n",
      "ICE2 did not match any columns in proteomics dataframe. ICE2_proteomics column inserted, but filled with NaN.\n",
      "FRMD1 did not match any columns in proteomics dataframe. FRMD1_proteomics column inserted, but filled with NaN.\n",
      "ESR2 did not match any columns in proteomics dataframe. ESR2_proteomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "        Comparison   P_Value\n",
      "0  PCK2_proteomics  0.000011\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GALNT8 did not match any columns in proteomics dataframe. GALNT8_proteomics column inserted, but filled with NaN.\n",
      "ST6GALNAC2 did not match any columns in proteomics dataframe. ST6GALNAC2_proteomics column inserted, but filled with NaN.\n",
      "ST3GAL3 did not match any columns in proteomics dataframe. ST3GAL3_proteomics column inserted, but filled with NaN.\n",
      "MUC17 did not match any columns in proteomics dataframe. MUC17_proteomics column inserted, but filled with NaN.\n",
      "MUC21 did not match any columns in proteomics dataframe. MUC21_proteomics column inserted, but filled with NaN.\n",
      "MUC3A did not match any columns in proteomics dataframe. MUC3A_proteomics column inserted, but filled with NaN.\n",
      "ST3GAL2 did not match any columns in proteomics dataframe. ST3GAL2_proteomics column inserted, but filled with NaN.\n",
      "GCNT3 did not match any columns in proteomics dataframe. GCNT3_proteomics column inserted, but filled with NaN.\n",
      "MUC6 did not match any columns in proteomics dataframe. MUC6_proteomics column inserted, but filled with NaN.\n",
      "MUC7 did not match any columns in proteomics dataframe. MUC7_proteomics column inserted, but filled with NaN.\n",
      "MUC20 did not match any columns in proteomics dataframe. MUC20_proteomics column inserted, but filled with NaN.\n",
      "MUC15 did not match any columns in proteomics dataframe. MUC15_proteomics column inserted, but filled with NaN.\n",
      "GALNTL6 did not match any columns in proteomics dataframe. GALNTL6_proteomics column inserted, but filled with NaN.\n",
      "MUC12 did not match any columns in proteomics dataframe. MUC12_proteomics column inserted, but filled with NaN.\n",
      "B3GNT6 did not match any columns in proteomics dataframe. B3GNT6_proteomics column inserted, but filled with NaN.\n",
      "MUC13 did not match any columns in proteomics dataframe. MUC13_proteomics column inserted, but filled with NaN.\n",
      "MUC5AC did not match any columns in proteomics dataframe. MUC5AC_proteomics column inserted, but filled with NaN.\n",
      "UBC did not match any columns in proteomics dataframe. UBC_proteomics column inserted, but filled with NaN.\n",
      "KIR2DS2 did not match any columns in proteomics dataframe. KIR2DS2_proteomics column inserted, but filled with NaN.\n",
      "ESR2 did not match any columns in proteomics dataframe. ESR2_proteomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.loc[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.loc[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "#protdf = protdf.drop_duplicates()\n",
    "protdf = protdf.loc[:,~protdf.columns.duplicated()]\n",
    "\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", True, all_significant_comparisons)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All Proteins: Proteomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  MUC5B\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  MUC5B\n",
      "Doing t-test comparisons\n",
      "\n",
      "            Comparison       P_Value\n",
      "0     MRE11_proteomics  1.144045e-10\n",
      "1      GPX2_proteomics  5.988006e-10\n",
      "2    ANP32E_proteomics  3.042503e-09\n",
      "3     KRT20_proteomics  3.969811e-09\n",
      "4     RAD50_proteomics  9.817991e-09\n",
      "5     PRDX5_proteomics  5.077905e-08\n",
      "6   ARHGEF2_proteomics  7.919124e-08\n",
      "7     NFKB2_proteomics  1.471187e-07\n",
      "8    MRPS31_proteomics  3.790579e-07\n",
      "9     PCSK9_proteomics  4.405533e-07\n",
      "10    HNF4A_proteomics  6.761123e-07\n",
      "11      VTN_proteomics  8.848484e-07\n",
      "12   MYO15B_proteomics  9.027894e-07\n",
      "13    NUBPL_proteomics  9.960365e-07\n",
      "14   FUNDC1_proteomics  1.025502e-06\n",
      "15    ACSL5_proteomics  1.289272e-06\n",
      "16     ACE2_proteomics  1.332648e-06\n",
      "17   GALNT1_proteomics  1.374055e-06\n",
      "18     A1CF_proteomics  1.503580e-06\n",
      "19  TFCP2L1_proteomics  1.666901e-06\n",
      "20  SLC16A3_proteomics  2.262895e-06\n",
      "21  PLA2G4A_proteomics  2.320262e-06\n",
      "22    AIFM1_proteomics  2.482703e-06\n",
      "23     GYS1_proteomics  2.763100e-06\n",
      "24   ENGASE_proteomics  2.794402e-06\n",
      "25      PKM_proteomics  3.085404e-06\n",
      "26   SMCHD1_proteomics  3.179521e-06\n",
      "27  ARHGEF1_proteomics  3.802987e-06\n",
      "28      C8A_proteomics  4.031319e-06\n",
      "29    ALDH2_proteomics  4.510004e-06\n",
      "30       F9_proteomics  4.895850e-06\n",
      "31     VAPB_proteomics  4.916239e-06\n",
      "32    SSFA2_proteomics  4.989103e-06\n",
      "33    MMS19_proteomics  5.148296e-06\n",
      "34    CFHR5_proteomics  5.171754e-06\n",
      "35       C7_proteomics  5.632118e-06\n",
      "36     RIDA_proteomics  6.171126e-06\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "    \n",
    "    '''Use all proteins'''\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the dataframe correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  MUC5B\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nGene: \", gene)\n",
    "\n",
    "'''Use all proteins'''\n",
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.loc[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.loc[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "protdf = protdf.loc[:,~protdf.columns.duplicated()]\n",
    "\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", False, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Print all significant comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Interacting_Protein</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>PCK2_proteomics</td>\n",
       "      <td>True</td>\n",
       "      <td>1.136850e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MRE11_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.144045e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>GPX2_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.988006e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ANP32E_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.042503e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>KRT20_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.969811e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>RAD50_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.817991e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>PRDX5_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.077905e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ARHGEF2_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>7.919124e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>NFKB2_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.471187e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MRPS31_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.790579e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>PCSK9_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.405533e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>HNF4A_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.761123e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>VTN_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>8.848484e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MYO15B_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.027894e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>NUBPL_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.960365e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>FUNDC1_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.025502e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ACSL5_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.289272e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ACE2_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.332648e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>GALNT1_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.374055e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>A1CF_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.503580e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>TFCP2L1_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.666901e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>SLC16A3_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.262895e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>PLA2G4A_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.320262e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>AIFM1_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.482703e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>GYS1_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.763100e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ENGASE_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.794402e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>PKM_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.085404e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>SMCHD1_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.179521e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ARHGEF1_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.802987e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>C8A_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.031319e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ALDH2_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.510004e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>F9_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.895850e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>VAPB_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.916239e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>SSFA2_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.989103e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MMS19_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.148296e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>CFHR5_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.171754e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>C7_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.632118e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>RIDA_proteomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.171126e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Cancer_Type   Gene          Comparison Interacting_Protein       P_Value\n",
       "0        Colon  MUC5B     PCK2_proteomics                True  1.136850e-05\n",
       "0        Colon  MUC5B    MRE11_proteomics               False  1.144045e-10\n",
       "1        Colon  MUC5B     GPX2_proteomics               False  5.988006e-10\n",
       "2        Colon  MUC5B   ANP32E_proteomics               False  3.042503e-09\n",
       "3        Colon  MUC5B    KRT20_proteomics               False  3.969811e-09\n",
       "4        Colon  MUC5B    RAD50_proteomics               False  9.817991e-09\n",
       "5        Colon  MUC5B    PRDX5_proteomics               False  5.077905e-08\n",
       "6        Colon  MUC5B  ARHGEF2_proteomics               False  7.919124e-08\n",
       "7        Colon  MUC5B    NFKB2_proteomics               False  1.471187e-07\n",
       "8        Colon  MUC5B   MRPS31_proteomics               False  3.790579e-07\n",
       "9        Colon  MUC5B    PCSK9_proteomics               False  4.405533e-07\n",
       "10       Colon  MUC5B    HNF4A_proteomics               False  6.761123e-07\n",
       "11       Colon  MUC5B      VTN_proteomics               False  8.848484e-07\n",
       "12       Colon  MUC5B   MYO15B_proteomics               False  9.027894e-07\n",
       "13       Colon  MUC5B    NUBPL_proteomics               False  9.960365e-07\n",
       "14       Colon  MUC5B   FUNDC1_proteomics               False  1.025502e-06\n",
       "15       Colon  MUC5B    ACSL5_proteomics               False  1.289272e-06\n",
       "16       Colon  MUC5B     ACE2_proteomics               False  1.332648e-06\n",
       "17       Colon  MUC5B   GALNT1_proteomics               False  1.374055e-06\n",
       "18       Colon  MUC5B     A1CF_proteomics               False  1.503580e-06\n",
       "19       Colon  MUC5B  TFCP2L1_proteomics               False  1.666901e-06\n",
       "20       Colon  MUC5B  SLC16A3_proteomics               False  2.262895e-06\n",
       "21       Colon  MUC5B  PLA2G4A_proteomics               False  2.320262e-06\n",
       "22       Colon  MUC5B    AIFM1_proteomics               False  2.482703e-06\n",
       "23       Colon  MUC5B     GYS1_proteomics               False  2.763100e-06\n",
       "24       Colon  MUC5B   ENGASE_proteomics               False  2.794402e-06\n",
       "25       Colon  MUC5B      PKM_proteomics               False  3.085404e-06\n",
       "26       Colon  MUC5B   SMCHD1_proteomics               False  3.179521e-06\n",
       "27       Colon  MUC5B  ARHGEF1_proteomics               False  3.802987e-06\n",
       "28       Colon  MUC5B      C8A_proteomics               False  4.031319e-06\n",
       "29       Colon  MUC5B    ALDH2_proteomics               False  4.510004e-06\n",
       "30       Colon  MUC5B       F9_proteomics               False  4.895850e-06\n",
       "31       Colon  MUC5B     VAPB_proteomics               False  4.916239e-06\n",
       "32       Colon  MUC5B    SSFA2_proteomics               False  4.989103e-06\n",
       "33       Colon  MUC5B    MMS19_proteomics               False  5.148296e-06\n",
       "34       Colon  MUC5B    CFHR5_proteomics               False  5.171754e-06\n",
       "35       Colon  MUC5B       C7_proteomics               False  5.632118e-06\n",
       "36       Colon  MUC5B     RIDA_proteomics               False  6.171126e-06"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if len(all_significant_comparisons) > 0:\n",
    "    display(all_significant_comparisons)\n",
    "    \n",
    "else:\n",
    "    print('No Significant Comparisons!')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Write significant comparisons (if any) to shared CSV file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "existing_results = pd.read_csv(gene+'_Trans_Results.csv')\n",
    "\n",
    "updated_results = pd.concat([existing_results, all_significant_comparisons], sort=False)\n",
    "\n",
    "updated_results.to_csv(path_or_buf = gene + '_Trans_Results.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
