{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MUC5B Mutation Trans Effect on Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook analyzes the trans effect of MUC5B mutation on interacting and other proteins Transcriptomics, in Endometrial, Colon, and Ovarian cancer."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Library Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "import cptac\n",
    "import cptac.algorithms as al\n",
    "\n",
    "en = cptac.Endometrial()\n",
    "co = cptac.Colon()\n",
    "ov = cptac.Ovarian()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = \"MUC5B\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Investigate Proteomics, Phosphoproteomics, Acetylproteomics, or Transcriptomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#omics = \"proteomics\"\n",
    "omics = \"transcriptomics\"\n",
    "#omics = \"phosphoproteomics\"\n",
    "#omics = \"acetylproteomics\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Track all significant comparisons in Dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_significant_comparisons = pd.DataFrame(columns=['Cancer_Type', 'Gene', 'Comparison','Interacting_Protein','P_Value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_to_all_significant_comparisons(df, cancer, interacting, all_sig_comp):\n",
    "    expanded = df\n",
    "    expanded['Gene'] = gene\n",
    "    expanded['Cancer_Type'] = cancer\n",
    "    expanded['Interacting_Protein'] = interacting\n",
    "    \n",
    "    updated_all_comparisons = pd.concat([all_sig_comp, expanded], sort=False)\n",
    "    \n",
    "    return updated_all_comparisons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Interacting Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate interacting protein list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a call to the cptac.algorithms get interacting proteins method, which interacts with the uniprot and string databases to generate a list of known interacting partners with the given gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Interacting Proteins:\n",
      "GALNT8\n",
      "ST6GAL1\n",
      "ST6GALNAC2\n",
      "ST3GAL3\n",
      "MUC17\n",
      "GALNT16\n",
      "GALNT12\n",
      "MUC21\n",
      "MUC3A\n",
      "ST3GAL2\n",
      "GCNT3\n",
      "MUC16\n",
      "C1GALT1\n",
      "MUC6\n",
      "MUC7\n",
      "MUC20\n",
      "MUC15\n",
      "MUC4\n",
      "GALNTL6\n",
      "ST3GAL4\n",
      "MUC5B\n",
      "MUC12\n",
      "B3GNT6\n",
      "MUC1\n",
      "MUC13\n",
      "MUC5AC\n",
      "UBC\n",
      "KIR2DS2\n",
      "C7orf25\n",
      "ICE2\n",
      "FRMD1\n",
      "PCK2\n",
      "HBM\n",
      "ARHGAP12\n",
      "TIMM50\n",
      "DDX31\n",
      "ESR2\n",
      "AGR2\n"
     ]
    }
   ],
   "source": [
    "'''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "'''Show interacting protein list'''\n",
    "print(\"Interacting Proteins:\")\n",
    "for interacting_protein in interacting_proteins:\n",
    "    print(interacting_protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", True, all_significant_comparisons)\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MUC17 did not match any columns in transcriptomics dataframe. MUC17_transcriptomics column inserted, but filled with NaN.\n",
      "GALNT16 did not match any columns in transcriptomics dataframe. GALNT16_transcriptomics column inserted, but filled with NaN.\n",
      "MUC21 did not match any columns in transcriptomics dataframe. MUC21_transcriptomics column inserted, but filled with NaN.\n",
      "MUC16 did not match any columns in transcriptomics dataframe. MUC16_transcriptomics column inserted, but filled with NaN.\n",
      "MUC6 did not match any columns in transcriptomics dataframe. MUC6_transcriptomics column inserted, but filled with NaN.\n",
      "MUC7 did not match any columns in transcriptomics dataframe. MUC7_transcriptomics column inserted, but filled with NaN.\n",
      "MUC15 did not match any columns in transcriptomics dataframe. MUC15_transcriptomics column inserted, but filled with NaN.\n",
      "GALNTL6 did not match any columns in transcriptomics dataframe. GALNTL6_transcriptomics column inserted, but filled with NaN.\n",
      "MUC5B did not match any columns in transcriptomics dataframe. MUC5B_transcriptomics column inserted, but filled with NaN.\n",
      "MUC5AC did not match any columns in transcriptomics dataframe. MUC5AC_transcriptomics column inserted, but filled with NaN.\n",
      "KIR2DS2 did not match any columns in transcriptomics dataframe. KIR2DS2_transcriptomics column inserted, but filled with NaN.\n",
      "HBM did not match any columns in transcriptomics dataframe. HBM_transcriptomics column inserted, but filled with NaN.\n",
      "ESR2 did not match any columns in transcriptomics dataframe. ESR2_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "                   Comparison   P_Value\n",
      "0       MUC20_transcriptomics  0.000004\n",
      "1       FRMD1_transcriptomics  0.000013\n",
      "2     ST6GAL1_transcriptomics  0.000067\n",
      "3       MUC12_transcriptomics  0.000111\n",
      "4  ST6GALNAC2_transcriptomics  0.000615\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GALNT8 did not match any columns in transcriptomics dataframe. GALNT8_transcriptomics column inserted, but filled with NaN.\n",
      "C1GALT1 did not match any columns in transcriptomics dataframe. C1GALT1_transcriptomics column inserted, but filled with NaN.\n",
      "MUC20 did not match any columns in transcriptomics dataframe. MUC20_transcriptomics column inserted, but filled with NaN.\n",
      "MUC1 did not match any columns in transcriptomics dataframe. MUC1_transcriptomics column inserted, but filled with NaN.\n",
      "MUC5AC did not match any columns in transcriptomics dataframe. MUC5AC_transcriptomics column inserted, but filled with NaN.\n",
      "ICE2 did not match any columns in transcriptomics dataframe. ICE2_transcriptomics column inserted, but filled with NaN.\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", True, all_significant_comparisons)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Expand our search for significant comparisons to all proteins in our dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Endometrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  MUC5B\n",
      "Doing t-test comparisons\n",
      "\n",
      "                Comparison   P_Value\n",
      "0  FAM86HP_transcriptomics  0.000001\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = en.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Endometrial\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  MUC5B\n",
      "Doing t-test comparisons\n",
      "\n",
      "                     Comparison       P_Value\n",
      "0          GPX2_transcriptomics  1.758001e-10\n",
      "1       POU5F1B_transcriptomics  4.688794e-10\n",
      "2        RUBCNL_transcriptomics  1.541604e-09\n",
      "3        ATP10B_transcriptomics  3.068562e-09\n",
      "4         NUBPL_transcriptomics  3.884908e-09\n",
      "5       CEACAM5_transcriptomics  6.870982e-09\n",
      "6       CEACAM6_transcriptomics  9.942373e-09\n",
      "7         ADNP2_transcriptomics  1.155737e-08\n",
      "8    DLGAP1-AS2_transcriptomics  1.221313e-08\n",
      "9      TOR1AIP2_transcriptomics  1.275600e-08\n",
      "10         GRM8_transcriptomics  1.851409e-08\n",
      "11       LY6G6D_transcriptomics  2.788723e-08\n",
      "12        CCSAP_transcriptomics  3.180643e-08\n",
      "13        EPDR1_transcriptomics  3.808045e-08\n",
      "14         ACE2_transcriptomics  4.376884e-08\n",
      "15        NR1I2_transcriptomics  4.771506e-08\n",
      "16       MAPK12_transcriptomics  5.517810e-08\n",
      "17       CASC21_transcriptomics  5.951666e-08\n",
      "18    LINC00513_transcriptomics  9.176024e-08\n",
      "19       GPR143_transcriptomics  1.071900e-07\n",
      "20        PIAS2_transcriptomics  1.264160e-07\n",
      "21    LINC01560_transcriptomics  1.514142e-07\n",
      "22  THUMPD3-AS1_transcriptomics  1.689576e-07\n",
      "23     SLC25A13_transcriptomics  1.864544e-07\n",
      "24        CXXC1_transcriptomics  2.088444e-07\n",
      "25        TNNC2_transcriptomics  2.443947e-07\n",
      "26        TDGF1_transcriptomics  2.727111e-07\n",
      "27         CFTR_transcriptomics  2.870300e-07\n",
      "28      CTTNBP2_transcriptomics  3.082151e-07\n",
      "29       TSPAN6_transcriptomics  3.718990e-07\n",
      "..                          ...           ...\n",
      "47        MRE11_transcriptomics  1.403404e-06\n",
      "48         PIGX_transcriptomics  1.418377e-06\n",
      "49          MBP_transcriptomics  1.546044e-06\n",
      "50        MYZAP_transcriptomics  1.561001e-06\n",
      "51        PRAP1_transcriptomics  1.692921e-06\n",
      "52       SEMA5A_transcriptomics  1.824961e-06\n",
      "53       THRAP3_transcriptomics  2.186983e-06\n",
      "54         MLH1_transcriptomics  2.270186e-06\n",
      "55        DDX27_transcriptomics  2.290637e-06\n",
      "56         LMNA_transcriptomics  2.326658e-06\n",
      "57       SLC3A1_transcriptomics  2.421983e-06\n",
      "58        LIN54_transcriptomics  2.553519e-06\n",
      "59         IER5_transcriptomics  2.572737e-06\n",
      "60         RAC3_transcriptomics  2.702800e-06\n",
      "61        KIF14_transcriptomics  2.727888e-06\n",
      "62       GUCY2C_transcriptomics  2.736175e-06\n",
      "63        SOCS6_transcriptomics  2.843641e-06\n",
      "64         VAPB_transcriptomics  2.907669e-06\n",
      "65         QPRT_transcriptomics  2.953795e-06\n",
      "66        ATP9A_transcriptomics  2.977994e-06\n",
      "67        CRAT8_transcriptomics  3.196294e-06\n",
      "68         TYMS_transcriptomics  3.330010e-06\n",
      "69      CXorf23_transcriptomics  3.369104e-06\n",
      "70      SPATA25_transcriptomics  3.383985e-06\n",
      "71       POFUT1_transcriptomics  3.506052e-06\n",
      "72         AAR2_transcriptomics  3.522057e-06\n",
      "73       ZNF737_transcriptomics  3.535392e-06\n",
      "74    LINC01006_transcriptomics  3.607070e-06\n",
      "75        RNF43_transcriptomics  3.665086e-06\n",
      "76        GSPT2_transcriptomics  3.699970e-06\n",
      "\n",
      "[77 rows x 2 columns]\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    '''Use all proteins'''\n",
    "\n",
    "    '''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "    protdf = co.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    '''Create the binary valued column needed to do the comparison'''\n",
    "    for ind, row in protdf.iterrows():\n",
    "        if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "            protdf.at[ind,'Label'] = 'Mutated'\n",
    "        else:\n",
    "            protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "    '''Format the datafram correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    '''Make list of columns to be compared using t-tests'''\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    '''Call wrap_ttest, pass in formatted dataframe'''\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    '''Print results, if anything significant was found'''\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Colon\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  MUC5B\n",
      "Doing t-test comparisons\n",
      "\n",
      "No significant comparisons.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(\"\\nGene: \", gene)\n",
    "\n",
    "'''Use all proteins'''\n",
    "\n",
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = ov.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "protdf = protdf.loc[:,~protdf.columns.duplicated()]\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the datafram correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Ovarian\", False, all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Print All Signififant Comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Interacting_Protein</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MUC20_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>4.319530e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>FRMD1_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>1.340157e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ST6GAL1_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>6.725471e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MUC12_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>1.108167e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ST6GALNAC2_transcriptomics</td>\n",
       "      <td>True</td>\n",
       "      <td>6.152779e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Endometrial</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>FAM86HP_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.094390e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>GPX2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.758001e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>POU5F1B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.688794e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>RUBCNL_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.541604e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ATP10B_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.068562e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>NUBPL_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.884908e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>CEACAM5_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>6.870982e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>CEACAM6_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.942373e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ADNP2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.155737e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>DLGAP1-AS2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.221313e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>TOR1AIP2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.275600e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>GRM8_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.851409e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>LY6G6D_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.788723e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>CCSAP_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.180643e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>EPDR1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.808045e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ACE2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.376884e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>NR1I2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>4.771506e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MAPK12_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.517810e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>CASC21_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>5.951666e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>LINC00513_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>9.176024e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>GPR143_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.071900e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>PIAS2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.264160e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>LINC01560_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.514142e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>THUMPD3-AS1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.689576e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>SLC25A13_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.864544e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MRE11_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.403404e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>PIGX_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.418377e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MBP_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.546044e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MYZAP_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.561001e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>PRAP1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.692921e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>SEMA5A_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>1.824961e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>THRAP3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.186983e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>MLH1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.270186e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>DDX27_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.290637e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>LMNA_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.326658e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>SLC3A1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.421983e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>LIN54_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.553519e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>IER5_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.572737e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>RAC3_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.702800e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>KIF14_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.727888e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>GUCY2C_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.736175e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>SOCS6_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.843641e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>VAPB_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.907669e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>QPRT_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.953795e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ATP9A_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>2.977994e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>CRAT8_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.196294e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>TYMS_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.330010e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>CXorf23_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.369104e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>SPATA25_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.383985e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>POFUT1_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.506052e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>AAR2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.522057e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>ZNF737_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.535392e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>LINC01006_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.607070e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>RNF43_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.665086e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>Colon</td>\n",
       "      <td>MUC5B</td>\n",
       "      <td>GSPT2_transcriptomics</td>\n",
       "      <td>False</td>\n",
       "      <td>3.699970e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>83 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Cancer_Type   Gene                   Comparison Interacting_Protein  \\\n",
       "0         Colon  MUC5B        MUC20_transcriptomics                True   \n",
       "1         Colon  MUC5B        FRMD1_transcriptomics                True   \n",
       "2         Colon  MUC5B      ST6GAL1_transcriptomics                True   \n",
       "3         Colon  MUC5B        MUC12_transcriptomics                True   \n",
       "4         Colon  MUC5B   ST6GALNAC2_transcriptomics                True   \n",
       "0   Endometrial  MUC5B      FAM86HP_transcriptomics               False   \n",
       "0         Colon  MUC5B         GPX2_transcriptomics               False   \n",
       "1         Colon  MUC5B      POU5F1B_transcriptomics               False   \n",
       "2         Colon  MUC5B       RUBCNL_transcriptomics               False   \n",
       "3         Colon  MUC5B       ATP10B_transcriptomics               False   \n",
       "4         Colon  MUC5B        NUBPL_transcriptomics               False   \n",
       "5         Colon  MUC5B      CEACAM5_transcriptomics               False   \n",
       "6         Colon  MUC5B      CEACAM6_transcriptomics               False   \n",
       "7         Colon  MUC5B        ADNP2_transcriptomics               False   \n",
       "8         Colon  MUC5B   DLGAP1-AS2_transcriptomics               False   \n",
       "9         Colon  MUC5B     TOR1AIP2_transcriptomics               False   \n",
       "10        Colon  MUC5B         GRM8_transcriptomics               False   \n",
       "11        Colon  MUC5B       LY6G6D_transcriptomics               False   \n",
       "12        Colon  MUC5B        CCSAP_transcriptomics               False   \n",
       "13        Colon  MUC5B        EPDR1_transcriptomics               False   \n",
       "14        Colon  MUC5B         ACE2_transcriptomics               False   \n",
       "15        Colon  MUC5B        NR1I2_transcriptomics               False   \n",
       "16        Colon  MUC5B       MAPK12_transcriptomics               False   \n",
       "17        Colon  MUC5B       CASC21_transcriptomics               False   \n",
       "18        Colon  MUC5B    LINC00513_transcriptomics               False   \n",
       "19        Colon  MUC5B       GPR143_transcriptomics               False   \n",
       "20        Colon  MUC5B        PIAS2_transcriptomics               False   \n",
       "21        Colon  MUC5B    LINC01560_transcriptomics               False   \n",
       "22        Colon  MUC5B  THUMPD3-AS1_transcriptomics               False   \n",
       "23        Colon  MUC5B     SLC25A13_transcriptomics               False   \n",
       "..          ...    ...                          ...                 ...   \n",
       "47        Colon  MUC5B        MRE11_transcriptomics               False   \n",
       "48        Colon  MUC5B         PIGX_transcriptomics               False   \n",
       "49        Colon  MUC5B          MBP_transcriptomics               False   \n",
       "50        Colon  MUC5B        MYZAP_transcriptomics               False   \n",
       "51        Colon  MUC5B        PRAP1_transcriptomics               False   \n",
       "52        Colon  MUC5B       SEMA5A_transcriptomics               False   \n",
       "53        Colon  MUC5B       THRAP3_transcriptomics               False   \n",
       "54        Colon  MUC5B         MLH1_transcriptomics               False   \n",
       "55        Colon  MUC5B        DDX27_transcriptomics               False   \n",
       "56        Colon  MUC5B         LMNA_transcriptomics               False   \n",
       "57        Colon  MUC5B       SLC3A1_transcriptomics               False   \n",
       "58        Colon  MUC5B        LIN54_transcriptomics               False   \n",
       "59        Colon  MUC5B         IER5_transcriptomics               False   \n",
       "60        Colon  MUC5B         RAC3_transcriptomics               False   \n",
       "61        Colon  MUC5B        KIF14_transcriptomics               False   \n",
       "62        Colon  MUC5B       GUCY2C_transcriptomics               False   \n",
       "63        Colon  MUC5B        SOCS6_transcriptomics               False   \n",
       "64        Colon  MUC5B         VAPB_transcriptomics               False   \n",
       "65        Colon  MUC5B         QPRT_transcriptomics               False   \n",
       "66        Colon  MUC5B        ATP9A_transcriptomics               False   \n",
       "67        Colon  MUC5B        CRAT8_transcriptomics               False   \n",
       "68        Colon  MUC5B         TYMS_transcriptomics               False   \n",
       "69        Colon  MUC5B      CXorf23_transcriptomics               False   \n",
       "70        Colon  MUC5B      SPATA25_transcriptomics               False   \n",
       "71        Colon  MUC5B       POFUT1_transcriptomics               False   \n",
       "72        Colon  MUC5B         AAR2_transcriptomics               False   \n",
       "73        Colon  MUC5B       ZNF737_transcriptomics               False   \n",
       "74        Colon  MUC5B    LINC01006_transcriptomics               False   \n",
       "75        Colon  MUC5B        RNF43_transcriptomics               False   \n",
       "76        Colon  MUC5B        GSPT2_transcriptomics               False   \n",
       "\n",
       "         P_Value  \n",
       "0   4.319530e-06  \n",
       "1   1.340157e-05  \n",
       "2   6.725471e-05  \n",
       "3   1.108167e-04  \n",
       "4   6.152779e-04  \n",
       "0   1.094390e-06  \n",
       "0   1.758001e-10  \n",
       "1   4.688794e-10  \n",
       "2   1.541604e-09  \n",
       "3   3.068562e-09  \n",
       "4   3.884908e-09  \n",
       "5   6.870982e-09  \n",
       "6   9.942373e-09  \n",
       "7   1.155737e-08  \n",
       "8   1.221313e-08  \n",
       "9   1.275600e-08  \n",
       "10  1.851409e-08  \n",
       "11  2.788723e-08  \n",
       "12  3.180643e-08  \n",
       "13  3.808045e-08  \n",
       "14  4.376884e-08  \n",
       "15  4.771506e-08  \n",
       "16  5.517810e-08  \n",
       "17  5.951666e-08  \n",
       "18  9.176024e-08  \n",
       "19  1.071900e-07  \n",
       "20  1.264160e-07  \n",
       "21  1.514142e-07  \n",
       "22  1.689576e-07  \n",
       "23  1.864544e-07  \n",
       "..           ...  \n",
       "47  1.403404e-06  \n",
       "48  1.418377e-06  \n",
       "49  1.546044e-06  \n",
       "50  1.561001e-06  \n",
       "51  1.692921e-06  \n",
       "52  1.824961e-06  \n",
       "53  2.186983e-06  \n",
       "54  2.270186e-06  \n",
       "55  2.290637e-06  \n",
       "56  2.326658e-06  \n",
       "57  2.421983e-06  \n",
       "58  2.553519e-06  \n",
       "59  2.572737e-06  \n",
       "60  2.702800e-06  \n",
       "61  2.727888e-06  \n",
       "62  2.736175e-06  \n",
       "63  2.843641e-06  \n",
       "64  2.907669e-06  \n",
       "65  2.953795e-06  \n",
       "66  2.977994e-06  \n",
       "67  3.196294e-06  \n",
       "68  3.330010e-06  \n",
       "69  3.369104e-06  \n",
       "70  3.383985e-06  \n",
       "71  3.506052e-06  \n",
       "72  3.522057e-06  \n",
       "73  3.535392e-06  \n",
       "74  3.607070e-06  \n",
       "75  3.665086e-06  \n",
       "76  3.699970e-06  \n",
       "\n",
       "[83 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if len(all_significant_comparisons) == 0:\n",
    "    print('No Significant Comparisons!')\n",
    "    \n",
    "else:\n",
    "    display(all_significant_comparisons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Write Significant Comparisons (if any) to Shared CSV file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "existing_results = pd.read_csv(gene+'_Trans_Results.csv')\n",
    "\n",
    "updated_results = pd.concat([existing_results, all_significant_comparisons], sort=False)\n",
    "\n",
    "updated_results.to_csv(path_or_buf = gene + '_Trans_Results.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
