{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get_frequently_mutated Final "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[Kding transcriptomics data.....ata....\r"
     ]
    }
   ],
   "source": [
    "import cptac\n",
    "#en = cptac.Endometrial()\n",
    "#cptac.sync('endometrial', version='2.1')\n",
    "#ov = cptac.Ovarian()\n",
    "#cptac.sync(dataset='ovarian', version='0.0')\n",
    "#cptac.sync(dataset='colon', version='0.0')\n",
    "co = cptac.Colon()\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_frequently_mutated(cancer_object, cutoff = 0.1):  \n",
    "    \"\"\"\n",
    "    take cancer object and find the frequently \n",
    "    mutated genes in the total tumors compared to the cutoff.\n",
    "\n",
    "    Parameters:\n",
    "    cancer_object (object): cancer class object from cptac module \n",
    "    cutoff (float): used as comparison to determine the \n",
    "                    status of gene mutation frequency\n",
    "\n",
    "    Returns:\n",
    "    freq_mutated_df (pd.DataFrame): DataFrame of frequently \n",
    "        mutated genes passing the cutoff. Columns contain the \n",
    "        fractions of total unique mutations,missence type \n",
    "        mutations, and truncation type mutations per gene.\n",
    "    \n",
    "    The Missence_Mut column includes: \n",
    "        In_Frame_Del, In_Frame_Ins, Missense_Mutation\n",
    "   \n",
    "   The Truncation_Mut column includes: \n",
    "        Frame_Shift_Del, Frame_Shift_Ins, Splice_Site, \n",
    "        Nonsense_Mutation, Nonstop_Mutation\n",
    "        \n",
    "    These columns count multiple mutations of one gene in the \n",
    "    same sample, so fractions in the last two columns may \n",
    "    exceed the Unique_Samples_Mut column which only counts if \n",
    "    the gene was mutated once per sample.\"\"\"    \n",
    "    \n",
    "    # Get mutations data frame (whole exome sequencing)\n",
    "    somatic_mutations = cancer_object.get_mutations()\n",
    "    sample_status_map = cancer_object._get_sample_status_map()\n",
    "         \n",
    "    # Get total tumors/patients\n",
    "    sample_status_count = sample_status_map.value_counts().reset_index()\n",
    "    tumors = sample_status_count.loc[sample_status_count['index'] == 'Tumor']\n",
    "    total_tumor_patients = tumors.iloc[0,1] #number of tumors \n",
    "    print(total_tumor_patients)\n",
    "    # Find frequently mutated genes and their fraction of unique mutated samples.\n",
    "    #move 'Sample_ID' from index to column\n",
    "    origin_df = somatic_mutations.reset_index()\n",
    "    \n",
    "    # Drop silent mutations for Ovarian dataset\n",
    "    if cancer_object.get_cancer_type() == 'ovarian':\n",
    "        origin_df = origin_df.loc[origin_df['Mutation'] != 'Silent']\n",
    "        #print(origin_df['Mutation'].unique())\n",
    "        \n",
    "    #group by gene and count unique samples\n",
    "    count_mutations = origin_df.groupby(['Gene']).nunique()\n",
    "\n",
    "    #format\n",
    "    count_mutations = count_mutations.rename(columns={\"Sample_ID\": \"Unique_Samples_Mut\"})\n",
    "    count_mutations = count_mutations.drop(['Gene', 'Mutation','Location'], axis = 1)\n",
    "    \n",
    "    #filter using the cutoff and create fraction\n",
    "    fraction_mutated = count_mutations.apply(lambda x: x / total_tumor_patients)\n",
    "    fraction_greater_than_cutoff = fraction_mutated.where(lambda x: x > cutoff) #na used when false\n",
    "    filtered_gene_df = fraction_greater_than_cutoff.dropna()\n",
    "    \n",
    "    \n",
    "    #Create Missence and Trucation data frame\n",
    "    \n",
    "    #create two categories in Mutation column\n",
    "    if cancer_object.get_cancer_type() == 'colon':\n",
    "        missence_truncation_groups = {'frameshift substitution': 'T', \n",
    "            'frameshift deletion': 'T', 'frameshift insertion': 'T', \n",
    "            'stopgain': 'T', 'stoploss': 'T', 'nonsynonymous SNV': 'M',\n",
    "            'nonframeshift insertion': 'M','nonframeshift deletion': 'M', \n",
    "            'nonframeshift substitution': 'M'}\n",
    "    else: \n",
    "        missence_truncation_groups = {'In_Frame_Del': 'M', 'In_Frame_Ins': 'M',\n",
    "            'Missense_Mutation': 'M', 'Frame_Shift_Del': 'T','Nonsense_Mutation': 'T', \n",
    "            'Splice_Site': 'T', 'Frame_Shift_Ins': 'T','Nonstop_Mutation':'T'}\n",
    "       \n",
    "    mutations_replaced_M_T = origin_df.replace(missence_truncation_groups)\n",
    "    \n",
    "    # group mutation categories\n",
    "    miss = mutations_replaced_M_T.loc[mutations_replaced_M_T['Mutation'] == 'M']\n",
    "    trunc = mutations_replaced_M_T.loc[mutations_replaced_M_T['Mutation'] == 'T']\n",
    "\n",
    "    # group by gene and count unique samples for both categories\n",
    "    count_miss = miss.groupby(['Gene']).nunique()\n",
    "    count_trunc = trunc.groupby(['Gene']).nunique()\n",
    "\n",
    "    #format\n",
    "    missence_df = count_miss.rename(columns={\"Sample_ID\": \"Missence_Mut\"})\n",
    "    missence_df = missence_df.drop(['Gene', 'Mutation', 'Location'], axis = 1)\n",
    "\n",
    "    truncation_df = count_trunc.rename(columns={\"Sample_ID\": \"Truncation_Mut\"})\n",
    "    truncation_df = truncation_df.drop(['Gene', 'Mutation', 'Location'], axis = 1)\n",
    "\n",
    "    #join miss and trunc and change nan to 0, then divide by total tumors\n",
    "    join_mutations = missence_df.join(truncation_df).fillna(0)\n",
    "    missence_and_truncation_df = join_mutations.apply(lambda x: x / total_tumor_patients)\n",
    "\n",
    "\n",
    "    #Join data frames, keeping only the genes that passed the cutoff \n",
    "    freq_mutated_df = filtered_gene_df.join(missence_and_truncation_df).reset_index()\n",
    "    freq_mutated_df.name = 'frequently_mutated'\n",
    "    \n",
    "    return freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "110\n"
     ]
    }
   ],
   "source": [
    "df = get_frequently_mutated(co, .1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Unique_Samples_Mut</th>\n",
       "      <th>Missence_Mut</th>\n",
       "      <th>Truncation_Mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA13</td>\n",
       "      <td>0.172727</td>\n",
       "      <td>0.145455</td>\n",
       "      <td>0.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCA2</td>\n",
       "      <td>0.154545</td>\n",
       "      <td>0.145455</td>\n",
       "      <td>0.027273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCA4</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.072727</td>\n",
       "      <td>0.054545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCB4</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.054545</td>\n",
       "      <td>0.063636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.009091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ACVR2A</td>\n",
       "      <td>0.227273</td>\n",
       "      <td>0.018182</td>\n",
       "      <td>0.227273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ADAMTS16</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.009091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ADAMTS5</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ADAMTS7</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ADAMTSL4</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.054545</td>\n",
       "      <td>0.063636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ADGRG4</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ADGRV1</td>\n",
       "      <td>0.172727</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.081818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AHDC1</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.081818</td>\n",
       "      <td>0.045455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AHNAK</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AHNAK2</td>\n",
       "      <td>0.190909</td>\n",
       "      <td>0.172727</td>\n",
       "      <td>0.027273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AKAP13</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.009091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AKAP9</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.072727</td>\n",
       "      <td>0.072727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ANK1</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.027273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ANK2</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.036364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ANK3</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.045455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ANKRD11</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.036364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>APC</td>\n",
       "      <td>0.745455</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.727273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>APC2</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.063636</td>\n",
       "      <td>0.054545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>APOB</td>\n",
       "      <td>0.172727</td>\n",
       "      <td>0.154545</td>\n",
       "      <td>0.054545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ARFGEF3</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ARHGAP23</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ARHGEF17</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.027273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ARHGEF5</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.036364</td>\n",
       "      <td>0.072727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>0.209091</td>\n",
       "      <td>0.063636</td>\n",
       "      <td>0.163636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ASCC3</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.054545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>TRPA1</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.036364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>TRPS1</td>\n",
       "      <td>0.163636</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.036364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>TTK</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.045455</td>\n",
       "      <td>0.100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.545455</td>\n",
       "      <td>0.527273</td>\n",
       "      <td>0.181818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311</th>\n",
       "      <td>UBR4</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.009091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>UBR5</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.036364</td>\n",
       "      <td>0.127273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>UNC80</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.027273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>USF3</td>\n",
       "      <td>0.218182</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.154545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315</th>\n",
       "      <td>USH2A</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.009091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316</th>\n",
       "      <td>VCAN</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317</th>\n",
       "      <td>VPS13A</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.072727</td>\n",
       "      <td>0.054545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>318</th>\n",
       "      <td>VPS13D</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319</th>\n",
       "      <td>VWDE</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.081818</td>\n",
       "      <td>0.045455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>VWF</td>\n",
       "      <td>0.154545</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.045455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>WDFY3</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.045455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>WDFY4</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.081818</td>\n",
       "      <td>0.036364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>WDR87</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.063636</td>\n",
       "      <td>0.054545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>324</th>\n",
       "      <td>XIRP2</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.054545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>YLPM1</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.036364</td>\n",
       "      <td>0.081818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>ZDBF2</td>\n",
       "      <td>0.163636</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.054545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>ZFHX3</td>\n",
       "      <td>0.145455</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.045455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>ZFHX4</td>\n",
       "      <td>0.172727</td>\n",
       "      <td>0.154545</td>\n",
       "      <td>0.036364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.036364</td>\n",
       "      <td>0.072727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>ZNF292</td>\n",
       "      <td>0.163636</td>\n",
       "      <td>0.081818</td>\n",
       "      <td>0.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>ZNF365</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.054545</td>\n",
       "      <td>0.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>ZNF384</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>ZNF462</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>ZNF469</td>\n",
       "      <td>0.227273</td>\n",
       "      <td>0.163636</td>\n",
       "      <td>0.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>ZNF536</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>0.109091</td>\n",
       "      <td>0.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>ZNF831</td>\n",
       "      <td>0.127273</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.045455</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>337 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  Unique_Samples_Mut  Missence_Mut  Truncation_Mut\n",
       "0      ABCA13            0.172727      0.145455        0.090909\n",
       "1       ABCA2            0.154545      0.145455        0.027273\n",
       "2       ABCA4            0.127273      0.072727        0.054545\n",
       "3       ABCB4            0.118182      0.054545        0.063636\n",
       "4       ACACB            0.109091      0.100000        0.009091\n",
       "5      ACVR2A            0.227273      0.018182        0.227273\n",
       "6    ADAMTS16            0.127273      0.118182        0.009091\n",
       "7     ADAMTS5            0.109091      0.100000        0.018182\n",
       "8     ADAMTS7            0.109091      0.109091        0.000000\n",
       "9    ADAMTSL4            0.109091      0.054545        0.063636\n",
       "10     ADGRG4            0.118182      0.109091        0.018182\n",
       "11     ADGRV1            0.172727      0.118182        0.081818\n",
       "12      AHDC1            0.109091      0.081818        0.045455\n",
       "13      AHNAK            0.118182      0.100000        0.018182\n",
       "14     AHNAK2            0.190909      0.172727        0.027273\n",
       "15     AKAP13            0.127273      0.118182        0.009091\n",
       "16      AKAP9            0.118182      0.072727        0.072727\n",
       "17       ANK1            0.136364      0.109091        0.027273\n",
       "18       ANK2            0.136364      0.118182        0.036364\n",
       "19       ANK3            0.118182      0.090909        0.045455\n",
       "20    ANKRD11            0.136364      0.109091        0.036364\n",
       "21        APC            0.745455      0.100000        0.727273\n",
       "22       APC2            0.118182      0.063636        0.054545\n",
       "23       APOB            0.172727      0.154545        0.054545\n",
       "24    ARFGEF3            0.109091      0.100000        0.018182\n",
       "25   ARHGAP23            0.118182      0.109091        0.018182\n",
       "26   ARHGEF17            0.118182      0.090909        0.027273\n",
       "27    ARHGEF5            0.109091      0.036364        0.072727\n",
       "28     ARID1A            0.209091      0.063636        0.163636\n",
       "29      ASCC3            0.118182      0.090909        0.054545\n",
       "..        ...                 ...           ...             ...\n",
       "307     TRPA1            0.118182      0.090909        0.036364\n",
       "308     TRPS1            0.163636      0.136364        0.036364\n",
       "309       TTK            0.127273      0.045455        0.100000\n",
       "310       TTN            0.545455      0.527273        0.181818\n",
       "311      UBR4            0.136364      0.127273        0.009091\n",
       "312      UBR5            0.127273      0.036364        0.127273\n",
       "313     UNC80            0.127273      0.109091        0.027273\n",
       "314      USF3            0.218182      0.136364        0.154545\n",
       "315     USH2A            0.136364      0.127273        0.009091\n",
       "316      VCAN            0.127273      0.118182        0.018182\n",
       "317    VPS13A            0.109091      0.072727        0.054545\n",
       "318    VPS13D            0.136364      0.118182        0.018182\n",
       "319      VWDE            0.109091      0.081818        0.045455\n",
       "320       VWF            0.154545      0.136364        0.045455\n",
       "321     WDFY3            0.118182      0.100000        0.045455\n",
       "322     WDFY4            0.109091      0.081818        0.036364\n",
       "323     WDR87            0.109091      0.063636        0.054545\n",
       "324     XIRP2            0.127273      0.090909        0.054545\n",
       "325     YLPM1            0.109091      0.036364        0.081818\n",
       "326     ZDBF2            0.163636      0.109091        0.054545\n",
       "327     ZFHX3            0.145455      0.127273        0.045455\n",
       "328     ZFHX4            0.172727      0.154545        0.036364\n",
       "329    ZNF106            0.109091      0.036364        0.072727\n",
       "330    ZNF292            0.163636      0.081818        0.090909\n",
       "331    ZNF365            0.127273      0.054545        0.090909\n",
       "332    ZNF384            0.136364      0.136364        0.000000\n",
       "333    ZNF462            0.118182      0.109091        0.018182\n",
       "334    ZNF469            0.227273      0.163636        0.090909\n",
       "335    ZNF536            0.118182      0.109091        0.018182\n",
       "336    ZNF831            0.127273      0.090909        0.045455\n",
       "\n",
       "[337 rows x 4 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          Gene           Mutation         Location\n",
      "Sample_ID                                         \n",
      "S002       TTN  Missense_Mutation        p.P25604T\n",
      "S009       TTN        Splice_Site   p.X8828_splice\n",
      "S009       TTN  Missense_Mutation         p.M2411K\n",
      "S013       TTN             Silent        p.S14806S\n",
      "S013       TTN  Nonsense_Mutation         p.R7040*\n",
      "S015       TTN  Missense_Mutation        p.P27473S\n",
      "S022       TTN    Frame_Shift_Ins    p.N25635Kfs*5\n",
      "S028       TTN             Silent         p.K6047K\n",
      "S029       TTN             Silent         p.V8411V\n",
      "S037       TTN  Missense_Mutation        p.F31025L\n",
      "S038       TTN  Missense_Mutation        p.S31594R\n",
      "S038       TTN    Frame_Shift_Del    p.S22176Lfs*7\n",
      "S038       TTN  Missense_Mutation        p.A12166P\n",
      "S041       TTN             Silent         p.A8541A\n",
      "S046       TTN  Missense_Mutation        p.L14165V\n",
      "S046       TTN  Missense_Mutation         p.R9840C\n",
      "S047       TTN  Missense_Mutation          p.E465Q\n",
      "S051       TTN  Missense_Mutation        p.P19878S\n",
      "S054       TTN             Silent        p.K12951K\n",
      "S054       TTN  Missense_Mutation        p.E12142K\n",
      "S054       TTN  Missense_Mutation        p.L12137S\n",
      "S056       TTN  Missense_Mutation        p.P13011T\n",
      "S062       TTN  Missense_Mutation        p.T20493I\n",
      "S063       TTN        Splice_Site  p.X11877_splice\n",
      "S066       TTN  Missense_Mutation        p.I32071T\n",
      "S066       TTN  Missense_Mutation         p.G5464V\n",
      "S071       TTN  Missense_Mutation         p.C8217S\n",
      "S071       TTN             Silent         p.S3863S\n",
      "S071       TTN    Frame_Shift_Del    p.M3392Lfs*25\n",
      "S074       TTN             Silent        p.E31795E\n",
      "...        ...                ...              ...\n",
      "S080       TTN  Missense_Mutation        p.P12083H\n",
      "S080       TTN  Nonsense_Mutation        p.E11171*\n",
      "S080       TTN  Missense_Mutation        p.R10072I\n",
      "S080       TTN             Silent         p.F8821F\n",
      "S080       TTN  Nonsense_Mutation         p.E8578*\n",
      "S080       TTN  Missense_Mutation         p.E7648K\n",
      "S080       TTN  Missense_Mutation         p.Q7466H\n",
      "S080       TTN  Nonsense_Mutation         p.E5886*\n",
      "S080       TTN  Missense_Mutation         p.R5812I\n",
      "S080       TTN  Missense_Mutation         p.D4317N\n",
      "S080       TTN  Missense_Mutation         p.T3210I\n",
      "S080       TTN  Missense_Mutation         p.E3071D\n",
      "S080       TTN  Missense_Mutation         p.Q2629H\n",
      "S080       TTN             Silent         p.I2485I\n",
      "S080       TTN  Missense_Mutation         p.S2409Y\n",
      "S080       TTN  Nonsense_Mutation         p.E2285*\n",
      "S080       TTN             Silent         p.V2099V\n",
      "S080       TTN  Missense_Mutation         p.Q1701H\n",
      "S080       TTN  Missense_Mutation         p.R1003H\n",
      "S084       TTN  Missense_Mutation         p.L8666P\n",
      "S090       TTN             Silent        p.L21646L\n",
      "S095       TTN             Silent        p.Q29169Q\n",
      "S096       TTN  Missense_Mutation         p.E9605Q\n",
      "S097       TTN  Missense_Mutation        p.P35903L\n",
      "S097       TTN  Missense_Mutation        p.D34870E\n",
      "S100       TTN  Missense_Mutation         p.G9393R\n",
      "S101       TTN  Missense_Mutation         p.L3021H\n",
      "S108       TTN  Missense_Mutation        p.P30018A\n",
      "S108       TTN  Missense_Mutation        p.T27422I\n",
      "S110       TTN  Missense_Mutation         p.G7058S\n",
      "\n",
      "[80 rows x 3 columns]\n",
      "31\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'Frame_Shift_Del': Index(['S038', 'S071'], dtype='object', name='Sample_ID'),\n",
       " 'Frame_Shift_Ins': Index(['S022'], dtype='object', name='Sample_ID'),\n",
       " 'Missense_Mutation': Index(['S002', 'S009', 'S015', 'S037', 'S038', 'S038', 'S046', 'S046', 'S047',\n",
       "        'S051', 'S054', 'S054', 'S056', 'S062', 'S066', 'S066', 'S071', 'S078',\n",
       "        'S080', 'S080', 'S080', 'S080', 'S080', 'S080', 'S080', 'S080', 'S080',\n",
       "        'S080', 'S080', 'S080', 'S080', 'S080', 'S080', 'S080', 'S080', 'S080',\n",
       "        'S080', 'S080', 'S080', 'S080', 'S080', 'S080', 'S080', 'S084', 'S096',\n",
       "        'S097', 'S097', 'S100', 'S101', 'S108', 'S108', 'S110'],\n",
       "       dtype='object', name='Sample_ID'),\n",
       " 'Nonsense_Mutation': Index(['S013', 'S080', 'S080', 'S080', 'S080', 'S080'], dtype='object', name='Sample_ID'),\n",
       " 'Silent': Index(['S013', 'S028', 'S029', 'S041', 'S054', 'S071', 'S074', 'S078', 'S080',\n",
       "        'S080', 'S080', 'S080', 'S080', 'S080', 'S090', 'S095'],\n",
       "       dtype='object', name='Sample_ID'),\n",
       " 'Splice_Site': Index(['S009', 'S063', 'S080'], dtype='object', name='Sample_ID')}"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#test results\n",
    "m = ov.get_mutations()\n",
    "gene = 'TTN'\n",
    "g = m.loc[m['Gene'] == gene]\n",
    "print(g)\n",
    "print(len(g.index.unique()))\n",
    "\n",
    "r = g.groupby('Mutation')\n",
    "\n",
    "r.groups\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colon_freq_mutated_df = get_frequently_mutated(cancer, .25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colon_freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ovarian_freq_mutated_df = get_frequently_mutated(ov,.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ovarian_freq_mutated_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
