{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get_frequently_mutated Final "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[Kding treatment data...ata........\r"
     ]
    }
   ],
   "source": [
    "import cptac\n",
    "#en = cptac.Endometrial()\n",
    "#cptac.sync('endometrial', version='2.1')\n",
    "ov = cptac.Ovarian()\n",
    "#cptac.sync(dataset='ovarian', version='0.0')\n",
    "#cptac.sync(dataset='colon', version='0.0')\n",
    "#co = cptac.Colon()\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_frequently_mutated(cancer_object, cutoff = 0.1):  \n",
    "    \"\"\"\n",
    "    take cancer object and find the frequently \n",
    "    mutated genes in the total tumors compared to the cutoff.\n",
    "\n",
    "    Parameters:\n",
    "    cancer_object (object): cancer class object from cptac module \n",
    "    cutoff (float): used as comparison to determine the \n",
    "                    status of gene mutation frequency\n",
    "\n",
    "    Returns:\n",
    "    freq_mutated_df (pd.DataFrame): DataFrame of frequently \n",
    "        mutated genes passing the cutoff. Columns contain the \n",
    "        fractions of total unique mutations,missence type \n",
    "        mutations, and truncation type mutations per gene.\n",
    "    \n",
    "    The Missence_Mut column includes: \n",
    "        In_Frame_Del, In_Frame_Ins, Missense_Mutation\n",
    "   \n",
    "   The Truncation_Mut column includes: \n",
    "        Frame_Shift_Del, Frame_Shift_Ins, Splice_Site, \n",
    "        Nonsense_Mutation, Nonstop_Mutation\n",
    "        \n",
    "    These columns count multiple mutations of one gene in the \n",
    "    same sample, so fractions in the last two columns may \n",
    "    exceed the Unique_Samples_Mut column which only counts if \n",
    "    the gene was mutated once per sample.\"\"\"    \n",
    "    \n",
    "    # Get mutations data frame (whole exome sequencing)\n",
    "    somatic_mutations = cancer_object.get_mutations()\n",
    "    sample_status_map = cancer_object._get_sample_status_map()\n",
    "         \n",
    "    # Get total tumors/patients\n",
    "    sample_status_count = sample_status_map.value_counts().reset_index()\n",
    "    tumors = sample_status_count.loc[sample_status_count['index'] == 'Tumor']\n",
    "    total_tumor_patients = tumors.iloc[0,1] #number of tumors \n",
    "        \n",
    "    # Find frequently mutated genes and their fraction of unique mutated samples.\n",
    "    #move 'Sample_ID' from index to column\n",
    "    origin_df = somatic_mutations.reset_index()\n",
    "\n",
    "    #group by gene and count unique samples\n",
    "    count_mutations = origin_df.groupby(['Gene']).nunique()\n",
    "\n",
    "    #format\n",
    "    count_mutations = count_mutations.rename(columns={\"Sample_ID\": \"Unique_Samples_Mut\"})\n",
    "    count_mutations = count_mutations.drop(['Gene', 'Mutation','Location'], axis = 1)\n",
    "    \n",
    "    #filter using the cutoff and create fraction\n",
    "    fraction_mutated = count_mutations.apply(lambda x: x / total_tumor_patients)\n",
    "    fraction_greater_than_cutoff = fraction_mutated.where(lambda x: x > cutoff) #na used when false\n",
    "    filtered_gene_df = fraction_greater_than_cutoff.dropna()\n",
    "    \n",
    "    \n",
    "    #Create Missence and Trucation data frame\n",
    "    #create two categories in Mutation column\n",
    "    if cancer_object.get_cancer_type() == 'colon':\n",
    "        missence_truncation_groups = {'frameshift substitution': 'T', \n",
    "            'frameshift deletion': 'T', 'frameshift insertion': 'T', \n",
    "            'stopgain': 'T', 'stoploss': 'T', 'nonsynonymous SNV': 'M',\n",
    "            'nonframeshift insertion': 'M','nonframeshift deletion': 'M', \n",
    "            'nonframeshift substitution': 'M'}\n",
    "    else: \n",
    "        missence_truncation_groups = {'In_Frame_Del': 'M', 'In_Frame_Ins': 'M',\n",
    "            'Missense_Mutation': 'M', 'Frame_Shift_Del': 'T','Nonsense_Mutation': 'T', \n",
    "            'Splice_Site': 'T', 'Frame_Shift_Ins': 'T','Nonstop_Mutation':'T'}\n",
    "       \n",
    "    mutations_replaced_M_T = origin_df.replace(missence_truncation_groups)\n",
    "    \n",
    "    # group mutation categories\n",
    "    miss = mutations_replaced_M_T.loc[mutations_replaced_M_T['Mutation'] == 'M']\n",
    "    trunc = mutations_replaced_M_T.loc[mutations_replaced_M_T['Mutation'] == 'T']\n",
    "\n",
    "    # group by gene and count unique samples for both categories\n",
    "    count_miss = miss.groupby(['Gene']).nunique()\n",
    "    count_trunc = trunc.groupby(['Gene']).nunique()\n",
    "\n",
    "    #format\n",
    "    missence_df = count_miss.rename(columns={\"Sample_ID\": \"Missence_Mut\"})\n",
    "    missence_df = missence_df.drop(['Gene', 'Mutation', 'Location'], axis = 1)\n",
    "\n",
    "    truncation_df = count_trunc.rename(columns={\"Sample_ID\": \"Truncation_Mut\"})\n",
    "    truncation_df = truncation_df.drop(['Gene', 'Mutation', 'Location'], axis = 1)\n",
    "\n",
    "    #join miss and trunc and change nan to 0, then divide by total tumors\n",
    "    join_mutations = missence_df.join(truncation_df).fillna(0)\n",
    "    missence_and_truncation_df = join_mutations.apply(lambda x: x / total_tumor_patients)\n",
    "\n",
    "\n",
    "    #Join data frames, keeping only the genes that passed the cutoff \n",
    "    freq_mutated_df = filtered_gene_df.join(missence_and_truncation_df).reset_index()\n",
    "    freq_mutated_df.name = 'frequently_mutated'\n",
    "    \n",
    "    return freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = get_frequently_mutated(ov, .1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Unique_Samples_Mut</th>\n",
       "      <th>Missence_Mut</th>\n",
       "      <th>Truncation_Mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FCGBP</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.063063</td>\n",
       "      <td>0.009009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>0.117117</td>\n",
       "      <td>0.090090</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MT-ND5</td>\n",
       "      <td>0.144144</td>\n",
       "      <td>0.063063</td>\n",
       "      <td>0.018018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MUC12</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.009009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.144144</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.009009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUC17</td>\n",
       "      <td>0.126126</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MUC4</td>\n",
       "      <td>0.306306</td>\n",
       "      <td>0.234234</td>\n",
       "      <td>0.009009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MUC5B</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.081081</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NCOR2</td>\n",
       "      <td>0.108108</td>\n",
       "      <td>0.027027</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.693694</td>\n",
       "      <td>0.450450</td>\n",
       "      <td>0.243243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.279279</td>\n",
       "      <td>0.198198</td>\n",
       "      <td>0.063063</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene  Unique_Samples_Mut  Missence_Mut  Truncation_Mut\n",
       "0    FCGBP            0.108108      0.063063        0.009009\n",
       "1   MT-CO1            0.117117      0.090090        0.000000\n",
       "2   MT-ND5            0.144144      0.063063        0.018018\n",
       "3    MUC12            0.108108      0.108108        0.009009\n",
       "4    MUC16            0.144144      0.108108        0.009009\n",
       "5    MUC17            0.126126      0.108108        0.000000\n",
       "6     MUC4            0.306306      0.234234        0.009009\n",
       "7    MUC5B            0.108108      0.081081        0.000000\n",
       "8    NCOR2            0.108108      0.027027        0.000000\n",
       "9     TP53            0.693694      0.450450        0.243243\n",
       "10     TTN            0.279279      0.198198        0.063063"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           Gene           Mutation        Location\n",
      "Sample_ID                                         \n",
      "S002       TP53    Frame_Shift_Del          p.D61*\n",
      "S006       TP53    Frame_Shift_Del     p.S90Vfs*55\n",
      "S007       TP53  Missense_Mutation         p.R273C\n",
      "S009       TP53  Missense_Mutation         p.I195T\n",
      "S011       TP53    Frame_Shift_Ins    p.C124Lfs*25\n",
      "S013       TP53        Splice_Site   p.X261_splice\n",
      "S015       TP53  Missense_Mutation         p.C141Y\n",
      "S016       TP53    Frame_Shift_Ins     p.T81Nfs*68\n",
      "S022       TP53  Nonsense_Mutation         p.S183*\n",
      "S023       TP53        Splice_Site   p.X307_splice\n",
      "S024       TP53  Missense_Mutation         p.V173M\n",
      "S026       TP53    Frame_Shift_Del    p.R110Lfs*13\n",
      "S027       TP53  Missense_Mutation         p.Y205D\n",
      "S028       TP53    Frame_Shift_Del     p.P75Lfs*48\n",
      "S029       TP53  Missense_Mutation         p.R282W\n",
      "S031       TP53  Nonsense_Mutation         p.Q136*\n",
      "S033       TP53        Splice_Site   p.X187_splice\n",
      "S035       TP53        Splice_Site   p.X224_splice\n",
      "S037       TP53       In_Frame_Del       p.S241del\n",
      "S038       TP53    Frame_Shift_Ins    p.L111Pfs*38\n",
      "S039       TP53  Missense_Mutation         p.M237I\n",
      "S040       TP53    Frame_Shift_Del     p.S241Pfs*6\n",
      "S041       TP53       In_Frame_Del  p.D148_S149del\n",
      "S042       TP53  Nonsense_Mutation         p.E286*\n",
      "S043       TP53  Missense_Mutation         p.R248Q\n",
      "S044       TP53  Missense_Mutation         p.Y234C\n",
      "S045       TP53    Frame_Shift_Del     p.P47Rfs*76\n",
      "S046       TP53  Missense_Mutation         p.K132N\n",
      "S047       TP53  Missense_Mutation         p.G266R\n",
      "S049       TP53  Missense_Mutation         p.R280K\n",
      "...         ...                ...             ...\n",
      "S070       TP53    Frame_Shift_Del     p.A84Pfs*39\n",
      "S071       TP53  Missense_Mutation         p.R337L\n",
      "S076       TP53  Missense_Mutation         p.R282W\n",
      "S078       TP53  Missense_Mutation         p.R175H\n",
      "S079       TP53    Frame_Shift_Del    p.Q192Rfs*55\n",
      "S080       TP53  Nonsense_Mutation         p.R213*\n",
      "S084       TP53  Missense_Mutation         p.V274A\n",
      "S084       TP53  Missense_Mutation         p.R248Q\n",
      "S085       TP53        Splice_Site   p.X224_splice\n",
      "S088       TP53  Missense_Mutation         p.Y163C\n",
      "S089       TP53    Frame_Shift_Del    p.D186Efs*61\n",
      "S090       TP53  Missense_Mutation         p.R175H\n",
      "S091       TP53    Frame_Shift_Del    p.V122Dfs*26\n",
      "S093       TP53  Missense_Mutation         p.C275Y\n",
      "S094       TP53  Missense_Mutation         p.R273C\n",
      "S095       TP53  Missense_Mutation         p.Y220C\n",
      "S096       TP53    Frame_Shift_Del    p.C182Afs*65\n",
      "S097       TP53  Missense_Mutation         p.R248L\n",
      "S098       TP53  Missense_Mutation         p.S127F\n",
      "S100       TP53    Frame_Shift_Del    p.R333Vfs*12\n",
      "S101       TP53  Missense_Mutation         p.G245D\n",
      "S102       TP53  Missense_Mutation         p.M237I\n",
      "S103       TP53  Missense_Mutation         p.G245D\n",
      "S104       TP53  Missense_Mutation         p.R248Q\n",
      "S106       TP53  Missense_Mutation         p.H193Y\n",
      "S106       TP53  Missense_Mutation         p.Q192H\n",
      "S107       TP53    Frame_Shift_Del     p.R337Pfs*8\n",
      "S108       TP53  Missense_Mutation         p.C238Y\n",
      "S110       TP53  Missense_Mutation         p.R248W\n",
      "S111       TP53  Missense_Mutation         p.R273H\n",
      "\n",
      "[79 rows x 3 columns]\n",
      "77\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'Frame_Shift_Del': Index(['S002', 'S006', 'S026', 'S028', 'S040', 'S045', 'S052', 'S070', 'S079',\n",
       "        'S089', 'S091', 'S096', 'S100', 'S107'],\n",
       "       dtype='object', name='Sample_ID'),\n",
       " 'Frame_Shift_Ins': Index(['S011', 'S016', 'S038'], dtype='object', name='Sample_ID'),\n",
       " 'In_Frame_Del': Index(['S037', 'S041', 'S064'], dtype='object', name='Sample_ID'),\n",
       " 'Missense_Mutation': Index(['S007', 'S009', 'S015', 'S024', 'S027', 'S029', 'S039', 'S043', 'S044',\n",
       "        'S046', 'S047', 'S049', 'S050', 'S051', 'S053', 'S054', 'S055', 'S056',\n",
       "        'S057', 'S059', 'S060', 'S061', 'S062', 'S063', 'S065', 'S066', 'S068',\n",
       "        'S069', 'S071', 'S076', 'S078', 'S084', 'S084', 'S088', 'S090', 'S093',\n",
       "        'S094', 'S095', 'S097', 'S098', 'S101', 'S102', 'S103', 'S104', 'S106',\n",
       "        'S106', 'S108', 'S110', 'S111'],\n",
       "       dtype='object', name='Sample_ID'),\n",
       " 'Nonsense_Mutation': Index(['S022', 'S031', 'S042', 'S067', 'S080'], dtype='object', name='Sample_ID'),\n",
       " 'Splice_Site': Index(['S013', 'S023', 'S033', 'S035', 'S085'], dtype='object', name='Sample_ID')}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#test results\n",
    "m = ov.get_mutations()\n",
    "gene = 'TP53'\n",
    "g = m.loc[m['Gene'] == gene]\n",
    "print(g)\n",
    "print(len(g.index.unique()))\n",
    "\n",
    "r = g.groupby('Mutation')\n",
    "\n",
    "r.groups\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TP53_proteomics</th>\n",
       "      <th>TP53_Mutation</th>\n",
       "      <th>TP53_Location</th>\n",
       "      <th>TP53_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S012</th>\n",
       "      <td>-1.017320</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S019</th>\n",
       "      <td>-1.195933</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S025</th>\n",
       "      <td>-1.072294</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S048</th>\n",
       "      <td>-1.429764</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S074</th>\n",
       "      <td>-0.963240</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S087</th>\n",
       "      <td>-0.547332</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           TP53_proteomics     TP53_Mutation  TP53_Location  \\\n",
       "Sample_ID                                                     \n",
       "S012             -1.017320  [Wildtype_Tumor]  [No_mutation]   \n",
       "S019             -1.195933  [Wildtype_Tumor]  [No_mutation]   \n",
       "S025             -1.072294  [Wildtype_Tumor]  [No_mutation]   \n",
       "S048             -1.429764  [Wildtype_Tumor]  [No_mutation]   \n",
       "S074             -0.963240  [Wildtype_Tumor]  [No_mutation]   \n",
       "S087             -0.547332  [Wildtype_Tumor]  [No_mutation]   \n",
       "\n",
       "          TP53_Mutation_Status Sample_Status  \n",
       "Sample_ID                                     \n",
       "S012            Wildtype_Tumor         Tumor  \n",
       "S019            Wildtype_Tumor         Tumor  \n",
       "S025            Wildtype_Tumor         Tumor  \n",
       "S048            Wildtype_Tumor         Tumor  \n",
       "S074            Wildtype_Tumor         Tumor  \n",
       "S087            Wildtype_Tumor         Tumor  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# count wt\n",
    "omics = ov.get_proteomics()\n",
    "somatic_mutations = ov.get_mutations()\n",
    "a = ov.append_mutations_to_omics(mutation_genes = gene, omics_df_name = omics.name, omics_genes = gene)\n",
    "\n",
    "p53 = a.loc[a['TP53_Mutation_Status'] == 'Wildtype_Tumor']\n",
    "p53"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colon_freq_mutated_df = get_frequently_mutated(cancer, .25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colon_freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ovarian_freq_mutated_df = get_frequently_mutated(ov,.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ovarian_freq_mutated_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
