{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get_frequently_mutated Final "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_frequently_mutated(cancer_type, cutoff=.1):  \n",
    "    \"\"\"take cancer type to import cptac and find the frequently mutated genes in total tumors compared to the cutoff.\n",
    "        \n",
    "        Parameters:\n",
    "        cancer_type (string): type of  cancer \n",
    "        cutoff (float): used as comparison to determine status of gene mutation frequency\n",
    "        \n",
    "        Returns:\n",
    "        freq_mutated_df (pd.DataFrame): DataFrame of frequently mutated genes passing the cutoff\n",
    "            and Total_Mutated (mutated genes / total tumors), percent Truncated, and percent Missence\n",
    "        \n",
    "        There are many types of mutations catagorized into the columns Truncated and Missence. \n",
    "        The Truncated column includes: Frame_Shift_Del, Frame_Shift_Ins, Splice_Site, Nonsense_Mutation, Nonstop_Mutation\n",
    "        The Missence column includes: In_Frame_Del, In_Frame_Ins, Missense_Mutation\n",
    "        These columns count multiple mutations of one gene in the same sample, so percentages in the last two columns may \n",
    "        exceed the Total_Mutated column(which only counts if the gene was mutated once)\"\"\"    \n",
    "    \n",
    "    # import CPTAC and pandas\n",
    "    \n",
    "    import pandas as pd\n",
    "    colon = False\n",
    "    if cancer_type == \"endometrial\" or cancer_type == \"Endometrial\":\n",
    "        import cptac.endometrial as cptac\n",
    "        \n",
    "    elif cancer_type == \"colon\" or cancer_type == \"Colon\":\n",
    "        import cptac.colon as cptac\n",
    "        colon = True\n",
    "        \n",
    "    elif cancer_type == \"ovarian\" or cancer_type == \"Ovarian\":\n",
    "        import cptac.ovarian as cptac\n",
    "    \n",
    "    else:\n",
    "        str_cancer_options = '\\n' + 'Options: endometrial, ovarian, colon'\n",
    "        print(\"Please enter a valid cancer type.\", str_cancer_options)\n",
    "        return 0\n",
    "    \n",
    "    # get data frames\n",
    "    somatic_mutations = cptac.get_mutations()\n",
    "    proteomics = cptac.get_proteomics()\n",
    "    sample_status_map = cptac.get_sample_status_map()\n",
    "    merged_mutations = somatic_mutations.join(sample_status_map, how=\"left\") \n",
    "    \n",
    "    # standardize mutation names \n",
    "    if colon == True:\n",
    "        mutation_equivalents = {'frameshift substitution': 'Frame_Shift_Del' , 'frameshift deletion': 'Frame_Shift_Del', \n",
    "            'frameshift insertion': 'Frame_Shift_Ins', 'stopgain': 'Nonsense_Mutation ', 'stoploss':'Nonstop_Mutation',\n",
    "            'nonsynonymous SNV': 'Missense_Mutation','nonframeshift insertion': 'In_Frame_Ins',\n",
    "            'nonframeshift deletion': 'In_Frame_Del', 'nonframeshift substitution': 'Missense_Mutation'}\n",
    "        merged_mutations = merged_mutations.replace(to_replace = mutation_equivalents)\n",
    "        \n",
    "    # get list of unique genes\n",
    "    unique_genes = somatic_mutations['Gene'].unique()\n",
    "    \n",
    "    # get total tumors/patients\n",
    "    sample_status_series = sample_status_map.value_counts()\n",
    "    total_tumor_patients = sample_status_series[0]\n",
    "        \n",
    "    # find frequently mutated genes and their percent mutated. Create lists for frequently mutated genes and percentage.\n",
    "    freq_mut = []\n",
    "    total_percent_mutated = []\n",
    "    for gene in unique_genes:\n",
    "        gene_mutated = merged_mutations.loc[merged_mutations['Gene'] == gene]\n",
    "        gene_mutated = gene_mutated.index.unique()\n",
    "        num_gene_mutated = len(gene_mutated)\n",
    "        percentage = (num_gene_mutated / total_tumor_patients)\n",
    "        if percentage > cutoff:\n",
    "            freq_mut.append(gene)\n",
    "            total_percent_mutated.append(percentage)\n",
    "    \n",
    "    # find truncated percentage\n",
    "    truncated = []\n",
    "    for gene in freq_mut:\n",
    "        gene_mutated = merged_mutations.loc[merged_mutations['Gene'] == gene]\n",
    "        trunc = gene_mutated.loc[gene_mutated['Mutation'] != 'In_Frame_Del']\n",
    "        trunc = trunc.loc[trunc['Mutation'] != 'In_Frame_Ins']\n",
    "        trunc = trunc.loc[trunc['Mutation'] != 'Missense_Mutation'] \n",
    "        num_gene_mutated = len(trunc.index.unique())\n",
    "        percentage = (num_gene_mutated / total_tumor_patients)\n",
    "        truncated.append(percentage)\n",
    "\n",
    "    # find missence percentage \n",
    "    missence = []\n",
    "    for gene in freq_mut:\n",
    "        num_gene_mutated = 0 \n",
    "        gene_mutated = merged_mutations.loc[merged_mutations['Gene'] == gene]\n",
    "        missence_mutations = gene_mutated.loc[gene_mutated['Mutation'] == 'In_Frame_Del']\n",
    "        num_gene_mutated = len(missence_mutations.index.unique())\n",
    "        missence_mutations = gene_mutated.loc[gene_mutated['Mutation'] == 'In_Frame_Ins']\n",
    "        num_gene_mutated += len(missence_mutations.index.unique())\n",
    "        missence_mutations = gene_mutated.loc[gene_mutated['Mutation'] == 'Missense_Mutation']\n",
    "        num_gene_mutated += len(missence_mutations.index.unique())\n",
    "        percentage = (num_gene_mutated / total_tumor_patients)\n",
    "        missence.append(percentage)\n",
    "        \n",
    "    # create dataframe\n",
    "    merged_lists = list(zip(freq_mut, total_percent_mutated, truncated, missence))\n",
    "    freq_mutated_df = pd.DataFrame(merged_lists, columns =['Gene', 'Fraction_Mutated', 'Truncated', 'Missence'])\n",
    "                   \n",
    "    return freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Endometrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "endo_freq_mutated_df = get_frequently_mutated(\"endometrial\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Fraction_Mutated</th>\n",
       "      <th>Truncated</th>\n",
       "      <th>Missence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RPL22</td>\n",
       "      <td>0.168421</td>\n",
       "      <td>0.157895</td>\n",
       "      <td>0.010526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SPEN</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>0.063158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>0.452632</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.136842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FLG</td>\n",
       "      <td>0.136842</td>\n",
       "      <td>0.010526</td>\n",
       "      <td>0.126316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIF26B</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.073684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PXDN</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ALK</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.031579</td>\n",
       "      <td>0.073684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>DYSF</td>\n",
       "      <td>0.147368</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NEB</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.115789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LRP2</td>\n",
       "      <td>0.136842</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>0.115789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ABCA12</td>\n",
       "      <td>0.147368</td>\n",
       "      <td>0.073684</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CELSR3</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.010526</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CACNA1D</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.021053</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PLXNA1</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.031579</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>0.494737</td>\n",
       "      <td>0.010526</td>\n",
       "      <td>0.494737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>LIMCH1</td>\n",
       "      <td>0.136842</td>\n",
       "      <td>0.073684</td>\n",
       "      <td>0.063158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>KIAA1211</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.031579</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>FAT4</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.031579</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>FBXW7</td>\n",
       "      <td>0.189474</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.157895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>DNAH5</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.084211</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ADGRV1</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>APC</td>\n",
       "      <td>0.136842</td>\n",
       "      <td>0.084211</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>RBM27</td>\n",
       "      <td>0.168421</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.052632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NSD1</td>\n",
       "      <td>0.168421</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PKHD1</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.021053</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>DST</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.073684</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MDN1</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.063158</td>\n",
       "      <td>0.063158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SYNE1</td>\n",
       "      <td>0.168421</td>\n",
       "      <td>0.063158</td>\n",
       "      <td>0.147368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MAP3K4</td>\n",
       "      <td>0.147368</td>\n",
       "      <td>0.073684</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>ARID5B</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.010526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>LYST</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>MYT1L</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.063158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>XIRP2</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.084211</td>\n",
       "      <td>0.073684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>SPEG</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.063158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>CADPS</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.073684</td>\n",
       "      <td>0.063158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>WDFY3</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.010526</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>PCDHA1</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.010526</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>DSP</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>COL12A1</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.073684</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>ASCC3</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>MUC17</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.031579</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>ZNF800</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.073684</td>\n",
       "      <td>0.063158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>MGAM</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.010526</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.021053</td>\n",
       "      <td>0.115789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>CUBN</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.031579</td>\n",
       "      <td>0.073684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>WDFY4</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.010526</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.063158</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>KCNH5</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.031579</td>\n",
       "      <td>0.073684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>SYNE2</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.052632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>MAP1A</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.031579</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>PKD1</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.010526</td>\n",
       "      <td>0.105263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>DNAH3</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.042105</td>\n",
       "      <td>0.094737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>ANKRD11</td>\n",
       "      <td>0.126316</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>0.084211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>DNAH17</td>\n",
       "      <td>0.157895</td>\n",
       "      <td>0.031579</td>\n",
       "      <td>0.136842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>PIEZO2</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.073684</td>\n",
       "      <td>0.073684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>HELZ2</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.063158</td>\n",
       "      <td>0.042105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>BCOR</td>\n",
       "      <td>0.115789</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.115789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>INPPL1</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.189474</td>\n",
       "      <td>0.021053</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>232 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  Fraction_Mutated  Truncated  Missence\n",
       "0       RPL22          0.168421   0.157895  0.010526\n",
       "1        SPEN          0.105263   0.052632  0.063158\n",
       "2      ARID1A          0.452632   0.400000  0.136842\n",
       "3         FLG          0.136842   0.010526  0.126316\n",
       "4      KIF26B          0.105263   0.042105  0.073684\n",
       "5        PXDN          0.126316   0.042105  0.084211\n",
       "6         ALK          0.105263   0.031579  0.073684\n",
       "7        DYSF          0.147368   0.042105  0.105263\n",
       "8         NEB          0.126316   0.042105  0.115789\n",
       "9        LRP2          0.136842   0.052632  0.115789\n",
       "10     ABCA12          0.147368   0.073684  0.094737\n",
       "11     CELSR3          0.105263   0.010526  0.094737\n",
       "12    CACNA1D          0.115789   0.021053  0.094737\n",
       "13     PLXNA1          0.115789   0.031579  0.084211\n",
       "14     PIK3CA          0.494737   0.010526  0.494737\n",
       "15     LIMCH1          0.136842   0.073684  0.063158\n",
       "16   KIAA1211          0.115789   0.031579  0.084211\n",
       "17       FAT4          0.105263   0.031579  0.105263\n",
       "18      FBXW7          0.189474   0.042105  0.157895\n",
       "19       FAT1          0.115789   0.052632  0.084211\n",
       "20      DNAH5          0.126316   0.084211  0.084211\n",
       "21     ADGRV1          0.115789   0.052632  0.094737\n",
       "22        APC          0.136842   0.084211  0.105263\n",
       "23      RBM27          0.168421   0.126316  0.052632\n",
       "24       NSD1          0.168421   0.115789  0.094737\n",
       "25      PKHD1          0.126316   0.021053  0.105263\n",
       "26        DST          0.115789   0.073684  0.084211\n",
       "27       MDN1          0.105263   0.063158  0.063158\n",
       "28      SYNE1          0.168421   0.063158  0.147368\n",
       "29     MAP3K4          0.147368   0.073684  0.105263\n",
       "..        ...               ...        ...       ...\n",
       "202    ARID5B          0.126316   0.115789  0.010526\n",
       "203      LYST          0.105263   0.042105  0.084211\n",
       "204      RYR2          0.115789   0.042105  0.105263\n",
       "205     MYT1L          0.105263   0.042105  0.063158\n",
       "206     XIRP2          0.115789   0.084211  0.073684\n",
       "207      SPEG          0.105263   0.042105  0.063158\n",
       "208     CADPS          0.105263   0.073684  0.063158\n",
       "209     WDFY3          0.105263   0.010526  0.094737\n",
       "210    PCDHA1          0.105263   0.010526  0.094737\n",
       "211       DSP          0.105263   0.052632  0.094737\n",
       "212   COL12A1          0.115789   0.073684  0.084211\n",
       "213     ASCC3          0.105263   0.042105  0.084211\n",
       "214     MUC17          0.115789   0.031579  0.105263\n",
       "215    ZNF800          0.115789   0.073684  0.063158\n",
       "216      MGAM          0.105263   0.010526  0.105263\n",
       "217   CACNA1B          0.126316   0.021053  0.115789\n",
       "218      CUBN          0.105263   0.031579  0.073684\n",
       "219     WDFY4          0.105263   0.010526  0.094737\n",
       "220      FAT3          0.126316   0.063158  0.094737\n",
       "221     KCNH5          0.105263   0.031579  0.073684\n",
       "222     SYNE2          0.105263   0.105263  0.052632\n",
       "223     MAP1A          0.105263   0.031579  0.084211\n",
       "224      PKD1          0.105263   0.010526  0.105263\n",
       "225     DNAH3          0.126316   0.042105  0.094737\n",
       "226   ANKRD11          0.126316   0.052632  0.084211\n",
       "227    DNAH17          0.157895   0.031579  0.136842\n",
       "228    PIEZO2          0.105263   0.073684  0.073684\n",
       "229     HELZ2          0.105263   0.063158  0.042105\n",
       "230      BCOR          0.115789   0.000000  0.115789\n",
       "231    INPPL1          0.200000   0.189474  0.021053\n",
       "\n",
       "[232 rows x 4 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "endo_freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Colon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You have loaded the cptac colon dataset. To view available dataframes,\n",
      "use cptac.colon.list_data(). To view available functions for accessing\n",
      "and manipulating the dataframes, use cptac.colon.list_api().\n",
      "colon data version: Most recent release\n",
      "\n",
      "Loading cptac colon data:\n",
      "Loading clinical data...\n",
      "Loading miRNA data...\n",
      "Loading mutation data...\n",
      "Loading mutation_binary data...\n",
      "Loading phosphoproteomics_normal data...\n",
      "Loading phosphoproteomics_tumor data...\n",
      "Loading proteomics_normal data...\n",
      "Loading proteomics_tumor data...\n",
      "Loading transcriptomics data...\n",
      "['Missense_Mutation' 'Frame_Shift_Del' 'Frame_Shift_Ins'\n",
      " 'Nonsense_Mutation ' 'In_Frame_Ins' 'In_Frame_Del' 'Nonstop_Mutation']\n",
      "total_tumor_patients 110\n"
     ]
    }
   ],
   "source": [
    "colon_freq_mutated_df = get_frequently_mutated(\"colon\", .25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Percent Mutated</th>\n",
       "      <th>Truncated</th>\n",
       "      <th>Missence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>APC</td>\n",
       "      <td>0.745455</td>\n",
       "      <td>3.136364</td>\n",
       "      <td>0.381818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.509091</td>\n",
       "      <td>2.145455</td>\n",
       "      <td>4.845455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.354545</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.627273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>0.254545</td>\n",
       "      <td>0.036364</td>\n",
       "      <td>0.381818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SYNE1</td>\n",
       "      <td>0.263636</td>\n",
       "      <td>0.281818</td>\n",
       "      <td>1.018182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.545455</td>\n",
       "      <td>1.409091</td>\n",
       "      <td>8.318182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>0.318182</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.636364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>OBSCN</td>\n",
       "      <td>0.263636</td>\n",
       "      <td>0.190909</td>\n",
       "      <td>1.127273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CCDC168</td>\n",
       "      <td>0.263636</td>\n",
       "      <td>0.263636</td>\n",
       "      <td>0.336364</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene  Percent Mutated  Truncated  Missence\n",
       "0      APC         0.745455   3.136364  0.381818\n",
       "1     TP53         0.509091   2.145455  4.845455\n",
       "2    MUC16         0.354545   0.100000  0.627273\n",
       "3     FAT3         0.254545   0.036364  0.381818\n",
       "4    SYNE1         0.263636   0.281818  1.018182\n",
       "5      TTN         0.545455   1.409091  8.318182\n",
       "6     KRAS         0.318182   0.000000  0.636364\n",
       "7    OBSCN         0.263636   0.190909  1.127273\n",
       "8  CCDC168         0.263636   0.263636  0.336364"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "colon_freq_mutated_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Ovarian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You have loaded the cptac ovarian dataset. To view available\n",
      "dataframes, use cptac.ovarian.list_data(). To view available functions\n",
      "for accessing and manipulating the dataframes, use\n",
      "cptac.ovarian.list_api().\n",
      "ovarian data version: Most recent release\n",
      "\n",
      "Loading cptac ovarian data:\n",
      "Loading clinical data...\n",
      "Loading cnv data...\n",
      "Loading phosphoproteomics data...\n",
      "Loading proteomics data...\n",
      "Loading somatic_38 data...\n",
      "Loading transcriptomics data...\n",
      "Loading treatment data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until June 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until June 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n",
      "total_tumor_patients 111\n"
     ]
    }
   ],
   "source": [
    "ovarian_freq_mutated_df = get_frequently_mutated(\"ovarian\", .25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Percent Mutated</th>\n",
       "      <th>Truncated</th>\n",
       "      <th>Missence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.279279</td>\n",
       "      <td>0.252252</td>\n",
       "      <td>0.468468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.693694</td>\n",
       "      <td>0.243243</td>\n",
       "      <td>0.468468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MUC4</td>\n",
       "      <td>0.306306</td>\n",
       "      <td>0.153153</td>\n",
       "      <td>0.432432</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene  Percent Mutated  Truncated  Missence\n",
       "0   TTN         0.279279   0.252252  0.468468\n",
       "1  TP53         0.693694   0.243243  0.468468\n",
       "2  MUC4         0.306306   0.153153  0.432432"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ovarian_freq_mutated_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
