{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from scipy import stats\n",
    "#import gseapy as gp\n",
    "#from gseapy.plot import barplot, dotplot\n",
    "# gseapy not found\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome to the CPTAC data service package. Available datasets may be\n",
      "viewed using CPTAC.list_data(). In order to access a specific data\n",
      "set, import a CPTAC subfolder using either 'import CPTAC.Dataset' or\n",
      "'from CPTAC import Dataset'.\n",
      "******\n",
      "Version: 0.2.7\n",
      "******\n",
      "Loading Endometrial CPTAC data:\n",
      "Loading Dictionary...\n",
      "Loading Clinical Data...\n",
      "Loading Acetylation Proteomics Data...\n",
      "Loading Proteomics Data...\n",
      "Loading Transcriptomics Data...\n",
      "Loading CNA Data...\n",
      "Loading Phosphoproteomics Data...\n",
      "Loading Somatic Mutation Data...\n",
      "\n",
      " ******PLEASE READ******\n",
      "WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but the data may not be in a\n",
      "publication until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter embargo() to open the webpage for more details.\n"
     ]
    }
   ],
   "source": [
    "import CPTAC.Endometrial as en"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical = en.get_clinical()\n",
    "proteomics = en.get_proteomics()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "tumorProt = en.compare_clinical(clinical_col='Proteomics_Tumor_Normal', omics_data=proteomics)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>idx</th>\n",
       "      <th>Proteomics_Tumor_Normal</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>-1.180</td>\n",
       "      <td>-0.8630</td>\n",
       "      <td>-0.802</td>\n",
       "      <td>0.222</td>\n",
       "      <td>0.2560</td>\n",
       "      <td>0.6650</td>\n",
       "      <td>1.2800</td>\n",
       "      <td>-0.3390</td>\n",
       "      <td>0.412</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.08770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0229</td>\n",
       "      <td>0.1090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.332</td>\n",
       "      <td>-0.43300</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>-0.1230</td>\n",
       "      <td>-0.0859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S002</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>-0.685</td>\n",
       "      <td>-1.0700</td>\n",
       "      <td>-0.684</td>\n",
       "      <td>0.984</td>\n",
       "      <td>0.1350</td>\n",
       "      <td>0.3340</td>\n",
       "      <td>1.3000</td>\n",
       "      <td>0.1390</td>\n",
       "      <td>1.330</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.03560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3630</td>\n",
       "      <td>1.0700</td>\n",
       "      <td>0.737</td>\n",
       "      <td>-0.564</td>\n",
       "      <td>-0.00461</td>\n",
       "      <td>-1.130</td>\n",
       "      <td>-0.0757</td>\n",
       "      <td>-0.4730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S003</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>-0.528</td>\n",
       "      <td>-1.3200</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2400</td>\n",
       "      <td>1.0400</td>\n",
       "      <td>-0.0213</td>\n",
       "      <td>-0.0479</td>\n",
       "      <td>0.419</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00112</td>\n",
       "      <td>-0.1450</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>-0.1160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.151</td>\n",
       "      <td>-0.07400</td>\n",
       "      <td>-0.540</td>\n",
       "      <td>0.3200</td>\n",
       "      <td>-0.4190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S005</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>-1.670</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>-0.443</td>\n",
       "      <td>0.243</td>\n",
       "      <td>-0.0993</td>\n",
       "      <td>0.7570</td>\n",
       "      <td>0.7400</td>\n",
       "      <td>-0.9290</td>\n",
       "      <td>0.229</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07250</td>\n",
       "      <td>-0.0552</td>\n",
       "      <td>-0.0714</td>\n",
       "      <td>0.0933</td>\n",
       "      <td>0.156</td>\n",
       "      <td>-0.398</td>\n",
       "      <td>-0.07520</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>-0.0301</td>\n",
       "      <td>-0.4670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S006</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>-0.374</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>-0.537</td>\n",
       "      <td>0.311</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>-1.1000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.565</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.17600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.2200</td>\n",
       "      <td>-0.5620</td>\n",
       "      <td>0.937</td>\n",
       "      <td>-0.646</td>\n",
       "      <td>0.20700</td>\n",
       "      <td>-1.850</td>\n",
       "      <td>-0.1760</td>\n",
       "      <td>0.0513</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11000 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "idx  Proteomics_Tumor_Normal   A1BG     A2M  A2ML1  A4GALT    AAAS    AACS  \\\n",
       "S001                   Tumor -1.180 -0.8630 -0.802   0.222  0.2560  0.6650   \n",
       "S002                   Tumor -0.685 -1.0700 -0.684   0.984  0.1350  0.3340   \n",
       "S003                   Tumor -0.528 -1.3200  0.435     NaN -0.2400  1.0400   \n",
       "S005                   Tumor -1.670 -1.1900 -0.443   0.243 -0.0993  0.7570   \n",
       "S006                   Tumor -0.374 -0.0206 -0.537   0.311  0.3750  0.0131   \n",
       "\n",
       "idx    AADAT   AAED1  AAGAB  ...   ZSWIM8  ZSWIM9    ZW10  ZWILCH  ZWINT  \\\n",
       "S001  1.2800 -0.3390  0.412  ... -0.08770     NaN  0.0229  0.1090    NaN   \n",
       "S002  1.3000  0.1390  1.330  ... -0.03560     NaN  0.3630  1.0700  0.737   \n",
       "S003 -0.0213 -0.0479  0.419  ...  0.00112 -0.1450  0.0105 -0.1160    NaN   \n",
       "S005  0.7400 -0.9290  0.229  ...  0.07250 -0.0552 -0.0714  0.0933  0.156   \n",
       "S006 -1.1000     NaN  0.565  ... -0.17600     NaN -1.2200 -0.5620  0.937   \n",
       "\n",
       "idx    ZXDC   ZYG11B    ZYX   ZZEF1    ZZZ3  \n",
       "S001 -0.332 -0.43300 -1.020 -0.1230 -0.0859  \n",
       "S002 -0.564 -0.00461 -1.130 -0.0757 -0.4730  \n",
       "S003  0.151 -0.07400 -0.540  0.3200 -0.4190  \n",
       "S005 -0.398 -0.07520 -0.797 -0.0301 -0.4670  \n",
       "S006 -0.646  0.20700 -1.850 -0.1760  0.0513  \n",
       "\n",
       "[5 rows x 11000 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumorProt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>idx</th>\n",
       "      <th>Proteomics_Tumor_Normal</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S105</th>\n",
       "      <td>Adjacent_normal</td>\n",
       "      <td>1.700</td>\n",
       "      <td>1.420</td>\n",
       "      <td>0.444</td>\n",
       "      <td>0.398</td>\n",
       "      <td>-0.0394</td>\n",
       "      <td>-0.604</td>\n",
       "      <td>-0.19500</td>\n",
       "      <td>0.209</td>\n",
       "      <td>-0.555</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2520</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.297</td>\n",
       "      <td>-0.130</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.721</td>\n",
       "      <td>0.309</td>\n",
       "      <td>1.08</td>\n",
       "      <td>-0.1370</td>\n",
       "      <td>0.14400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S106</th>\n",
       "      <td>Adjacent_normal</td>\n",
       "      <td>0.109</td>\n",
       "      <td>-0.579</td>\n",
       "      <td>0.226</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1350</td>\n",
       "      <td>-0.363</td>\n",
       "      <td>-0.10400</td>\n",
       "      <td>0.328</td>\n",
       "      <td>-1.110</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0236</td>\n",
       "      <td>0.283</td>\n",
       "      <td>-0.135</td>\n",
       "      <td>0.189</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.340</td>\n",
       "      <td>0.275</td>\n",
       "      <td>1.87</td>\n",
       "      <td>0.1530</td>\n",
       "      <td>0.47800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S107</th>\n",
       "      <td>Adjacent_normal</td>\n",
       "      <td>0.437</td>\n",
       "      <td>0.456</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0377</td>\n",
       "      <td>-0.720</td>\n",
       "      <td>0.02180</td>\n",
       "      <td>0.488</td>\n",
       "      <td>-0.469</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.281</td>\n",
       "      <td>-0.551</td>\n",
       "      <td>-0.2960</td>\n",
       "      <td>0.258</td>\n",
       "      <td>0.346</td>\n",
       "      <td>1.57</td>\n",
       "      <td>0.1300</td>\n",
       "      <td>0.00555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S108</th>\n",
       "      <td>Adjacent_normal</td>\n",
       "      <td>0.937</td>\n",
       "      <td>0.283</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1950</td>\n",
       "      <td>-0.798</td>\n",
       "      <td>0.00836</td>\n",
       "      <td>1.270</td>\n",
       "      <td>-0.850</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0741</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.352</td>\n",
       "      <td>-0.499</td>\n",
       "      <td>-0.0512</td>\n",
       "      <td>0.562</td>\n",
       "      <td>0.314</td>\n",
       "      <td>1.23</td>\n",
       "      <td>-0.0285</td>\n",
       "      <td>0.46100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S109</th>\n",
       "      <td>Adjacent_normal</td>\n",
       "      <td>0.581</td>\n",
       "      <td>-0.784</td>\n",
       "      <td>0.418</td>\n",
       "      <td>-0.123</td>\n",
       "      <td>-0.2490</td>\n",
       "      <td>-0.845</td>\n",
       "      <td>-0.12100</td>\n",
       "      <td>0.352</td>\n",
       "      <td>-0.482</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.283</td>\n",
       "      <td>-0.454</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.557</td>\n",
       "      <td>0.186</td>\n",
       "      <td>1.58</td>\n",
       "      <td>-0.1590</td>\n",
       "      <td>0.40000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11000 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "idx  Proteomics_Tumor_Normal   A1BG    A2M  A2ML1  A4GALT    AAAS   AACS  \\\n",
       "S105         Adjacent_normal  1.700  1.420  0.444   0.398 -0.0394 -0.604   \n",
       "S106         Adjacent_normal  0.109 -0.579  0.226     NaN -0.1350 -0.363   \n",
       "S107         Adjacent_normal  0.437  0.456    NaN     NaN -0.0377 -0.720   \n",
       "S108         Adjacent_normal  0.937  0.283    NaN     NaN  0.1950 -0.798   \n",
       "S109         Adjacent_normal  0.581 -0.784  0.418  -0.123 -0.2490 -0.845   \n",
       "\n",
       "idx     AADAT  AAED1  AAGAB  ...  ZSWIM8  ZSWIM9   ZW10  ZWILCH   ZWINT  \\\n",
       "S105 -0.19500  0.209 -0.555  ... -0.2520     NaN -0.297  -0.130     NaN   \n",
       "S106 -0.10400  0.328 -1.110  ...  0.0236   0.283 -0.135   0.189     NaN   \n",
       "S107  0.02180  0.488 -0.469  ...  0.0162     NaN -0.281  -0.551 -0.2960   \n",
       "S108  0.00836  1.270 -0.850  ... -0.0741     NaN -0.352  -0.499 -0.0512   \n",
       "S109 -0.12100  0.352 -0.482  ...  0.1450     NaN -0.283  -0.454     NaN   \n",
       "\n",
       "idx    ZXDC  ZYG11B   ZYX   ZZEF1     ZZZ3  \n",
       "S105  0.721   0.309  1.08 -0.1370  0.14400  \n",
       "S106  0.340   0.275  1.87  0.1530  0.47800  \n",
       "S107  0.258   0.346  1.57  0.1300  0.00555  \n",
       "S108  0.562   0.314  1.23 -0.0285  0.46100  \n",
       "S109  0.557   0.186  1.58 -0.1590  0.40000  \n",
       "\n",
       "[5 rows x 11000 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumor_bool =tumorProt['Proteomics_Tumor_Normal'] == \"Tumor\"\n",
    "normal_bool =tumorProt['Proteomics_Tumor_Normal'] != \"Tumor\"\n",
    "tumor = tumorProt[tumor_bool]\n",
    "#tumor.head()\n",
    "normal = tumorProt[normal_bool]\n",
    "normal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tumor_genes = []\n",
    "normal_genes = []\n",
    "genes = tumor.column[1:]\n",
    "threshold = .05 / len(genes)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
