{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that index is up-to-date...Loading clinical data...Loading cnv data...Loading definitions data...Loading phosphoproteomics data...Loading proteomics data...Loading somatic_38 data...Loading transcriptomics data...Loading treatment data...Formatting dataframes...Checking that index is up-to-date...Loading acetylproteomics data...Loading clinical data...Loading CNA data...Loading definitions data...Loading miRNA data...Loading phosphoproteomics_gene data...Loading phosphoproteomics_site data...Loading proteomics data...Loading somatic_binary data...Loading somatic data...Loading transcriptomics_circular data...Loading transcriptomics_linear data...Formatting dataframes...WARNING: This data is under a publication embargo until July 1, 2019.\n",
      "CPTAC is a community resource project and data are made available\n",
      "rapidly after generation for community research use. The embargo\n",
      "allows exploring and utilizing the data, but analysis may not be\n",
      "published until July 1, 2019. Please see\n",
      "https://proteomics.cancer.gov/data-portal/about/data-use-agreement or\n",
      "enter cptac.embargo() to open the webpage for more details.\n",
      "Checking that index is up-to-date...Loading clinical data...Loading miRNA data...Loading mutation_binary data...Loading mutation data...Loading phosphoproteomics_normal data...Loading phosphoproteomics_tumor data...Loading proteomics_normal data...Loading proteomics_tumor data...Loading transcriptomics data...Formatting dataframes..."
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import collections\n",
    "import re\n",
    "import gseapy as gp\n",
    "from gseapy.plot import barplot, dotplot\n",
    "\n",
    "import cptac\n",
    "import operator\n",
    "\n",
    "ov = cptac.Ovarian()\n",
    "en = cptac.Endometrial()\n",
    "co = cptac.Colon()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_frequently_mutated(cancer_type, cutoff=.1):  \n",
    "    \"\"\"\n",
    "    take cancer object and find the frequently \n",
    "    mutated genes in total tumors compared to the cutoff.\n",
    "        \n",
    "        Parameters:\n",
    "        cancer_type (object): object of cancer class from cptac module \n",
    "        cutoff (float): used as comparison to determine status of gene mutation frequency\n",
    "        \n",
    "        Returns:\n",
    "        freq_mutated_df (pd.DataFrame): DataFrame of frequently mutated genes passing the cutoff\n",
    "            and Total_Mutated (mutated genes / total tumors), percent Truncated, and percent Missence\n",
    "        \n",
    "        There are many types of mutations catagorized into the columns Truncated and Missence. \n",
    "        The Truncated column includes: Frame_Shift_Del, Frame_Shift_Ins, Splice_Site, Nonsense_Mutation, Nonstop_Mutation\n",
    "        The Missence column includes: In_Frame_Del, In_Frame_Ins, Missense_Mutation\n",
    "        These columns count multiple mutations of one gene in the same sample, so percentages in the last two columns may \n",
    "        exceed the Total_Mutated column(which only counts if the gene was mutated once)\"\"\"    \n",
    "    \n",
    "    # get data frames\n",
    "    somatic_mutations = cancer_type.get_mutations()\n",
    "    sample_status_map = cancer_type._get_sample_status_map()\n",
    "    merged_mutations = somatic_mutations.join(sample_status_map, how=\"left\") \n",
    "    \"\"\"\n",
    "    # standardize mutation names \n",
    "    if cancer_type.get_cancer_type() == 'Colon':\n",
    "        mutation_equivalents = {'frameshift substitution': 'Frame_Shift_Del' , 'frameshift deletion': 'Frame_Shift_Del', \n",
    "            'frameshift insertion': 'Frame_Shift_Ins', 'stopgain': 'Nonsense_Mutation ', 'stoploss':'Nonstop_Mutation',\n",
    "            'nonsynonymous SNV': 'Missense_Mutation','nonframeshift insertion': 'In_Frame_Ins',\n",
    "            'nonframeshift deletion': 'In_Frame_Del', 'nonframeshift substitution': 'Missense_Mutation'}\n",
    "        merged_mutations = merged_mutations.replace(to_replace = mutation_equivalents)\n",
    "    \"\"\"  \n",
    "    # get list of unique genes\n",
    "    unique_genes = somatic_mutations['Gene'].unique()\n",
    "    \n",
    "    # get total tumors/patients\n",
    "    sample_status_series = sample_status_map.value_counts()\n",
    "    total_tumor_patients = sample_status_series[0]\n",
    "        \n",
    "    # find frequently mutated genes and their total mutated fraction. Create lists for frequently mutated genes and fraction.\n",
    "    freq_mut = [] \n",
    "    total_fraction_mutated = []\n",
    "    for gene in unique_genes:\n",
    "        gene_mutated = merged_mutations.loc[merged_mutations['Gene'] == gene]\n",
    "        gene_mutated = gene_mutated.index.unique()\n",
    "        num_gene_mutated = len(gene_mutated)\n",
    "        fraction = (num_gene_mutated / total_tumor_patients)\n",
    "        if fraction > cutoff:\n",
    "            freq_mut.append(gene)\n",
    "            total_fraction_mutated.append(fraction)\n",
    "    \n",
    "    # find truncated fraction\n",
    "    truncated = []\n",
    "    for gene in freq_mut:\n",
    "        gene_mutated = merged_mutations.loc[merged_mutations['Gene'] == gene] #possibly change var name\n",
    "        #truncated_df = gene_mutated.loc[(gene_mutated['Mutation'] not in my_list)]\n",
    "        truncated_df = gene_mutated.loc[(gene_mutated['Mutation'] != 'In_Frame_Del') & \n",
    "            (gene_mutated['Mutation'] != 'In_Frame_Ins') & (gene_mutated['Mutation'] != 'Missense_Mutation')] \n",
    "        samples_trunc = truncated_df.index.unique()\n",
    "        num_trunc_mut = len(samples_trunc)\n",
    "        fraction_trunc = (num_trunc_mut / total_tumor_patients)\n",
    "        truncated.append(fraction_trunc)\n",
    "\n",
    "    # find missence fraction #combine for loop\n",
    "    missence = []\n",
    "    for gene in freq_mut:\n",
    "        gene_mutated = merged_mutations.loc[merged_mutations['Gene'] == gene]\n",
    "        #missence_mutations = gene_mutated.loc[(gene_mutated['Mutation'] in my_list)]\n",
    "        missence_mutations = gene_mutated.loc[(gene_mutated['Mutation'] == 'In_Frame_Ins') | (gene_mutated['Mutation'] == 'In_Frame_Del') | (gene_mutated['Mutation'] == 'Missense_Mutation')]\n",
    "        samples_miss = missence_mutations.index.unique()\n",
    "        num_miss_mut = len(samples_miss)\n",
    "        fraction_miss = (num_miss_mut / total_tumor_patients)\n",
    "        missence.append(fraction_miss)\n",
    "        \n",
    "    # create dataframe\n",
    "    merged_lists = list(zip(freq_mut, total_fraction_mutated, truncated, missence))\n",
    "    freq_mutated_df = pd.DataFrame(merged_lists, columns =['Gene', 'Fraction_Mutated', 'Truncation', 'Missence'])\n",
    "    freq_mutated_df.name = 'frequently_mutated'\n",
    "                   \n",
    "    return freq_mutated_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "fmo = get_frequently_mutated(ov, .15)\n",
    "fme = get_frequently_mutated(en, .15)\n",
    "fmc = get_frequently_mutated(co, .15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "APC :  [('COLON', 0.7454545454545455)]\n",
      "NRXN1 :  [('COLON', 0.18181818181818182)]\n",
      "FBXW7 :  [('COLON', 0.15454545454545454), ('ENDO', 0.18947368421052632)]\n",
      "TP53 :  [('COLON', 0.509090909090909), ('ENDO', 0.22105263157894736), ('OVARIAN', 0.6936936936936937)]\n",
      "MUC16 :  [('COLON', 0.35454545454545455), ('ENDO', 0.18947368421052632)]\n",
      "FAT3 :  [('COLON', 0.2545454545454545)]\n",
      "PCDH17 :  [('COLON', 0.15454545454545454)]\n",
      "HMCN1 :  [('COLON', 0.20909090909090908)]\n",
      "APOB :  [('COLON', 0.17272727272727273)]\n",
      "SYNE1 :  [('COLON', 0.2636363636363636), ('ENDO', 0.16842105263157894)]\n",
      "DNAH7 :  [('COLON', 0.15454545454545454)]\n",
      "TTN :  [('COLON', 0.5454545454545454), ('OVARIAN', 0.27927927927927926)]\n",
      "KMT2B :  [('COLON', 0.15454545454545454), ('ENDO', 0.24210526315789474)]\n",
      "BSN :  [('COLON', 0.15454545454545454)]\n",
      "NBEA :  [('COLON', 0.19090909090909092)]\n",
      "RYR3 :  [('COLON', 0.17272727272727273)]\n",
      "KMT2D :  [('COLON', 0.17272727272727273), ('ENDO', 0.17894736842105263)]\n",
      "ARID1A :  [('COLON', 0.20909090909090908), ('ENDO', 0.45263157894736844)]\n",
      "CSMD3 :  [('COLON', 0.23636363636363636)]\n",
      "RNF43 :  [('COLON', 0.16363636363636364)]\n",
      "LRP1B :  [('COLON', 0.15454545454545454)]\n",
      "HELZ2 :  [('COLON', 0.16363636363636364)]\n",
      "IGFN1 :  [('COLON', 0.15454545454545454)]\n",
      "KRAS :  [('COLON', 0.3181818181818182), ('ENDO', 0.3263157894736842)]\n",
      "TCF7L2 :  [('COLON', 0.21818181818181817)]\n",
      "OBSCN :  [('COLON', 0.2636363636363636), ('ENDO', 0.15789473684210525)]\n",
      "MYCBP2 :  [('COLON', 0.19090909090909092)]\n",
      "ZFHX4 :  [('COLON', 0.17272727272727273), ('ENDO', 0.2)]\n",
      "SETD1B :  [('COLON', 0.17272727272727273)]\n",
      "MUC5B :  [('COLON', 0.24545454545454545)]\n",
      "NCOR2 :  [('COLON', 0.21818181818181817)]\n",
      "ZNF469 :  [('COLON', 0.22727272727272727), ('ENDO', 0.15789473684210525)]\n",
      "ADGRV1 :  [('COLON', 0.17272727272727273)]\n",
      "RYR2 :  [('COLON', 0.19090909090909092)]\n",
      "FAT2 :  [('COLON', 0.16363636363636364)]\n",
      "RYR1 :  [('COLON', 0.22727272727272727)]\n",
      "SSPO :  [('COLON', 0.19090909090909092)]\n",
      "COL5A1 :  [('COLON', 0.19090909090909092)]\n",
      "ZDBF2 :  [('COLON', 0.16363636363636364)]\n",
      "SPEG :  [('COLON', 0.18181818181818182)]\n",
      "ZNF292 :  [('COLON', 0.16363636363636364)]\n",
      "PIK3CA :  [('COLON', 0.21818181818181817), ('ENDO', 0.49473684210526314)]\n",
      "PLEC :  [('COLON', 0.20909090909090908)]\n",
      "KMT2C :  [('COLON', 0.18181818181818182)]\n",
      "TRPS1 :  [('COLON', 0.16363636363636364)]\n",
      "ACVR2A :  [('COLON', 0.22727272727272727)]\n",
      "HERC2 :  [('COLON', 0.19090909090909092), ('ENDO', 0.17894736842105263)]\n",
      "DOCK3 :  [('COLON', 0.19090909090909092), ('ENDO', 0.2)]\n",
      "USF3 :  [('COLON', 0.21818181818181817)]\n",
      "TENM4 :  [('COLON', 0.15454545454545454)]\n",
      "CASP5 :  [('COLON', 0.17272727272727273)]\n",
      "ROBO2 :  [('COLON', 0.15454545454545454)]\n",
      "BRAF :  [('COLON', 0.16363636363636364)]\n",
      "SOX9 :  [('COLON', 0.17272727272727273)]\n",
      "PCLO :  [('COLON', 0.21818181818181817), ('ENDO', 0.16842105263157894)]\n",
      "FAT4 :  [('COLON', 0.23636363636363636)]\n",
      "SLC4A3 :  [('COLON', 0.18181818181818182)]\n",
      "NEB :  [('COLON', 0.15454545454545454)]\n",
      "HSPG2 :  [('COLON', 0.16363636363636364)]\n",
      "CCDC168 :  [('COLON', 0.2636363636363636), ('ENDO', 0.16842105263157894)]\n",
      "RPL22 :  [('COLON', 0.17272727272727273), ('ENDO', 0.16842105263157894)]\n",
      "FSIP2 :  [('COLON', 0.2)]\n",
      "LAMA5 :  [('COLON', 0.15454545454545454)]\n",
      "AHNAK2 :  [('COLON', 0.19090909090909092)]\n",
      "SACS :  [('COLON', 0.18181818181818182)]\n",
      "DST :  [('COLON', 0.15454545454545454)]\n",
      "SDK1 :  [('COLON', 0.15454545454545454)]\n",
      "ABCA13 :  [('COLON', 0.17272727272727273)]\n",
      "ABCA2 :  [('COLON', 0.15454545454545454)]\n",
      "DNAH10 :  [('COLON', 0.15454545454545454)]\n",
      "DMD :  [('COLON', 0.17272727272727273)]\n",
      "DNAH5 :  [('COLON', 0.23636363636363636)]\n",
      "PKD1 :  [('COLON', 0.16363636363636364)]\n",
      "FLG :  [('COLON', 0.15454545454545454)]\n",
      "VWF :  [('COLON', 0.15454545454545454)]\n",
      "MUC12 :  [('COLON', 0.16363636363636364)]\n",
      "DCHS2 :  [('COLON', 0.17272727272727273)]\n",
      "MUC4 :  [('COLON', 0.18181818181818182), ('OVARIAN', 0.3063063063063063)]\n",
      "KCNMA1 :  [('COLON', 0.15454545454545454)]\n",
      "TENM3 :  [('COLON', 0.16363636363636364)]\n",
      "HYDIN :  [('COLON', 0.16363636363636364)]\n",
      "DNHD1 :  [('COLON', 0.16363636363636364)]\n",
      "DNAH3 :  [('COLON', 0.17272727272727273)]\n",
      "RBM27 :  [('ENDO', 0.16842105263157894)]\n",
      "NSD1 :  [('ENDO', 0.16842105263157894)]\n",
      "PTEN :  [('ENDO', 0.7894736842105263)]\n",
      "CTCF :  [('ENDO', 0.28421052631578947)]\n",
      "LMAN1 :  [('ENDO', 0.15789473684210525)]\n",
      "SCAF4 :  [('ENDO', 0.16842105263157894)]\n",
      "PIK3R1 :  [('ENDO', 0.3894736842105263)]\n",
      "CTNNB1 :  [('ENDO', 0.30526315789473685)]\n",
      "ZFHX3 :  [('ENDO', 0.22105263157894736)]\n",
      "JAK1 :  [('ENDO', 0.16842105263157894)]\n",
      "AHNAK :  [('ENDO', 0.15789473684210525)]\n",
      "HUWE1 :  [('ENDO', 0.15789473684210525)]\n",
      "DNAH17 :  [('ENDO', 0.15789473684210525)]\n",
      "INPPL1 :  [('ENDO', 0.2)]\n"
     ]
    }
   ],
   "source": [
    "genes = {}\n",
    "for ind, row in fmc.iterrows():\n",
    "    if row['Gene'] not in genes.keys():\n",
    "        genes[row['Gene']] = [(\"COLON\",row['Fraction_Mutated'])]\n",
    "    else:\n",
    "        genes[row['Gene']].append((\"COLON\",row['Fraction_Mutated']))\n",
    "        \n",
    "\n",
    "for ind, row in fme.iterrows():\n",
    "    if row['Gene'] not in genes.keys():\n",
    "        genes[row['Gene']] = [(\"ENDO\",row['Fraction_Mutated'])]\n",
    "    else:\n",
    "        genes[row['Gene']].append((\"ENDO\",row['Fraction_Mutated']))\n",
    "        \n",
    "        \n",
    "for ind, row in fmo.iterrows():\n",
    "    if row['Gene'] not in genes.keys():\n",
    "        genes[row['Gene']] = [(\"OVARIAN\",row['Fraction_Mutated'])]\n",
    "    else:\n",
    "        genes[row['Gene']].append((\"OVARIAN\",row['Fraction_Mutated']))\n",
    "\n",
    "        \n",
    "for gene in genes.keys():\n",
    "    print(gene, \": \", genes[gene])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
